{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Programming in life sciences"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5611412d",
   "metadata": {},
   "source": [
    "[Anaconda](https://www.anaconda.com/download/success)\n",
    "\n",
    "`conda create --name new_env`\n",
    "\n",
    "`conda activate new_env`\n",
    "\n",
    "Install libraries you want in isolated env"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Predicting open/close conformation of a kinase model\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edef8232",
   "metadata": {},
   "source": [
    "<b>Lecture 1.1: Introduction to alignments - exercise: align two homologous kinases (given as fasta files)</b>\n",
    "\n",
    "Lecture 1.2: Introduction to superpositions - exercise: get RMSD of two structures\n",
    "\n",
    "Lecture 2: Introduction to AlphaFold - exercise: get models from AlphaFold and classify them as open/close, do the same for models from Swiss-Model and AlphaFold-dropout\n",
    "\n",
    "Lecture 3: Ligand docking (DiffDock) - exercise: get docking from DiffDock colab, find interacting atoms and visualize docking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe148b03",
   "metadata": {
    "tags": [
     "lesson1"
    ]
   },
   "source": [
    "### 1. Lecture: Needleman-Wunsch algorithm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "933ffbba",
   "metadata": {},
   "source": [
    "In this exercise, we want to align two sequences obtained from fasta files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6d97f4c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# numpy is a library providing a lot of numerical mathematical functionalities\n",
    "# here we only use it to store stuff in matrices\n",
    "import numpy as np\n",
    "\n",
    "# we create a new class that will be used to define substitution matrices\n",
    "class SubstitutionMatrix:\n",
    "  \"\"\"Read, store and analyse substitution matrices, default = BLOSUM62\"\"\"\n",
    "\n",
    "  # this is the constructor function! When you create a new instance of type\n",
    "  # Substitution Matrix, this function gets called. By default it reads in\n",
    "  # the provided file: blosum62.txt... you can create your own matrix if\n",
    "  # you like ;)\n",
    "  def __init__(self, data_file: str = \"blosum62.txt\"): # takes datafile downloaded from adam\n",
    "    \"\"\" Constructor function for the SubstitutionMatrix class\n",
    "    Parameters\n",
    "    ----------\n",
    "    data_file : string\n",
    "        Filename of the substitution matrix to read in, must contain 21 lines,\n",
    "        with the first line containing the 20 amino acid one letter codes,\n",
    "        by default reads the blosum62.txt matrix in the current folder\n",
    "    \"\"\"\n",
    "    \n",
    "    # read all the lines from the file at once  ('r' = read)\n",
    "    with open(data_file,'r') as fh:\n",
    "      data = fh.readlines()\n",
    "\n",
    "    # the size of subsitution matrices must be 20x20\n",
    "    # the first line of the file gives the amino acid one letter codes\n",
    "    # which gives a total of 21 lines in the file\n",
    "    if len(data) != 21:\n",
    "      err = \"Expected exactly 21 lines in matrix file!\"\n",
    "      err += \" One line defining the amino acid one letter codes and \"\n",
    "      err += \"20 lines defining the substitution matrix\"\n",
    "      raise RuntimeError(err)    \n",
    "\n",
    "    # get the amino acid one letter codes\n",
    "    self.one_letter_codes = data[0].strip()\n",
    "\n",
    "    # check that 20 amino acids are defined\n",
    "    if len(self.one_letter_codes) != 20:\n",
    "      err = \"First line must contain exactly 20 letters representing \"\n",
    "      err += \"the amino acids!\"\n",
    "      raise RuntimeError(err)\n",
    "\n",
    "    # initialize the substitution matrix with zeros\n",
    "    self.matrix = np.zeros((20,20))\n",
    "\n",
    "    # fill in the matrix by reading the file line by line\n",
    "    # note that range returns a vector starting at 0 to the integer in argument - 1 \n",
    "    for i in range(20):\n",
    "      data_line = data[1+i].split()\n",
    "      if len(data_line) != 20:\n",
    "        raise RuntimeError(\"Each data line must contain exactly 20 elements!\")\n",
    "      for j in range(20):\n",
    "        self.matrix[(i,j)] = int(data_line[j])\n",
    "\n",
    "\n",
    "  # The function where you can extract the score for a specific amino acid substitution...\n",
    "  # e.g. mat.get_score('W', 'A')\n",
    "  def get_score(self, aa_one: str, aa_two: str):\n",
    "    \"\"\" Extract the substitution score for two amino acids from the substitution matrix\n",
    "    Parameters\n",
    "    ----------\n",
    "    aa_one : string\n",
    "        One letter amino acid code\n",
    "    aa_two : string\n",
    "        One letter amino acid code\n",
    "    Returns\n",
    "    -------\n",
    "    self.matrix[(idx_one,idx_two)] : float\n",
    "        Substitution matrix entry for the substitution of amino acid one and amino acid two,\n",
    "        note that the matrices are symmetric, so get_score(aa_one, aa_two) is equal to get_score(aa_two, aa_one)\n",
    "    \"\"\"\n",
    "\n",
    "    # get the indices of the amino acids in the matrix\n",
    "    idx_one = self.one_letter_codes.find(aa_one)\n",
    "    idx_two = self.one_letter_codes.find(aa_two)\n",
    "\n",
    "    if idx_one == -1:\n",
    "      raise RuntimeError(\"aa_one must be one of: \" + self.one_letter_codes)\n",
    "    \n",
    "    if idx_two == -1:\n",
    "      raise RuntimeError(\"aa_two must be one of: \" + self.one_letter_codes)\n",
    "\n",
    "    return self.matrix[(idx_one,idx_two)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0f8b4798",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function get_score in module __main__:\n",
      "\n",
      "get_score(self, aa_one: str, aa_two: str)\n",
      "    Extract the substitution score for two amino acids from the substitution matrix\n",
      "    Parameters\n",
      "    ----------\n",
      "    aa_one : string\n",
      "        One letter amino acid code\n",
      "    aa_two : string\n",
      "        One letter amino acid code\n",
      "    Returns\n",
      "    -------\n",
      "    self.matrix[(idx_one,idx_two)] : float\n",
      "        Substitution matrix entry for the substitution of amino acid one and amino acid two,\n",
      "        note that the matrices are symmetric, so get_score(aa_one, aa_two) is equal to get_score(aa_two, aa_one)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Decorating functions with docstrings is not only best practice,\n",
    "# it also helps automated documentation generation with tools\n",
    "# like Sphinx (https://www.sphinx-doc.org) or simply get more\n",
    "# information using the Python help function:\n",
    "help(SubstitutionMatrix.get_score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7bedee41",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Score D->N: 1.0\n",
      "Score G->P: -2.0\n"
     ]
    }
   ],
   "source": [
    "# Construct a default substitution matrix\n",
    "subst_matrix = SubstitutionMatrix()\n",
    "\n",
    "# get score of substituting ASP with ASN and GLY with PRO\n",
    "print(\"Score D->N:\", subst_matrix.get_score('N', 'D'))\n",
    "print(\"Score G->P:\", subst_matrix.get_score('G', 'P'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a94cabbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def ReadFasta(filename: str):\n",
    "    \"\"\" A simplistic reader that only checks if the first line starts with '>' and assumes that the whole rest is the input sequence\n",
    "    Parameters\n",
    "    ----------\n",
    "    filename : string\n",
    "        Filename of the FASTA file to read\n",
    "    Returns\n",
    "    -------\n",
    "    s : string\n",
    "        Sequence read from the input FASTA file\n",
    "    \"\"\"\n",
    "    # TODO Fill the function\n",
    "\n",
    "    f = open(filename)\n",
    "    s = ''\n",
    "\n",
    "    first_line = f.readline().strip()\n",
    "\n",
    "    if first_line.startswith(\">\"):\n",
    "        for line in f: \n",
    "            s += line.strip()\n",
    "    \n",
    "    else: \n",
    "        print(\"Error: File doesn't seem to be Fasta, or just doesnt start with '>'. \")\n",
    "        s += ''\n",
    "\n",
    "    f.close()\n",
    "    return s\n",
    "\n",
    "    pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2f651f70",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VDRVG\n",
      "VERMAG\n"
     ]
    }
   ],
   "source": [
    "# TODO\n",
    "# read in the two sequences to align s1 & s2\n",
    "# the two files are located in the same directory as the notebook\n",
    "\n",
    "s1 = ReadFasta(\"s1.fasta\")\n",
    "s2 = ReadFasta(\"s2.fasta\")\n",
    "\n",
    "print(s1)\n",
    "print(s2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44b39e0a",
   "metadata": {},
   "source": [
    "#### Write the code for running the Needleman-Wunsch algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "09b8cc4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# define a penalty for introducing a gap in the sequence alignment\n",
    "gap_penalty = -8"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d455926",
   "metadata": {},
   "source": [
    "Setup scoring and backtracking matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ce456d98",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "row sequence:  MRIVLLGAPGAGKGTQAQFLMAKFGIPQISTGDMLRAAIKAGSELGNKAKAVMDAGQLVSDDLIIGLVKERVAQEDCKAGFLLDGFPRTIPQADAMKESGIVVDHVLEFDVPDEVIVERMAGRRVHSGSGRVYHLVYNPPKVEGKDDVSGDDLSIRPDDEEATVRKRLAIYHEQTKPLVDFYQAEAKSGSCSYLTIDGTQAVEKVNQLLSAQLA\n",
      "col sequence:  MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n",
      "empty scoring matrix: \n",
      "[[    0.    -8.   -16. ... -1696. -1704. -1712.]\n",
      " [   -8.     0.     0. ...     0.     0.     0.]\n",
      " [  -16.     0.     0. ...     0.     0.     0.]\n",
      " ...\n",
      " [-1696.     0.     0. ...     0.     0.     0.]\n",
      " [-1704.     0.     0. ...     0.     0.     0.]\n",
      " [-1712.     0.     0. ...     0.     0.     0.]]\n",
      "empty backtracking matrix: \n",
      "[[0. 2. 2. ... 2. 2. 2.]\n",
      " [3. 0. 0. ... 0. 0. 0.]\n",
      " [3. 0. 0. ... 0. 0. 0.]\n",
      " ...\n",
      " [3. 0. 0. ... 0. 0. 0.]\n",
      " [3. 0. 0. ... 0. 0. 0.]\n",
      " [3. 0. 0. ... 0. 0. 0.]]\n"
     ]
    }
   ],
   "source": [
    "# Keeps track of what path we chose going through the matrix\n",
    "\n",
    "# s1 = ReadFasta(\"s1.fasta\")\n",
    "# s2 = ReadFasta(\"s2.fasta\")\n",
    "\n",
    "s1 = ReadFasta(\"Q47XA8.fasta\")\n",
    "s2 = ReadFasta(\"P69441.fasta\")\n",
    "\n",
    "n_rows = len(s1)+1 # s1 goes from top to bottom (in the algorith we access i-1, so we need to add 1)\n",
    "n_cols = len(s2)+1 # s2 goes from left to right\n",
    "\n",
    "scoring_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "# backtrack encoding:\n",
    "# 1 => aligned\n",
    "# 2 => deletion in s1 (insertion in s2 respectively)\n",
    "# 3 => deletion in s2 (insertion in s1 respectively)\n",
    "backtrack_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "# the first row and the first column can already be prefilled\n",
    "for i in range(1, n_rows):\n",
    "    scoring_matrix[(i,0)] = i*gap_penalty\n",
    "    backtrack_matrix[(i,0)] = 3\n",
    "for i in range(1, n_cols):\n",
    "    scoring_matrix[(0,i)] = i*gap_penalty\n",
    "    backtrack_matrix[(0,i)] = 2\n",
    "\n",
    "print(\"row sequence: \", s1)\n",
    "print(\"col sequence: \", s2)\n",
    "print(\"empty scoring matrix: \")\n",
    "print(scoring_matrix)\n",
    "print(\"empty backtracking matrix: \")\n",
    "print(backtrack_matrix)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "710f66ae",
   "metadata": {},
   "source": [
    "#### Fill scoring and backtracking matrices\n",
    "\n",
    "Every position in the scoring matrix represents the best local solution given the\n",
    "steps I can take\n",
    "1. Align two residues => Move along the diagonal (consume a letter from s1 and s2),\n",
    "                        the value of the scoring matrix to the upper left plus the corresponding score from the scoring matrix\n",
    "2. Apply deletion/gap in s1 => Move right in the scoring matrix (consume a letter from s2), \n",
    "                              the value from the scoring matrix to the left plus the penalty for a gap\n",
    "3. Apply deletion/gap in s2 => Move down in the scoring matrix (consume a letter from s1),\n",
    "                              the value from the scoring matrix above plus the penalty for a gap\n",
    "\n",
    "The goal is to identify the best scoring solution from those three,\n",
    "set the according score in the scoring matrix and keep track of this local\n",
    "step in the backtracking matrix. \n",
    "\n",
    "The backtracking matrix stores the path I took, e.g. 1 for the first option\n",
    "(see backtracking encoding in previous code cell)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "fda9db23",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "filled scoring matrix:\n",
      "[[    0.    -8.   -16. ... -1696. -1704. -1712.]\n",
      " [   -8.     5.    -3. ... -1683. -1691. -1699.]\n",
      " [  -16.    -3.    10. ... -1670. -1678. -1686.]\n",
      " ...\n",
      " [-1696. -1683. -1670. ...   785.   780.   772.]\n",
      " [-1704. -1691. -1678. ...   784.   789.   781.]\n",
      " [-1712. -1699. -1686. ...   776.   783.   789.]]\n",
      "filled backtracking matrix:\n",
      "[[0. 2. 2. ... 2. 2. 2.]\n",
      " [3. 1. 2. ... 2. 2. 2.]\n",
      " [3. 3. 1. ... 2. 2. 2.]\n",
      " ...\n",
      " [3. 3. 3. ... 1. 1. 1.]\n",
      " [3. 3. 3. ... 1. 1. 2.]\n",
      " [3. 3. 3. ... 3. 1. 1.]]\n"
     ]
    }
   ],
   "source": [
    "for r_idx in range(1, n_rows):\n",
    "    for c_idx in range(1, n_cols):\n",
    "\n",
    "        # TODO calculate the scores\n",
    "\n",
    "\n",
    "        score_match = scoring_matrix[(r_idx - 1, c_idx - 1)] + subst_matrix.get_score(s1[r_idx-1], s2[c_idx-1])\n",
    "\n",
    "        score_del_1 = scoring_matrix[(r_idx, c_idx - 1)] + gap_penalty\n",
    "        score_del_2 = scoring_matrix[(r_idx - 1, c_idx)] + gap_penalty\n",
    "        \n",
    "        # TODO Fill the values at position (r_idx, c_idx) in scoring_matrix and backtracking_matrix\n",
    "        # you can assign an element with\n",
    "        # scoring_matrix[(r_idx, c_idx)] = my_awesome_value\n",
    "        if score_match >= score_del_1 and score_match >= score_del_2: \n",
    "            scoring_matrix[(r_idx, c_idx)] = score_match #fill score into the matrix\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 1 #fill with corresponding number \n",
    "        elif score_del_1 >= score_match and score_del_1 >= score_del_2:\n",
    "            scoring_matrix[(r_idx, c_idx)] = score_del_1\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 2\n",
    "        else:\n",
    "            scoring_matrix[(r_idx, c_idx)] = score_del_2\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 3\n",
    "        \n",
    "print(\"filled scoring matrix:\")\n",
    "print(scoring_matrix)\n",
    "print(\"filled backtracking matrix:\")\n",
    "print(backtrack_matrix)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f7572fe",
   "metadata": {},
   "source": [
    "#### Perform backtracking to get final alignment\n",
    "\n",
    "Remember the encoding:\n",
    "\n",
    "1 => aligned, move along the diagonal\n",
    "\n",
    "2 => deletion in s1 (insertion in s2 respectively), move right\n",
    "\n",
    "3 => deletion in s2 (insertion in s1 respectively), move down"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9b8e7212",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input sequence s1: MRIVLLGAPGAGKGTQAQFLMAKFGIPQISTGDMLRAAIKAGSELGNKAKAVMDAGQLVSDDLIIGLVKERVAQEDCKAGFLLDGFPRTIPQADAMKESGIVVDHVLEFDVPDEVIVERMAGRRVHSGSGRVYHLVYNPPKVEGKDDVSGDDLSIRPDDEEATVRKRLAIYHEQTKPLVDFYQAEAKSGSCSYLTIDGTQAVEKVNQLLSAQLA\n",
      "input sequence s2: MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n",
      "\n",
      "optimal alignment:\n",
      "s1: MRIVLLGAPGAGKGTQAQFLMAKFGIPQISTGDMLRAAIKAGSELGNKAKAVMDAGQLVSDDLIIGLVKERVAQEDCKAGFLLDGFPRTIPQADAMKESGIVVDHVLEFDVPDEVIVERMAGRRVHSGSGRVYHLVYNPPKVEGKDDVSGDDLSIRPDDEEATVRKRLAIYHEQTKPLVDFYQAEAKSGSCSYLTIDGTQAVEKVNQLLSAQLA\n",
      "s2: MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n"
     ]
    }
   ],
   "source": [
    "# In principle we start at the bottom right of the backtracking matrix and\n",
    "# work our way through the matrix until we hit the upper left\n",
    "\n",
    "r_idx = n_rows-1\n",
    "c_idx = n_cols-1\n",
    "path = list()\n",
    "\n",
    "# if we hit zero, we're at the upper left corner and can stop\n",
    "while backtrack_matrix[(r_idx, c_idx)] != 0:\n",
    "    path.append(backtrack_matrix[(r_idx, c_idx)])\n",
    "    if backtrack_matrix[(r_idx, c_idx)] == 1: \n",
    "        c_idx = c_idx - 1\n",
    "        r_idx = r_idx - 1\n",
    "        \n",
    "    elif backtrack_matrix[(r_idx, c_idx)] == 2:\n",
    "        c_idx = c_idx - 1\n",
    "\n",
    "    elif backtrack_matrix[(r_idx, c_idx)] == 3:\n",
    "        r_idx = r_idx - 1\n",
    "        \n",
    "# backtracking comes from the back, so lets reverse...\n",
    "path.reverse()\n",
    "\n",
    "aln_s1 = []\n",
    "aln_s2 = []\n",
    "s1_idx = 0\n",
    "s2_idx = 0\n",
    "\n",
    "for p in path:\n",
    "    if p == 1:\n",
    "        aln_s1.append(s1[s1_idx])\n",
    "        aln_s2.append(s2[s2_idx])\n",
    "        s1_idx += 1\n",
    "        s2_idx += 1\n",
    "    elif p == 2:\n",
    "        aln_s1.append('-')\n",
    "        aln_s2.append(s2[s2_idx])\n",
    "        s2_idx += 1\n",
    "    else:\n",
    "        aln_s1.append(s1[s1_idx])\n",
    "        aln_s2.append('-')\n",
    "        s1_idx += 1\n",
    "        \n",
    "aln_s1 = ''.join(aln_s1)\n",
    "aln_s2 = ''.join(aln_s2)\n",
    "        \n",
    "print(\"input sequence s1:\", s1)\n",
    "print(\"input sequence s2:\", s2)\n",
    "print(\"\\noptimal alignment:\")\n",
    "print(\"s1:\", aln_s1)\n",
    "print(\"s2:\", aln_s2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7169eb3c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VDRV-G\n",
      "VERMAG\n"
     ]
    }
   ],
   "source": [
    "# let's put everything in one function\n",
    "def Align(s1: str, s2: str):\n",
    "    import numpy as np\n",
    "    \"\"\" Alignment of two strings of amino acids according to the Needleman-Wunsch algorithm\n",
    "    Parameters\n",
    "    ----------\n",
    "    s1 : string\n",
    "        First amino acid string to align\n",
    "    s2 : string\n",
    "        Second amino acid string to align\n",
    "    Returns\n",
    "    -------\n",
    "    aln_s1 : string\n",
    "        First amino acid string, aligned\n",
    "    aln_s2 : string\n",
    "        Second amino acid string, aligned\n",
    "    \"\"\"\n",
    "    \n",
    "    # Are we meant to copy paste all code so far into here? \n",
    "\n",
    "    subst_matrix = SubstitutionMatrix() # redefine / create new one\n",
    "\n",
    "    # Write the code for running the Needleman-Wunsch algorithm\n",
    "    \n",
    "    gap_penalty = -8\n",
    "\n",
    "    n_rows = len(s1)+1 # s1 goes from top to bottom (in the algorith we access i-1, so we need to add 1)\n",
    "    n_cols = len(s2)+1\n",
    "\n",
    "    scoring_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "    # backtrack encoding:\n",
    "    # 1 => aligned\n",
    "    # 2 => deletion in s1 (insertion in s2 respectively)\n",
    "    # 3 => deletion in s2 (insertion in s1 respectively)\n",
    "    backtrack_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "    # the first row and the first column can already be prefilled\n",
    "    for i in range(1, n_rows):\n",
    "        scoring_matrix[(i,0)] = i*gap_penalty\n",
    "        backtrack_matrix[(i,0)] = 3\n",
    "    for i in range(1, n_cols):\n",
    "        scoring_matrix[(0,i)] = i*gap_penalty\n",
    "        backtrack_matrix[(0,i)] = 2\n",
    "\n",
    "    # From textblock #### Fill scoring and backtracking matrices\n",
    "    \n",
    "    for r_idx in range(1, n_rows):\n",
    "        for c_idx in range(1, n_cols):\n",
    "\n",
    "            score_match = scoring_matrix[(r_idx - 1, c_idx - 1)] + subst_matrix.get_score(s1[r_idx-1], s2[c_idx-1])\n",
    "\n",
    "            score_del_1 = scoring_matrix[(r_idx, c_idx - 1)] + gap_penalty\n",
    "            score_del_2 = scoring_matrix[(r_idx - 1, c_idx)] + gap_penalty\n",
    "            \n",
    "            # TODO Fill the values at position (r_idx, c_idx) in scoring_matrix and backtracking_matrix\n",
    "            # you can assign an element with\n",
    "            # scoring_matrix[(r_idx, c_idx)] = my_awesome_value\n",
    "            if score_match >= score_del_1 and score_match >= score_del_2: \n",
    "                scoring_matrix[(r_idx, c_idx)] = score_match #fill score into the matrix\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 1 #fill with corresponding number \n",
    "            elif score_del_1 >= score_match and score_del_1 >= score_del_2:\n",
    "                scoring_matrix[(r_idx, c_idx)] = score_del_1\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 2\n",
    "            else:\n",
    "                scoring_matrix[(r_idx, c_idx)] = score_del_2\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 3\n",
    "\n",
    "\n",
    "    # From textblock #### Perform backtracking to get final alignment\n",
    "    \n",
    "    r_idx = n_rows-1\n",
    "    c_idx = n_cols-1\n",
    "    path = list()\n",
    "\n",
    "    # if we hit zero, we're at the upper left corner and can stop\n",
    "    while backtrack_matrix[(r_idx, c_idx)] != 0:\n",
    "        path.append(backtrack_matrix[(r_idx, c_idx)])\n",
    "        if backtrack_matrix[(r_idx, c_idx)] == 1: \n",
    "            c_idx = c_idx - 1\n",
    "            r_idx = r_idx - 1\n",
    "            \n",
    "        elif backtrack_matrix[(r_idx, c_idx)] == 2:\n",
    "            c_idx = c_idx - 1\n",
    "\n",
    "        elif backtrack_matrix[(r_idx, c_idx)] == 3:\n",
    "            r_idx = r_idx - 1\n",
    "            \n",
    "    # backtracking comes from the back, so lets reverse...\n",
    "    path.reverse()\n",
    "\n",
    "    aln_s1 = []\n",
    "    aln_s2 = []\n",
    "    s1_idx = 0\n",
    "    s2_idx = 0\n",
    "\n",
    "    for p in path:\n",
    "        if p == 1:\n",
    "            aln_s1.append(s1[s1_idx])\n",
    "            aln_s2.append(s2[s2_idx])\n",
    "            s1_idx += 1\n",
    "            s2_idx += 1\n",
    "        elif p == 2:\n",
    "            aln_s1.append('-')\n",
    "            aln_s2.append(s2[s2_idx])\n",
    "            s2_idx += 1\n",
    "        else:\n",
    "            aln_s1.append(s1[s1_idx])\n",
    "            aln_s2.append('-')\n",
    "            s1_idx += 1\n",
    "            \n",
    "    aln_s1 = ''.join(aln_s1)\n",
    "    aln_s2 = ''.join(aln_s2)\n",
    "\n",
    "    # print(\"Input sequence 1: \", s1)\n",
    "    # print(\"Input sequence 2: \", s2)\n",
    "    # print(\"Aligned sequence 1: \", aln_s1)\n",
    "    # print(\"Aligned sequence 2: \", aln_s2)\n",
    "\n",
    "\n",
    "    return aln_s1, aln_s2\n",
    "\n",
    "\n",
    "alignments = Align(\"VDRVG\", \"VERMAG\")\n",
    "print(alignments[0])\n",
    "print(alignments[1])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "dfc5ff1d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MRIVLLGAPGAGKGTQAQFLMAKFGIPQISTGDMLRAAIKAGSELGNKAKAVMDAGQLVSDDLIIGLVKERVAQEDCKAGFLLDGFPRTIPQADAMKESGIVVDHVLEFDVPDEVIVERMAGRRVHSGSGRVYHLVYNPPKVEGKDDVSGDDLSIRPDDEEATVRKRLAIYHEQTKPLVDFYQAEAKSGSCSYLTIDGTQAVEKVNQLLSAQLA\n",
      "MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n",
      "MRIVLLGAPGAGKGTQAQFLMAKFGIPQISTGDMLRAAIKAGSELGNKAKAVMDAGQLVSDDLIIGLVKERVAQEDCKAGFLLDGFPRTIPQADAMKESGIVVDHVLEFDVPDEVIVERMAGRRVHSGSGRVYHLVYNPPKVEGKDDVSGDDLSIRPDDEEATVRKRLAIYHEQTKPLVDFYQAEAKSGSCSYLTIDGTQAVEKVNQLLSAQLA\n",
      "MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n",
      "[[0. 2. 2. ... 2. 2. 2.]\n",
      " [3. 1. 2. ... 2. 2. 2.]\n",
      " [3. 3. 1. ... 2. 2. 2.]\n",
      " ...\n",
      " [3. 3. 3. ... 1. 1. 1.]\n",
      " [3. 3. 3. ... 1. 1. 2.]\n",
      " [3. 3. 3. ... 3. 1. 1.]]\n"
     ]
    }
   ],
   "source": [
    "# Read two fasta files of bacterial kinase homologs (Q47XA8 & P69441) and get the alignment\\\n",
    "\n",
    "subst_matrix = SubstitutionMatrix()\n",
    "\n",
    "bac1_kinase = ReadFasta(\"Q47XA8.fasta\")\n",
    "bac2_kinase = ReadFasta(\"P69441.fasta\")\n",
    "\n",
    "print(bac1_kinase)\n",
    "print(bac2_kinase)\n",
    "\n",
    "#print(bac1_kinase)\n",
    "\n",
    "# TODO get alignment\n",
    "alignments = Align(bac1_kinase, bac2_kinase)\n",
    "bac1_aln = alignments[0]\n",
    "bac2_aln = alignments[1]\n",
    "\n",
    "print(bac1_aln)\n",
    "print(bac2_aln)\n",
    "\n",
    "print(backtrack_matrix)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3de2ec78",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequence identity is a simple proxy for evolutionary distance\n",
    "def SeqID(aln_s1: str, aln_s2: str):\n",
    "    \"\"\" Computes sequence identity - the percentage of conserved\n",
    "    amino acids\n",
    "    ----------\n",
    "    aln_s1 : string\n",
    "        First aligned amino acid string\n",
    "    aln_s2 : string\n",
    "        Second aligned amino acid string\n",
    "    Returns\n",
    "    -------\n",
    "    seqid : float\n",
    "        The sequence identity\n",
    "    \"\"\"\n",
    "    # must have same length\n",
    "    assert(len(aln_s1) == len(aln_s2))\n",
    "    n_match = 0\n",
    "    n_aligned = 0\n",
    "    for a, b in zip(aln_s1, aln_s2):\n",
    "        if a!='-' and b!='-':\n",
    "            n_aligned += 1\n",
    "            if a == b:\n",
    "                n_match += 1\n",
    "    if n_aligned != 0:\n",
    "        # explicitely cast to float here\n",
    "        # This is not required in Python3 but Python2 would perform\n",
    "        # an integer division otherwise\n",
    "        return float(n_match)/n_aligned*100\n",
    "    return 0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "bc470f41",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SeqID: 0.00%\n"
     ]
    }
   ],
   "source": [
    "# Python provides several ways of string formatting\n",
    "# one example are so called f-strings\n",
    "print(f\"SeqID: {SeqID(bac1_aln, bac2_aln):.2f}%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90a57d64",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c2fa713",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebd17c94",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bff34d83",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69fa72c5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8777cf0c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f9c5c0d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b852b1e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2eb8570",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a616c56a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb9d7a11",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f5218593",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b92eeba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1435846",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1446966a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bbc2c1e6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6de44738",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05499200",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79a2a415",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b28e586c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c693d24",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5e48935",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4439f29",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89555a1a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d6ef537",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6e3f416",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46c87caf",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
