{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "eaf4e3e4",
   "metadata": {},
   "source": [
    "### Programming in life sciences"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "89abeed2",
   "metadata": {},
   "source": [
    "### Predicting open/close conformation of a kinase model\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edef8232",
   "metadata": {},
   "source": [
    "<b>Lecture 1: Introduction to alignments - exercise: align two homologous kinases (given as fasta files)</b>\n",
    "\n",
    "Lecture 2: Introduction to superpositions - exercise: get RMSD of two structures\n",
    "\n",
    "Lecture 3: Introduction to AlphaFold - exercise: get models from AlphaFold and classify them as open/close, do the same for models from Swiss-Model and AlphaFold-dropout\n",
    "\n",
    "Lecture 4: Ligand docking (DiffDock) - exercise: get docking from DiffDock colab, find interacting atoms and visualize docking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe148b03",
   "metadata": {
    "tags": [
     "lesson1"
    ]
   },
   "source": [
    "### 1. Lecture: Needleman-Wunsch algorithm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "933ffbba",
   "metadata": {},
   "source": [
    "In this exercise, we want to align two sequences obtained from fasta files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6d97f4c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# numpy is a library providing a lot of numerical mathematical functionalities\n",
    "# here we only use it to store stuff in matrices\n",
    "import numpy as np\n",
    "\n",
    "# we create a new class that will be used to define substitution matrices\n",
    "class SubstitutionMatrix:\n",
    "  \"\"\"Read, store and analyse substitution matrices, default = BLOSUM62\"\"\"\n",
    "\n",
    "  # this is the constructor function! When you create a new instance of type\n",
    "  # Substitution Matrix, this function gets called. By default it reads in\n",
    "  # the provided file: blosum62.txt... you can create your own matrix if\n",
    "  # you like ;)\n",
    "  def __init__(self, data_file: str = \"blosum62.txt\"):\n",
    "    \"\"\" Constructor function for the SubstitutionMatrix class\n",
    "    Parameters\n",
    "    ----------\n",
    "    data_file : string\n",
    "        Filename of the substitution matrix to read in, must contain 21 lines,\n",
    "        with the first line containing the 20 amino acid one letter codes,\n",
    "        by default reads the blosum62.txt matrix in the current folder\n",
    "    \"\"\"\n",
    "    \n",
    "    # read all the lines from the file at once  ('r' = read)\n",
    "    with open(data_file,'r') as fh:\n",
    "      data = fh.readlines()\n",
    "\n",
    "    # the size of subsitution matrices must be 20x20\n",
    "    # the first line of the file gives the amino acid one letter codes\n",
    "    # which gives a total of 21 lines in the file\n",
    "    if len(data) != 21:\n",
    "      err = \"Expected exactly 21 lines in matrix file!\"\n",
    "      err += \" One line defining the amino acid one letter codes and \"\n",
    "      err += \"20 lines defining the substitution matrix\"\n",
    "      raise RuntimeError(err)    \n",
    "\n",
    "    # get the amino acid one letter codes\n",
    "    self.one_letter_codes = data[0].strip()\n",
    "\n",
    "    # check that 20 amino acids are defined\n",
    "    if len(self.one_letter_codes) != 20:\n",
    "      err = \"First line must contain exactly 20 letters representing \"\n",
    "      err += \"the amino acids!\"\n",
    "      raise RuntimeError(err)\n",
    "\n",
    "    # initialize the substitution matrix with zeros\n",
    "    self.matrix = np.zeros((20,20))\n",
    "\n",
    "    # fill in the matrix by reading the file line by line\n",
    "    # note that range returns a vector starting at 0 to the integer in argument - 1 \n",
    "    for i in range(20):\n",
    "      data_line = data[1+i].split()\n",
    "      if len(data_line) != 20:\n",
    "        raise RuntimeError(\"Each data line must contain exactly 20 elements!\")\n",
    "      for j in range(20):\n",
    "        self.matrix[(i,j)] = int(data_line[j])\n",
    "\n",
    "\n",
    "  # The function where you can extract the score for a specific amino acid substitution...\n",
    "  # e.g. mat.get_score('W', 'A')\n",
    "  def get_score(self, aa_one: str, aa_two: str):\n",
    "    \"\"\" Extract the substitution score for two amino acids from the substitution matrix\n",
    "    Parameters\n",
    "    ----------\n",
    "    aa_one : string\n",
    "        One letter amino acid code\n",
    "    aa_two : string\n",
    "        One letter amino acid code\n",
    "    Returns\n",
    "    -------\n",
    "    self.matrix[(idx_one,idx_two)] : float\n",
    "        Substitution matrix entry for the substitution of amino acid one and amino acid two,\n",
    "        note that the matrices are symmetric, so get_score(aa_one, aa_two) is equal to get_score(aa_two, aa_one)\n",
    "    \"\"\"\n",
    "\n",
    "    # get the indices of the amino acids in the matrix\n",
    "    idx_one = self.one_letter_codes.find(aa_one)\n",
    "    idx_two = self.one_letter_codes.find(aa_two)\n",
    "\n",
    "    if idx_one == -1:\n",
    "      raise RuntimeError(\"aa_one must be one of: \" + self.one_letter_codes)\n",
    "    \n",
    "    if idx_two == -1:\n",
    "      raise RuntimeError(\"aa_two must be one of: \" + self.one_letter_codes)\n",
    "\n",
    "    return self.matrix[(idx_one,idx_two)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f8b4798",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Decorating functions with docstrings is not only best practice,\n",
    "# it also helps automated documentation generation with tools\n",
    "# like Sphinx (https://www.sphinx-doc.org) or simply get more\n",
    "# information using the Python help function:\n",
    "help(SubstitutionMatrix.get_score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7bedee41",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Construct a default substitution matrix\n",
    "subst_matrix = SubstitutionMatrix()\n",
    "\n",
    "# get score of substituting ASP with ASN and GLY with PRO\n",
    "print(\"Score D->N:\", subst_matrix.get_score('N', 'D'))\n",
    "print(\"Score G->P:\", subst_matrix.get_score('G', 'P'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a94cabbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def ReadFasta(filename: str):\n",
    "    \"\"\" A simplistic reader that only checks if the first line starts with '>' and assumes that the whole rest is the input sequence\n",
    "    Parameters\n",
    "    ----------\n",
    "    filename : string\n",
    "        Filename of the FASTA file to read\n",
    "    Returns\n",
    "    -------\n",
    "    s : string\n",
    "        Sequence read from the input FASTA file\n",
    "    \"\"\"\n",
    "    \n",
    "    # store all the lines in a single list  ('r' = read)\n",
    "    with open(filename, 'r') as fh:\n",
    "        lines = fh.readlines()\n",
    "    \n",
    "    # handle errors\n",
    "    if len(lines) <= 1:\n",
    "        raise RuntimeError(\"Expect multiple lines in FASTA file\")\n",
    "    if not lines[0].startswith('>'):\n",
    "        raise RuntimeError(\"Assume first line in FASTA file to start with '>'\")\n",
    "\n",
    "    # read the sequence lines, while removing all extra characters\n",
    "    s = \"\"\n",
    "    for line in lines[1:]:\n",
    "        s += line.strip()\n",
    "        \n",
    "    return s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3786a28b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in the two sequences to align \n",
    "# the two files are located in the same directory as the notebook\n",
    "s1 = ReadFasta(\"s1.fasta\")\n",
    "s2 = ReadFasta(\"s2.fasta\")\n",
    "print(s1)\n",
    "print(s2)\n",
    "\n",
    "## old s1: GAWGHEE, old s2: PAWHEA"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44b39e0a",
   "metadata": {},
   "source": [
    "#### Write the code for running the Needleman-Wunsch algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09b8cc4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# define a penalty for introducing a gap in the sequence alignment\n",
    "gap_penalty = -8"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d455926",
   "metadata": {},
   "source": [
    "Setup scoring and backtracking matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce456d98",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_rows = len(s1)+1 # s1 goes from top to bottom\n",
    "n_cols = len(s2)+1 # s2 goes from left to right\n",
    "\n",
    "scoring_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "# backtrack encoding:\n",
    "# 1 => aligned\n",
    "# 2 => deletion in s1 (insertion in s2 respectively)\n",
    "# 3 => deletion in s2 (insertion in s1 respectively)\n",
    "backtrack_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "# the first row and the first column can already be prefilled\n",
    "for i in range(1, n_rows):\n",
    "    scoring_matrix[(i,0)] = i*gap_penalty\n",
    "    backtrack_matrix[(i,0)] = 3\n",
    "for i in range(1, n_cols):\n",
    "    scoring_matrix[(0,i)] = i*gap_penalty\n",
    "    backtrack_matrix[(0,i)] = 2\n",
    "\n",
    "print(\"row sequence: \", s1)\n",
    "print(\"col sequence: \", s2)\n",
    "print(\"empty scoring matrix: \")\n",
    "print(scoring_matrix)\n",
    "print(\"empty backtracking matrix: \")\n",
    "print(backtrack_matrix)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "710f66ae",
   "metadata": {},
   "source": [
    "#### Fill scoring and backtracking matrices\n",
    "\n",
    "Every position in the scoring matrix represents the best local solution given the\n",
    "steps I can take\n",
    "- Align two residues => Move along the diagonal (consume a letter from s1 and s2),\n",
    "                        the value of the scoring matrix to the upper left plus the corresponding score from the scoring matrix\n",
    "- Apply deletion/gap in s1 => Move right in the scoring matrix (consume a letter from s2), \n",
    "                              the value from the scoring matrix to the left plus the penalty for a gap\n",
    "- Apply deletion/gap in s2 => Move down in the scoring matrix (consume a letter from s1),\n",
    "                              the value from the scoring matrix above plus the penalty for a gap\n",
    "\n",
    "The goal is to identify the best scoring solution from those three,\n",
    "set the according score in the scoring matrix and keep track of this local\n",
    "step in the backtracking matrix. \n",
    "\n",
    "The backtracking matrix stores the path I took, e.g. 1 for the first option\n",
    "(see backtracking encoding in previous code cell)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fda9db23",
   "metadata": {},
   "outputs": [],
   "source": [
    "for r_idx in range(1, n_rows):\n",
    "    for c_idx in range(1, n_cols):\n",
    "        score_match = scoring_matrix[(r_idx-1, c_idx-1)] + subst_matrix.get_score(s1[r_idx-1], s2[c_idx-1])\n",
    "        score_del_1 = scoring_matrix[(r_idx, c_idx-1)] + gap_penalty\n",
    "        score_del_2 = scoring_matrix[(r_idx-1, c_idx)] + gap_penalty        \n",
    "        # TODO Fill the values at position (r_idx, c_idx) in scoring_matrix and backtracking_matrix\n",
    "        # you can assign an element with\n",
    "        # scoring_matrix[(r_idx, c_idx)] = my_awesome_value\n",
    "        if score_match >= score_del_1 and score_match >= score_del_2:\n",
    "            scoring_matrix[(r_idx, c_idx)] = score_match\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 1\n",
    "        elif score_del_1 >= score_match and score_del_1 >= score_del_2:\n",
    "            scoring_matrix[(r_idx, c_idx)] = score_del_1\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 2\n",
    "        else:\n",
    "            scoring_matrix[(r_idx, c_idx)] = score_del_2\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 3\n",
    "        \n",
    "print(\"filled scoring matrix:\")\n",
    "print(scoring_matrix)\n",
    "print(\"filled backtracking matrix:\")\n",
    "print(backtrack_matrix)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f7572fe",
   "metadata": {},
   "source": [
    "#### Perform backtracking to get final alignment\n",
    "\n",
    "Remember the encoding:\n",
    "\n",
    "1 => aligned, move along the diagonal\n",
    "\n",
    "2 => deletion in s1 (insertion in s2 respectively), move right\n",
    "\n",
    "3 => deletion in s2 (insertion in s1 respectively), move down"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b8e7212",
   "metadata": {},
   "outputs": [],
   "source": [
    "# In principle we start at the bottom right of the backtracking matrix and\n",
    "# work our way through the matrix until we hit the upper left\n",
    "\n",
    "r_idx = n_rows-1\n",
    "c_idx = n_cols-1\n",
    "path = list()\n",
    "\n",
    "# if we hit zero, we're at the upper left corner and can stop\n",
    "while backtrack_matrix[(r_idx, c_idx)] != 0:\n",
    "    path.append(backtrack_matrix[(r_idx, c_idx)])\n",
    "    if backtrack_matrix[(r_idx, c_idx)] == 1:\n",
    "        r_idx -= 1\n",
    "        c_idx -= 1\n",
    "    elif backtrack_matrix[(r_idx, c_idx)] == 2:\n",
    "        c_idx -= 1\n",
    "    else:\n",
    "        r_idx -= 1\n",
    "        \n",
    "# backtracking comes from the back, so lets reverse...\n",
    "path.reverse()\n",
    "\n",
    "aln_s1 = []\n",
    "aln_s2 = []\n",
    "s1_idx = 0\n",
    "s2_idx = 0\n",
    "\n",
    "for p in path:\n",
    "    if p == 1:\n",
    "        aln_s1.append(s1[s1_idx])\n",
    "        aln_s2.append(s2[s2_idx])\n",
    "        s1_idx += 1\n",
    "        s2_idx += 1\n",
    "    elif p == 2:\n",
    "        aln_s1.append('-')\n",
    "        aln_s2.append(s2[s2_idx])\n",
    "        s2_idx += 1\n",
    "    else:\n",
    "        aln_s1.append(s1[s1_idx])\n",
    "        aln_s2.append('-')\n",
    "        s1_idx += 1\n",
    "        \n",
    "aln_s1 = ''.join(aln_s1)\n",
    "aln_s2 = ''.join(aln_s2)\n",
    "        \n",
    "print(\"input sequence s1:\", s1)\n",
    "print(\"input sequence s2:\", s2)\n",
    "print(\"\\noptimal alignment:\")\n",
    "print(\"s1:\", aln_s1)\n",
    "print(\"s2:\", aln_s2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7169eb3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#let's put everything in one function\n",
    "def Align(s1: str, s2: str):\n",
    "    \"\"\" Alignment of two strings of amino acids according to the Needleman-Wunsch algorithm\n",
    "    Parameters\n",
    "    ----------\n",
    "    s1 : string\n",
    "        First amino acid string to align\n",
    "    s2 : string\n",
    "        Second amino acid string to align\n",
    "    Returns\n",
    "    -------\n",
    "    aln_s1 : string\n",
    "        First amino acid string, aligned\n",
    "    aln_s2 : string\n",
    "        Second amino acid string, aligned\n",
    "    \"\"\"\n",
    "    \n",
    "    # initialize the substitution matrix (default = BLOSUM62)\n",
    "    subst_matrix = SubstitutionMatrix()\n",
    "    \n",
    "    # define the gap penalty value\n",
    "    gap_penalty = -8\n",
    "\n",
    "    # initialize the scoring and backtrack matrices\n",
    "    n_rows = len(s1)+1 # s1 goes from top to bottom\n",
    "    n_cols = len(s2)+1 # s2 goes from left to right\n",
    "    scoring_matrix = np.zeros((n_rows, n_cols))\n",
    "    backtrack_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "    # the first row and the first column can already be prefilled\n",
    "    for i in range(1, n_rows):\n",
    "        scoring_matrix[(i,0)] = i*gap_penalty\n",
    "        backtrack_matrix[(i,0)] = 3\n",
    "    for i in range(1, n_cols):\n",
    "        scoring_matrix[(0,i)] = i*gap_penalty\n",
    "        backtrack_matrix[(0,i)] = 2\n",
    "\n",
    "    # fill the scoring and backtrack matrices\n",
    "    for r_idx in range(1, n_rows):\n",
    "        for c_idx in range(1, n_cols):\n",
    "            score_match = scoring_matrix[(r_idx-1, c_idx-1)] + subst_matrix.get_score(s1[r_idx-1], s2[c_idx-1])\n",
    "            score_del_1 = scoring_matrix[(r_idx, c_idx-1)] + gap_penalty\n",
    "            score_del_2 = scoring_matrix[(r_idx-1, c_idx)] + gap_penalty        \n",
    "\n",
    "            if score_match >= score_del_1 and score_match >= score_del_2:\n",
    "                scoring_matrix[(r_idx, c_idx)] = score_match\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 1\n",
    "            elif score_del_1 >= score_match and score_del_1 >= score_del_2:\n",
    "                scoring_matrix[(r_idx, c_idx)] = score_del_1\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 2\n",
    "            else:\n",
    "                scoring_matrix[(r_idx, c_idx)] = score_del_2\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 3\n",
    "\n",
    "    # track back from the lower right corner to the top left corner of the backtrack matrix to get the path\n",
    "    r_idx = n_rows-1\n",
    "    c_idx = n_cols-1\n",
    "    path = list()\n",
    "\n",
    "    # if we hit zero, we're at the upper left corner and can stop\n",
    "    while backtrack_matrix[(r_idx, c_idx)] != 0:\n",
    "        path.append(backtrack_matrix[(r_idx, c_idx)])\n",
    "        if backtrack_matrix[(r_idx, c_idx)] == 1:\n",
    "            r_idx -= 1\n",
    "            c_idx -= 1\n",
    "        elif backtrack_matrix[(r_idx, c_idx)] == 2:\n",
    "            c_idx -= 1\n",
    "        else:\n",
    "            r_idx -= 1\n",
    "\n",
    "    # backtracking comes from the back, so lets reverse...\n",
    "    path.reverse()\n",
    "\n",
    "    # get the aligned sequences by going through the path\n",
    "    aln_s1 = []\n",
    "    aln_s2 = []\n",
    "    s1_idx = 0\n",
    "    s2_idx = 0\n",
    "\n",
    "    for p in path:\n",
    "        if p == 1:\n",
    "            aln_s1.append(s1[s1_idx])\n",
    "            aln_s2.append(s2[s2_idx])\n",
    "            s1_idx += 1\n",
    "            s2_idx += 1\n",
    "        elif p == 2:\n",
    "            aln_s1.append('-')\n",
    "            aln_s2.append(s2[s2_idx])\n",
    "            s2_idx += 1\n",
    "        else:\n",
    "            aln_s1.append(s1[s1_idx])\n",
    "            aln_s2.append('-')\n",
    "            s1_idx += 1\n",
    "\n",
    "    aln_s1 = ''.join(aln_s1)\n",
    "    aln_s2 = ''.join(aln_s2)\n",
    "    \n",
    "    return aln_s1, aln_s2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "dfc5ff1d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MRIVLLGAPGAGKGTQAQFLMAKFGIPQISTGDMLRAAIKAGSELGNKAKAVMDAGQLVSDDLIIGLVKERVAQEDCKAGFLLDGFPRTIPQADAMKESGIVVDHVLEFDVPDEVIVERMAGRRVHSGSGRVYHLVYNPPKVEGKDDVSGDDLSIRPDDEEATVRKRLAIYHEQTKPLVDFYQAEAKSGSCSYLTIDGTQAVEKVNQLLSAQLA\n",
      "MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n"
     ]
    }
   ],
   "source": [
    "# Read two fasta files of bacterial kinase homologs and get the alignment\n",
    "bac1_kinase = ReadFasta('Q47XA8.fasta')\n",
    "print(bac1_kinase)\n",
    "\n",
    "bac2_kinase = ReadFasta('P69441.fasta')\n",
    "print(bac2_kinase)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a3c90bf2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MRIVLLGAPGAGKGTQAQFLMAKFGIPQISTGDMLRAAIKAGSELGNKAKAVMDAGQLVSDDLIIGLVKERVAQEDCKAGFLLDGFPRTIPQADAMKESGIVVDHVLEFDVPDEVIVERMAGRRVHSGSGRVYHLVYNPPKVEGKDDVSGDDLSIRPDDEEATVRKRLAIYHEQTKPLVDFYQAEAKSGSCSYLTIDGTQAVEKVNQLLSAQLA\n",
      "MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n"
     ]
    }
   ],
   "source": [
    "bac1_aln, bac2_aln = Align(bac1_kinase, bac2_kinase)\n",
    "print(bac1_aln)\n",
    "print(bac2_aln)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3de2ec78",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequence identity is a simple proxy for evolutionary distance\n",
    "def SeqID(aln_s1: str, aln_s2: str):\n",
    "    \"\"\" Computes sequence identity - the percentage of conserved\n",
    "    amino acids\n",
    "    ----------\n",
    "    aln_s1 : string\n",
    "        First aligned amino acid string\n",
    "    aln_s2 : string\n",
    "        Second aligned amino acid string\n",
    "    Returns\n",
    "    -------\n",
    "    seqid : float\n",
    "        The sequence identity\n",
    "    \"\"\"\n",
    "    # must have same length\n",
    "    assert(len(aln_s1) == len(aln_s2))\n",
    "    n_match = 0\n",
    "    n_aligned = 0\n",
    "    for a,b in zip(aln_s1, aln_s2):\n",
    "        if a!='-' and b!='-':\n",
    "            n_aligned += 1\n",
    "            if a == b:\n",
    "                n_match += 1\n",
    "    if n_aligned != 0:\n",
    "        # explicitely cast to float here\n",
    "        # This is not required in Python3 but Python2 would perform\n",
    "        # an integer division otherwise\n",
    "        return float(n_match)/n_aligned*100\n",
    "    return 0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc470f41",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Python provides several ways of string formatting\n",
    "# one example are so called f-strings\n",
    "print(f\"SeqID: {SeqID(bac1_aln, bac2_aln):.2f}%\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7cd9005f",
   "metadata": {},
   "source": [
    "Lecture 1: Introduction to alignments - exercise: align two homologous kinases (given as fasta files)\n",
    "\n",
    "<b>Lecture 2: Introduction to superpositions - exercise: get RMSD of two structures</b>\n",
    "\n",
    "Lecture 3: Introduction to AlphaFold - exercise: get models from AlphaFold and classify them as open/close, do the same for models from Swiss-Model and AlphaFold-dropout\n",
    "\n",
    "Lecture 4: Ligand docking (DiffDock) - exercise: get docking from DiffDock colab, find interacting atoms and visualize docking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bebea35",
   "metadata": {},
   "source": [
    "### 2. Lecture: superpositions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c46b9172",
   "metadata": {},
   "outputs": [],
   "source": [
    "# we create a new class that will be used to define atoms in 3D space\n",
    "class Vec3:\n",
    "    \"\"\"Store and print 3D coordinates\"\"\"\n",
    "    \n",
    "    def __init__(self, x: float, y: float, z: float):\n",
    "        \"\"\" Constructor function for the Vec3 class\n",
    "        Parameters\n",
    "        ----------\n",
    "        x, y, z : float\n",
    "            Coordinates of the atom in 3D space\n",
    "        \"\"\"\n",
    "\n",
    "        # define member variables here\n",
    "        self.x = x\n",
    "        self.y = y\n",
    "        self.z = z\n",
    "        \n",
    "    def __str__(self):\n",
    "        \"\"\" Print function for the Vec3 class \"\"\"\n",
    "        return '[{:.2f}, {:.2f}, {:.2f}]'.format(self.x, self.y, self.z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9d6d2117",
   "metadata": {},
   "outputs": [],
   "source": [
    "# we create a new class that will be used to describe a single residue of a protein chain\n",
    "class Residue:\n",
    "    \"\"\"Store and print a protein residue\"\"\"\n",
    "    def __init__(self, rname: str, rnum: int):\n",
    "        \"\"\" Constructor function for the Residue class\n",
    "        Parameters\n",
    "        ----------\n",
    "        rname : string\n",
    "            Name of the residue (amino acid 3 letter code)\n",
    "        rnum : int\n",
    "            Residue sequence number, position in the chain\n",
    "        \"\"\"\n",
    "\n",
    "        # define member variables here\n",
    "        self.rname = rname\n",
    "        self.rnum = rnum\n",
    "        self.n_pos = None\n",
    "        self.ca_pos = None\n",
    "        self.c_pos = None\n",
    "        \n",
    "    def __str__(self):\n",
    "        \"\"\" Print function for the Residue class \"\"\"    \n",
    "        pos_strings = list()\n",
    "        pos_strings.append('N: ' + str(self.n_pos))\n",
    "        pos_strings.append('CA: ' + str(self.ca_pos))\n",
    "        pos_strings.append('C: ' + str(self.c_pos))\n",
    "        return '\\t'.join([self.rname, self.rnum, ' '.join(pos_strings)])\n",
    "    \n",
    "    def get_coords(self):\n",
    "        \"\"\" Get all the coordinates of the atoms in the residue as a list\n",
    "        Parameters\n",
    "        ----------\n",
    "\n",
    "        Returns\n",
    "        -------\n",
    "        list\n",
    "            x, y and z Ã… coordinates of the N (amino group atom), CA (carbon atom) and C (carboxyl group atom) positions in a single list\n",
    "        \"\"\"\n",
    "        return [self.n_pos.x, self.n_pos.y, self.n_pos.z,self.ca_pos.x, self.ca_pos.y, self.ca_pos.z,self.c_pos.x, self.c_pos.y, self.c_pos.z]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "dca21a99",
   "metadata": {},
   "outputs": [],
   "source": [
    "# define a Chain class, that contains following information as member variables:\n",
    "# - name (e.g. \"A\")\n",
    "# - a list of residues\n",
    "#\n",
    "#AWESOME IMPLEMENTATION COMES HERE:\n",
    "\n",
    "class Chain:\n",
    "    \"\"\"Store and print a protein chain = a list of residues\"\"\"\n",
    "\n",
    "    def __init__(self, cname: str):\n",
    "        \"\"\" Constructor function for the Chain class\n",
    "        Parameters\n",
    "        ----------\n",
    "        cname : string\n",
    "            Name of the chain, e.g. 'A'\n",
    "        \"\"\"\n",
    "\n",
    "        self.cname = cname\n",
    "        self.residues = list()\n",
    "        \n",
    "    def __str__(self):\n",
    "        \"\"\" Print function for the Chain class \"\"\"    \n",
    "        return_list = list()\n",
    "        return_list.append('chain name: ' + self.cname)\n",
    "        for r in self.residues:\n",
    "            return_list.append(' ' + str(r))\n",
    "        return '\\n'.join(return_list)\n",
    "\n",
    "# define an Assembly class, that contains following information as member variables:\n",
    "# - a list of chains\n",
    "#\n",
    "# AWESOME IMPLEMENTATION COMES HERE:\n",
    "\n",
    "class Assembly:\n",
    "    \"\"\"Store and print a protein assembly = a list of chains\"\"\"\n",
    "\n",
    "    def __init__(self):\n",
    "        \"\"\" Constructor function for the Assembly class\n",
    "        Parameters\n",
    "        ----------\n",
    "        \"\"\"\n",
    "\n",
    "        self.chains = list()\n",
    "        \n",
    "    def __str__(self):\n",
    "        \"\"\" Print function for the Assembly class \"\"\"    \n",
    "        return_list = list()\n",
    "        for c in self.chains:\n",
    "            return_list.append(str(c))\n",
    "        return '\\n\\n'.join(return_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea073e88",
   "metadata": {},
   "source": [
    "The PDB is a very specific file format for protein structures:\n",
    "https://www.cgl.ucsf.edu/chimera/docs/UsersGuide/tutorials/pdbintro.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0c3eeda9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def LoadPDB(filename: str):\n",
    "    \"\"\" Read and save a protein structure from a PDB file\n",
    "    Parameters\n",
    "    ----------\n",
    "    filename : string\n",
    "        Name of the PDB file to be read\n",
    "    Returns\n",
    "    -------\n",
    "    assembly : Assembly\n",
    "        The full Assembly of the protein read from the PDB file, containing all chains with all residues\n",
    "    \"\"\"\n",
    "    \n",
    "    #AWESOME IMPLEMENTATION COMES HERE:\n",
    "    \n",
    "    # directly load full content in one list ('r' = read)\n",
    "    with open(filename, 'r') as file:\n",
    "        content = file.readlines()\n",
    "        \n",
    "    # define variables for everything we want to read, start with dummy/temporary names for the residue and chain\n",
    "    assembly = Assembly()\n",
    "    current_res = Residue(rname = 'dummy_res', rnum = None)\n",
    "    current_chain = Chain(cname = 'dummy_chain')\n",
    "    \n",
    "    # go through each describing an atom in the PDB file\n",
    "    for line in content:\n",
    "        \n",
    "        if line.startswith('ATOM'):\n",
    "            \n",
    "            # get all the non-empty values\n",
    "            line_values = line.split(' ')\n",
    "            line_values = [c for c in line_values if c != '']\n",
    "            aname = line_values[2].strip() #13:17\n",
    "\n",
    "            # we're only interested in those three backbone atoms\n",
    "            if aname in ['N', 'CA', 'C']:\n",
    "\n",
    "                # parse current line\n",
    "                rname = line_values[3] #[17:21]\n",
    "                rnum = line_values[5].strip() #23:27\n",
    "                cname = line_values[4] #21\n",
    "                x_coord = float(line_values[6].strip()) #30:38\n",
    "                y_coord = float(line_values[7].strip()) #38:46\n",
    "                z_coord = float(line_values[8].strip()) #46:54\n",
    "                \n",
    "                # check if there is a new residue\n",
    "                if rnum != current_res.rnum or cname != current_chain.cname:\n",
    "                    # we add the previous one to the current chain if all positions are valid\n",
    "                    if current_res.n_pos and current_res.ca_pos and current_res.c_pos:\n",
    "                        current_chain.residues.append(current_res)\n",
    "                    # and generate the new residue\n",
    "                    current_res = Residue(rname, rnum)\n",
    "                \n",
    "                # check if there is a new chain\n",
    "                if cname != current_chain.cname:\n",
    "                    # we add the previous one if it is not empty\n",
    "                    if len(current_chain.residues) > 0:\n",
    "                        assembly.chains.append(current_chain)\n",
    "                    # and generate the new chain\n",
    "                    current_chain = Chain(cname)\n",
    "                \n",
    "                # for the current residue, save the coordinates of the three backbone atoms as a Vec3 instance\n",
    "                if aname == 'N':\n",
    "                    current_res.n_pos = Vec3(x_coord, y_coord, z_coord)\n",
    "                if aname == 'CA':\n",
    "                    current_res.ca_pos = Vec3(x_coord, y_coord, z_coord)\n",
    "                if aname == 'C':\n",
    "                    current_res.c_pos = Vec3(x_coord, y_coord, z_coord)\n",
    "                    \n",
    "    # at the end, there still might be a complete residue and a complete chain that haven't\n",
    "    # been added to the assembly\n",
    "    if current_res.n_pos and current_res.ca_pos and current_res.c_pos:\n",
    "        current_chain.residues.append(current_res)\n",
    "    if len(current_chain.residues) > 0:\n",
    "        assembly.chains.append(current_chain)\n",
    "                \n",
    "    return assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6cd8a95b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_aa(assembly: Assembly):\n",
    "    ''' Get the three-letter amino acid sequence from an Assembly\n",
    "    Warning: we assume here that our protein is a monomer (only one chain) and all the sequence is in the first chain\n",
    "    Parameters\n",
    "    ----------\n",
    "    assembly : Assembly\n",
    "        Assembly of a monomeric protein with the entire sequence in the first chain\n",
    "    Returns\n",
    "    -------\n",
    "    aa_sequence : string\n",
    "        The entire three-letter amino acid sequence from the first chain\n",
    "    '''\n",
    "\n",
    "    # initiate the value we want to compute\n",
    "    aa_sequence = ''\n",
    "\n",
    "    # extra only the sequence from the first chain\n",
    "    first_chain = assembly.chains[0]\n",
    "\n",
    "    for res in first_chain.residues:\n",
    "        aa_sequence += res.rname + ' '\n",
    "\n",
    "    return aa_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "160096ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "def convert_sequence_to_one_letter(three_letter_sequence: str, delimiter: str = ' '):\n",
    "    ''' Convert a three-letter code amino acid sequence, separated by the delimiter, to a one-letter code amino acid sequence\n",
    "    Parameters\n",
    "    ----------\n",
    "    three_letter_sequence : string\n",
    "        Three-letter code amino acid sequence to be converted, separated by the delimiter\n",
    "    delimiter : string\n",
    "        Delimiter for the three-letter code amino acid sequence\n",
    "    Returns\n",
    "    -------\n",
    "    one_letter_sequence : string\n",
    "        One-letter code amino acid sequence\n",
    "    '''\n",
    "\n",
    "    # define a dictionary mapping the three-letter codes to the one-letter codes\n",
    "    mapping = {\n",
    "    'ALA': 'A',\n",
    "    'ARG': 'R',\n",
    "    'ASN': 'N',\n",
    "    'ASP': 'D',\n",
    "    'CYS': 'C',\n",
    "    'GLN': 'Q',\n",
    "    'GLU': 'E',\n",
    "    'GLY': 'G',\n",
    "    'HIS': 'H',\n",
    "    'ILE': 'I',\n",
    "    'LEU': 'L',\n",
    "    'LYS': 'K',\n",
    "    'MET': 'M',\n",
    "    'PHE': 'F',\n",
    "    'PRO': 'P',\n",
    "    'SER': 'S',\n",
    "    'THR': 'T',\n",
    "    'TRP': 'W',\n",
    "    'TYR': 'Y',\n",
    "    'VAL': 'V'\n",
    "    }\n",
    "\n",
    "    # intialize the value we want to compute\n",
    "    one_letter_sequence = ''\n",
    "\n",
    "    # remove the empty spaces at the beginning and end of the full sequence\n",
    "    three_letter_sequence = three_letter_sequence.strip()\n",
    "\n",
    "    # process and map each three letter code in the sequence, otherwise raise an error\n",
    "    for three_letter_code in three_letter_sequence.split(delimiter):\n",
    "        three_letter_code = three_letter_code.upper().strip()\n",
    "\n",
    "        if three_letter_code in mapping:\n",
    "            one_letter_sequence += mapping[three_letter_code]\n",
    "        else:\n",
    "            raise KeyError(f\"Three-letter amino acid code '{three_letter_code}' not found.\")\n",
    "\n",
    "    return one_letter_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "504f470f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def GetAlignedCaCoords(assembly1: Assembly, assembly2: Assembly, alignment1: str, alignment2: str):\n",
    "    \"\"\" Given two assemblies and their alignment strings, return two arrays of the CA coordinates of aligned (not gaps) residues\n",
    "    Warning: we assume here that our proteins are monomers (only one chain) and all the sequences are in the first chains\n",
    "    Parameters\n",
    "    ----------\n",
    "    assembly1, assembly2 : Assembly\n",
    "        Protein assemblies from which to get the CA coordinates of the aligned residues\n",
    "    alignment1, alignment2 : string\n",
    "        Aligned sequences (by Needleman-Wunsch algorithm for example) of the two assembly sequences\n",
    "    Returns\n",
    "    -------\n",
    "    aligned_ca_coords1, aligned_ca_coords2 : numpy.array\n",
    "        Numpy arrays of the CA coordinates of the aligned (not gaps) residues of the two input assemblies\n",
    "    \"\"\"\n",
    "\n",
    "    # get the lists of all residues from the first chains in the assemblies\n",
    "    residues1 = [residue for residue in assembly1.chains[0].residues]\n",
    "    residues2 = [residue for residue in assembly2.chains[0].residues]\n",
    "\n",
    "    # Initialize the variables we want to define\n",
    "    aligned_ca_coords1 = []\n",
    "    aligned_ca_coords2 = []\n",
    "    res_index1 = 0\n",
    "    res_index2 = 0\n",
    "\n",
    "    # Iterate through the alignments\n",
    "    for a1, a2 in zip(alignment1, alignment2):\n",
    "        \n",
    "        #check for the gaps and length\n",
    "        if a1 != '-' and res_index1 < len(residues1):\n",
    "            residue1 = residues1[res_index1]\n",
    "            res_index1 += 1\n",
    "        else:\n",
    "            residue1 = None\n",
    "            \n",
    "        if a2 != '-' and res_index2 < len(residues2):\n",
    "            residue2 = residues2[res_index2]\n",
    "            res_index2 += 1\n",
    "        else:\n",
    "            residue2 = None\n",
    "        \n",
    "        # If both residues are not None, record CA coordinates\n",
    "        if residue1 is not None and residue2 is not None:\n",
    "            aligned_ca_coords1.append([residue1.ca_pos.x, residue1.ca_pos.y, residue1.ca_pos.z])\n",
    "            aligned_ca_coords2.append([residue2.ca_pos.x, residue2.ca_pos.y, residue2.ca_pos.z])\n",
    "\n",
    "    # Convert the lists to numpy arrays for easier manipulation.\n",
    "    aligned_ca_coords1 = np.array(aligned_ca_coords1)\n",
    "    aligned_ca_coords2 = np.array(aligned_ca_coords2)\n",
    "\n",
    "    return aligned_ca_coords1, aligned_ca_coords2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "80bf4bbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "def kabsch_superimpose(P: np.ndarray, Q: np.ndarray):\n",
    "    \"\"\" Return the optimal rotation and translation matrices for superimposing the points P onto Q\n",
    "    Parameters\n",
    "    ----------\n",
    "    P, Q : numpy ndarray\n",
    "        Set of points: Nx3 matrix for N points in 3 dimensions to superimpose\n",
    "    Returns\n",
    "    -------\n",
    "    rotation : numpy ndarray\n",
    "        Optimal rotation matrix to align the set of point P and Q\n",
    "    translation : numpy array\n",
    "        Optimal translation vector to align the set of point P and Q\n",
    "    \"\"\"\n",
    "    # this assertion is to make sure the number of points in P and Q are the same\n",
    "    assert len(P) == len(Q)\n",
    "\n",
    "    # np.mean calculates the average of the points in P and Q along the axis specified (axis=0 = per-column)\n",
    "    # the centroid is the average position of all the points in each set\n",
    "    centroid_P = np.mean(P, axis=0)\n",
    "    centroid_Q = np.mean(Q, axis=0)\n",
    "\n",
    "    # subtract the centroid from each set of points\n",
    "    # this is done to move all points in both sets to the origin\n",
    "    # which is a standard procedure before applying the Kabsch algorithm\n",
    "    P_centered = P - centroid_P\n",
    "    Q_centered = Q - centroid_Q\n",
    "\n",
    "    # the dot product of the transpose of P and Q forms the covariance matrix\n",
    "    # np.dot is a function that computes the dot product of two arrays\n",
    "    H = np.dot(np.transpose(P_centered), Q_centered)\n",
    "\n",
    "    # use Singular Value Decomposition (SVD) to decompose H\n",
    "    # this is used to find the rotation matrix that will align the points in P with the points in Q\n",
    "    U, S, Vt = np.linalg.svd(H)\n",
    "\n",
    "    # To ensure a right-handed coordinate system, the determinant of U*Vt must be 1. If it's -1, we need to correct it\n",
    "    d = (np.linalg.det(U) * np.linalg.det(Vt)) < 0.0\n",
    "\n",
    "    if d:\n",
    "        S[-1] = -S[-1]\n",
    "        U[:, -1] = -U[:, -1]\n",
    "\n",
    "    # Calculate the rotation matrix as the product of U and Vt\n",
    "    rotation = np.dot(U, Vt)\n",
    "    \n",
    "    # The translation vector is calculated as the difference of centroids, transformed by the rotation matrix\n",
    "    translation = centroid_Q - np.dot(centroid_P, rotation)\n",
    "\n",
    "    return rotation, translation\n",
    "\n",
    "def rmsd(V: np.ndarray, W: np.ndarray):\n",
    "    \"\"\" Calculate Root-mean-square deviation (RMSD) from two sets of vectors V and W.\n",
    "    Parameters\n",
    "    ----------\n",
    "    V, W : numpy ndarray\n",
    "        (N,D) matrix, where N is points and D is dimension\n",
    "    Returns\n",
    "    -------\n",
    "    rmsd : float\n",
    "        Root-mean-square deviation\n",
    "    \"\"\"\n",
    "\n",
    "    # initialize the variables\n",
    "    N = len(V)\n",
    "    D = len(V[0])\n",
    "    \n",
    "    if len(W[0]) != D or len(W) != N:\n",
    "        raise ValueError(f\"Error: The input vectors V and W do not have the same dimensions.\")\n",
    "    \n",
    "    # compute the RMSD\n",
    "    rmsd = 0.0\n",
    "\n",
    "    for v, w in zip(V, W):\n",
    "        rmsd += sum([(v[i] - w[i])**2.0 for i in range(D)])\n",
    "\n",
    "    return np.sqrt(rmsd/N)\n",
    "\n",
    "def transform(v: np.ndarray, rotation: np.ndarray, translation: np.array):\n",
    "    \"\"\" Apply rotation and translation to the points v\n",
    "    Parameters\n",
    "    ----------\n",
    "    v : numpy ndarray\n",
    "        Matrix of points to be tranformed\n",
    "    rotation : numpy ndarray\n",
    "        Rotation matrix to transform v\n",
    "    translation : numpy array\n",
    "        Translation vector transform v\n",
    "    Returns\n",
    "    -------\n",
    "    numpy : ndarray\n",
    "        Tranformed matrix of v\n",
    "    \"\"\"\n",
    "    return np.dot(v, rotation) + translation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "30e7c0c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RMSD: 51.69238810933071\n"
     ]
    }
   ],
   "source": [
    "# Example usage:\n",
    "\n",
    "# These are random examples\n",
    "P = np.random.rand(10,3)*100 \n",
    "Q = np.random.rand(10,3)*100  \n",
    "\n",
    "rotation, translation = kabsch_superimpose(P, Q)\n",
    "\n",
    "# Apply the rotation and translation to P\n",
    "P_tr = transform(P, rotation=rotation, translation=translation)\n",
    "\n",
    "print('RMSD:', rmsd(P_tr, Q))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fd78a828",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_superposition(P: np.ndarray, Q: np.ndarray):\n",
    "    \"\"\" Combine all previous functions to get the RMSD of the superposition of P and Q\n",
    "    Parameters\n",
    "    ----------\n",
    "    P, Q : numpy ndarray\n",
    "        Set of points: Nx3 matrix for N points in 3 dimensions to superimpose\n",
    "    Returns\n",
    "    -------\n",
    "    rmsd : float\n",
    "        Root-mean-square deviation of the superimposed points P and Q\n",
    "    \"\"\"\n",
    "    \n",
    "    rotation, translation = kabsch_superimpose(P, Q)\n",
    "    P = transform(P, rotation, translation)\n",
    "\n",
    "    return rmsd(P, Q)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "aceacdf6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RMSD: 51.69238810933071\n"
     ]
    }
   ],
   "source": [
    "rmsd_value = get_superposition(P, Q)\n",
    "print('RMSD:', rmsd_value)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10f50d7a",
   "metadata": {},
   "source": [
    "Exercise: \n",
    "\n",
    "From the reference .pdb files for open (4ake) and closed (1ake) conformations, align their structures, get the matrices of the CA coordinates and print the RMSD."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d648585f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligned sequence of the open structure:\n",
      " MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n",
      "Aligned sequence of the closed structure:\n",
      " MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVTDELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRIVGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIGYYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG\n"
     ]
    }
   ],
   "source": [
    "# get the assembly objects from the PDB files\n",
    "assembly_open = LoadPDB('4ake.pdb')\n",
    "assembly_close = LoadPDB('1ake.pdb')\n",
    "\n",
    "# get the three-letter code amino acid sequence from the assemblies and convert them to one-letter code sequences\n",
    "open_three_letter = get_aa(assembly_open)\n",
    "close_three_letter = get_aa(assembly_close)\n",
    "\n",
    "open_one_letter = convert_sequence_to_one_letter(open_three_letter)\n",
    "close_one_letter = convert_sequence_to_one_letter(close_three_letter)\n",
    "\n",
    "# align the one-letter code sequences\n",
    "aln_open, aln_close = Align(open_one_letter, close_one_letter)\n",
    "\n",
    "print('Aligned sequence of the open structure:\\n', aln_open)\n",
    "print('Aligned sequence of the closed structure:\\n', aln_close)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "1fb50b05",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RMSD: 7.130700152775839\n"
     ]
    }
   ],
   "source": [
    "# get the arrays of the CA coordinates of aligned residues\n",
    "ca_coord_open, ca_coord_close = GetAlignedCaCoords(assembly_open, assembly_close, aln_open, aln_close)\n",
    "\n",
    "# compute and print the RMSD value of the superposition\n",
    "rmsd_value = get_superposition(ca_coord_open, ca_coord_close)\n",
    "print('RMSD:', rmsd_value)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "7c29d2e6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[26.091 52.849 39.889]\n",
      " [27.437 49.969 37.786]\n",
      " [24.961 47.988 35.671]\n",
      " [25.194 44.925 33.36 ]\n",
      " [22.428 44.503 30.712]\n",
      " [21.933 40.811 29.752]\n",
      " [19.604 39.512 26.973]\n",
      " [19.324 37.742 23.567]\n",
      " [20.108 39.651 20.294]\n",
      " [17.487 42.426 19.756]\n",
      " [15.96  42.201 23.284]\n",
      " [16.082 46.035 23.547]\n",
      " [19.303 46.427 25.595]\n",
      " [20.816 49.254 23.528]\n",
      " [17.421 51.124 23.373]\n",
      " [17.135 51.071 27.21 ]\n",
      " [20.821 51.825 27.722]\n",
      " [20.249 55.314 26.08 ]\n",
      " [17.474 56.05  28.712]\n",
      " [19.532 54.877 31.696]\n",
      " [22.441 56.903 30.479]\n",
      " [20.467 60.117 30.02 ]\n",
      " [18.497 59.71  33.252]\n",
      " [21.269 58.502 35.566]\n",
      " [24.253 60.217 33.938]\n",
      " [26.346 56.989 33.707]\n",
      " [28.231 56.198 30.472]\n",
      " [27.118 53.394 28.303]\n",
      " [30.122 51.105 27.381]\n",
      " [28.78  49.07  24.408]\n",
      " [31.156 46.587 22.987]\n",
      " [29.191 46.253 19.704]\n",
      " [29.487 50.042 19.273]\n",
      " [33.184 50.074 20.161]\n",
      " [33.94  47.195 17.763]\n",
      " [31.961 48.619 14.693]\n",
      " [33.778 51.895 15.373]\n",
      " [37.207 50.146 15.498]\n",
      " [36.426 48.279 12.266]\n",
      " [35.384 51.553 10.65 ]\n",
      " [38.507 53.491 11.905]\n",
      " [40.955 50.617 11.185]\n",
      " [41.976 50.997 14.916]\n",
      " [44.618 48.368 15.772]\n",
      " [42.604 47.255 18.836]\n",
      " [39.438 45.942 17.303]\n",
      " [39.679 46.226 13.467]\n",
      " [40.193 42.426 13.556]\n",
      " [36.488 41.903 14.418]\n",
      " [35.308 42.759 10.917]\n",
      " [34.849 39.271  9.552]\n",
      " [33.759 37.591 12.753]\n",
      " [30.854 40.075 13.253]\n",
      " [29.903 39.62   9.595]\n",
      " [29.654 35.907 10.111]\n",
      " [27.438 36.396 13.278]\n",
      " [30.045 35.011 15.563]\n",
      " [31.439 36.126 18.922]\n",
      " [34.805 38.036 18.787]\n",
      " [37.608 36.412 20.972]\n",
      " [37.3   37.089 24.665]\n",
      " [40.905 38.219 24.879]\n",
      " [40.206 41.06  22.376]\n",
      " [36.989 42.314 23.858]\n",
      " [38.291 42.205 27.487]\n",
      " [41.229 44.428 26.326]\n",
      " [38.822 47.067 24.857]\n",
      " [36.653 47.149 28.003]\n",
      " [39.562 47.475 30.424]\n",
      " [40.826 50.385 28.39 ]\n",
      " [37.492 52.114 28.408]\n",
      " [36.751 51.628 32.046]\n",
      " [40.118 53.39  32.667]\n",
      " [38.879 56.7   31.105]\n",
      " [37.942 59.347 33.644]\n",
      " [34.169 59.072 32.914]\n",
      " [34.023 55.549 34.377]\n",
      " [35.295 56.377 37.93 ]\n",
      " [31.688 57.065 38.942]\n",
      " [29.874 54.162 37.301]\n",
      " [29.106 52.875 33.933]\n",
      " [26.566 50.72  32.18 ]\n",
      " [27.916 47.525 30.373]\n",
      " [25.834 46.813 27.329]\n",
      " [26.933 43.697 25.483]\n",
      " [29.653 42.677 27.895]\n",
      " [30.201 40.135 29.493]\n",
      " [29.055 37.92  26.723]\n",
      " [30.715 34.709 27.886]\n",
      " [31.743 33.11  31.221]\n",
      " [35.535 33.921 30.568]\n",
      " [34.575 37.605 30.14 ]\n",
      " [32.541 37.446 33.423]\n",
      " [35.444 35.776 35.246]\n",
      " [37.851 38.413 33.908]\n",
      " [35.66  41.143 35.472]\n",
      " [35.733 39.148 38.737]\n",
      " [39.569 38.768 38.72 ]\n",
      " [39.899 42.49  37.999]\n",
      " [37.775 43.86  40.924]\n",
      " [34.864 45.066 38.723]\n",
      " [31.793 44.278 40.733]\n",
      " [28.372 45.707 39.643]\n",
      " [25.332 46.877 41.404]\n",
      " [22.614 45.492 39.142]\n",
      " [22.348 42.878 36.473]\n",
      " [19.163 43.331 34.358]\n",
      " [17.976 40.535 32.145]\n",
      " [15.795 41.545 29.219]\n",
      " [13.724 38.46  28.356]\n",
      " [12.261 38.125 24.898]\n",
      " [11.073 34.932 23.064]\n",
      " [13.347 33.748 20.186]\n",
      " [10.481 34.    17.694]\n",
      " [10.26  37.817 17.934]\n",
      " [13.933 38.508 17.208]\n",
      " [14.225 38.291 13.469]\n",
      " [11.453 40.992 12.84 ]\n",
      " [12.823 43.313 15.486]\n",
      " [16.144 43.201 13.823]\n",
      " [14.99  43.687 10.181]\n",
      " [13.285 46.97  11.062]\n",
      " [16.37  48.198 12.993]\n",
      " [18.075 51.418 11.759]\n",
      " [20.966 53.077 13.531]\n",
      " [22.804 56.339 13.735]\n",
      " [26.322 54.841 13.688]\n",
      " [28.355 57.764 15.058]\n",
      " [26.333 57.776 18.239]\n",
      " [25.086 54.264 18.402]\n",
      " [21.452 55.561 18.766]\n",
      " [18.996 52.896 17.538]\n",
      " [15.513 53.093 16.026]\n",
      " [12.92  50.671 14.575]\n",
      " [10.854 51.6   11.529]\n",
      " [ 7.667 50.304 13.237]\n",
      " [ 8.263 49.977 16.941]\n",
      " [10.221 53.056 17.749]\n",
      " [10.584 55.204 14.637]\n",
      " [12.639 58.484 14.325]\n",
      " [10.533 61.82  14.45 ]\n",
      " [11.505 62.37  10.828]\n",
      " [11.718 59.172  8.801]\n",
      " [15.26  58.064  8.028]\n",
      " [17.037 60.722 10.089]\n",
      " [18.527 60.716 13.497]\n",
      " [16.517 62.933 15.961]\n",
      " [19.53  64.752 17.344]\n",
      " [21.763 65.36  14.392]\n",
      " [19.389 65.023 11.444]\n",
      " [21.826 62.57   9.898]\n",
      " [20.683 59.723  7.72 ]\n",
      " [20.09  56.412  9.516]\n",
      " [21.658 53.19   8.095]\n",
      " [21.171 49.444  8.553]\n",
      " [23.818 47.27  10.158]\n",
      " [25.618 44.885  7.852]\n",
      " [25.037 41.923 10.23 ]\n",
      " [21.262 42.561 10.522]\n",
      " [20.054 40.082  7.944]\n",
      " [17.942 37.066  8.799]\n",
      " [20.712 34.443  8.506]\n",
      " [23.378 36.312 10.514]\n",
      " [20.732 36.952 13.256]\n",
      " [20.001 33.162 13.347]\n",
      " [23.644 32.45  13.793]\n",
      " [23.859 35.071 16.525]\n",
      " [20.894 33.428 18.379]\n",
      " [22.603 29.993 18.47 ]\n",
      " [25.889 31.549 19.798]\n",
      " [23.794 33.424 22.374]\n",
      " [22.188 30.172 23.627]\n",
      " [25.463 28.312 23.673]\n",
      " [27.78  30.924 25.231]\n",
      " [25.882 33.824 26.706]\n",
      " [22.709 32.326 28.101]\n",
      " [24.84  30.655 30.859]\n",
      " [25.494 34.116 32.326]\n",
      " [21.847 34.121 33.617]\n",
      " [22.743 31.291 36.059]\n",
      " [26.209 32.813 36.742]\n",
      " [24.857 36.175 37.96 ]\n",
      " [21.773 34.685 39.55 ]\n",
      " [24.305 32.796 41.74 ]\n",
      " [26.114 36.11  42.295]\n",
      " [22.815 37.702 43.488]\n",
      " [22.141 34.782 45.857]\n",
      " [25.497 35.366 47.468]\n",
      " [24.765 39.076 48.082]\n",
      " [27.363 40.312 45.461]\n",
      " [24.855 42.119 43.112]\n",
      " [21.066 42.643 42.614]\n",
      " [19.512 40.627 39.611]\n",
      " [16.181 41.854 38.027]\n",
      " [14.39  40.266 35.066]\n",
      " [12.218 42.589 32.815]\n",
      " [ 9.688 41.664 30.175]\n",
      " [11.316 42.835 26.964]\n",
      " [ 8.045 42.211 24.904]\n",
      " [ 6.346 45.271 26.476]\n",
      " [ 6.324 48.459 24.505]\n",
      " [ 9.572 50.527 24.721]\n",
      " [ 8.001 53.132 27.123]\n",
      " [ 6.592 50.429 29.321]\n",
      " [ 9.958 48.634 29.563]\n",
      " [11.381 52.102 30.354]\n",
      " [ 8.907 52.492 33.247]\n",
      " [ 9.716 48.993 34.79 ]\n",
      " [13.339 49.966 34.772]\n",
      " [12.663 53.23  36.684]\n",
      " [10.927 51.111 39.364]\n",
      " [14.037 48.992 39.744]\n",
      " [16.624 51.868 39.661]\n",
      " [15.055 55.001 41.318]]\n"
     ]
    }
   ],
   "source": [
    "# you  can also print the matrices to see if it matches the coordinates of ca atoms in the pdb files\n",
    "print(ca_coord_close)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bf745edf",
   "metadata": {},
   "source": [
    "Lecture 1: Introduction to alignments - exercise: align two homologous kinases (given as fasta files)\n",
    "\n",
    "Lecture 2: Introduction to superpositions - exercise: get RMSD of two structures\n",
    "\n",
    "<b>Lecture 3: Introduction to AlphaFold - exercise: get models from AlphaFold and classify them as open/close, do the same for models from Swiss-Model and AlphaFold-dropout</b>\n",
    "\n",
    "Lecture 4: Ligand docking (DiffDock) - exercise: get docking from DiffDock colab, find interacting atoms and visualize docking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "447155c2",
   "metadata": {},
   "source": [
    "### 3. Lecture: Alphafold modelling, alignments, superposition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ab798ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "# first start with getting the alphafold structures from colabfold (seq in previous ex) \n",
    "# and assign them as open/close\n",
    "# then check if all 21 models from swiss-model and 40 models from alphafold-dropout also classify as close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "8740d1e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def compare_with_assembly(model_assembly: Assembly, exp_assembly: Assembly):\n",
    "    \"\"\" Compare a modeled assembly to the experimentally determined assembly\n",
    "    Parameters\n",
    "    ----------\n",
    "    model_assembly, exp_assembly : Assembly\n",
    "        Structure of a protein, saved as an Assembly object\n",
    "    Returns\n",
    "    -------\n",
    "    rmsd_value : float\n",
    "        Root-mean-square deviation of the superimposed of the model and experimental proteins\n",
    "    \"\"\"\n",
    "\n",
    "    # use the same methods as in exercise 2 to get the RMSD of the two assemblies\n",
    "    model_aa = get_aa(model_assembly)\n",
    "    exp_aa = get_aa(exp_assembly)\n",
    "    \n",
    "    model_one_letter = convert_sequence_to_one_letter(model_aa)\n",
    "    exp_one_letter = convert_sequence_to_one_letter(exp_aa)\n",
    "    \n",
    "    model_aln, exp_aln = Align(model_one_letter, exp_one_letter)\n",
    "    \n",
    "    model_aligned_ca_coords, exp_aligned_ca_coords = GetAlignedCaCoords(model_assembly, exp_assembly, model_aln, exp_aln)\n",
    "    \n",
    "    rmsd_value = get_superposition(model_aligned_ca_coords, exp_aligned_ca_coords)\n",
    "    \n",
    "    return rmsd_value"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca485d42",
   "metadata": {},
   "source": [
    "Classify the models obtained with standard AlphaFold to open or closed conformation using the experimentally determined structures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "73046368",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "test_27f14_unrelaxed_rank_001_alphafold2_ptm_model_3_seed_000.pdb\n",
      "RMSD_open: 7.222333381741465\n",
      "RMSD_close: 0.778000547020972 \n",
      "\n",
      "test_27f14_unrelaxed_rank_004_alphafold2_ptm_model_2_seed_000.pdb\n",
      "RMSD_open: 5.606497270907733\n",
      "RMSD_close: 2.039854829835252 \n",
      "\n",
      "test_27f14_unrelaxed_rank_003_alphafold2_ptm_model_5_seed_000.pdb\n",
      "RMSD_open: 7.284754137480984\n",
      "RMSD_close: 0.776160871996097 \n",
      "\n",
      "test_27f14_unrelaxed_rank_005_alphafold2_ptm_model_4_seed_000.pdb\n",
      "RMSD_open: 7.0824691618723525\n",
      "RMSD_close: 0.9261907145828077 \n",
      "\n",
      "test_27f14_unrelaxed_rank_002_alphafold2_ptm_model_1_seed_000.pdb\n",
      "RMSD_open: 6.054032248832644\n",
      "RMSD_close: 1.8550082076994587 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "#these are the models they will get with standard alphafold (AF)\n",
    "# also save the data in csv format for analysis in the fourth lecture\n",
    "data = list()\n",
    "\n",
    "# the os package allows you to easily navigate folders and files with the use of paths\n",
    "import os\n",
    "\n",
    "alphafold_model_dir = './alphafold_models_standard'\n",
    "open_exp_structure = LoadPDB('4ake.pdb')\n",
    "close_exp_structure = LoadPDB('1ake.pdb')\n",
    "\n",
    "# get the list of all files in a directory\n",
    "for f in os.listdir(alphafold_model_dir):\n",
    "\n",
    "    # get the full path of the file f in the directory\n",
    "    path = os.path.join(alphafold_model_dir, f)\n",
    "\n",
    "    # get the assembly of the structure using the PDB file saved from AlphaFold\n",
    "    af_assembly = LoadPDB(path)\n",
    "\n",
    "    # compare the RMSD of the AF-structure with the experimentally-determined open and closed stuctures\n",
    "    rmsd_open = compare_with_assembly(af_assembly, open_exp_structure) # How far away is Protein from open state\n",
    "    rmsd_close = compare_with_assembly(af_assembly, close_exp_structure) # How far away is Protein from closed state\n",
    "    \n",
    "    print(f)\n",
    "    print('RMSD_open:', rmsd_open)\n",
    "    print('RMSD_close:', rmsd_close,'\\n')\n",
    "\n",
    "    if rmsd_open < rmsd_close: # closer to open state: Label = 0\n",
    "        label = 0\n",
    "        \n",
    "    if rmsd_open > rmsd_close: # closer to closed state: label = 1\n",
    "        label = 1\n",
    "\n",
    "    data.append([f, rmsd_open, rmsd_close, label])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "631dde49",
   "metadata": {},
   "source": [
    "Now classify the models obtained with AlphaFold dropout to open or closed conformations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "6d8465ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "test_27f14_0_unrelaxed_rank_039_alphafold2_ptm_model_2_seed_003.pdb\n",
      "RMSD_open: 4.770099326785428\n",
      "RMSD_close: 2.677812763271696 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_029_alphafold2_ptm_model_2_seed_005.pdb\n",
      "RMSD_open: 5.61008960338677\n",
      "RMSD_close: 1.920229129822461 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_001_alphafold2_ptm_model_3_seed_007.pdb\n",
      "RMSD_open: 7.2742157869092\n",
      "RMSD_close: 0.8157055949847961 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_035_alphafold2_ptm_model_4_seed_003.pdb\n",
      "RMSD_open: 6.67057196924278\n",
      "RMSD_close: 1.2081497148127518 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_015_alphafold2_ptm_model_1_seed_002.pdb\n",
      "RMSD_open: 5.741800398190161\n",
      "RMSD_close: 2.0964445535937206 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_036_alphafold2_ptm_model_4_seed_004.pdb\n",
      "RMSD_open: 6.858423328334636\n",
      "RMSD_close: 1.0451039974365202 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_009_alphafold2_ptm_model_1_seed_005.pdb\n",
      "RMSD_open: 6.084841328196615\n",
      "RMSD_close: 1.8228666503514361 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_002_alphafold2_ptm_model_3_seed_006.pdb\n",
      "RMSD_open: 7.252437297511559\n",
      "RMSD_close: 0.7830221329700296 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_016_alphafold2_ptm_model_1_seed_003.pdb\n",
      "RMSD_open: 5.748457161038184\n",
      "RMSD_close: 2.019089919723996 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_025_alphafold2_ptm_model_2_seed_004.pdb\n",
      "RMSD_open: 5.803695239103886\n",
      "RMSD_close: 1.820709479130244 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_017_alphafold2_ptm_model_5_seed_005.pdb\n",
      "RMSD_open: 6.537660948458078\n",
      "RMSD_close: 1.0879458885339344 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_030_alphafold2_ptm_model_4_seed_005.pdb\n",
      "RMSD_open: 6.837481386594573\n",
      "RMSD_close: 0.9875198479639014 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_023_alphafold2_ptm_model_5_seed_003.pdb\n",
      "RMSD_open: 6.657227349654999\n",
      "RMSD_close: 0.957154282442785 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_010_alphafold2_ptm_model_1_seed_004.pdb\n",
      "RMSD_open: 6.122996915805754\n",
      "RMSD_close: 1.7213230307877032 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_014_alphafold2_ptm_model_5_seed_002.pdb\n",
      "RMSD_open: 7.00020684097926\n",
      "RMSD_close: 0.7650223462234337 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_040_alphafold2_ptm_model_4_seed_000.pdb\n",
      "RMSD_open: 6.763522663327822\n",
      "RMSD_close: 1.122114279080696 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_020_alphafold2_ptm_model_5_seed_000.pdb\n",
      "RMSD_open: 6.434457110476071\n",
      "RMSD_close: 1.225198460951337 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_033_alphafold2_ptm_model_4_seed_007.pdb\n",
      "RMSD_open: 6.720250868067107\n",
      "RMSD_close: 1.2247236080696196 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_013_alphafold2_ptm_model_1_seed_007.pdb\n",
      "RMSD_open: 6.350533063578687\n",
      "RMSD_close: 1.411226716069355 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_004_alphafold2_ptm_model_3_seed_005.pdb\n",
      "RMSD_open: 7.134281159264053\n",
      "RMSD_close: 0.7775928792457528 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_018_alphafold2_ptm_model_5_seed_001.pdb\n",
      "RMSD_open: 7.095994181638488\n",
      "RMSD_close: 0.7589575325237611 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_008_alphafold2_ptm_model_3_seed_001.pdb\n",
      "RMSD_open: 7.169818909208843\n",
      "RMSD_close: 0.7961610384753505 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_007_alphafold2_ptm_model_3_seed_002.pdb\n",
      "RMSD_open: 7.15861707562667\n",
      "RMSD_close: 0.7569407137689645 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_003_alphafold2_ptm_model_3_seed_000.pdb\n",
      "RMSD_open: 7.198074709866434\n",
      "RMSD_close: 0.794640043083965 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_038_alphafold2_ptm_model_4_seed_006.pdb\n",
      "RMSD_open: 6.888149602437507\n",
      "RMSD_close: 1.0159516252702951 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_037_alphafold2_ptm_model_2_seed_002.pdb\n",
      "RMSD_open: 5.099751498883069\n",
      "RMSD_close: 2.3641235832987983 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_028_alphafold2_ptm_model_2_seed_007.pdb\n",
      "RMSD_open: 6.20157849049337\n",
      "RMSD_close: 1.3489212746268415 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_027_alphafold2_ptm_model_4_seed_002.pdb\n",
      "RMSD_open: 6.858072566945148\n",
      "RMSD_close: 1.0414824984908455 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_024_alphafold2_ptm_model_5_seed_007.pdb\n",
      "RMSD_open: 7.290068931109881\n",
      "RMSD_close: 0.8123744309170317 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_034_alphafold2_ptm_model_2_seed_001.pdb\n",
      "RMSD_open: 5.697986810218451\n",
      "RMSD_close: 1.7958020552967342 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_031_alphafold2_ptm_model_2_seed_000.pdb\n",
      "RMSD_open: 5.347545373150292\n",
      "RMSD_close: 2.1576725430779837 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_012_alphafold2_ptm_model_1_seed_001.pdb\n",
      "RMSD_open: 6.012647948004681\n",
      "RMSD_close: 1.8464654718506162 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_032_alphafold2_ptm_model_4_seed_001.pdb\n",
      "RMSD_open: 7.0519199093860205\n",
      "RMSD_close: 0.9066186575190128 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_011_alphafold2_ptm_model_1_seed_006.pdb\n",
      "RMSD_open: 6.171363166909062\n",
      "RMSD_close: 1.6655402063239615 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_019_alphafold2_ptm_model_5_seed_004.pdb\n",
      "RMSD_open: 6.60426409496824\n",
      "RMSD_close: 1.0331075668019443 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_021_alphafold2_ptm_model_5_seed_006.pdb\n",
      "RMSD_open: 7.151382140707422\n",
      "RMSD_close: 0.782939999600177 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_006_alphafold2_ptm_model_3_seed_004.pdb\n",
      "RMSD_open: 7.191635004014734\n",
      "RMSD_close: 0.7862757224249118 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_005_alphafold2_ptm_model_3_seed_003.pdb\n",
      "RMSD_open: 7.219738952168957\n",
      "RMSD_close: 0.7863218257713439 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_022_alphafold2_ptm_model_2_seed_006.pdb\n",
      "RMSD_open: 6.1268808639414685\n",
      "RMSD_close: 1.511019497559219 \n",
      "\n",
      "test_27f14_0_unrelaxed_rank_026_alphafold2_ptm_model_1_seed_000.pdb\n",
      "RMSD_open: 4.8533406270767845\n",
      "RMSD_close: 2.76390043249238 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# then check if alphafold-dropout does a better job\n",
    "\n",
    "\n",
    "\n",
    "#TODO: calculate rmsd of models in \"./alphafold_models_dropout\" to experimental structures\n",
    "\n",
    "# the os package allows you to easily navigate folders and files with the use of paths\n",
    "import os\n",
    "\n",
    "alphafold_model_dir = \"./alphafold_models_dropout\"\n",
    "open_exp_structure = LoadPDB('4ake.pdb')\n",
    "close_exp_structure = LoadPDB('1ake.pdb')\n",
    "\n",
    "# get the list of all files in a directory\n",
    "for f in os.listdir(alphafold_model_dir):\n",
    "\n",
    "    # get the full path of the file f in the directory\n",
    "    path = os.path.join(alphafold_model_dir, f)\n",
    "\n",
    "    # get the assembly of the structure using the PDB file saved from AlphaFold\n",
    "    af_assembly = LoadPDB(path)\n",
    "\n",
    "    # compare the RMSD of the AF-structure with the experimentally-determined open and closed stuctures\n",
    "    rmsd_open = compare_with_assembly(af_assembly, open_exp_structure) # How far away is Protein from open state\n",
    "    rmsd_close = compare_with_assembly(af_assembly, close_exp_structure) # How far away is Protein from closed state\n",
    "    \n",
    "    print(f)\n",
    "    print('RMSD_open:', rmsd_open)\n",
    "    print('RMSD_close:', rmsd_close,'\\n')\n",
    "\n",
    "    if rmsd_open < rmsd_close: # closer to open state: Label = 0\n",
    "        label = 0\n",
    "        \n",
    "    if rmsd_open > rmsd_close: # closer to closed state: label = 1\n",
    "        label = 1\n",
    "\n",
    "    data.append([f, rmsd_open, rmsd_close, label])\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "04305621",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "model_09.pdb\n",
      "RMSD_open: 7.457632384668484\n",
      "RMSD_close: 0.9839113192898898 \n",
      "\n",
      "model_21.pdb\n",
      "RMSD_open: 5.962369229489549\n",
      "RMSD_close: 3.64936521277075 \n",
      "\n",
      "model_20.pdb\n",
      "RMSD_open: 6.253750223410452\n",
      "RMSD_close: 3.5694162818283193 \n",
      "\n",
      "model_08.pdb\n",
      "RMSD_open: 7.44378471068204\n",
      "RMSD_close: 0.9997453061916688 \n",
      "\n",
      "model_18.pdb\n",
      "RMSD_open: 6.435332163548642\n",
      "RMSD_close: 3.20034708709186 \n",
      "\n",
      "model_19.pdb\n",
      "RMSD_open: 6.173371313254284\n",
      "RMSD_close: 3.3895320498911197 \n",
      "\n",
      "model_14.pdb\n",
      "RMSD_open: 5.234569482374888\n",
      "RMSD_close: 4.811048292722425 \n",
      "\n",
      "model_15.pdb\n",
      "RMSD_open: 6.401471409976921\n",
      "RMSD_close: 3.456491047767172 \n",
      "\n",
      "model_01.pdb\n",
      "RMSD_open: 7.240747404312423\n",
      "RMSD_close: 0.4205283572155502 \n",
      "\n",
      "model_17.pdb\n",
      "RMSD_open: 6.2638229760471305\n",
      "RMSD_close: 3.633737054697014 \n",
      "\n",
      "model_03.pdb\n",
      "RMSD_open: 6.9853361921947545\n",
      "RMSD_close: 0.8049712535618541 \n",
      "\n",
      "model_02.pdb\n",
      "RMSD_open: 6.865318540805919\n",
      "RMSD_close: 0.7999470553907604 \n",
      "\n",
      "model_16.pdb\n",
      "RMSD_open: 6.321605575647794\n",
      "RMSD_close: 3.456564769612247 \n",
      "\n",
      "model_12.pdb\n",
      "RMSD_open: 2.0910367039344537\n",
      "RMSD_close: 6.132128951003093 \n",
      "\n",
      "model_06.pdb\n",
      "RMSD_open: 7.1183760217129946\n",
      "RMSD_close: 0.44391913793543075 \n",
      "\n",
      "model_07.pdb\n",
      "RMSD_open: 1.0799970602552633\n",
      "RMSD_close: 6.330783299726932 \n",
      "\n",
      "model_13.pdb\n",
      "RMSD_open: 2.0118793040911753\n",
      "RMSD_close: 6.5132796620933195 \n",
      "\n",
      "model_05.pdb\n",
      "RMSD_open: 7.16418038503085\n",
      "RMSD_close: 0.501611447517743 \n",
      "\n",
      "model_11.pdb\n",
      "RMSD_open: 2.0184769019426434\n",
      "RMSD_close: 6.523277683109816 \n",
      "\n",
      "model_10.pdb\n",
      "RMSD_open: 2.0895667218327008\n",
      "RMSD_close: 6.130181706890457 \n",
      "\n",
      "model_04.pdb\n",
      "RMSD_open: 6.329755220866905\n",
      "RMSD_close: 1.4070459914189704 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# then they will check how swiss-model is doing\n",
    "\n",
    "#TODO: calculate rmsd of models in \"./swiss_models\" to experimental structures\n",
    "\n",
    "# the os package allows you to easily navigate folders and files with the use of paths\n",
    "import os\n",
    "\n",
    "alphafold_model_dir = \"./swiss_models\"\n",
    "open_exp_structure = LoadPDB('4ake.pdb')\n",
    "close_exp_structure = LoadPDB('1ake.pdb')\n",
    "\n",
    "# get the list of all files in a directory\n",
    "for f in os.listdir(alphafold_model_dir):\n",
    "\n",
    "    # get the full path of the file f in the directory\n",
    "    path = os.path.join(alphafold_model_dir, f)\n",
    "\n",
    "    # get the assembly of the structure using the PDB file saved from AlphaFold\n",
    "    af_assembly = LoadPDB(path)\n",
    "\n",
    "    # compare the RMSD of the AF-structure with the experimentally-determined open and closed stuctures\n",
    "    rmsd_open = compare_with_assembly(af_assembly, open_exp_structure) # How far away is Protein from open state\n",
    "    rmsd_close = compare_with_assembly(af_assembly, close_exp_structure) # How far away is Protein from closed state\n",
    "    \n",
    "    print(f)\n",
    "    print('RMSD_open:', rmsd_open)\n",
    "    print('RMSD_close:', rmsd_close,'\\n')\n",
    "\n",
    "    if rmsd_open < rmsd_close: # closer to open state: Label = 0\n",
    "        label = 0\n",
    "        \n",
    "    if rmsd_open > rmsd_close: # closer to closed state: label = 1\n",
    "        label = 1\n",
    "\n",
    "    data.append([f, rmsd_open, rmsd_close, label])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "521aeeec",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save all the data to a csv file for the next lecture\n",
    "np.savetxt('data.csv',\n",
    "        data,\n",
    "        delimiter =', ',\n",
    "        header='filename, rmsd_open, rmsd_close, label',\n",
    "        fmt ='% s')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78d299d0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
