{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "eaf4e3e4",
   "metadata": {},
   "source": [
    "### Programming in life sciences"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "89abeed2",
   "metadata": {},
   "source": [
    "### Predicting open/close conformation of a kinase model\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edef8232",
   "metadata": {},
   "source": [
    "<b>Lecture 1: Introduction to alignments - exercise: align two homologous kinases (given as fasta files)</b>\n",
    "\n",
    "Lecture 2: Introduction to superpositions - exercise: get RMSD of two structures\n",
    "\n",
    "Lecture 3: Introduction to AlphaFold - exercise: get models from AlphaFold and classify them as open/close, do the same for models from Swiss-Model and AlphaFold-dropout\n",
    "\n",
    "Lecture 4: Ligand docking (DiffDock) - exercise: get docking from DiffDock colab, find interacting atoms and visualize docking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe148b03",
   "metadata": {
    "tags": [
     "lesson1"
    ]
   },
   "source": [
    "### 1. Lecture: Needleman-Wunsch algorithm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "933ffbba",
   "metadata": {},
   "source": [
    "In this exercise, we want to align two sequences obtained from fasta files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "6d97f4c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# numpy is a library providing a lot of numerical mathematical functionalities\n",
    "# here we only use it to store stuff in matrices\n",
    "import numpy as np\n",
    "\n",
    "# we create a new class that will be used to define substitution matrices\n",
    "class SubstitutionMatrix:\n",
    "  \"\"\"Read, store and analyse substitution matrices, default = BLOSUM62\"\"\"\n",
    "\n",
    "  # this is the constructor function! When you create a new instance of type\n",
    "  # Substitution Matrix, this function gets called. By default it reads in\n",
    "  # the provided file: blosum62.txt... you can create your own matrix if\n",
    "  # you like ;)\n",
    "  def __init__(self, data_file: str = \"blosum62.txt\"):\n",
    "    \"\"\" Constructor function for the SubstitutionMatrix class\n",
    "    Parameters\n",
    "    ----------\n",
    "    data_file : string\n",
    "        Filename of the substitution matrix to read in, must contain 21 lines,\n",
    "        with the first line containing the 20 amino acid one letter codes,\n",
    "        by default reads the blosum62.txt matrix in the current folder\n",
    "    \"\"\"\n",
    "    \n",
    "    # read all the lines from the file at once  ('r' = read)\n",
    "    with open(data_file,'r') as fh:\n",
    "      data = fh.readlines()\n",
    "\n",
    "    # the size of subsitution matrices must be 20x20\n",
    "    # the first line of the file gives the amino acid one letter codes\n",
    "    # which gives a total of 21 lines in the file\n",
    "    if len(data) != 21:\n",
    "      err = \"Expected exactly 21 lines in matrix file!\"\n",
    "      err += \" One line defining the amino acid one letter codes and \"\n",
    "      err += \"20 lines defining the substitution matrix\"\n",
    "      raise RuntimeError(err)    \n",
    "\n",
    "    # get the amino acid one letter codes\n",
    "    self.one_letter_codes = data[0].strip()\n",
    "\n",
    "    # check that 20 amino acids are defined\n",
    "    if len(self.one_letter_codes) != 20:\n",
    "      err = \"First line must contain exactly 20 letters representing \"\n",
    "      err += \"the amino acids!\"\n",
    "      raise RuntimeError(err)\n",
    "\n",
    "    # initialize the substitution matrix with zeros\n",
    "    self.matrix = np.zeros((20,20))\n",
    "\n",
    "    # fill in the matrix by reading the file line by line\n",
    "    # note that range returns a vector starting at 0 to the integer in argument - 1 \n",
    "    for i in range(20):\n",
    "      data_line = data[1+i].split()\n",
    "      if len(data_line) != 20:\n",
    "        raise RuntimeError(\"Each data line must contain exactly 20 elements!\")\n",
    "      for j in range(20):\n",
    "        self.matrix[(i,j)] = int(data_line[j])\n",
    "\n",
    "\n",
    "  # The function where you can extract the score for a specific amino acid substitution...\n",
    "  # e.g. mat.get_score('W', 'A')\n",
    "  def get_score(self, aa_one: str, aa_two: str):\n",
    "    \"\"\" Extract the substitution score for two amino acids from the substitution matrix\n",
    "    Parameters\n",
    "    ----------\n",
    "    aa_one : string\n",
    "        One letter amino acid code\n",
    "    aa_two : string\n",
    "        One letter amino acid code\n",
    "    Returns\n",
    "    -------\n",
    "    self.matrix[(idx_one,idx_two)] : float\n",
    "        Substitution matrix entry for the substitution of amino acid one and amino acid two,\n",
    "        note that the matrices are symmetric, so get_score(aa_one, aa_two) is equal to get_score(aa_two, aa_one)\n",
    "    \"\"\"\n",
    "\n",
    "    # get the indices of the amino acids in the matrix\n",
    "    idx_one = self.one_letter_codes.find(aa_one)\n",
    "    idx_two = self.one_letter_codes.find(aa_two)\n",
    "\n",
    "    if idx_one == -1:\n",
    "      raise RuntimeError(\"aa_one must be one of: \" + self.one_letter_codes)\n",
    "    \n",
    "    if idx_two == -1:\n",
    "      raise RuntimeError(\"aa_two must be one of: \" + self.one_letter_codes)\n",
    "\n",
    "    return self.matrix[(idx_one,idx_two)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f8b4798",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Decorating functions with docstrings is not only best practice,\n",
    "# it also helps automated documentation generation with tools\n",
    "# like Sphinx (https://www.sphinx-doc.org) or simply get more\n",
    "# information using the Python help function:\n",
    "help(SubstitutionMatrix.get_score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7bedee41",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Construct a default substitution matrix\n",
    "subst_matrix = SubstitutionMatrix()\n",
    "\n",
    "# get score of substituting ASP with ASN and GLY with PRO\n",
    "print(\"Score D->N:\", subst_matrix.get_score('N', 'D'))\n",
    "print(\"Score G->P:\", subst_matrix.get_score('G', 'P'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a94cabbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def ReadFasta(filename: str):\n",
    "    \"\"\" A simplistic reader that only checks if the first line starts with '>' and assumes that the whole rest is the input sequence\n",
    "    Parameters\n",
    "    ----------\n",
    "    filename : string\n",
    "        Filename of the FASTA file to read\n",
    "    Returns\n",
    "    -------\n",
    "    s : string\n",
    "        Sequence read from the input FASTA file\n",
    "    \"\"\"\n",
    "    \n",
    "    # store all the lines in a single list  ('r' = read)\n",
    "    with open(filename, 'r') as fh:\n",
    "        lines = fh.readlines()\n",
    "    \n",
    "    # handle errors\n",
    "    if len(lines) <= 1:\n",
    "        raise RuntimeError(\"Expect multiple lines in FASTA file\")\n",
    "    if not lines[0].startswith('>'):\n",
    "        raise RuntimeError(\"Assume first line in FASTA file to start with '>'\")\n",
    "\n",
    "    # read the sequence lines, while removing all extra characters\n",
    "    s = \"\"\n",
    "    for line in lines[1:]:\n",
    "        s += line.strip()\n",
    "        \n",
    "    return s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3786a28b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in the two sequences to align \n",
    "# the two files are located in the same directory as the notebook\n",
    "s1 = ReadFasta(\"s1.fasta\")\n",
    "s2 = ReadFasta(\"s2.fasta\")\n",
    "print(s1)\n",
    "print(s2)\n",
    "\n",
    "## old s1: GAWGHEE, old s2: PAWHEA"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44b39e0a",
   "metadata": {},
   "source": [
    "#### Write the code for running the Needleman-Wunsch algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09b8cc4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# define a penalty for introducing a gap in the sequence alignment\n",
    "gap_penalty = -8"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d455926",
   "metadata": {},
   "source": [
    "Setup scoring and backtracking matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce456d98",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_rows = len(s1)+1 # s1 goes from top to bottom\n",
    "n_cols = len(s2)+1 # s2 goes from left to right\n",
    "\n",
    "scoring_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "# backtrack encoding:\n",
    "# 1 => aligned\n",
    "# 2 => deletion in s1 (insertion in s2 respectively)\n",
    "# 3 => deletion in s2 (insertion in s1 respectively)\n",
    "backtrack_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "# the first row and the first column can already be prefilled\n",
    "for i in range(1, n_rows):\n",
    "    scoring_matrix[(i,0)] = i*gap_penalty\n",
    "    backtrack_matrix[(i,0)] = 3\n",
    "for i in range(1, n_cols):\n",
    "    scoring_matrix[(0,i)] = i*gap_penalty\n",
    "    backtrack_matrix[(0,i)] = 2\n",
    "\n",
    "print(\"row sequence: \", s1)\n",
    "print(\"col sequence: \", s2)\n",
    "print(\"empty scoring matrix: \")\n",
    "print(scoring_matrix)\n",
    "print(\"empty backtracking matrix: \")\n",
    "print(backtrack_matrix)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "710f66ae",
   "metadata": {},
   "source": [
    "#### Fill scoring and backtracking matrices\n",
    "\n",
    "Every position in the scoring matrix represents the best local solution given the\n",
    "steps I can take\n",
    "- Align two residues => Move along the diagonal (consume a letter from s1 and s2),\n",
    "                        the value of the scoring matrix to the upper left plus the corresponding score from the scoring matrix\n",
    "- Apply deletion/gap in s1 => Move right in the scoring matrix (consume a letter from s2), \n",
    "                              the value from the scoring matrix to the left plus the penalty for a gap\n",
    "- Apply deletion/gap in s2 => Move down in the scoring matrix (consume a letter from s1),\n",
    "                              the value from the scoring matrix above plus the penalty for a gap\n",
    "\n",
    "The goal is to identify the best scoring solution from those three,\n",
    "set the according score in the scoring matrix and keep track of this local\n",
    "step in the backtracking matrix. \n",
    "\n",
    "The backtracking matrix stores the path I took, e.g. 1 for the first option\n",
    "(see backtracking encoding in previous code cell)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fda9db23",
   "metadata": {},
   "outputs": [],
   "source": [
    "for r_idx in range(1, n_rows):\n",
    "    for c_idx in range(1, n_cols):\n",
    "        score_match = scoring_matrix[(r_idx-1, c_idx-1)] + subst_matrix.get_score(s1[r_idx-1], s2[c_idx-1])\n",
    "        score_del_1 = scoring_matrix[(r_idx, c_idx-1)] + gap_penalty\n",
    "        score_del_2 = scoring_matrix[(r_idx-1, c_idx)] + gap_penalty        \n",
    "        # TODO Fill the values at position (r_idx, c_idx) in scoring_matrix and backtracking_matrix\n",
    "        # you can assign an element with\n",
    "        # scoring_matrix[(r_idx, c_idx)] = my_awesome_value\n",
    "        if score_match >= score_del_1 and score_match >= score_del_2:\n",
    "            scoring_matrix[(r_idx, c_idx)] = score_match\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 1\n",
    "        elif score_del_1 >= score_match and score_del_1 >= score_del_2:\n",
    "            scoring_matrix[(r_idx, c_idx)] = score_del_1\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 2\n",
    "        else:\n",
    "            scoring_matrix[(r_idx, c_idx)] = score_del_2\n",
    "            backtrack_matrix[(r_idx, c_idx)] = 3\n",
    "        \n",
    "print(\"filled scoring matrix:\")\n",
    "print(scoring_matrix)\n",
    "print(\"filled backtracking matrix:\")\n",
    "print(backtrack_matrix)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f7572fe",
   "metadata": {},
   "source": [
    "#### Perform backtracking to get final alignment\n",
    "\n",
    "Remember the encoding:\n",
    "\n",
    "1 => aligned, move along the diagonal\n",
    "\n",
    "2 => deletion in s1 (insertion in s2 respectively), move right\n",
    "\n",
    "3 => deletion in s2 (insertion in s1 respectively), move down"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b8e7212",
   "metadata": {},
   "outputs": [],
   "source": [
    "# In principle we start at the bottom right of the backtracking matrix and\n",
    "# work our way through the matrix until we hit the upper left\n",
    "\n",
    "r_idx = n_rows-1\n",
    "c_idx = n_cols-1\n",
    "path = list()\n",
    "\n",
    "# if we hit zero, we're at the upper left corner and can stop\n",
    "while backtrack_matrix[(r_idx, c_idx)] != 0:\n",
    "    path.append(backtrack_matrix[(r_idx, c_idx)])\n",
    "    if backtrack_matrix[(r_idx, c_idx)] == 1:\n",
    "        r_idx -= 1\n",
    "        c_idx -= 1\n",
    "    elif backtrack_matrix[(r_idx, c_idx)] == 2:\n",
    "        c_idx -= 1\n",
    "    else:\n",
    "        r_idx -= 1\n",
    "        \n",
    "# backtracking comes from the back, so lets reverse...\n",
    "path.reverse()\n",
    "\n",
    "aln_s1 = []\n",
    "aln_s2 = []\n",
    "s1_idx = 0\n",
    "s2_idx = 0\n",
    "\n",
    "for p in path:\n",
    "    if p == 1:\n",
    "        aln_s1.append(s1[s1_idx])\n",
    "        aln_s2.append(s2[s2_idx])\n",
    "        s1_idx += 1\n",
    "        s2_idx += 1\n",
    "    elif p == 2:\n",
    "        aln_s1.append('-')\n",
    "        aln_s2.append(s2[s2_idx])\n",
    "        s2_idx += 1\n",
    "    else:\n",
    "        aln_s1.append(s1[s1_idx])\n",
    "        aln_s2.append('-')\n",
    "        s1_idx += 1\n",
    "        \n",
    "aln_s1 = ''.join(aln_s1)\n",
    "aln_s2 = ''.join(aln_s2)\n",
    "        \n",
    "print(\"input sequence s1:\", s1)\n",
    "print(\"input sequence s2:\", s2)\n",
    "print(\"\\noptimal alignment:\")\n",
    "print(\"s1:\", aln_s1)\n",
    "print(\"s2:\", aln_s2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7169eb3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#let's put everything in one function\n",
    "def Align(s1: str, s2: str):\n",
    "    \"\"\" Alignment of two strings of amino acids according to the Needleman-Wunsch algorithm\n",
    "    Parameters\n",
    "    ----------\n",
    "    s1 : string\n",
    "        First amino acid string to align\n",
    "    s2 : string\n",
    "        Second amino acid string to align\n",
    "    Returns\n",
    "    -------\n",
    "    aln_s1 : string\n",
    "        First amino acid string, aligned\n",
    "    aln_s2 : string\n",
    "        Second amino acid string, aligned\n",
    "    \"\"\"\n",
    "    \n",
    "    # initialize the substitution matrix (default = BLOSUM62)\n",
    "    subst_matrix = SubstitutionMatrix()\n",
    "    \n",
    "    # define the gap penalty value\n",
    "    gap_penalty = -8\n",
    "\n",
    "    # initialize the scoring and backtrack matrices\n",
    "    n_rows = len(s1)+1 # s1 goes from top to bottom\n",
    "    n_cols = len(s2)+1 # s2 goes from left to right\n",
    "    scoring_matrix = np.zeros((n_rows, n_cols))\n",
    "    backtrack_matrix = np.zeros((n_rows, n_cols))\n",
    "\n",
    "    # the first row and the first column can already be prefilled\n",
    "    for i in range(1, n_rows):\n",
    "        scoring_matrix[(i,0)] = i*gap_penalty\n",
    "        backtrack_matrix[(i,0)] = 3\n",
    "    for i in range(1, n_cols):\n",
    "        scoring_matrix[(0,i)] = i*gap_penalty\n",
    "        backtrack_matrix[(0,i)] = 2\n",
    "\n",
    "    # fill the scoring and backtrack matrices\n",
    "    for r_idx in range(1, n_rows):\n",
    "        for c_idx in range(1, n_cols):\n",
    "            score_match = scoring_matrix[(r_idx-1, c_idx-1)] + subst_matrix.get_score(s1[r_idx-1], s2[c_idx-1])\n",
    "            score_del_1 = scoring_matrix[(r_idx, c_idx-1)] + gap_penalty\n",
    "            score_del_2 = scoring_matrix[(r_idx-1, c_idx)] + gap_penalty        \n",
    "\n",
    "            if score_match >= score_del_1 and score_match >= score_del_2:\n",
    "                scoring_matrix[(r_idx, c_idx)] = score_match\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 1\n",
    "            elif score_del_1 >= score_match and score_del_1 >= score_del_2:\n",
    "                scoring_matrix[(r_idx, c_idx)] = score_del_1\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 2\n",
    "            else:\n",
    "                scoring_matrix[(r_idx, c_idx)] = score_del_2\n",
    "                backtrack_matrix[(r_idx, c_idx)] = 3\n",
    "\n",
    "    # track back from the lower right corner to the top left corner of the backtrack matrix to get the path\n",
    "    r_idx = n_rows-1\n",
    "    c_idx = n_cols-1\n",
    "    path = list()\n",
    "\n",
    "    # if we hit zero, we're at the upper left corner and can stop\n",
    "    while backtrack_matrix[(r_idx, c_idx)] != 0:\n",
    "        path.append(backtrack_matrix[(r_idx, c_idx)])\n",
    "        if backtrack_matrix[(r_idx, c_idx)] == 1:\n",
    "            r_idx -= 1\n",
    "            c_idx -= 1\n",
    "        elif backtrack_matrix[(r_idx, c_idx)] == 2:\n",
    "            c_idx -= 1\n",
    "        else:\n",
    "            r_idx -= 1\n",
    "\n",
    "    # backtracking comes from the back, so lets reverse...\n",
    "    path.reverse()\n",
    "\n",
    "    # get the aligned sequences by going through the path\n",
    "    aln_s1 = []\n",
    "    aln_s2 = []\n",
    "    s1_idx = 0\n",
    "    s2_idx = 0\n",
    "\n",
    "    for p in path:\n",
    "        if p == 1:\n",
    "            aln_s1.append(s1[s1_idx])\n",
    "            aln_s2.append(s2[s2_idx])\n",
    "            s1_idx += 1\n",
    "            s2_idx += 1\n",
    "        elif p == 2:\n",
    "            aln_s1.append('-')\n",
    "            aln_s2.append(s2[s2_idx])\n",
    "            s2_idx += 1\n",
    "        else:\n",
    "            aln_s1.append(s1[s1_idx])\n",
    "            aln_s2.append('-')\n",
    "            s1_idx += 1\n",
    "\n",
    "    aln_s1 = ''.join(aln_s1)\n",
    "    aln_s2 = ''.join(aln_s2)\n",
    "    \n",
    "    return aln_s1, aln_s2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfc5ff1d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Read two fasta files of bacterial kinase homologs and get the alignment\n",
    "bac1_kinase = ReadFasta('Q47XA8.fasta')\n",
    "print(bac1_kinase)\n",
    "\n",
    "bac2_kinase = ReadFasta('P69441.fasta')\n",
    "print(bac2_kinase)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3c90bf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "bac1_aln, bac2_aln = Align(bac1_kinase, bac2_kinase)\n",
    "print(bac1_aln)\n",
    "print(bac2_aln)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3de2ec78",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequence identity is a simple proxy for evolutionary distance\n",
    "def SeqID(aln_s1: str, aln_s2: str):\n",
    "    \"\"\" Computes sequence identity - the percentage of conserved\n",
    "    amino acids\n",
    "    ----------\n",
    "    aln_s1 : string\n",
    "        First aligned amino acid string\n",
    "    aln_s2 : string\n",
    "        Second aligned amino acid string\n",
    "    Returns\n",
    "    -------\n",
    "    seqid : float\n",
    "        The sequence identity\n",
    "    \"\"\"\n",
    "    # must have same length\n",
    "    assert(len(aln_s1) == len(aln_s2))\n",
    "    n_match = 0\n",
    "    n_aligned = 0\n",
    "    for a,b in zip(aln_s1, aln_s2):\n",
    "        if a!='-' and b!='-':\n",
    "            n_aligned += 1\n",
    "            if a == b:\n",
    "                n_match += 1\n",
    "    if n_aligned != 0:\n",
    "        # explicitely cast to float here\n",
    "        # This is not required in Python3 but Python2 would perform\n",
    "        # an integer division otherwise\n",
    "        return float(n_match)/n_aligned*100\n",
    "    return 0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc470f41",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Python provides several ways of string formatting\n",
    "# one example are so called f-strings\n",
    "print(f\"SeqID: {SeqID(bac1_aln, bac2_aln):.2f}%\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7cd9005f",
   "metadata": {},
   "source": [
    "Lecture 1: Introduction to alignments - exercise: align two homologous kinases (given as fasta files)\n",
    "\n",
    "<b>Lecture 2: Introduction to superpositions - exercise: get RMSD of two structures</b>\n",
    "\n",
    "Lecture 3: Introduction to AlphaFold - exercise: get models from AlphaFold and classify them as open/close, do the same for models from Swiss-Model and AlphaFold-dropout\n",
    "\n",
    "Lecture 4: Ligand docking (DiffDock) - exercise: get docking from DiffDock colab, find interacting atoms and visualize docking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bebea35",
   "metadata": {},
   "source": [
    "### 2. Lecture: superpositions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c46b9172",
   "metadata": {},
   "outputs": [],
   "source": [
    "# we create a new class that will be used to define atoms in 3D space\n",
    "class Vec3:\n",
    "    \"\"\"Store and print 3D coordinates\"\"\"\n",
    "    \n",
    "    def __init__(self, x: float, y: float, z: float):\n",
    "        \"\"\" Constructor function for the Vec3 class\n",
    "        Parameters\n",
    "        ----------\n",
    "        x, y, z : float\n",
    "            Coordinates of the atom in 3D space\n",
    "        \"\"\"\n",
    "\n",
    "        # define member variables here\n",
    "        self.x = x\n",
    "        self.y = y\n",
    "        self.z = z\n",
    "        \n",
    "    def __str__(self):\n",
    "        \"\"\" Print function for the Vec3 class \"\"\"\n",
    "        return '[{:.2f}, {:.2f}, {:.2f}]'.format(self.x, self.y, self.z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9d6d2117",
   "metadata": {},
   "outputs": [],
   "source": [
    "# we create a new class that will be used to describe a single residue of a protein chain\n",
    "class Residue:\n",
    "    \"\"\"Store and print a protein residue\"\"\"\n",
    "    def __init__(self, rname: str, rnum: int):\n",
    "        \"\"\" Constructor function for the Residue class\n",
    "        Parameters\n",
    "        ----------\n",
    "        rname : string\n",
    "            Name of the residue (amino acid 3 letter code)\n",
    "        rnum : int\n",
    "            Residue sequence number, position in the chain\n",
    "        \"\"\"\n",
    "\n",
    "        # define member variables here\n",
    "        self.rname = rname\n",
    "        self.rnum = rnum\n",
    "        self.n_pos = None\n",
    "        self.ca_pos = None\n",
    "        self.c_pos = None\n",
    "        \n",
    "    def __str__(self):\n",
    "        \"\"\" Print function for the Residue class \"\"\"    \n",
    "        pos_strings = list()\n",
    "        pos_strings.append('N: ' + str(self.n_pos))\n",
    "        pos_strings.append('CA: ' + str(self.ca_pos))\n",
    "        pos_strings.append('C: ' + str(self.c_pos))\n",
    "        return '\\t'.join([self.rname, self.rnum, ' '.join(pos_strings)])\n",
    "    \n",
    "    def get_coords(self):\n",
    "        \"\"\" Get all the coordinates of the atoms in the residue as a list\n",
    "        Parameters\n",
    "        ----------\n",
    "\n",
    "        Returns\n",
    "        -------\n",
    "        list\n",
    "            x, y and z Ã… coordinates of the N (amino group atom), CA (carbon atom) and C (carboxyl group atom) positions in a single list\n",
    "        \"\"\"\n",
    "        return [self.n_pos.x, self.n_pos.y, self.n_pos.z,self.ca_pos.x, self.ca_pos.y, self.ca_pos.z,self.c_pos.x, self.c_pos.y, self.c_pos.z]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "dca21a99",
   "metadata": {},
   "outputs": [],
   "source": [
    "# define a Chain class, that contains following information as member variables:\n",
    "# - name (e.g. \"A\")\n",
    "# - a list of residues\n",
    "#\n",
    "#AWESOME IMPLEMENTATION COMES HERE:\n",
    "\n",
    "class Chain:\n",
    "    \"\"\"Store and print a protein chain = a list of residues\"\"\"\n",
    "\n",
    "    def __init__(self, cname: str):\n",
    "        \"\"\" Constructor function for the Chain class\n",
    "        Parameters\n",
    "        ----------\n",
    "        cname : string\n",
    "            Name of the chain, e.g. 'A'\n",
    "        \"\"\"\n",
    "\n",
    "        self.cname = cname\n",
    "        self.residues = list()\n",
    "        \n",
    "    def __str__(self):\n",
    "        \"\"\" Print function for the Chain class \"\"\"    \n",
    "        return_list = list()\n",
    "        return_list.append('chain name: ' + self.cname)\n",
    "        for r in self.residues:\n",
    "            return_list.append(' ' + str(r))\n",
    "        return '\\n'.join(return_list)\n",
    "\n",
    "# define an Assembly class, that contains following information as member variables:\n",
    "# - a list of chains\n",
    "#\n",
    "# AWESOME IMPLEMENTATION COMES HERE:\n",
    "\n",
    "class Assembly:\n",
    "    \"\"\"Store and print a protein assembly = a list of chains\"\"\"\n",
    "\n",
    "    def __init__(self):\n",
    "        \"\"\" Constructor function for the Assembly class\n",
    "        Parameters\n",
    "        ----------\n",
    "        \"\"\"\n",
    "\n",
    "        self.chains = list()\n",
    "        \n",
    "    def __str__(self):\n",
    "        \"\"\" Print function for the Assembly class \"\"\"    \n",
    "        return_list = list()\n",
    "        for c in self.chains:\n",
    "            return_list.append(str(c))\n",
    "        return '\\n\\n'.join(return_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea073e88",
   "metadata": {},
   "source": [
    "The PDB is a very specific file format for protein structures:\n",
    "https://www.cgl.ucsf.edu/chimera/docs/UsersGuide/tutorials/pdbintro.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0c3eeda9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def LoadPDB(filename: str):\n",
    "    \"\"\" Read and save a protein structure from a PDB file\n",
    "    Parameters\n",
    "    ----------\n",
    "    filename : string\n",
    "        Name of the PDB file to be read\n",
    "    Returns\n",
    "    -------\n",
    "    assembly : Assembly\n",
    "        The full Assembly of the protein read from the PDB file, containing all chains with all residues\n",
    "    \"\"\"\n",
    "    \n",
    "    #AWESOME IMPLEMENTATION COMES HERE:\n",
    "    \n",
    "    # directly load full content in one list ('r' = read)\n",
    "    with open(filename, 'r') as file:\n",
    "        content = file.readlines()\n",
    "        \n",
    "    # define variables for everything we want to read, start with dummy/temporary names for the residue and chain\n",
    "    assembly = Assembly()\n",
    "    current_res = Residue(rname = 'dummy_res', rnum = None)\n",
    "    current_chain = Chain(cname = 'dummy_chain')\n",
    "    \n",
    "    # go through each describing an atom in the PDB file\n",
    "    for line in content:\n",
    "        \n",
    "        if line.startswith('ATOM'):\n",
    "            \n",
    "            # get all the non-empty values\n",
    "            line_values = line.split(' ')\n",
    "            line_values = [c for c in line_values if c != '']\n",
    "            aname = line_values[2].strip() #13:17\n",
    "\n",
    "            # we're only interested in those three backbone atoms\n",
    "            if aname in ['N', 'CA', 'C']:\n",
    "\n",
    "                # parse current line\n",
    "                rname = line_values[3] #[17:21]\n",
    "                rnum = line_values[5].strip() #23:27\n",
    "                cname = line_values[4] #21\n",
    "                x_coord = float(line_values[6].strip()) #30:38\n",
    "                y_coord = float(line_values[7].strip()) #38:46\n",
    "                z_coord = float(line_values[8].strip()) #46:54\n",
    "                \n",
    "                # check if there is a new residue\n",
    "                if rnum != current_res.rnum or cname != current_chain.cname:\n",
    "                    # we add the previous one to the current chain if all positions are valid\n",
    "                    if current_res.n_pos and current_res.ca_pos and current_res.c_pos:\n",
    "                        current_chain.residues.append(current_res)\n",
    "                    # and generate the new residue\n",
    "                    current_res = Residue(rname, rnum)\n",
    "                \n",
    "                # check if there is a new chain\n",
    "                if cname != current_chain.cname:\n",
    "                    # we add the previous one if it is not empty\n",
    "                    if len(current_chain.residues) > 0:\n",
    "                        assembly.chains.append(current_chain)\n",
    "                    # and generate the new chain\n",
    "                    current_chain = Chain(cname)\n",
    "                \n",
    "                # for the current residue, save the coordinates of the three backbone atoms as a Vec3 instance\n",
    "                if aname == 'N':\n",
    "                    current_res.n_pos = Vec3(x_coord, y_coord, z_coord)\n",
    "                if aname == 'CA':\n",
    "                    current_res.ca_pos = Vec3(x_coord, y_coord, z_coord)\n",
    "                if aname == 'C':\n",
    "                    current_res.c_pos = Vec3(x_coord, y_coord, z_coord)\n",
    "                    \n",
    "    # at the end, there still might be a complete residue and a complete chain that haven't\n",
    "    # been added to the assembly\n",
    "    if current_res.n_pos and current_res.ca_pos and current_res.c_pos:\n",
    "        current_chain.residues.append(current_res)\n",
    "    if len(current_chain.residues) > 0:\n",
    "        assembly.chains.append(current_chain)\n",
    "                \n",
    "    return assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cd8a95b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_aa(assembly: Assembly):\n",
    "    ''' Get the three-letter amino acid sequence from an Assembly\n",
    "    Warning: we assume here that our protein is a monomer (only one chain) and all the sequence is in the first chain\n",
    "    Parameters\n",
    "    ----------\n",
    "    assembly : Assembly\n",
    "        Assembly of a monomeric protein with the entire sequence in the first chain\n",
    "    Returns\n",
    "    -------\n",
    "    aa_sequence : string\n",
    "        The entire three-letter amino acid sequence from the first chain\n",
    "    '''\n",
    "\n",
    "    # initiate the value we want to compute\n",
    "    aa_sequence = ''\n",
    "\n",
    "    # extra only the sequence from the first chain\n",
    "    first_chain = assembly.chains[0]\n",
    "\n",
    "    for res in first_chain.residues:\n",
    "        aa_sequence += res.rname + ' '\n",
    "\n",
    "    return aa_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "160096ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "def convert_sequence_to_one_letter(three_letter_sequence: str, delimiter: str = ' '):\n",
    "    ''' Convert a three-letter code amino acid sequence, separated by the delimiter, to a one-letter code amino acid sequence\n",
    "    Parameters\n",
    "    ----------\n",
    "    three_letter_sequence : string\n",
    "        Three-letter code amino acid sequence to be converted, separated by the delimiter\n",
    "    delimiter : string\n",
    "        Delimiter for the three-letter code amino acid sequence\n",
    "    Returns\n",
    "    -------\n",
    "    one_letter_sequence : string\n",
    "        One-letter code amino acid sequence\n",
    "    '''\n",
    "\n",
    "    # define a dictionary mapping the three-letter codes to the one-letter codes\n",
    "    mapping = {\n",
    "    'ALA': 'A',\n",
    "    'ARG': 'R',\n",
    "    'ASN': 'N',\n",
    "    'ASP': 'D',\n",
    "    'CYS': 'C',\n",
    "    'GLN': 'Q',\n",
    "    'GLU': 'E',\n",
    "    'GLY': 'G',\n",
    "    'HIS': 'H',\n",
    "    'ILE': 'I',\n",
    "    'LEU': 'L',\n",
    "    'LYS': 'K',\n",
    "    'MET': 'M',\n",
    "    'PHE': 'F',\n",
    "    'PRO': 'P',\n",
    "    'SER': 'S',\n",
    "    'THR': 'T',\n",
    "    'TRP': 'W',\n",
    "    'TYR': 'Y',\n",
    "    'VAL': 'V'\n",
    "    }\n",
    "\n",
    "    # intialize the value we want to compute\n",
    "    one_letter_sequence = ''\n",
    "\n",
    "    # remove the empty spaces at the beginning and end of the full sequence\n",
    "    three_letter_sequence = three_letter_sequence.strip()\n",
    "\n",
    "    # process and map each three letter code in the sequence, otherwise raise an error\n",
    "    for three_letter_code in three_letter_sequence.split(delimiter):\n",
    "        three_letter_code = three_letter_code.upper().strip()\n",
    "\n",
    "        if three_letter_code in mapping:\n",
    "            one_letter_sequence += mapping[three_letter_code]\n",
    "        else:\n",
    "            raise KeyError(f\"Three-letter amino acid code '{three_letter_code}' not found.\")\n",
    "\n",
    "    return one_letter_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "504f470f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def GetAlignedCaCoords(assembly1: Assembly, assembly2: Assembly, alignment1: str, alignment2: str):\n",
    "    \"\"\" Given two assemblies and their alignment strings, return two arrays of the CA coordinates of aligned (not gaps) residues\n",
    "    Warning: we assume here that our proteins are monomers (only one chain) and all the sequences are in the first chains\n",
    "    Parameters\n",
    "    ----------\n",
    "    assembly1, assembly2 : Assembly\n",
    "        Protein assemblies from which to get the CA coordinates of the aligned residues\n",
    "    alignment1, alignment2 : string\n",
    "        Aligned sequences (by Needleman-Wunsch algorithm for example) of the two assembly sequences\n",
    "    Returns\n",
    "    -------\n",
    "    aligned_ca_coords1, aligned_ca_coords2 : numpy.array\n",
    "        Numpy arrays of the CA coordinates of the aligned (not gaps) residues of the two input assemblies\n",
    "    \"\"\"\n",
    "\n",
    "    # get the lists of all residues from the first chains in the assemblies\n",
    "    residues1 = [residue for residue in assembly1.chains[0].residues]\n",
    "    residues2 = [residue for residue in assembly2.chains[0].residues]\n",
    "\n",
    "    # Initialize the variables we want to define\n",
    "    aligned_ca_coords1 = []\n",
    "    aligned_ca_coords2 = []\n",
    "    res_index1 = 0\n",
    "    res_index2 = 0\n",
    "\n",
    "    # Iterate through the alignments\n",
    "    for a1, a2 in zip(alignment1, alignment2):\n",
    "        \n",
    "        #check for the gaps and length\n",
    "        if a1 != '-' and res_index1 < len(residues1):\n",
    "            residue1 = residues1[res_index1]\n",
    "            res_index1 += 1\n",
    "        else:\n",
    "            residue1 = None\n",
    "            \n",
    "        if a2 != '-' and res_index2 < len(residues2):\n",
    "            residue2 = residues2[res_index2]\n",
    "            res_index2 += 1\n",
    "        else:\n",
    "            residue2 = None\n",
    "        \n",
    "        # If both residues are not None, record CA coordinates\n",
    "        if residue1 is not None and residue2 is not None:\n",
    "            aligned_ca_coords1.append([residue1.ca_pos.x, residue1.ca_pos.y, residue1.ca_pos.z])\n",
    "            aligned_ca_coords2.append([residue2.ca_pos.x, residue2.ca_pos.y, residue2.ca_pos.z])\n",
    "\n",
    "    # Convert the lists to numpy arrays for easier manipulation.\n",
    "    aligned_ca_coords1 = np.array(aligned_ca_coords1)\n",
    "    aligned_ca_coords2 = np.array(aligned_ca_coords2)\n",
    "\n",
    "    return aligned_ca_coords1, aligned_ca_coords2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80bf4bbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "def kabsch_superimpose(P: np.ndarray, Q: np.ndarray):\n",
    "    \"\"\" Return the optimal rotation and translation matrices for superimposing the points P onto Q\n",
    "    Parameters\n",
    "    ----------\n",
    "    P, Q : numpy ndarray\n",
    "        Set of points: Nx3 matrix for N points in 3 dimensions to superimpose\n",
    "    Returns\n",
    "    -------\n",
    "    rotation : numpy ndarray\n",
    "        Optimal rotation matrix to align the set of point P and Q\n",
    "    translation : numpy array\n",
    "        Optimal translation vector to align the set of point P and Q\n",
    "    \"\"\"\n",
    "    # this assertion is to make sure the number of points in P and Q are the same\n",
    "    assert len(P) == len(Q)\n",
    "\n",
    "    # np.mean calculates the average of the points in P and Q along the axis specified (axis=0 = per-column)\n",
    "    # the centroid is the average position of all the points in each set\n",
    "    centroid_P = np.mean(P, axis=0)\n",
    "    centroid_Q = np.mean(Q, axis=0)\n",
    "\n",
    "    # subtract the centroid from each set of points\n",
    "    # this is done to move all points in both sets to the origin\n",
    "    # which is a standard procedure before applying the Kabsch algorithm\n",
    "    P_centered = P - centroid_P\n",
    "    Q_centered = Q - centroid_Q\n",
    "\n",
    "    # the dot product of the transpose of P and Q forms the covariance matrix\n",
    "    # np.dot is a function that computes the dot product of two arrays\n",
    "    H = np.dot(np.transpose(P_centered), Q_centered)\n",
    "\n",
    "    # use Singular Value Decomposition (SVD) to decompose H\n",
    "    # this is used to find the rotation matrix that will align the points in P with the points in Q\n",
    "    U, S, Vt = np.linalg.svd(H)\n",
    "\n",
    "    # To ensure a right-handed coordinate system, the determinant of U*Vt must be 1. If it's -1, we need to correct it\n",
    "    d = (np.linalg.det(U) * np.linalg.det(Vt)) < 0.0\n",
    "\n",
    "    if d:\n",
    "        S[-1] = -S[-1]\n",
    "        U[:, -1] = -U[:, -1]\n",
    "\n",
    "    # Calculate the rotation matrix as the product of U and Vt\n",
    "    rotation = np.dot(U, Vt)\n",
    "    \n",
    "    # The translation vector is calculated as the difference of centroids, transformed by the rotation matrix\n",
    "    translation = centroid_Q - np.dot(centroid_P, rotation)\n",
    "\n",
    "    return rotation, translation\n",
    "\n",
    "def rmsd(V: np.ndarray, W: np.ndarray):\n",
    "    \"\"\" Calculate Root-mean-square deviation (RMSD) from two sets of vectors V and W.\n",
    "    Parameters\n",
    "    ----------\n",
    "    V, W : numpy ndarray\n",
    "        (N,D) matrix, where N is points and D is dimension\n",
    "    Returns\n",
    "    -------\n",
    "    rmsd : float\n",
    "        Root-mean-square deviation\n",
    "    \"\"\"\n",
    "\n",
    "    # initialize the variables\n",
    "    N = len(V)\n",
    "    D = len(V[0])\n",
    "    \n",
    "    if len(W[0]) != D or len(W) != N:\n",
    "        raise ValueError(f\"Error: The input vectors V and W do not have the same dimensions.\")\n",
    "    \n",
    "    # compute the RMSD\n",
    "    rmsd = 0.0\n",
    "\n",
    "    for v, w in zip(V, W):\n",
    "        rmsd += sum([(v[i] - w[i])**2.0 for i in range(D)])\n",
    "\n",
    "    return np.sqrt(rmsd/N)\n",
    "\n",
    "def transform(v: np.ndarray, rotation: np.ndarray, translation: np.array):\n",
    "    \"\"\" Apply rotation and translation to the points v\n",
    "    Parameters\n",
    "    ----------\n",
    "    v : numpy ndarray\n",
    "        Matrix of points to be tranformed\n",
    "    rotation : numpy ndarray\n",
    "        Rotation matrix to transform v\n",
    "    translation : numpy array\n",
    "        Translation vector transform v\n",
    "    Returns\n",
    "    -------\n",
    "    numpy : ndarray\n",
    "        Tranformed matrix of v\n",
    "    \"\"\"\n",
    "    return np.dot(v, rotation) + translation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30e7c0c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Example usage:\n",
    "\n",
    "# These are random examples\n",
    "P = np.random.rand(10,3)*100 \n",
    "Q = np.random.rand(10,3)*100  \n",
    "\n",
    "rotation, translation = kabsch_superimpose(P, Q)\n",
    "\n",
    "# Apply the rotation and translation to P\n",
    "P_tr = transform(P, rotation=rotation, translation=translation)\n",
    "\n",
    "print('RMSD:', rmsd(P_tr, Q))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd78a828",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_superposition(P: np.ndarray, Q: np.ndarray):\n",
    "    \"\"\" Combine all previous functions to get the RMSD of the superposition of P and Q\n",
    "    Parameters\n",
    "    ----------\n",
    "    P, Q : numpy ndarray\n",
    "        Set of points: Nx3 matrix for N points in 3 dimensions to superimpose\n",
    "    Returns\n",
    "    -------\n",
    "    rmsd : float\n",
    "        Root-mean-square deviation of the superimposed points P and Q\n",
    "    \"\"\"\n",
    "    \n",
    "    rotation, translation = kabsch_superimpose(P, Q)\n",
    "    P = transform(P, rotation, translation)\n",
    "\n",
    "    return rmsd(P, Q)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aceacdf6",
   "metadata": {},
   "outputs": [],
   "source": [
    "rmsd_value = get_superposition(P, Q)\n",
    "print('RMSD:', rmsd_value)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10f50d7a",
   "metadata": {},
   "source": [
    "Exercise: \n",
    "\n",
    "From the reference .pdb files for open (4ake) and closed (1ake) conformations, align their structures, get the matrices of the CA coordinates and print the RMSD."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d648585f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get the assembly objects from the PDB files\n",
    "assembly_open = LoadPDB('4ake.pdb')\n",
    "assembly_close = LoadPDB('1ake.pdb')\n",
    "\n",
    "# get the three-letter code amino acid sequence from the assemblies and convert them to one-letter code sequences\n",
    "open_three_letter = get_aa(assembly_open)\n",
    "close_three_letter = get_aa(assembly_close)\n",
    "\n",
    "open_one_letter = convert_sequence_to_one_letter(open_three_letter)\n",
    "close_one_letter = convert_sequence_to_one_letter(close_three_letter)\n",
    "\n",
    "# align the one-letter code sequences\n",
    "aln_open, aln_close = Align(open_one_letter, close_one_letter)\n",
    "\n",
    "print('Aligned sequence of the open structure:\\n', aln_open)\n",
    "print('Aligned sequence of the closed structure:\\n', aln_close)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1fb50b05",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get the arrays of the CA coordinates of aligned residues\n",
    "ca_coord_open, ca_coord_close = GetAlignedCaCoords(assembly_open, assembly_close, aln_open, aln_close)\n",
    "\n",
    "# compute and print the RMSD value of the superposition\n",
    "rmsd_value = get_superposition(ca_coord_open, ca_coord_close)\n",
    "print('RMSD:', rmsd_value)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c29d2e6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# you  can also print the matrices to see if it matches the coordinates of ca atoms in the pdb files\n",
    "print(ca_coord_close)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bf745edf",
   "metadata": {},
   "source": [
    "Lecture 1: Introduction to alignments - exercise: align two homologous kinases (given as fasta files)\n",
    "\n",
    "Lecture 2: Introduction to superpositions - exercise: get RMSD of two structures\n",
    "\n",
    "<b>Lecture 3: Introduction to AlphaFold - exercise: get models from AlphaFold and classify them as open/close, do the same for models from Swiss-Model and AlphaFold-dropout</b>\n",
    "\n",
    "Lecture 4: Ligand docking (DiffDock) - exercise: get docking from DiffDock colab, find interacting atoms and visualize docking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "447155c2",
   "metadata": {},
   "source": [
    "### 3. Lecture: Alphafold modelling, alignments, superposition"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8fc82189",
   "metadata": {},
   "source": [
    "In this exercise, you will use functions from previous two exercises for the comparison of predicted structures to experimental structures. \n",
    "As experimental structures you will use open and close conformation of the adenylate kinase (pdb: 4ake/1ake). The task is to classify predicted structures as open or close conformation based on the rmsd to experimental structures. You will use models from different methods:\n",
    "\n",
    "1. Unmodified AlphaFold2 (5 models) - folder \"./alphafold_models_standard\"\n",
    "2. AlphaFold2 with dropout layers (40 models) - folder \"./alphafold_models_dropout\"\n",
    "3. Swiss-Model (21 models) - folder \"./swiss_models\"\n",
    "\n",
    "Explain the results!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8740d1e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# combined function from the previous exercise\n",
    "\n",
    "def compare_with_assembly(model_assembly: Assembly, exp_assembly: Assembly):\n",
    "    \"\"\" Compare a modeled assembly to the experimentally determined assembly\n",
    "    Parameters\n",
    "    ----------\n",
    "    model_assembly, exp_assembly : Assembly\n",
    "        Structure of a protein, saved as an Assembly object\n",
    "    Returns\n",
    "    -------\n",
    "    rmsd_value : float\n",
    "        Root-mean-square deviation of the superimposed of the model and experimental proteins\n",
    "    \"\"\"\n",
    "\n",
    "    # use the same methods as in exercise 2 to get the RMSD of the two assemblies\n",
    "    model_aa = get_aa(model_assembly)\n",
    "    exp_aa = get_aa(exp_assembly)\n",
    "    \n",
    "    model_one_letter = convert_sequence_to_one_letter(model_aa)\n",
    "    exp_one_letter = convert_sequence_to_one_letter(exp_aa)\n",
    "    \n",
    "    model_aln, exp_aln = Align(model_one_letter, exp_one_letter)\n",
    "    \n",
    "    model_aligned_ca_coords, exp_aligned_ca_coords = GetAlignedCaCoords(model_assembly, exp_assembly, model_aln, exp_aln)\n",
    "    \n",
    "    rmsd_value = get_superposition(model_aligned_ca_coords, exp_aligned_ca_coords)\n",
    "    \n",
    "    return rmsd_value"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca485d42",
   "metadata": {},
   "source": [
    "Classify the models obtained with standard AlphaFold to open or closed conformation using the experimentally determined structures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73046368",
   "metadata": {},
   "outputs": [],
   "source": [
    "#to visualize results later we will save them in a list of lists\n",
    "data = list()\n",
    "\n",
    "col_names = ['model', 'rmsd_open', 'rmsd_close', 'label', 'method'] \n",
    "\n",
    "data.append(col_names)\n",
    "\n",
    "# the os package allows you to easily navigate folders and files with the use of paths\n",
    "import os\n",
    "\n",
    "alphafold_model_dir = './alphafold_models_standard'\n",
    "open_exp_structure = LoadPDB('4ake.pdb')\n",
    "close_exp_structure = LoadPDB('1ake.pdb')\n",
    "\n",
    "# get the list of all files in a directory\n",
    "for f in os.listdir(alphafold_model_dir):\n",
    "    method = \"AF_vanilla\"\n",
    "\n",
    "    # get the full path of the file f in the directory\n",
    "    path = os.path.join(alphafold_model_dir, f)\n",
    "\n",
    "    # get the assembly of the structure using the PDB file saved from AlphaFold\n",
    "    af_assembly = LoadPDB(path)\n",
    "\n",
    "    # compare the RMSD of the AF-structure with the experimentally-determined open and closed stuctures\n",
    "    rmsd_open = compare_with_assembly(af_assembly, open_exp_structure)\n",
    "    rmsd_close = compare_with_assembly(af_assembly, close_exp_structure)\n",
    "    \n",
    "    print(f)\n",
    "    print('RMSD_open:', rmsd_open)\n",
    "    print('RMSD_close:', rmsd_close,'\\n')\n",
    "\n",
    "    if rmsd_open < rmsd_close:\n",
    "        label = 0\n",
    "        \n",
    "    if rmsd_open > rmsd_close:\n",
    "        label = 1\n",
    "\n",
    "    data.append([f, rmsd_open, rmsd_close, label, method])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a893e152",
   "metadata": {},
   "outputs": [],
   "source": [
    "### use matplotlib library to visualize results\n",
    "\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5269325d",
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(10, 6))\n",
    "\n",
    "#Extract names of columns\n",
    "columns = data[0]\n",
    "\n",
    "# Extract the values\n",
    "rmsd_open = [row[1] for row in data[1:]]\n",
    "rmsd_close = [row[2] for row in data[1:]]\n",
    "labels = [row[3] for row in data[1:]]\n",
    "method = [row[4] for row in data[1:]]\n",
    "\n",
    "# Map the values to method (we only have 1 for now)\n",
    "vanilla_rmsd_open = [rmsd_open[i] for i in range(len(method)) if method[i] == \"AF_vanilla\"]\n",
    "vanilla_rmsd_close = [rmsd_close[i] for i in range(len(method)) if method[i] == \"AF_vanilla\"]\n",
    "\n",
    "# Add diagonal and set axis limits to start from 0\n",
    "max_value = max(vanilla_rmsd_close + vanilla_rmsd_open) + 0.2\n",
    "plt.plot([0, max_value], [0, max_value], linestyle='--', color='gray')\n",
    "\n",
    "plt.xlim(0, max_value)\n",
    "plt.ylim(0, max_value)\n",
    "\n",
    "# Plot data\n",
    "plt.scatter(vanilla_rmsd_open, vanilla_rmsd_close, label='AF-vanilla', alpha=0.7, marker='o')\n",
    "\n",
    "# Add titles and labels\n",
    "plt.title('RMSD Open vs. RMSD Close - unmodified AlphaFold2')\n",
    "plt.xlabel('RMSD to 4AKE (0pen conf)')\n",
    "plt.ylabel('RMSD to 1AKE (Close conf)')\n",
    "plt.legend()\n",
    "plt.grid(True)\n",
    "\n",
    "# Show the plot\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1b0c622",
   "metadata": {},
   "outputs": [],
   "source": [
    "### we can count the labels of open/close to confirm\n",
    "vanilla_labels = [labels[i] for i in range(len(method)) if method[i] == \"AF_vanilla\"]\n",
    "\n",
    "print(\"Vanilla-AF\")\n",
    "print(\"Models with close conformation:\", vanilla_labels.count(1))\n",
    "print(\"Models with open conformation:\", vanilla_labels.count(0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d7c69e60",
   "metadata": {},
   "outputs": [],
   "source": [
    "### TODO: explain the results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "631dde49",
   "metadata": {},
   "source": [
    "Now classify the models obtained with AlphaFold dropout and Swiss-Model as open or closed conformations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d8465ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "### TODO: classify models from AF2-dropout method (folder: './alphafold_models_dropout') as open/close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04305621",
   "metadata": {},
   "outputs": [],
   "source": [
    "### TODO: classify models from Swiss-Model method (folder: './alphafold_models_dropout') as open/close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef5a9280",
   "metadata": {},
   "outputs": [],
   "source": [
    "### TODO: visualize the results and explain your observations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df7b0146",
   "metadata": {},
   "source": [
    "Lecture 1: Introduction to alignments - exercise: align two homologous kinases (given as fasta files)\n",
    "\n",
    "Lecture 2: Introduction to superpositions - exercise: get RMSD of two structures\n",
    "\n",
    "Lecture 3: Introduction to AlphaFold - exercise: get models from AlphaFold and classify them as open/close, do the same for models from Swiss-Model and AlphaFold-dropout\n",
    "\n",
    "<b>Lecture 4: Ligand docking (DiffDock) - exercise: get docking from DiffDock colab, find interacting atoms and visualize docking</b>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21954283",
   "metadata": {},
   "source": [
    "### 4. Lecture: Introduction to ligand docking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ee6c19b",
   "metadata": {},
   "source": [
    "This exercise is a demonstration of how the predicted models can be used in drug discovery, explaining protein function, enzymology etc. You will visualize the results of ligand docking (in our case ligand is ATP) to open/close conformation of adenylate kinase and make conclusions about the final results."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33e82b95",
   "metadata": {},
   "source": [
    "Some useful functions:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "99fda47b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def LoadSDF(filename: str):\n",
    "    \"\"\" Load all ligand atom coordinates from an SDF file\n",
    "    Parameters\n",
    "    ----------\n",
    "    filename : string\n",
    "        Filename to read, sdf file format\n",
    "    Returns\n",
    "    -------\n",
    "    assembly : Assembly\n",
    "        Assembly object of the ligand\n",
    "    \"\"\"\n",
    "\n",
    "    # load full content in one list  ('r' = read)\n",
    "    with open(filename, 'r') as file:\n",
    "        content = file.readlines()\n",
    "        \n",
    "    # define variables for everything we want to read\n",
    "    assembly = Assembly()\n",
    "    current_res = Residue('dummy_res', None)\n",
    "    current_chain = Chain('dummy_chain')    \n",
    "    count=0\n",
    "\n",
    "    for line in content:\n",
    "\n",
    "        # get all the non-empty values\n",
    "        line_values = line.split(' ')\n",
    "        line_values = [c for c in line_values if c != '']\n",
    "\n",
    "        # we're only interested in lines describing atoms\n",
    "        if len(line_values) >= 4 and line_values[3] in ['N', 'C', 'O', 'P']:\n",
    "            \n",
    "            # parse current line\n",
    "            aname = str(line_values[3])\n",
    "            rname = str(count)\n",
    "            rnum = str(count)\n",
    "            cname = \"0\"\n",
    "            x_coord = float(line_values[0].strip())\n",
    "            y_coord = float(line_values[1].strip())\n",
    "            z_coord = float(line_values[2].strip())\n",
    "            count += 1\n",
    "                \n",
    "            # check if there is a new residue\n",
    "            if rnum != current_res.rnum or cname != current_chain.cname:\n",
    "                # we add it to the current chain if position is valid\n",
    "                if current_res.ca_pos:\n",
    "                    current_chain.residues.append(current_res)\n",
    "                # and generate the new residue\n",
    "                current_res = Residue(rname, rnum)\n",
    "            \n",
    "            # check if there is a new chain\n",
    "            if cname != current_chain.cname:\n",
    "                # its a new chain! add if not empty\n",
    "                if len(current_chain.residues) > 0:\n",
    "                    assembly.chains.append(current_chain)\n",
    "                # and generate the new chain\n",
    "                current_chain = Chain(cname)\n",
    "\n",
    "            # add the position of the atom to the current residue (save in the ca_pos for retrieval)\n",
    "            current_res.ca_pos = Vec3(x_coord, y_coord, z_coord)\n",
    "            current_res.n_pos = Vec3(None, None, None)\n",
    "            current_res.c_pos = Vec3(None, None, None)\n",
    "                    \n",
    "    # there still might be a complete residue and a complete chain, that havent\n",
    "    # been added to the assembly\n",
    "    if current_res.ca_pos:\n",
    "        current_chain.residues.append(current_res)\n",
    "    if len(current_chain.residues) > 0:\n",
    "        assembly.chains.append(current_chain)\n",
    "                \n",
    "    return assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "84442fda",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_coords(protein_file: str):\n",
    "    \"\"\" Load all atom coordinates of a protein from a PDB file\n",
    "    Parameters\n",
    "    ----------\n",
    "    protein : string\n",
    "        Filename to read, pdb file format\n",
    "    Returns\n",
    "    -------\n",
    "    protein_coords : dict[int, dict[str, list]]\n",
    "        Dictionary of all chains (line[21]), each containing a dictionary of the coordinates [x,y,z] for all residues (line[23:27])\n",
    "    \"\"\"\n",
    "\n",
    "    # load full content in one list  ('r' = read)\n",
    "    with open(protein_file, 'r') as file:\n",
    "        content = file.readlines()\n",
    "    \n",
    "    # define variables for everything we want to read\n",
    "    protein_coords = {}\n",
    "\n",
    "    for line in content:\n",
    "        \n",
    "        if line.startswith('ATOM'):\n",
    "\n",
    "            # for each new chain, add a new entry in the dictionary\n",
    "            if line[21] not in protein_coords:\n",
    "                protein_coords[line[21]] = {}\n",
    "\n",
    "            # for each new residue of a chain, add a new entry in the dictionary\n",
    "            if line[23:27].strip() not in protein_coords[line[21]]:\n",
    "                protein_coords[line[21]][line[23:27].strip()] = [[float(line[30:38].strip()),float(line[38:46].strip()),float(line[46:54].strip())]]\n",
    "            # for already started residues, add the new atom coordinates to the dictionary entry\n",
    "            else:\n",
    "                protein_coords[line[21]][line[23:27].strip()].append([float(line[30:38].strip()),float(line[38:46].strip()),float(line[46:54].strip())])   \n",
    "\n",
    "    return protein_coords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "62fd31af",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import PyMOL python package, if module not available, run 'pip install py3Dmol'\n",
    "import py3Dmol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8475f2c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def display_protein_ligand(file_protein: str, binding_residues_idx: list([int]), file_ligand: str, main_chain: str = ''):\n",
    "    \"\"\" Display a protein, its docked ligand and the highlighted binding residues between the two instances\n",
    "    Parameters\n",
    "    ----------\n",
    "    file_protein : string\n",
    "        Filename of protein to display, pdb file format\n",
    "    binding_residues_idx : list\n",
    "        List of all binding residue indices of the protein\n",
    "    file_ligand : string\n",
    "        Filename of ligand to display, sdf file format\n",
    "    main_chain : string\n",
    "        Identifier of which chain of the protein should be displayed, one of either ['A', 'B']\n",
    "    Returns\n",
    "    -------\n",
    "    view : 3Dmol viewer\n",
    "        Interactive window with all modules displayed\n",
    "    \"\"\"\n",
    "\n",
    "    # define the viewer window size and camera parameters: fov = field of view, z = distance to the origin\n",
    "    view = py3Dmol.view(width=800, height=800)\n",
    "    view.setCameraParameters({'fov': 35, 'z': 100})\n",
    "\n",
    "    # display the ligand in red\n",
    "    view.addModel(open(file_ligand).read(), \"sdf\")\n",
    "    view.setStyle({\"model\": 0}, {'stick':{\"color\":\"red\"}})\n",
    "    view.zoomTo()\n",
    "\n",
    "    # display the main chain of the protein\n",
    "    chains = set(['A', 'B'])\n",
    "    chain_remove = list(chains - {main_chain})[0]\n",
    "    # only use lines containing atom information\n",
    "    with open(file_protein) as file:\n",
    "        pdb_atom = \"\".join([x for x in file if x.startswith('ATOM')])\n",
    "    view.addModel(pdb_atom, \"pdb\")\n",
    "    # hide the other chain\n",
    "    view.setStyle({\"chain\": chain_remove}, {'style':{}})\n",
    "    # display the main chain in transparent grey\n",
    "    view.setStyle({\"chain\": main_chain}, {'cartoon':{\"color\":\"grey\",\"opacity\":0.75}})\n",
    "    # display the main chain binding residues in green\n",
    "    view.setStyle({\"chain\": main_chain, \"resi\":binding_residues_idx}, {'stick':{\"color\":\"green\"}, 'cartoon':{\"color\":\"grey\",\"opacity\":0.75}})\n",
    "    \n",
    "    \n",
    "    return view"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8e4cb6fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_binding_residues(protein_coords: dict(), ligand: Assembly, cutoff: float = 4.5):\n",
    "    \"\"\" Compute the binding residues of the protein to the ligand\n",
    "    Parameters\n",
    "    ----------\n",
    "    protein_coords : dict[int, dict[str, list]]\n",
    "        Dictionary of all chains, each containing a dictionary of the coordinates [x,y,z] for all residues\n",
    "    ligand : Assembly\n",
    "        Assembly object of the ligand\n",
    "    cutoff : float\n",
    "        Cutoff distance in Angstroms between atoms below which atoms are considered to be binding to the ligand\n",
    "    Returns\n",
    "    -------\n",
    "    binding_residues_idx : list[int]\n",
    "        List of all indices of residues of the protein that bind to the ligand\n",
    "    \"\"\"\n",
    "\n",
    "    # define variable for the data we want to read\n",
    "    binding_residues_idx = list()\n",
    "    \n",
    "    for chain in protein_coords:\n",
    "\n",
    "        # define a new list for each chain\n",
    "        binding_residues_idx_ch = list()\n",
    "\n",
    "        for res in protein_coords[chain]:\n",
    "\n",
    "            # define a new list of distances for each residue\n",
    "            distances = list()\n",
    "\n",
    "            for atom in ligand.chains[0].residues:\n",
    "                ligand_atom_coords = np.array([i for i in atom.get_coords() if i is not None])\n",
    "\n",
    "                # get the smallest distance between the chain's residues and the ligand atoms\n",
    "                dist = np.inf\n",
    "\n",
    "                for r in range(len(protein_coords[chain][res])):\n",
    "\n",
    "                    # compute the Euclidean distance in 3D between the atoms in the chain's residues and the ligand atoms \n",
    "                    diff_vector = np.subtract(np.array(protein_coords[chain][res][r]), ligand_atom_coords)\n",
    "                    dist_new = np.sqrt(np.sum(diff_vector * diff_vector))\n",
    "\n",
    "                    if dist_new <= dist:\n",
    "                        dist = dist_new\n",
    "\n",
    "                distances.append(dist)\n",
    "\n",
    "            # we save the index of the residue if the smallest distance between any atom of the residue and ligand is smaller than the cutoff distance\n",
    "            if min(distances) < cutoff:\n",
    "                binding_residues_idx_ch.append(int(res))\n",
    "\n",
    "        # we save the chain with the highest number of binding residues\n",
    "        if len(binding_residues_idx_ch) >= len(binding_residues_idx):\n",
    "            binding_residues_idx = binding_residues_idx_ch\n",
    "\n",
    "    return binding_residues_idx"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e7f6d43",
   "metadata": {},
   "source": [
    "Analyze the ligand docking on the open conformation (best model from the last exercise)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "106f6059",
   "metadata": {},
   "outputs": [],
   "source": [
    "best_model_open = 'model_07.pdb' # enter closest model to open conformation - model_07.pdb\n",
    "# we take the ligand model with the highest rank (1), e.g. rank1_confidence-2.76.sdf\n",
    "ligand_file_open = '/Users/lucas/Desktop/School/UniversitaÌˆt Basel/3. Semester - HS 2025/Programming in Bioinformatics 2025/2. Schwede Daniil Wei/Exercise 8/Downloaded Results from Adam (Template)/folders/diffdock_results_open/docking/rank1_confidence-2.76.sdf'\n",
    "ligand_open = LoadSDF(ligand_file_open)\n",
    "\n",
    "# we take the protein used for the docking \n",
    "file_protein_open = f\"/Users/lucas/Desktop/School/UniversitaÌˆt Basel/3. Semester - HS 2025/Programming in Bioinformatics 2025/2. Schwede Daniil Wei/Exercise 8/Downloaded Results from Adam (Template)/folders/diffdock_results_open/{best_model_open}\"\n",
    "protein_coords_open = get_coords(file_protein_open)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "36efbbcd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[5, 6, 7, 8, 9, 10, 12, 13, 14, 28, 84, 120, 123, 158, 167]\n"
     ]
    }
   ],
   "source": [
    "### TODO: get the binding residues for the complex with open conformation\n",
    "\n",
    "binding_residues = get_binding_residues(protein_coords_open, ligand_open)\n",
    "print(binding_residues)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a5dd8e08",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/3dmoljs_load.v0": "<div id=\"3dmolviewer_1762785482869049\"  style=\"position: relative; width: 800px; height: 800px;\">\n        <p id=\"3dmolwarning_1762785482869049\" style=\"background-color:#ffcccc;color:black\">3Dmol.js failed to load for some reason.  Please check your browser console for error messages.<br></p>\n        </div>\n<script>\n\nvar loadScriptAsync = function(uri){\n  return new Promise((resolve, reject) => {\n    //this is to ignore the existence of requirejs amd\n    var savedexports, savedmodule;\n    if (typeof exports !== 'undefined') savedexports = exports;\n    else exports = {}\n    if (typeof module !== 'undefined') savedmodule = module;\n    else module = {}\n\n    var tag = document.createElement('script');\n    tag.src = uri;\n    tag.async = true;\n    tag.onload = () => {\n        exports = savedexports;\n        module = savedmodule;\n        resolve();\n    };\n  var firstScriptTag = document.getElementsByTagName('script')[0];\n  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n});\n};\n\nif(typeof $3Dmolpromise === 'undefined') {\n$3Dmolpromise = null;\n  $3Dmolpromise = loadScriptAsync('https://cdn.jsdelivr.net/npm/3dmol@2.5.3/build/3Dmol-min.js');\n}\n\nvar viewer_1762785482869049 = null;\nvar warn = document.getElementById(\"3dmolwarning_1762785482869049\");\nif(warn) {\n    warn.parentNode.removeChild(warn);\n}\n$3Dmolpromise.then(function() {\nviewer_1762785482869049 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_1762785482869049\"),{backgroundColor:\"white\"});\nviewer_1762785482869049.zoomTo();\n\tviewer_1762785482869049.setCameraParameters({\"fov\": 35, \"z\": 100});\n\tviewer_1762785482869049.addModel(\"\\n     RDKit          3D\\n\\n 31 33  0  0  0  0  0  0  0  0999 V2000\\n   10.6521   10.3343   14.0011 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   10.1833   11.3892   14.6904 N   0  0  0  0  0  0  0  0  0  0  0  0\\n   10.6380   12.6361   14.5690 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   11.6702   12.8291   13.6578 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.1618   11.7567   12.9432 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   11.6487   10.5360   13.1250 N   0  0  0  0  0  0  0  0  0  0  0  0\\n   13.1405   12.1864   12.1387 N   0  0  0  0  0  0  0  0  0  0  0  0\\n   13.2527   13.5084   12.3533 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.3594   13.9108   13.2791 N   0  0  0  0  0  0  0  0  0  0  0  0\\n   13.9404   11.4328   11.2147 C   0  0  1  0  0  0  0  0  0  0  0  0\\n   13.0970   10.5567   10.3101 C   0  0  1  0  0  0  0  0  0  0  0  0\\n   14.0970   10.2330    9.2200 C   0  0  1  0  0  0  0  0  0  0  0  0\\n   15.1022   11.3646    9.3602 C   0  0  2  0  0  0  0  0  0  0  0  0\\n   14.5365   12.2749   10.2455 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   15.3278   11.9485    7.9957 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   14.1414   12.3144    7.3719 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   14.1306   13.9155    6.8790 P   0  0  1  0  0  5  0  0  0  0  0  0\\n   15.3201   14.5881    7.5527 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   14.3876   14.0796    5.2137 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.7538   14.7495    7.3818 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.5295   16.1870    6.5309 P   0  0  2  0  0  5  0  0  0  0  0  0\\n   13.8242   16.9944    6.5506 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   11.2268   17.0794    7.0973 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.2339   15.8127    4.9069 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.2035   17.1749    3.9374 P   0  0  0  0  0  5  0  0  0  0  0  0\\n   13.4776   17.9913    4.0286 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   11.8921   16.7016    2.3363 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   10.8691   18.1237    4.4366 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   14.7604    9.0468    9.4777 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.6152    9.4366   10.9550 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   10.1436   13.7407   15.2941 N   0  0  0  0  0  0  0  0  0  0  0  0\\n  1  2  2  0\\n  2  3  1  0\\n  3  4  2  0\\n  4  5  1  0\\n  5  6  2  0\\n  5  7  1  0\\n  7  8  1  0\\n  8  9  2  0\\n 10  7  1  1\\n 10 11  1  0\\n 11 12  1  0\\n 12 13  1  0\\n 13 14  1  0\\n 13 15  1  6\\n 15 16  1  0\\n 16 17  1  0\\n 17 18  2  0\\n 17 19  1  6\\n 17 20  1  0\\n 20 21  1  0\\n 21 22  2  0\\n 21 23  1  1\\n 21 24  1  0\\n 24 25  1  0\\n 25 26  2  0\\n 25 27  1  0\\n 25 28  1  0\\n 12 29  1  1\\n 11 30  1  1\\n  3 31  1  0\\n  6  1  1  0\\n  9  4  1  0\\n 14 10  1  0\\nM  END\\n$$$$\\n\",\"sdf\");\n\tviewer_1762785482869049.setStyle({\"model\": 0},{\"stick\": {\"color\": \"red\"}});\n\tviewer_1762785482869049.zoomTo();\n\tviewer_1762785482869049.addModel(\"ATOM      1  N   MET A   1      -0.707   5.264  27.934  1.00  0.84           N  \\nATOM      2  CA  MET A   1       0.264   6.383  27.765  1.00  0.84           C  \\nATOM      3  C   MET A   1       0.229   7.063  26.413  1.00  0.84           C  \\nATOM      4  O   MET A   1       0.446   6.432  25.379  1.00  0.84           O  \\nATOM      5  CB  MET A   1       1.689   5.853  28.038  1.00  0.84           C  \\nATOM      6  CG  MET A   1       2.845   6.837  27.760  1.00  0.84           C  \\nATOM      7  SD  MET A   1       3.113   8.251  28.874  1.00  0.84           S  \\nATOM      8  CE  MET A   1       1.860   8.129  30.182  1.00  0.84           C  \\nATOM      9  N   ARG A   2      -0.057   8.357  26.364  1.00  0.83           N  \\nATOM     10  CA  ARG A   2      -0.271   9.136  25.174  1.00  0.83           C  \\nATOM     11  C   ARG A   2       0.821  10.187  25.088  1.00  0.83           C  \\nATOM     12  O   ARG A   2       0.955  11.051  25.951  1.00  0.83           O  \\nATOM     13  CB  ARG A   2      -1.662   9.798  25.293  1.00  0.83           C  \\nATOM     14  CG  ARG A   2      -2.814   8.789  25.500  1.00  0.83           C  \\nATOM     15  CD  ARG A   2      -4.011   9.369  26.262  1.00  0.83           C  \\nATOM     16  NE  ARG A   2      -4.715  10.321  25.357  1.00  0.83           N  \\nATOM     17  CZ  ARG A   2      -5.729  10.021  24.526  1.00  0.83           C  \\nATOM     18  NH1 ARG A   2      -6.216   8.798  24.395  1.00  0.83           N  \\nATOM     19  NH2 ARG A   2      -6.235  11.025  23.812  1.00  0.83           N  \\nATOM     20  N   ILE A   3       1.662  10.110  24.045  1.00  0.95           N  \\nATOM     21  CA  ILE A   3       2.851  10.935  23.939  1.00  0.95           C  \\nATOM     22  C   ILE A   3       2.838  11.666  22.614  1.00  0.95           C  \\nATOM     23  O   ILE A   3       2.588  11.083  21.565  1.00  0.95           O  \\nATOM     24  CB  ILE A   3       4.130  10.105  24.079  1.00  0.95           C  \\nATOM     25  CG1 ILE A   3       4.218   9.476  25.485  1.00  0.95           C  \\nATOM     26  CG2 ILE A   3       5.387  10.958  23.817  1.00  0.95           C  \\nATOM     27  CD1 ILE A   3       5.269   8.369  25.609  1.00  0.95           C  \\nATOM     28  N   ILE A   4       3.124  12.980  22.639  1.00  0.95           N  \\nATOM     29  CA  ILE A   4       3.469  13.750  21.458  1.00  0.95           C  \\nATOM     30  C   ILE A   4       4.976  13.891  21.462  1.00  0.95           C  \\nATOM     31  O   ILE A   4       5.582  14.290  22.454  1.00  0.95           O  \\nATOM     32  CB  ILE A   4       2.787  15.114  21.428  1.00  0.95           C  \\nATOM     33  CG1 ILE A   4       1.297  14.924  21.088  1.00  0.95           C  \\nATOM     34  CG2 ILE A   4       3.440  16.059  20.400  1.00  0.95           C  \\nATOM     35  CD1 ILE A   4       0.440  16.165  21.345  1.00  0.95           C  \\nATOM     36  N   LEU A   5       5.625  13.543  20.340  1.00  0.94           N  \\nATOM     37  CA  LEU A   5       7.059  13.667  20.205  1.00  0.94           C  \\nATOM     38  C   LEU A   5       7.365  14.698  19.126  1.00  0.94           C  \\nATOM     39  O   LEU A   5       7.021  14.544  17.955  1.00  0.94           O  \\nATOM     40  CB  LEU A   5       7.683  12.298  19.874  1.00  0.94           C  \\nATOM     41  CG  LEU A   5       9.218  12.250  19.775  1.00  0.94           C  \\nATOM     42  CD1 LEU A   5       9.935  12.652  21.071  1.00  0.94           C  \\nATOM     43  CD2 LEU A   5       9.654  10.846  19.339  1.00  0.94           C  \\nATOM     44  N   LEU A   6       8.016  15.809  19.517  1.00  0.92           N  \\nATOM     45  CA  LEU A   6       8.330  16.918  18.638  1.00  0.92           C  \\nATOM     46  C   LEU A   6       9.827  16.961  18.409  1.00  0.92           C  \\nATOM     47  O   LEU A   6      10.626  16.514  19.222  1.00  0.92           O  \\nATOM     48  CB  LEU A   6       7.895  18.282  19.226  1.00  0.92           C  \\nATOM     49  CG  LEU A   6       6.400  18.390  19.577  1.00  0.92           C  \\nATOM     50  CD1 LEU A   6       6.091  19.775  20.160  1.00  0.92           C  \\nATOM     51  CD2 LEU A   6       5.480  18.089  18.388  1.00  0.92           C  \\nATOM     52  N   GLY A   7      10.268  17.491  17.257  1.00  0.93           N  \\nATOM     53  CA  GLY A   7      11.694  17.583  17.001  1.00  0.93           C  \\nATOM     54  C   GLY A   7      11.928  18.120  15.625  1.00  0.93           C  \\nATOM     55  O   GLY A   7      11.191  17.815  14.690  1.00  0.93           O  \\nATOM     56  N   ALA A   8      12.976  18.951  15.484  1.00  0.95           N  \\nATOM     57  CA  ALA A   8      13.364  19.596  14.249  1.00  0.95           C  \\nATOM     58  C   ALA A   8      13.677  18.618  13.110  1.00  0.95           C  \\nATOM     59  O   ALA A   8      14.126  17.495  13.376  1.00  0.95           O  \\nATOM     60  CB  ALA A   8      14.550  20.541  14.530  1.00  0.95           C  \\nATOM     61  N   PRO A   9      13.433  18.937  11.830  1.00  0.90           N  \\nATOM     62  CA  PRO A   9      13.850  18.086  10.722  1.00  0.90           C  \\nATOM     63  C   PRO A   9      15.341  17.815  10.751  1.00  0.90           C  \\nATOM     64  O   PRO A   9      16.148  18.739  10.639  1.00  0.90           O  \\nATOM     65  CB  PRO A   9      13.400  18.846   9.464  1.00  0.90           C  \\nATOM     66  CG  PRO A   9      13.403  20.311   9.898  1.00  0.90           C  \\nATOM     67  CD  PRO A   9      12.964  20.242  11.359  1.00  0.90           C  \\nATOM     68  N   GLY A  10      15.730  16.539  10.870  1.00  0.86           N  \\nATOM     69  CA  GLY A  10      17.128  16.157  10.977  1.00  0.86           C  \\nATOM     70  C   GLY A  10      17.620  15.900  12.363  1.00  0.86           C  \\nATOM     71  O   GLY A  10      18.674  15.270  12.509  1.00  0.86           O  \\nATOM     72  N   ALA A  11      16.885  16.278  13.414  1.00  0.89           N  \\nATOM     73  CA  ALA A  11      17.268  16.014  14.794  1.00  0.89           C  \\nATOM     74  C   ALA A  11      17.355  14.525  15.073  1.00  0.89           C  \\nATOM     75  O   ALA A  11      18.306  14.038  15.723  1.00  0.89           O  \\nATOM     76  CB  ALA A  11      16.314  16.732  15.768  1.00  0.89           C  \\nATOM     77  N   GLY A  12      16.434  13.716  14.534  1.00  0.80           N  \\nATOM     78  CA  GLY A  12      16.474  12.268  14.694  1.00  0.80           C  \\nATOM     79  C   GLY A  12      15.686  11.741  15.852  1.00  0.80           C  \\nATOM     80  O   GLY A  12      16.034  10.686  16.389  1.00  0.80           O  \\nATOM     81  N   LYS A  13      14.572  12.374  16.227  1.00  0.78           N  \\nATOM     82  CA  LYS A  13      13.675  11.954  17.307  1.00  0.78           C  \\nATOM     83  C   LYS A  13      13.165  10.524  17.204  1.00  0.78           C  \\nATOM     84  O   LYS A  13      12.868   9.860  18.212  1.00  0.78           O  \\nATOM     85  CB  LYS A  13      12.467  12.918  17.391  1.00  0.78           C  \\nATOM     86  CG  LYS A  13      11.535  12.914  16.168  1.00  0.78           C  \\nATOM     87  CD  LYS A  13      10.395  13.937  16.299  1.00  0.78           C  \\nATOM     88  CE  LYS A  13       9.875  14.523  14.982  1.00  0.78           C  \\nATOM     89  NZ  LYS A  13       9.513  13.437  14.061  1.00  0.78           N  \\nATOM     90  N   GLY A  14      13.112   9.980  15.981  1.00  0.85           N  \\nATOM     91  CA  GLY A  14      12.661   8.626  15.680  1.00  0.85           C  \\nATOM     92  C   GLY A  14      13.592   7.544  16.179  1.00  0.85           C  \\nATOM     93  O   GLY A  14      13.216   6.365  16.209  1.00  0.85           O  \\nATOM     94  N   THR A  15      14.801   7.894  16.648  1.00  0.79           N  \\nATOM     95  CA  THR A  15      15.762   6.987  17.287  1.00  0.79           C  \\nATOM     96  C   THR A  15      15.247   6.421  18.596  1.00  0.79           C  \\nATOM     97  O   THR A  15      15.692   5.353  19.040  1.00  0.79           O  \\nATOM     98  CB  THR A  15      17.137   7.609  17.565  1.00  0.79           C  \\nATOM     99  OG1 THR A  15      17.040   8.777  18.374  1.00  0.79           O  \\nATOM    100  CG2 THR A  15      17.824   8.003  16.247  1.00  0.79           C  \\nATOM    101  N   GLN A  16      14.286   7.096  19.246  1.00  0.83           N  \\nATOM    102  CA  GLN A  16      13.679   6.632  20.477  1.00  0.83           C  \\nATOM    103  C   GLN A  16      12.272   6.088  20.295  1.00  0.83           C  \\nATOM    104  O   GLN A  16      11.631   5.672  21.265  1.00  0.83           O  \\nATOM    105  CB  GLN A  16      13.666   7.765  21.525  1.00  0.83           C  \\nATOM    106  CG  GLN A  16      15.084   8.224  21.927  1.00  0.83           C  \\nATOM    107  CD  GLN A  16      15.889   7.105  22.592  1.00  0.83           C  \\nATOM    108  OE1 GLN A  16      15.377   6.152  23.173  1.00  0.83           O  \\nATOM    109  NE2 GLN A  16      17.237   7.249  22.529  1.00  0.83           N  \\nATOM    110  N   ALA A  17      11.745   6.023  19.055  1.00  0.87           N  \\nATOM    111  CA  ALA A  17      10.391   5.552  18.816  1.00  0.87           C  \\nATOM    112  C   ALA A  17      10.161   4.101  19.233  1.00  0.87           C  \\nATOM    113  O   ALA A  17       9.195   3.792  19.935  1.00  0.87           O  \\nATOM    114  CB  ALA A  17      10.013   5.743  17.337  1.00  0.87           C  \\nATOM    115  N   GLN A  18      11.076   3.181  18.875  1.00  0.79           N  \\nATOM    116  CA  GLN A  18      11.001   1.770  19.244  1.00  0.79           C  \\nATOM    117  C   GLN A  18      11.091   1.522  20.736  1.00  0.79           C  \\nATOM    118  O   GLN A  18      10.329   0.707  21.279  1.00  0.79           O  \\nATOM    119  CB  GLN A  18      12.022   0.916  18.464  1.00  0.79           C  \\nATOM    120  CG  GLN A  18      11.613   0.780  16.981  1.00  0.79           C  \\nATOM    121  CD  GLN A  18      12.646  -0.023  16.183  1.00  0.79           C  \\nATOM    122  OE1 GLN A  18      13.470  -0.741  16.707  1.00  0.79           O  \\nATOM    123  NE2 GLN A  18      12.575   0.118  14.829  1.00  0.79           N  \\nATOM    124  N   PHE A  19      11.949   2.264  21.452  1.00  0.87           N  \\nATOM    125  CA  PHE A  19      12.016   2.252  22.905  1.00  0.87           C  \\nATOM    126  C   PHE A  19      10.667   2.619  23.528  1.00  0.87           C  \\nATOM    127  O   PHE A  19      10.157   1.912  24.403  1.00  0.87           O  \\nATOM    128  CB  PHE A  19      13.143   3.233  23.367  1.00  0.87           C  \\nATOM    129  CG  PHE A  19      13.035   3.617  24.820  1.00  0.87           C  \\nATOM    130  CD1 PHE A  19      13.117   2.639  25.823  1.00  0.87           C  \\nATOM    131  CD2 PHE A  19      12.670   4.930  25.173  1.00  0.87           C  \\nATOM    132  CE1 PHE A  19      12.836   2.967  27.154  1.00  0.87           C  \\nATOM    133  CE2 PHE A  19      12.374   5.252  26.502  1.00  0.87           C  \\nATOM    134  CZ  PHE A  19      12.465   4.272  27.491  1.00  0.87           C  \\nATOM    135  N   ILE A  20      10.033   3.713  23.072  1.00  0.88           N  \\nATOM    136  CA  ILE A  20       8.736   4.143  23.577  1.00  0.88           C  \\nATOM    137  C   ILE A  20       7.626   3.143  23.265  1.00  0.88           C  \\nATOM    138  O   ILE A  20       6.806   2.803  24.128  1.00  0.88           O  \\nATOM    139  CB  ILE A  20       8.395   5.542  23.074  1.00  0.88           C  \\nATOM    140  CG1 ILE A  20       9.373   6.565  23.696  1.00  0.88           C  \\nATOM    141  CG2 ILE A  20       6.932   5.909  23.397  1.00  0.88           C  \\nATOM    142  CD1 ILE A  20       9.305   7.952  23.050  1.00  0.88           C  \\nATOM    143  N   MET A  21       7.589   2.608  22.031  1.00  0.86           N  \\nATOM    144  CA  MET A  21       6.621   1.603  21.624  1.00  0.86           C  \\nATOM    145  C   MET A  21       6.715   0.318  22.432  1.00  0.86           C  \\nATOM    146  O   MET A  21       5.694  -0.205  22.893  1.00  0.86           O  \\nATOM    147  CB  MET A  21       6.770   1.241  20.128  1.00  0.86           C  \\nATOM    148  CG  MET A  21       6.336   2.349  19.152  1.00  0.86           C  \\nATOM    149  SD  MET A  21       6.084   1.750  17.454  1.00  0.86           S  \\nATOM    150  CE  MET A  21       7.834   1.578  17.025  1.00  0.86           C  \\nATOM    151  N   GLU A  22       7.931  -0.204  22.653  1.00  0.84           N  \\nATOM    152  CA  GLU A  22       8.172  -1.371  23.489  1.00  0.84           C  \\nATOM    153  C   GLU A  22       7.847  -1.127  24.954  1.00  0.84           C  \\nATOM    154  O   GLU A  22       7.208  -1.970  25.610  1.00  0.84           O  \\nATOM    155  CB  GLU A  22       9.612  -1.898  23.310  1.00  0.84           C  \\nATOM    156  CG  GLU A  22       9.895  -3.204  24.093  1.00  0.84           C  \\nATOM    157  CD  GLU A  22      11.235  -3.859  23.749  1.00  0.84           C  \\nATOM    158  OE1 GLU A  22      11.879  -3.446  22.751  1.00  0.84           O  \\nATOM    159  OE2 GLU A  22      11.614  -4.797  24.496  1.00  0.84           O  \\nATOM    160  N   LYS A  23       8.206   0.041  25.514  1.00  0.84           N  \\nATOM    161  CA  LYS A  23       7.943   0.402  26.904  1.00  0.84           C  \\nATOM    162  C   LYS A  23       6.470   0.387  27.289  1.00  0.84           C  \\nATOM    163  O   LYS A  23       6.112  -0.045  28.398  1.00  0.84           O  \\nATOM    164  CB  LYS A  23       8.534   1.800  27.239  1.00  0.84           C  \\nATOM    165  CG  LYS A  23       8.352   2.219  28.715  1.00  0.84           C  \\nATOM    166  CD  LYS A  23       9.080   3.520  29.098  1.00  0.84           C  \\nATOM    167  CE  LYS A  23       8.801   4.046  30.524  1.00  0.84           C  \\nATOM    168  NZ  LYS A  23       9.332   3.167  31.585  1.00  0.84           N  \\nATOM    169  N   TYR A  24       5.579   0.874  26.415  1.00  0.84           N  \\nATOM    170  CA  TYR A  24       4.167   1.009  26.733  1.00  0.84           C  \\nATOM    171  C   TYR A  24       3.237   0.110  25.942  1.00  0.84           C  \\nATOM    172  O   TYR A  24       2.035   0.040  26.242  1.00  0.84           O  \\nATOM    173  CB  TYR A  24       3.739   2.460  26.446  1.00  0.84           C  \\nATOM    174  CG  TYR A  24       4.457   3.383  27.371  1.00  0.84           C  \\nATOM    175  CD1 TYR A  24       4.123   3.389  28.730  1.00  0.84           C  \\nATOM    176  CD2 TYR A  24       5.463   4.241  26.905  1.00  0.84           C  \\nATOM    177  CE1 TYR A  24       4.746   4.282  29.603  1.00  0.84           C  \\nATOM    178  CE2 TYR A  24       6.106   5.126  27.781  1.00  0.84           C  \\nATOM    179  CZ  TYR A  24       5.716   5.158  29.125  1.00  0.84           C  \\nATOM    180  OH  TYR A  24       6.230   6.080  30.040  1.00  0.84           O  \\nATOM    181  N   GLY A  25       3.721  -0.599  24.914  1.00  0.90           N  \\nATOM    182  CA  GLY A  25       2.877  -1.415  24.044  1.00  0.90           C  \\nATOM    183  C   GLY A  25       1.942  -0.579  23.209  1.00  0.90           C  \\nATOM    184  O   GLY A  25       0.736  -0.856  23.153  1.00  0.90           O  \\nATOM    185  N   ILE A  26       2.463   0.496  22.598  1.00  0.89           N  \\nATOM    186  CA  ILE A  26       1.692   1.510  21.898  1.00  0.89           C  \\nATOM    187  C   ILE A  26       2.224   1.680  20.486  1.00  0.89           C  \\nATOM    188  O   ILE A  26       3.413   1.447  20.267  1.00  0.89           O  \\nATOM    189  CB  ILE A  26       1.688   2.867  22.609  1.00  0.89           C  \\nATOM    190  CG1 ILE A  26       3.093   3.497  22.794  1.00  0.89           C  \\nATOM    191  CG2 ILE A  26       0.914   2.682  23.924  1.00  0.89           C  \\nATOM    192  CD1 ILE A  26       3.097   4.819  23.573  1.00  0.89           C  \\nATOM    193  N   PRO A  27       1.434   2.060  19.486  1.00  0.94           N  \\nATOM    194  CA  PRO A  27       1.948   2.276  18.143  1.00  0.94           C  \\nATOM    195  C   PRO A  27       2.489   3.684  17.979  1.00  0.94           C  \\nATOM    196  O   PRO A  27       2.091   4.615  18.677  1.00  0.94           O  \\nATOM    197  CB  PRO A  27       0.722   2.055  17.242  1.00  0.94           C  \\nATOM    198  CG  PRO A  27      -0.466   2.458  18.113  1.00  0.94           C  \\nATOM    199  CD  PRO A  27      -0.031   2.036  19.516  1.00  0.94           C  \\nATOM    200  N   GLN A  28       3.419   3.846  17.023  1.00  0.89           N  \\nATOM    201  CA  GLN A  28       3.857   5.130  16.526  1.00  0.89           C  \\nATOM    202  C   GLN A  28       2.894   5.530  15.421  1.00  0.89           C  \\nATOM    203  O   GLN A  28       2.607   4.752  14.515  1.00  0.89           O  \\nATOM    204  CB  GLN A  28       5.302   5.026  15.994  1.00  0.89           C  \\nATOM    205  CG  GLN A  28       5.917   6.330  15.435  1.00  0.89           C  \\nATOM    206  CD  GLN A  28       7.346   6.045  14.944  1.00  0.89           C  \\nATOM    207  OE1 GLN A  28       7.785   4.908  14.869  1.00  0.89           O  \\nATOM    208  NE2 GLN A  28       8.120   7.104  14.594  1.00  0.89           N  \\nATOM    209  N   ILE A  29       2.310   6.737  15.512  1.00  0.88           N  \\nATOM    210  CA  ILE A  29       1.329   7.227  14.567  1.00  0.88           C  \\nATOM    211  C   ILE A  29       1.792   8.584  14.071  1.00  0.88           C  \\nATOM    212  O   ILE A  29       2.444   9.347  14.785  1.00  0.88           O  \\nATOM    213  CB  ILE A  29      -0.102   7.280  15.132  1.00  0.88           C  \\nATOM    214  CG1 ILE A  29      -0.282   8.339  16.245  1.00  0.88           C  \\nATOM    215  CG2 ILE A  29      -0.487   5.867  15.619  1.00  0.88           C  \\nATOM    216  CD1 ILE A  29      -1.710   8.472  16.780  1.00  0.88           C  \\nATOM    217  N   SER A  30       1.473   8.933  12.815  1.00  0.90           N  \\nATOM    218  CA  SER A  30       1.903  10.187  12.227  1.00  0.90           C  \\nATOM    219  C   SER A  30       0.840  10.618  11.255  1.00  0.90           C  \\nATOM    220  O   SER A  30       0.062   9.802  10.765  1.00  0.90           O  \\nATOM    221  CB  SER A  30       3.295  10.149  11.545  1.00  0.90           C  \\nATOM    222  OG  SER A  30       3.243   9.504  10.236  1.00  0.90           O  \\nATOM    223  N   THR A  31       0.771  11.923  10.947  1.00  0.89           N  \\nATOM    224  CA  THR A  31      -0.174  12.471   9.977  1.00  0.89           C  \\nATOM    225  C   THR A  31       0.011  11.885   8.596  1.00  0.89           C  \\nATOM    226  O   THR A  31      -0.959  11.630   7.884  1.00  0.89           O  \\nATOM    227  CB  THR A  31      -0.139  13.986   9.877  1.00  0.89           C  \\nATOM    228  OG1 THR A  31       1.192  14.510   9.668  1.00  0.89           O  \\nATOM    229  CG2 THR A  31      -0.617  14.571  11.208  1.00  0.89           C  \\nATOM    230  N   GLY A  32       1.265  11.621   8.197  1.00  0.85           N  \\nATOM    231  CA  GLY A  32       1.586  10.969   6.933  1.00  0.85           C  \\nATOM    232  C   GLY A  32       1.125   9.532   6.846  1.00  0.85           C  \\nATOM    233  O   GLY A  32       0.557   9.125   5.828  1.00  0.85           O  \\nATOM    234  N   ASP A  33       1.334   8.722   7.895  1.00  0.83           N  \\nATOM    235  CA  ASP A  33       0.866   7.341   7.952  1.00  0.83           C  \\nATOM    236  C   ASP A  33      -0.646   7.227   7.978  1.00  0.83           C  \\nATOM    237  O   ASP A  33      -1.239   6.419   7.250  1.00  0.83           O  \\nATOM    238  CB  ASP A  33       1.479   6.597   9.165  1.00  0.83           C  \\nATOM    239  CG  ASP A  33       2.934   6.244   8.906  1.00  0.83           C  \\nATOM    240  OD1 ASP A  33       3.440   6.587   7.796  1.00  0.83           O  \\nATOM    241  OD2 ASP A  33       3.559   5.634   9.807  1.00  0.83           O  \\nATOM    242  N   MET A  34      -1.320   8.078   8.771  1.00  0.81           N  \\nATOM    243  CA  MET A  34      -2.771   8.142   8.836  1.00  0.81           C  \\nATOM    244  C   MET A  34      -3.384   8.470   7.495  1.00  0.81           C  \\nATOM    245  O   MET A  34      -4.330   7.816   7.041  1.00  0.81           O  \\nATOM    246  CB  MET A  34      -3.231   9.267   9.792  1.00  0.81           C  \\nATOM    247  CG  MET A  34      -3.002   8.989  11.279  1.00  0.81           C  \\nATOM    248  SD  MET A  34      -3.515  10.396  12.292  1.00  0.81           S  \\nATOM    249  CE  MET A  34      -2.144  10.358  13.475  1.00  0.81           C  \\nATOM    250  N   LEU A  35      -2.819   9.469   6.811  1.00  0.83           N  \\nATOM    251  CA  LEU A  35      -3.248   9.900   5.503  1.00  0.83           C  \\nATOM    252  C   LEU A  35      -3.103   8.818   4.443  1.00  0.83           C  \\nATOM    253  O   LEU A  35      -4.048   8.532   3.695  1.00  0.83           O  \\nATOM    254  CB  LEU A  35      -2.415  11.145   5.148  1.00  0.83           C  \\nATOM    255  CG  LEU A  35      -2.800  11.866   3.858  1.00  0.83           C  \\nATOM    256  CD1 LEU A  35      -4.256  12.323   3.889  1.00  0.83           C  \\nATOM    257  CD2 LEU A  35      -1.881  13.072   3.633  1.00  0.83           C  \\nATOM    258  N   ARG A  36      -1.952   8.127   4.393  1.00  0.75           N  \\nATOM    259  CA  ARG A  36      -1.722   6.997   3.506  1.00  0.75           C  \\nATOM    260  C   ARG A  36      -2.640   5.811   3.766  1.00  0.75           C  \\nATOM    261  O   ARG A  36      -3.193   5.217   2.828  1.00  0.75           O  \\nATOM    262  CB  ARG A  36      -0.259   6.506   3.621  1.00  0.75           C  \\nATOM    263  CG  ARG A  36       0.765   7.442   2.956  1.00  0.75           C  \\nATOM    264  CD  ARG A  36       2.196   6.883   2.958  1.00  0.75           C  \\nATOM    265  NE  ARG A  36       2.798   7.053   4.325  1.00  0.75           N  \\nATOM    266  CZ  ARG A  36       3.498   8.139   4.706  1.00  0.75           C  \\nATOM    267  NH1 ARG A  36       3.613   9.204   3.930  1.00  0.75           N  \\nATOM    268  NH2 ARG A  36       4.066   8.178   5.907  1.00  0.75           N  \\nATOM    269  N   ALA A  37      -2.833   5.435   5.039  1.00  0.86           N  \\nATOM    270  CA  ALA A  37      -3.729   4.367   5.437  1.00  0.86           C  \\nATOM    271  C   ALA A  37      -5.193   4.659   5.128  1.00  0.86           C  \\nATOM    272  O   ALA A  37      -5.906   3.788   4.603  1.00  0.86           O  \\nATOM    273  CB  ALA A  37      -3.542   4.060   6.935  1.00  0.86           C  \\nATOM    274  N   ALA A  38      -5.678   5.880   5.399  1.00  0.85           N  \\nATOM    275  CA  ALA A  38      -7.026   6.307   5.070  1.00  0.85           C  \\nATOM    276  C   ALA A  38      -7.300   6.315   3.571  1.00  0.85           C  \\nATOM    277  O   ALA A  38      -8.311   5.771   3.105  1.00  0.85           O  \\nATOM    278  CB  ALA A  38      -7.287   7.699   5.676  1.00  0.85           C  \\nATOM    279  N   VAL A  39      -6.397   6.861   2.741  1.00  0.83           N  \\nATOM    280  CA  VAL A  39      -6.548   6.862   1.283  1.00  0.83           C  \\nATOM    281  C   VAL A  39      -6.618   5.460   0.708  1.00  0.83           C  \\nATOM    282  O   VAL A  39      -7.464   5.154  -0.149  1.00  0.83           O  \\nATOM    283  CB  VAL A  39      -5.427   7.640   0.604  1.00  0.83           C  \\nATOM    284  CG1 VAL A  39      -5.445   7.455  -0.926  1.00  0.83           C  \\nATOM    285  CG2 VAL A  39      -5.606   9.131   0.931  1.00  0.83           C  \\nATOM    286  N   LYS A  40      -5.753   4.558   1.184  1.00  0.80           N  \\nATOM    287  CA  LYS A  40      -5.744   3.158   0.810  1.00  0.80           C  \\nATOM    288  C   LYS A  40      -6.986   2.389   1.235  1.00  0.80           C  \\nATOM    289  O   LYS A  40      -7.427   1.466   0.535  1.00  0.80           O  \\nATOM    290  CB  LYS A  40      -4.483   2.470   1.373  1.00  0.80           C  \\nATOM    291  CG  LYS A  40      -4.310   1.026   0.877  1.00  0.80           C  \\nATOM    292  CD  LYS A  40      -2.999   0.376   1.337  1.00  0.80           C  \\nATOM    293  CE  LYS A  40      -2.869  -1.070   0.856  1.00  0.80           C  \\nATOM    294  NZ  LYS A  40      -1.624  -1.667   1.387  1.00  0.80           N  \\nATOM    295  N   SER A  41      -7.569   2.710   2.400  1.00  0.83           N  \\nATOM    296  CA  SER A  41      -8.744   2.030   2.934  1.00  0.83           C  \\nATOM    297  C   SER A  41     -10.009   2.305   2.156  1.00  0.83           C  \\nATOM    298  O   SER A  41     -10.924   1.462   2.115  1.00  0.83           O  \\nATOM    299  CB  SER A  41      -8.974   2.307   4.451  1.00  0.83           C  \\nATOM    300  OG  SER A  41      -9.526   3.597   4.720  1.00  0.83           O  \\nATOM    301  N   GLY A  42     -10.127   3.488   1.536  1.00  0.84           N  \\nATOM    302  CA  GLY A  42     -11.339   3.915   0.854  1.00  0.84           C  \\nATOM    303  C   GLY A  42     -12.409   4.388   1.809  1.00  0.84           C  \\nATOM    304  O   GLY A  42     -13.570   4.525   1.422  1.00  0.84           O  \\nATOM    305  N   SER A  43     -12.049   4.626   3.093  1.00  0.83           N  \\nATOM    306  CA  SER A  43     -12.886   5.270   4.112  1.00  0.83           C  \\nATOM    307  C   SER A  43     -13.446   6.615   3.681  1.00  0.83           C  \\nATOM    308  O   SER A  43     -12.953   7.238   2.752  1.00  0.83           O  \\nATOM    309  CB  SER A  43     -12.235   5.412   5.532  1.00  0.83           C  \\nATOM    310  OG  SER A  43     -11.199   6.400   5.617  1.00  0.83           O  \\nATOM    311  N   GLU A  44     -14.484   7.141   4.370  1.00  0.82           N  \\nATOM    312  CA  GLU A  44     -15.048   8.446   4.048  1.00  0.82           C  \\nATOM    313  C   GLU A  44     -14.029   9.585   4.144  1.00  0.82           C  \\nATOM    314  O   GLU A  44     -13.885  10.417   3.241  1.00  0.82           O  \\nATOM    315  CB  GLU A  44     -16.256   8.723   4.980  1.00  0.82           C  \\nATOM    316  CG  GLU A  44     -17.462   7.789   4.717  1.00  0.82           C  \\nATOM    317  CD  GLU A  44     -17.938   7.934   3.274  1.00  0.82           C  \\nATOM    318  OE1 GLU A  44     -18.288   9.077   2.884  1.00  0.82           O  \\nATOM    319  OE2 GLU A  44     -17.923   6.907   2.551  1.00  0.82           O  \\nATOM    320  N   LEU A  45     -13.216   9.596   5.218  1.00  0.85           N  \\nATOM    321  CA  LEU A  45     -12.085  10.502   5.366  1.00  0.85           C  \\nATOM    322  C   LEU A  45     -10.989  10.256   4.344  1.00  0.85           C  \\nATOM    323  O   LEU A  45     -10.399  11.195   3.791  1.00  0.85           O  \\nATOM    324  CB  LEU A  45     -11.485  10.428   6.786  1.00  0.85           C  \\nATOM    325  CG  LEU A  45     -12.424  10.877   7.922  1.00  0.85           C  \\nATOM    326  CD1 LEU A  45     -11.679  10.764   9.255  1.00  0.85           C  \\nATOM    327  CD2 LEU A  45     -12.937  12.308   7.733  1.00  0.85           C  \\nATOM    328  N   GLY A  46     -10.702   8.985   4.034  1.00  0.81           N  \\nATOM    329  CA  GLY A  46      -9.751   8.595   3.007  1.00  0.81           C  \\nATOM    330  C   GLY A  46     -10.116   8.984   1.614  1.00  0.81           C  \\nATOM    331  O   GLY A  46      -9.259   9.378   0.821  1.00  0.81           O  \\nATOM    332  N   LYS A  47     -11.407   8.923   1.270  1.00  0.79           N  \\nATOM    333  CA  LYS A  47     -11.904   9.420   0.008  1.00  0.79           C  \\nATOM    334  C   LYS A  47     -11.755  10.930  -0.110  1.00  0.79           C  \\nATOM    335  O   LYS A  47     -11.256  11.432  -1.127  1.00  0.79           O  \\nATOM    336  CB  LYS A  47     -13.381   9.037  -0.236  1.00  0.79           C  \\nATOM    337  CG  LYS A  47     -13.750   9.249  -1.712  1.00  0.79           C  \\nATOM    338  CD  LYS A  47     -15.222   9.594  -1.957  1.00  0.79           C  \\nATOM    339  CE  LYS A  47     -15.539   9.852  -3.439  1.00  0.79           C  \\nATOM    340  NZ  LYS A  47     -14.629  10.844  -4.046  1.00  0.79           N  \\nATOM    341  N   GLN A  48     -12.117  11.694   0.937  1.00  0.79           N  \\nATOM    342  CA  GLN A  48     -11.940  13.141   0.956  1.00  0.79           C  \\nATOM    343  C   GLN A  48     -10.482  13.543   0.835  1.00  0.79           C  \\nATOM    344  O   GLN A  48     -10.119  14.457   0.074  1.00  0.79           O  \\nATOM    345  CB  GLN A  48     -12.504  13.766   2.256  1.00  0.79           C  \\nATOM    346  CG  GLN A  48     -12.382  15.310   2.280  1.00  0.79           C  \\nATOM    347  CD  GLN A  48     -13.011  15.929   3.531  1.00  0.79           C  \\nATOM    348  OE1 GLN A  48     -13.595  15.271   4.372  1.00  0.79           O  \\nATOM    349  NE2 GLN A  48     -12.874  17.281   3.642  1.00  0.79           N  \\nATOM    350  N   ALA A  49      -9.598  12.854   1.562  1.00  0.88           N  \\nATOM    351  CA  ALA A  49      -8.169  13.030   1.480  1.00  0.88           C  \\nATOM    352  C   ALA A  49      -7.596  12.706   0.116  1.00  0.88           C  \\nATOM    353  O   ALA A  49      -6.781  13.472  -0.416  1.00  0.88           O  \\nATOM    354  CB  ALA A  49      -7.489  12.147   2.532  1.00  0.88           C  \\nATOM    355  N   LYS A  50      -8.025  11.598  -0.504  1.00  0.79           N  \\nATOM    356  CA  LYS A  50      -7.604  11.213  -1.837  1.00  0.79           C  \\nATOM    357  C   LYS A  50      -7.973  12.228  -2.906  1.00  0.79           C  \\nATOM    358  O   LYS A  50      -7.108  12.654  -3.677  1.00  0.79           O  \\nATOM    359  CB  LYS A  50      -8.195   9.837  -2.218  1.00  0.79           C  \\nATOM    360  CG  LYS A  50      -7.549   9.265  -3.485  1.00  0.79           C  \\nATOM    361  CD  LYS A  50      -8.056   7.869  -3.869  1.00  0.79           C  \\nATOM    362  CE  LYS A  50      -7.331   7.288  -5.086  1.00  0.79           C  \\nATOM    363  NZ  LYS A  50      -7.591   8.104  -6.280  1.00  0.79           N  \\nATOM    364  N   ASP A  51      -9.223  12.716  -2.930  1.00  0.80           N  \\nATOM    365  CA  ASP A  51      -9.688  13.726  -3.868  1.00  0.80           C  \\nATOM    366  C   ASP A  51      -8.878  15.035  -3.750  1.00  0.80           C  \\nATOM    367  O   ASP A  51      -8.513  15.663  -4.760  1.00  0.80           O  \\nATOM    368  CB  ASP A  51     -11.202  13.993  -3.627  1.00  0.80           C  \\nATOM    369  CG  ASP A  51     -12.189  12.889  -4.006  1.00  0.80           C  \\nATOM    370  OD1 ASP A  51     -11.875  11.845  -4.635  1.00  0.80           O  \\nATOM    371  OD2 ASP A  51     -13.381  13.088  -3.632  1.00  0.80           O  \\nATOM    372  N   ILE A  52      -8.532  15.478  -2.526  1.00  0.79           N  \\nATOM    373  CA  ILE A  52      -7.645  16.619  -2.285  1.00  0.79           C  \\nATOM    374  C   ILE A  52      -6.217  16.398  -2.776  1.00  0.79           C  \\nATOM    375  O   ILE A  52      -5.645  17.238  -3.490  1.00  0.79           O  \\nATOM    376  CB  ILE A  52      -7.631  16.974  -0.794  1.00  0.79           C  \\nATOM    377  CG1 ILE A  52      -8.994  17.571  -0.383  1.00  0.79           C  \\nATOM    378  CG2 ILE A  52      -6.499  17.959  -0.443  1.00  0.79           C  \\nATOM    379  CD1 ILE A  52      -9.166  17.747   1.130  1.00  0.79           C  \\nATOM    380  N   MET A  53      -5.599  15.263  -2.425  1.00  0.76           N  \\nATOM    381  CA  MET A  53      -4.230  14.937  -2.797  1.00  0.76           C  \\nATOM    382  C   MET A  53      -4.038  14.688  -4.285  1.00  0.76           C  \\nATOM    383  O   MET A  53      -3.056  15.156  -4.881  1.00  0.76           O  \\nATOM    384  CB  MET A  53      -3.730  13.706  -2.027  1.00  0.76           C  \\nATOM    385  CG  MET A  53      -3.567  13.914  -0.514  1.00  0.76           C  \\nATOM    386  SD  MET A  53      -3.318  12.341   0.360  1.00  0.76           S  \\nATOM    387  CE  MET A  53      -1.898  11.706  -0.580  1.00  0.76           C  \\nATOM    388  N   ASP A  54      -4.986  13.996  -4.937  1.00  0.79           N  \\nATOM    389  CA  ASP A  54      -4.996  13.726  -6.367  1.00  0.79           C  \\nATOM    390  C   ASP A  54      -5.092  15.018  -7.190  1.00  0.79           C  \\nATOM    391  O   ASP A  54      -4.652  15.073  -8.349  1.00  0.79           O  \\nATOM    392  CB  ASP A  54      -6.156  12.762  -6.739  1.00  0.79           C  \\nATOM    393  CG  ASP A  54      -5.940  11.303  -6.360  1.00  0.79           C  \\nATOM    394  OD1 ASP A  54      -4.884  10.889  -5.828  1.00  0.79           O  \\nATOM    395  OD2 ASP A  54      -6.895  10.527  -6.646  1.00  0.79           O  \\nATOM    396  N   ALA A  55      -5.612  16.116  -6.609  1.00  0.82           N  \\nATOM    397  CA  ALA A  55      -5.656  17.425  -7.232  1.00  0.82           C  \\nATOM    398  C   ALA A  55      -4.455  18.303  -6.873  1.00  0.82           C  \\nATOM    399  O   ALA A  55      -4.425  19.495  -7.213  1.00  0.82           O  \\nATOM    400  CB  ALA A  55      -6.951  18.157  -6.821  1.00  0.82           C  \\nATOM    401  N   GLY A  56      -3.428  17.767  -6.183  1.00  0.81           N  \\nATOM    402  CA  GLY A  56      -2.221  18.514  -5.814  1.00  0.81           C  \\nATOM    403  C   GLY A  56      -2.447  19.576  -4.779  1.00  0.81           C  \\nATOM    404  O   GLY A  56      -1.771  20.613  -4.758  1.00  0.81           O  \\nATOM    405  N   LYS A  57      -3.416  19.363  -3.888  1.00  0.76           N  \\nATOM    406  CA  LYS A  57      -3.765  20.286  -2.841  1.00  0.76           C  \\nATOM    407  C   LYS A  57      -3.400  19.702  -1.494  1.00  0.76           C  \\nATOM    408  O   LYS A  57      -3.217  18.501  -1.314  1.00  0.76           O  \\nATOM    409  CB  LYS A  57      -5.274  20.620  -2.877  1.00  0.76           C  \\nATOM    410  CG  LYS A  57      -5.685  21.435  -4.113  1.00  0.76           C  \\nATOM    411  CD  LYS A  57      -7.204  21.669  -4.162  1.00  0.76           C  \\nATOM    412  CE  LYS A  57      -7.699  22.401  -5.410  1.00  0.76           C  \\nATOM    413  NZ  LYS A  57      -7.070  23.735  -5.473  1.00  0.76           N  \\nATOM    414  N   LEU A  58      -3.278  20.583  -0.487  1.00  0.79           N  \\nATOM    415  CA  LEU A  58      -3.057  20.164   0.875  1.00  0.79           C  \\nATOM    416  C   LEU A  58      -4.380  19.852   1.539  1.00  0.79           C  \\nATOM    417  O   LEU A  58      -5.377  20.546   1.375  1.00  0.79           O  \\nATOM    418  CB  LEU A  58      -2.324  21.244   1.702  1.00  0.79           C  \\nATOM    419  CG  LEU A  58      -0.793  21.100   1.726  1.00  0.79           C  \\nATOM    420  CD1 LEU A  58      -0.180  22.317   2.427  1.00  0.79           C  \\nATOM    421  CD2 LEU A  58      -0.357  19.816   2.444  1.00  0.79           C  \\nATOM    422  N   VAL A  59      -4.394  18.757   2.326  1.00  0.80           N  \\nATOM    423  CA  VAL A  59      -5.463  18.452   3.249  1.00  0.80           C  \\nATOM    424  C   VAL A  59      -5.616  19.539   4.304  1.00  0.80           C  \\nATOM    425  O   VAL A  59      -4.651  20.116   4.795  1.00  0.80           O  \\nATOM    426  CB  VAL A  59      -5.312  17.073   3.888  1.00  0.80           C  \\nATOM    427  CG1 VAL A  59      -5.448  15.991   2.799  1.00  0.80           C  \\nATOM    428  CG2 VAL A  59      -3.975  16.942   4.640  1.00  0.80           C  \\nATOM    429  N   THR A  60      -6.873  19.874   4.644  1.00  0.81           N  \\nATOM    430  CA  THR A  60      -7.205  20.858   5.664  1.00  0.81           C  \\nATOM    431  C   THR A  60      -6.826  20.395   7.062  1.00  0.81           C  \\nATOM    432  O   THR A  60      -6.830  19.202   7.351  1.00  0.81           O  \\nATOM    433  CB  THR A  60      -8.687  21.248   5.653  1.00  0.81           C  \\nATOM    434  OG1 THR A  60      -9.547  20.129   5.856  1.00  0.81           O  \\nATOM    435  CG2 THR A  60      -9.048  21.843   4.285  1.00  0.81           C  \\nATOM    436  N   ASP A  61      -6.535  21.336   7.998  1.00  0.82           N  \\nATOM    437  CA  ASP A  61      -6.327  21.015   9.407  1.00  0.82           C  \\nATOM    438  C   ASP A  61      -7.511  20.249  10.006  1.00  0.82           C  \\nATOM    439  O   ASP A  61      -7.327  19.291  10.754  1.00  0.82           O  \\nATOM    440  CB  ASP A  61      -6.065  22.294  10.253  1.00  0.82           C  \\nATOM    441  CG  ASP A  61      -4.612  22.762  10.222  1.00  0.82           C  \\nATOM    442  OD1 ASP A  61      -3.727  22.015   9.729  1.00  0.82           O  \\nATOM    443  OD2 ASP A  61      -4.359  23.872  10.757  1.00  0.82           O  \\nATOM    444  N   GLU A  62      -8.760  20.599   9.635  1.00  0.80           N  \\nATOM    445  CA  GLU A  62      -9.954  19.882  10.068  1.00  0.80           C  \\nATOM    446  C   GLU A  62      -9.955  18.407   9.679  1.00  0.80           C  \\nATOM    447  O   GLU A  62     -10.196  17.523  10.514  1.00  0.80           O  \\nATOM    448  CB  GLU A  62     -11.210  20.541   9.458  1.00  0.80           C  \\nATOM    449  CG  GLU A  62     -12.548  19.884   9.878  1.00  0.80           C  \\nATOM    450  CD  GLU A  62     -13.756  20.521   9.189  1.00  0.80           C  \\nATOM    451  OE1 GLU A  62     -13.566  21.535   8.466  1.00  0.80           O  \\nATOM    452  OE2 GLU A  62     -14.872  19.973   9.373  1.00  0.80           O  \\nATOM    453  N   LEU A  63      -9.615  18.089   8.417  1.00  0.84           N  \\nATOM    454  CA  LEU A  63      -9.479  16.720   7.950  1.00  0.84           C  \\nATOM    455  C   LEU A  63      -8.342  15.974   8.638  1.00  0.84           C  \\nATOM    456  O   LEU A  63      -8.492  14.815   9.043  1.00  0.84           O  \\nATOM    457  CB  LEU A  63      -9.316  16.674   6.416  1.00  0.84           C  \\nATOM    458  CG  LEU A  63      -9.176  15.265   5.810  1.00  0.84           C  \\nATOM    459  CD1 LEU A  63     -10.315  14.328   6.215  1.00  0.84           C  \\nATOM    460  CD2 LEU A  63      -9.109  15.354   4.287  1.00  0.84           C  \\nATOM    461  N   VAL A  64      -7.183  16.627   8.843  1.00  0.86           N  \\nATOM    462  CA  VAL A  64      -6.071  16.053   9.598  1.00  0.86           C  \\nATOM    463  C   VAL A  64      -6.464  15.711  11.031  1.00  0.86           C  \\nATOM    464  O   VAL A  64      -6.204  14.601  11.505  1.00  0.86           O  \\nATOM    465  CB  VAL A  64      -4.841  16.962   9.582  1.00  0.86           C  \\nATOM    466  CG1 VAL A  64      -3.715  16.428  10.489  1.00  0.86           C  \\nATOM    467  CG2 VAL A  64      -4.319  17.069   8.140  1.00  0.86           C  \\nATOM    468  N   ILE A  65      -7.160  16.608  11.749  1.00  0.84           N  \\nATOM    469  CA  ILE A  65      -7.674  16.358  13.096  1.00  0.84           C  \\nATOM    470  C   ILE A  65      -8.691  15.226  13.151  1.00  0.84           C  \\nATOM    471  O   ILE A  65      -8.647  14.373  14.046  1.00  0.84           O  \\nATOM    472  CB  ILE A  65      -8.252  17.624  13.718  1.00  0.84           C  \\nATOM    473  CG1 ILE A  65      -7.159  18.701  13.894  1.00  0.84           C  \\nATOM    474  CG2 ILE A  65      -8.968  17.347  15.061  1.00  0.84           C  \\nATOM    475  CD1 ILE A  65      -6.103  18.387  14.947  1.00  0.84           C  \\nATOM    476  N   ALA A  66      -9.615  15.152  12.174  1.00  0.87           N  \\nATOM    477  CA  ALA A  66     -10.570  14.063  12.053  1.00  0.87           C  \\nATOM    478  C   ALA A  66      -9.887  12.703  11.884  1.00  0.87           C  \\nATOM    479  O   ALA A  66     -10.249  11.727  12.565  1.00  0.87           O  \\nATOM    480  CB  ALA A  66     -11.545  14.361  10.894  1.00  0.87           C  \\nATOM    481  N   LEU A  67      -8.834  12.622  11.054  1.00  0.82           N  \\nATOM    482  CA  LEU A  67      -7.967  11.457  10.913  1.00  0.82           C  \\nATOM    483  C   LEU A  67      -7.241  11.074  12.197  1.00  0.82           C  \\nATOM    484  O   LEU A  67      -7.156   9.884  12.537  1.00  0.82           O  \\nATOM    485  CB  LEU A  67      -6.923  11.668   9.788  1.00  0.82           C  \\nATOM    486  CG  LEU A  67      -7.504  11.652   8.361  1.00  0.82           C  \\nATOM    487  CD1 LEU A  67      -6.483  12.185   7.344  1.00  0.82           C  \\nATOM    488  CD2 LEU A  67      -7.955  10.242   7.970  1.00  0.82           C  \\nATOM    489  N   VAL A  68      -6.717  12.038  12.981  1.00  0.82           N  \\nATOM    490  CA  VAL A  68      -6.115  11.764  14.290  1.00  0.82           C  \\nATOM    491  C   VAL A  68      -7.123  11.173  15.253  1.00  0.82           C  \\nATOM    492  O   VAL A  68      -6.858  10.155  15.913  1.00  0.82           O  \\nATOM    493  CB  VAL A  68      -5.466  12.989  14.944  1.00  0.82           C  \\nATOM    494  CG1 VAL A  68      -4.857  12.624  16.314  1.00  0.82           C  \\nATOM    495  CG2 VAL A  68      -4.348  13.551  14.056  1.00  0.82           C  \\nATOM    496  N   LYS A  69      -8.329  11.750  15.325  1.00  0.77           N  \\nATOM    497  CA  LYS A  69      -9.410  11.272  16.170  1.00  0.77           C  \\nATOM    498  C   LYS A  69      -9.866   9.861  15.819  1.00  0.77           C  \\nATOM    499  O   LYS A  69     -10.051   9.012  16.706  1.00  0.77           O  \\nATOM    500  CB  LYS A  69     -10.617  12.235  16.076  1.00  0.77           C  \\nATOM    501  CG  LYS A  69     -11.838  11.755  16.871  1.00  0.77           C  \\nATOM    502  CD  LYS A  69     -12.969  12.784  16.952  1.00  0.77           C  \\nATOM    503  CE  LYS A  69     -14.188  12.205  17.671  1.00  0.77           C  \\nATOM    504  NZ  LYS A  69     -15.175  13.271  17.924  1.00  0.77           N  \\nATOM    505  N   GLU A  70     -10.032   9.567  14.522  1.00  0.76           N  \\nATOM    506  CA  GLU A  70     -10.337   8.242  13.996  1.00  0.76           C  \\nATOM    507  C   GLU A  70      -9.227   7.245  14.295  1.00  0.76           C  \\nATOM    508  O   GLU A  70      -9.473   6.115  14.765  1.00  0.76           O  \\nATOM    509  CB  GLU A  70     -10.548   8.333  12.461  1.00  0.76           C  \\nATOM    510  CG  GLU A  70     -10.899   6.994  11.762  1.00  0.76           C  \\nATOM    511  CD  GLU A  70     -11.006   7.112  10.236  1.00  0.76           C  \\nATOM    512  OE1 GLU A  70      -9.975   7.425   9.586  1.00  0.76           O  \\nATOM    513  OE2 GLU A  70     -12.115   6.863   9.695  1.00  0.76           O  \\nATOM    514  N   ARG A  71      -7.961   7.626  14.095  1.00  0.75           N  \\nATOM    515  CA  ARG A  71      -6.793   6.788  14.316  1.00  0.75           C  \\nATOM    516  C   ARG A  71      -6.616   6.308  15.745  1.00  0.75           C  \\nATOM    517  O   ARG A  71      -6.241   5.142  15.965  1.00  0.75           O  \\nATOM    518  CB  ARG A  71      -5.484   7.502  13.905  1.00  0.75           C  \\nATOM    519  CG  ARG A  71      -4.208   6.643  14.043  1.00  0.75           C  \\nATOM    520  CD  ARG A  71      -4.168   5.448  13.094  1.00  0.75           C  \\nATOM    521  NE  ARG A  71      -3.022   4.587  13.506  1.00  0.75           N  \\nATOM    522  CZ  ARG A  71      -3.146   3.516  14.286  1.00  0.75           C  \\nATOM    523  NH1 ARG A  71      -4.250   3.232  14.985  1.00  0.75           N  \\nATOM    524  NH2 ARG A  71      -2.120   2.669  14.392  1.00  0.75           N  \\nATOM    525  N   ILE A  72      -6.853   7.164  16.745  1.00  0.80           N  \\nATOM    526  CA  ILE A  72      -6.619   6.844  18.147  1.00  0.80           C  \\nATOM    527  C   ILE A  72      -7.801   6.152  18.787  1.00  0.80           C  \\nATOM    528  O   ILE A  72      -7.702   5.683  19.932  1.00  0.80           O  \\nATOM    529  CB  ILE A  72      -6.211   8.062  18.966  1.00  0.80           C  \\nATOM    530  CG1 ILE A  72      -7.349   9.096  19.114  1.00  0.80           C  \\nATOM    531  CG2 ILE A  72      -4.934   8.653  18.333  1.00  0.80           C  \\nATOM    532  CD1 ILE A  72      -7.005  10.257  20.046  1.00  0.80           C  \\nATOM    533  N   ALA A  73      -8.925   6.030  18.066  1.00  0.82           N  \\nATOM    534  CA  ALA A  73     -10.076   5.244  18.461  1.00  0.82           C  \\nATOM    535  C   ALA A  73      -9.932   3.782  18.051  1.00  0.82           C  \\nATOM    536  O   ALA A  73     -10.702   2.925  18.499  1.00  0.82           O  \\nATOM    537  CB  ALA A  73     -11.351   5.820  17.811  1.00  0.82           C  \\nATOM    538  N   GLN A  74      -8.943   3.457  17.199  1.00  0.78           N  \\nATOM    539  CA  GLN A  74      -8.627   2.091  16.811  1.00  0.78           C  \\nATOM    540  C   GLN A  74      -8.084   1.254  17.970  1.00  0.78           C  \\nATOM    541  O   GLN A  74      -7.470   1.738  18.913  1.00  0.78           O  \\nATOM    542  CB  GLN A  74      -7.665   1.995  15.598  1.00  0.78           C  \\nATOM    543  CG  GLN A  74      -8.083   2.792  14.335  1.00  0.78           C  \\nATOM    544  CD  GLN A  74      -9.524   2.511  13.902  1.00  0.78           C  \\nATOM    545  OE1 GLN A  74      -9.859   1.421  13.457  1.00  0.78           O  \\nATOM    546  NE2 GLN A  74     -10.402   3.539  14.007  1.00  0.78           N  \\nATOM    547  N   GLU A  75      -8.311  -0.074  17.911  1.00  0.78           N  \\nATOM    548  CA  GLU A  75      -8.023  -1.032  18.961  1.00  0.78           C  \\nATOM    549  C   GLU A  75      -6.561  -1.104  19.375  1.00  0.78           C  \\nATOM    550  O   GLU A  75      -6.262  -1.421  20.537  1.00  0.78           O  \\nATOM    551  CB  GLU A  75      -8.479  -2.459  18.550  1.00  0.78           C  \\nATOM    552  CG  GLU A  75      -9.999  -2.624  18.289  1.00  0.78           C  \\nATOM    553  CD  GLU A  75     -10.484  -2.008  16.972  1.00  0.78           C  \\nATOM    554  OE1 GLU A  75      -9.626  -1.627  16.136  1.00  0.78           O  \\nATOM    555  OE2 GLU A  75     -11.723  -1.892  16.834  1.00  0.78           O  \\nATOM    556  N   ASP A  76      -5.623  -0.805  18.456  1.00  0.86           N  \\nATOM    557  CA  ASP A  76      -4.188  -0.802  18.691  1.00  0.86           C  \\nATOM    558  C   ASP A  76      -3.744   0.340  19.604  1.00  0.86           C  \\nATOM    559  O   ASP A  76      -2.708   0.247  20.275  1.00  0.86           O  \\nATOM    560  CB  ASP A  76      -3.377  -0.902  17.368  1.00  0.86           C  \\nATOM    561  CG  ASP A  76      -3.480   0.285  16.453  1.00  0.86           C  \\nATOM    562  OD1 ASP A  76      -4.435   1.104  16.526  1.00  0.86           O  \\nATOM    563  OD2 ASP A  76      -2.554   0.405  15.600  1.00  0.86           O  \\nATOM    564  N   CYS A  77      -4.555   1.407  19.729  1.00  0.88           N  \\nATOM    565  CA  CYS A  77      -4.266   2.568  20.552  1.00  0.88           C  \\nATOM    566  C   CYS A  77      -4.921   2.497  21.930  1.00  0.88           C  \\nATOM    567  O   CYS A  77      -4.973   3.482  22.673  1.00  0.88           O  \\nATOM    568  CB  CYS A  77      -4.663   3.888  19.841  1.00  0.88           C  \\nATOM    569  SG  CYS A  77      -3.657   4.226  18.354  1.00  0.88           S  \\nATOM    570  N   ARG A  78      -5.394   1.304  22.354  1.00  0.78           N  \\nATOM    571  CA  ARG A  78      -5.975   1.053  23.673  1.00  0.78           C  \\nATOM    572  C   ARG A  78      -5.043   1.366  24.835  1.00  0.78           C  \\nATOM    573  O   ARG A  78      -5.463   1.904  25.865  1.00  0.78           O  \\nATOM    574  CB  ARG A  78      -6.470  -0.413  23.789  1.00  0.78           C  \\nATOM    575  CG  ARG A  78      -7.220  -0.709  25.107  1.00  0.78           C  \\nATOM    576  CD  ARG A  78      -8.148  -1.931  25.115  1.00  0.78           C  \\nATOM    577  NE  ARG A  78      -9.204  -1.703  24.068  1.00  0.78           N  \\nATOM    578  CZ  ARG A  78      -9.289  -2.356  22.900  1.00  0.78           C  \\nATOM    579  NH1 ARG A  78     -10.192  -1.983  21.994  1.00  0.78           N  \\nATOM    580  NH2 ARG A  78      -8.451  -3.336  22.586  1.00  0.78           N  \\nATOM    581  N   ASN A  79      -3.738   1.093  24.681  1.00  0.82           N  \\nATOM    582  CA  ASN A  79      -2.751   1.370  25.706  1.00  0.82           C  \\nATOM    583  C   ASN A  79      -2.205   2.787  25.550  1.00  0.82           C  \\nATOM    584  O   ASN A  79      -1.388   3.253  26.352  1.00  0.82           O  \\nATOM    585  CB  ASN A  79      -1.551   0.398  25.590  1.00  0.82           C  \\nATOM    586  CG  ASN A  79      -1.979  -1.052  25.792  1.00  0.82           C  \\nATOM    587  OD1 ASN A  79      -3.028  -1.371  26.347  1.00  0.82           O  \\nATOM    588  ND2 ASN A  79      -1.115  -1.977  25.308  1.00  0.82           N  \\nATOM    589  N   GLY A  80      -2.656   3.537  24.526  1.00  0.91           N  \\nATOM    590  CA  GLY A  80      -2.143   4.850  24.178  1.00  0.91           C  \\nATOM    591  C   GLY A  80      -1.504   4.880  22.820  1.00  0.91           C  \\nATOM    592  O   GLY A  80      -1.747   4.028  21.974  1.00  0.91           O  \\nATOM    593  N   PHE A  81      -0.673   5.899  22.567  1.00  0.93           N  \\nATOM    594  CA  PHE A  81      -0.081   6.090  21.262  1.00  0.93           C  \\nATOM    595  C   PHE A  81       1.117   7.007  21.365  1.00  0.93           C  \\nATOM    596  O   PHE A  81       1.245   7.804  22.296  1.00  0.93           O  \\nATOM    597  CB  PHE A  81      -1.079   6.630  20.197  1.00  0.93           C  \\nATOM    598  CG  PHE A  81      -1.783   7.884  20.637  1.00  0.93           C  \\nATOM    599  CD1 PHE A  81      -1.189   9.145  20.462  1.00  0.93           C  \\nATOM    600  CD2 PHE A  81      -3.052   7.807  21.232  1.00  0.93           C  \\nATOM    601  CE1 PHE A  81      -1.846  10.305  20.891  1.00  0.93           C  \\nATOM    602  CE2 PHE A  81      -3.711   8.967  21.648  1.00  0.93           C  \\nATOM    603  CZ  PHE A  81      -3.110  10.216  21.481  1.00  0.93           C  \\nATOM    604  N   LEU A  82       2.017   6.918  20.377  1.00  0.95           N  \\nATOM    605  CA  LEU A  82       3.108   7.848  20.205  1.00  0.95           C  \\nATOM    606  C   LEU A  82       2.823   8.637  18.943  1.00  0.95           C  \\nATOM    607  O   LEU A  82       2.886   8.123  17.827  1.00  0.95           O  \\nATOM    608  CB  LEU A  82       4.440   7.076  20.106  1.00  0.95           C  \\nATOM    609  CG  LEU A  82       5.688   7.902  19.758  1.00  0.95           C  \\nATOM    610  CD1 LEU A  82       5.988   8.973  20.807  1.00  0.95           C  \\nATOM    611  CD2 LEU A  82       6.896   6.977  19.559  1.00  0.95           C  \\nATOM    612  N   LEU A  83       2.467   9.920  19.085  1.00  0.94           N  \\nATOM    613  CA  LEU A  83       2.127  10.790  17.980  1.00  0.94           C  \\nATOM    614  C   LEU A  83       3.374  11.552  17.579  1.00  0.94           C  \\nATOM    615  O   LEU A  83       3.913  12.379  18.309  1.00  0.94           O  \\nATOM    616  CB  LEU A  83       0.941  11.712  18.351  1.00  0.94           C  \\nATOM    617  CG  LEU A  83       0.492  12.756  17.307  1.00  0.94           C  \\nATOM    618  CD1 LEU A  83       0.144  12.153  15.943  1.00  0.94           C  \\nATOM    619  CD2 LEU A  83      -0.715  13.537  17.847  1.00  0.94           C  \\nATOM    620  N   ASP A  84       3.890  11.221  16.386  1.00  0.91           N  \\nATOM    621  CA  ASP A  84       5.167  11.698  15.919  1.00  0.91           C  \\nATOM    622  C   ASP A  84       5.015  12.944  15.044  1.00  0.91           C  \\nATOM    623  O   ASP A  84       4.424  12.922  13.962  1.00  0.91           O  \\nATOM    624  CB  ASP A  84       5.860  10.516  15.195  1.00  0.91           C  \\nATOM    625  CG  ASP A  84       7.319  10.776  14.897  1.00  0.91           C  \\nATOM    626  OD1 ASP A  84       7.990   9.855  14.373  1.00  0.91           O  \\nATOM    627  OD2 ASP A  84       7.810  11.898  15.178  1.00  0.91           O  \\nATOM    628  N   GLY A  85       5.567  14.089  15.510  1.00  0.85           N  \\nATOM    629  CA  GLY A  85       5.732  15.285  14.683  1.00  0.85           C  \\nATOM    630  C   GLY A  85       4.484  16.064  14.390  1.00  0.85           C  \\nATOM    631  O   GLY A  85       4.410  16.786  13.379  1.00  0.85           O  \\nATOM    632  N   PHE A  86       3.465  15.952  15.240  1.00  0.88           N  \\nATOM    633  CA  PHE A  86       2.234  16.691  15.108  1.00  0.88           C  \\nATOM    634  C   PHE A  86       1.711  16.961  16.512  1.00  0.88           C  \\nATOM    635  O   PHE A  86       1.875  16.090  17.370  1.00  0.88           O  \\nATOM    636  CB  PHE A  86       1.218  15.882  14.260  1.00  0.88           C  \\nATOM    637  CG  PHE A  86      -0.038  16.655  13.968  1.00  0.88           C  \\nATOM    638  CD1 PHE A  86      -0.064  17.582  12.915  1.00  0.88           C  \\nATOM    639  CD2 PHE A  86      -1.191  16.490  14.752  1.00  0.88           C  \\nATOM    640  CE1 PHE A  86      -1.234  18.286  12.615  1.00  0.88           C  \\nATOM    641  CE2 PHE A  86      -2.348  17.231  14.483  1.00  0.88           C  \\nATOM    642  CZ  PHE A  86      -2.376  18.115  13.401  1.00  0.88           C  \\nATOM    643  N   PRO A  87       1.066  18.082  16.830  1.00  0.91           N  \\nATOM    644  CA  PRO A  87       0.908  19.314  16.053  1.00  0.91           C  \\nATOM    645  C   PRO A  87       2.165  19.988  15.608  1.00  0.91           C  \\nATOM    646  O   PRO A  87       3.234  19.742  16.152  1.00  0.91           O  \\nATOM    647  CB  PRO A  87       0.132  20.254  16.974  1.00  0.91           C  \\nATOM    648  CG  PRO A  87      -0.702  19.268  17.764  1.00  0.91           C  \\nATOM    649  CD  PRO A  87       0.273  18.129  18.042  1.00  0.91           C  \\nATOM    650  N   ARG A  88       2.027  20.858  14.602  1.00  0.77           N  \\nATOM    651  CA  ARG A  88       3.120  21.668  14.132  1.00  0.77           C  \\nATOM    652  C   ARG A  88       2.849  23.137  14.378  1.00  0.77           C  \\nATOM    653  O   ARG A  88       3.755  23.969  14.298  1.00  0.77           O  \\nATOM    654  CB  ARG A  88       3.271  21.401  12.622  1.00  0.77           C  \\nATOM    655  CG  ARG A  88       3.620  19.928  12.330  1.00  0.77           C  \\nATOM    656  CD  ARG A  88       3.870  19.626  10.853  1.00  0.77           C  \\nATOM    657  NE  ARG A  88       2.561  19.264  10.219  1.00  0.77           N  \\nATOM    658  CZ  ARG A  88       2.129  18.010  10.011  1.00  0.77           C  \\nATOM    659  NH1 ARG A  88       2.806  16.955  10.461  1.00  0.77           N  \\nATOM    660  NH2 ARG A  88       0.959  17.841   9.405  1.00  0.77           N  \\nATOM    661  N   THR A  89       1.607  23.510  14.732  1.00  0.81           N  \\nATOM    662  CA  THR A  89       1.218  24.897  14.938  1.00  0.81           C  \\nATOM    663  C   THR A  89       0.318  24.991  16.158  1.00  0.81           C  \\nATOM    664  O   THR A  89      -0.258  23.993  16.592  1.00  0.81           O  \\nATOM    665  CB  THR A  89       0.501  25.531  13.732  1.00  0.81           C  \\nATOM    666  OG1 THR A  89      -0.804  25.002  13.517  1.00  0.81           O  \\nATOM    667  CG2 THR A  89       1.292  25.296  12.439  1.00  0.81           C  \\nATOM    668  N   ILE A  90       0.148  26.203  16.752  1.00  0.84           N  \\nATOM    669  CA  ILE A  90      -0.893  26.436  17.765  1.00  0.84           C  \\nATOM    670  C   ILE A  90      -2.311  26.151  17.240  1.00  0.84           C  \\nATOM    671  O   ILE A  90      -2.998  25.380  17.906  1.00  0.84           O  \\nATOM    672  CB  ILE A  90      -0.793  27.805  18.460  1.00  0.84           C  \\nATOM    673  CG1 ILE A  90       0.492  27.892  19.314  1.00  0.84           C  \\nATOM    674  CG2 ILE A  90      -2.042  28.072  19.331  1.00  0.84           C  \\nATOM    675  CD1 ILE A  90       0.662  29.240  20.027  1.00  0.84           C  \\nATOM    676  N   PRO A  91      -2.793  26.603  16.068  1.00  0.85           N  \\nATOM    677  CA  PRO A  91      -4.132  26.257  15.581  1.00  0.85           C  \\nATOM    678  C   PRO A  91      -4.429  24.770  15.483  1.00  0.85           C  \\nATOM    679  O   PRO A  91      -5.574  24.365  15.715  1.00  0.85           O  \\nATOM    680  CB  PRO A  91      -4.205  26.957  14.218  1.00  0.85           C  \\nATOM    681  CG  PRO A  91      -3.361  28.215  14.406  1.00  0.85           C  \\nATOM    682  CD  PRO A  91      -2.223  27.715  15.295  1.00  0.85           C  \\nATOM    683  N   GLN A  92      -3.438  23.931  15.144  1.00  0.83           N  \\nATOM    684  CA  GLN A  92      -3.554  22.484  15.191  1.00  0.83           C  \\nATOM    685  C   GLN A  92      -3.678  21.917  16.603  1.00  0.83           C  \\nATOM    686  O   GLN A  92      -4.494  21.024  16.860  1.00  0.83           O  \\nATOM    687  CB  GLN A  92      -2.365  21.849  14.443  1.00  0.83           C  \\nATOM    688  CG  GLN A  92      -2.458  22.087  12.924  1.00  0.83           C  \\nATOM    689  CD  GLN A  92      -1.177  21.703  12.196  1.00  0.83           C  \\nATOM    690  OE1 GLN A  92      -0.142  21.355  12.782  1.00  0.83           O  \\nATOM    691  NE2 GLN A  92      -1.241  21.756  10.847  1.00  0.83           N  \\nATOM    692  N   ALA A  93      -2.893  22.426  17.570  1.00  0.92           N  \\nATOM    693  CA  ALA A  93      -3.009  22.087  18.980  1.00  0.92           C  \\nATOM    694  C   ALA A  93      -4.347  22.509  19.589  1.00  0.92           C  \\nATOM    695  O   ALA A  93      -4.990  21.732  20.311  1.00  0.92           O  \\nATOM    696  CB  ALA A  93      -1.824  22.672  19.770  1.00  0.92           C  \\nATOM    697  N   ASP A  94      -4.839  23.712  19.246  1.00  0.87           N  \\nATOM    698  CA  ASP A  94      -6.165  24.202  19.591  1.00  0.87           C  \\nATOM    699  C   ASP A  94      -7.272  23.297  19.065  1.00  0.87           C  \\nATOM    700  O   ASP A  94      -8.185  22.899  19.802  1.00  0.87           O  \\nATOM    701  CB  ASP A  94      -6.379  25.614  18.996  1.00  0.87           C  \\nATOM    702  CG  ASP A  94      -5.529  26.673  19.684  1.00  0.87           C  \\nATOM    703  OD1 ASP A  94      -5.095  26.446  20.846  1.00  0.87           O  \\nATOM    704  OD2 ASP A  94      -5.335  27.739  19.046  1.00  0.87           O  \\nATOM    705  N   ALA A  95      -7.188  22.880  17.794  1.00  0.89           N  \\nATOM    706  CA  ALA A  95      -8.115  21.954  17.175  1.00  0.89           C  \\nATOM    707  C   ALA A  95      -8.141  20.563  17.818  1.00  0.89           C  \\nATOM    708  O   ALA A  95      -9.207  19.941  17.941  1.00  0.89           O  \\nATOM    709  CB  ALA A  95      -7.840  21.872  15.666  1.00  0.89           C  \\nATOM    710  N   MET A  96      -6.993  20.031  18.274  1.00  0.83           N  \\nATOM    711  CA  MET A  96      -6.935  18.821  19.087  1.00  0.83           C  \\nATOM    712  C   MET A  96      -7.641  18.960  20.415  1.00  0.83           C  \\nATOM    713  O   MET A  96      -8.362  18.045  20.843  1.00  0.83           O  \\nATOM    714  CB  MET A  96      -5.488  18.373  19.369  1.00  0.83           C  \\nATOM    715  CG  MET A  96      -4.822  17.801  18.116  1.00  0.83           C  \\nATOM    716  SD  MET A  96      -3.154  17.137  18.349  1.00  0.83           S  \\nATOM    717  CE  MET A  96      -3.335  15.709  19.446  1.00  0.83           C  \\nATOM    718  N   LYS A  97      -7.487  20.095  21.102  1.00  0.81           N  \\nATOM    719  CA  LYS A  97      -8.231  20.369  22.320  1.00  0.81           C  \\nATOM    720  C   LYS A  97      -9.743  20.449  22.101  1.00  0.81           C  \\nATOM    721  O   LYS A  97     -10.507  19.794  22.829  1.00  0.81           O  \\nATOM    722  CB  LYS A  97      -7.719  21.654  23.007  1.00  0.81           C  \\nATOM    723  CG  LYS A  97      -8.451  21.960  24.323  1.00  0.81           C  \\nATOM    724  CD  LYS A  97      -7.872  23.156  25.092  1.00  0.81           C  \\nATOM    725  CE  LYS A  97      -6.496  22.876  25.687  1.00  0.81           C  \\nATOM    726  NZ  LYS A  97      -5.994  24.039  26.436  1.00  0.81           N  \\nATOM    727  N   GLU A  98     -10.203  21.166  21.061  1.00  0.79           N  \\nATOM    728  CA  GLU A  98     -11.604  21.284  20.654  1.00  0.79           C  \\nATOM    729  C   GLU A  98     -12.224  19.938  20.313  1.00  0.79           C  \\nATOM    730  O   GLU A  98     -13.396  19.654  20.630  1.00  0.79           O  \\nATOM    731  CB  GLU A  98     -11.746  22.248  19.444  1.00  0.79           C  \\nATOM    732  CG  GLU A  98     -11.464  23.734  19.775  1.00  0.79           C  \\nATOM    733  CD  GLU A  98     -12.552  24.323  20.671  1.00  0.79           C  \\nATOM    734  OE1 GLU A  98     -13.658  24.611  20.146  1.00  0.79           O  \\nATOM    735  OE2 GLU A  98     -12.273  24.488  21.887  1.00  0.79           O  \\nATOM    736  N   ALA A  99     -11.468  19.033  19.682  1.00  0.83           N  \\nATOM    737  CA  ALA A  99     -11.898  17.691  19.348  1.00  0.83           C  \\nATOM    738  C   ALA A  99     -11.873  16.703  20.518  1.00  0.83           C  \\nATOM    739  O   ALA A  99     -12.441  15.598  20.412  1.00  0.83           O  \\nATOM    740  CB  ALA A  99     -11.017  17.163  18.204  1.00  0.83           C  \\nATOM    741  N   GLY A 100     -11.235  17.060  21.647  1.00  0.83           N  \\nATOM    742  CA  GLY A 100     -11.119  16.238  22.852  1.00  0.83           C  \\nATOM    743  C   GLY A 100      -9.917  15.338  22.902  1.00  0.83           C  \\nATOM    744  O   GLY A 100      -9.936  14.283  23.551  1.00  0.83           O  \\nATOM    745  N   ILE A 101      -8.811  15.701  22.242  1.00  0.80           N  \\nATOM    746  CA  ILE A 101      -7.614  14.886  22.187  1.00  0.80           C  \\nATOM    747  C   ILE A 101      -6.540  15.473  23.081  1.00  0.80           C  \\nATOM    748  O   ILE A 101      -5.733  16.333  22.692  1.00  0.80           O  \\nATOM    749  CB  ILE A 101      -7.104  14.677  20.765  1.00  0.80           C  \\nATOM    750  CG1 ILE A 101      -8.252  14.226  19.829  1.00  0.80           C  \\nATOM    751  CG2 ILE A 101      -5.953  13.646  20.786  1.00  0.80           C  \\nATOM    752  CD1 ILE A 101      -7.870  14.253  18.349  1.00  0.80           C  \\nATOM    753  N   ASN A 102      -6.465  14.985  24.324  1.00  0.84           N  \\nATOM    754  CA  ASN A 102      -5.418  15.289  25.275  1.00  0.84           C  \\nATOM    755  C   ASN A 102      -4.344  14.193  25.203  1.00  0.84           C  \\nATOM    756  O   ASN A 102      -4.615  13.069  24.766  1.00  0.84           O  \\nATOM    757  CB  ASN A 102      -6.082  15.414  26.669  1.00  0.84           C  \\nATOM    758  CG  ASN A 102      -5.097  15.644  27.812  1.00  0.84           C  \\nATOM    759  OD1 ASN A 102      -4.773  14.694  28.494  1.00  0.84           O  \\nATOM    760  ND2 ASN A 102      -4.608  16.895  27.993  1.00  0.84           N  \\nATOM    761  N   VAL A 103      -3.097  14.499  25.618  1.00  0.90           N  \\nATOM    762  CA  VAL A 103      -2.009  13.549  25.748  1.00  0.90           C  \\nATOM    763  C   VAL A 103      -1.363  13.782  27.097  1.00  0.90           C  \\nATOM    764  O   VAL A 103      -1.479  14.870  27.659  1.00  0.90           O  \\nATOM    765  CB  VAL A 103      -0.936  13.631  24.654  1.00  0.90           C  \\nATOM    766  CG1 VAL A 103      -1.601  13.413  23.284  1.00  0.90           C  \\nATOM    767  CG2 VAL A 103      -0.122  14.937  24.719  1.00  0.90           C  \\nATOM    768  N   ASP A 104      -0.668  12.753  27.624  1.00  0.91           N  \\nATOM    769  CA  ASP A 104      -0.039  12.735  28.926  1.00  0.91           C  \\nATOM    770  C   ASP A 104       1.285  13.497  28.931  1.00  0.91           C  \\nATOM    771  O   ASP A 104       1.622  14.209  29.886  1.00  0.91           O  \\nATOM    772  CB  ASP A 104       0.223  11.258  29.334  1.00  0.91           C  \\nATOM    773  CG  ASP A 104      -1.029  10.384  29.311  1.00  0.91           C  \\nATOM    774  OD1 ASP A 104      -2.010  10.676  30.029  1.00  0.91           O  \\nATOM    775  OD2 ASP A 104      -0.999   9.354  28.579  1.00  0.91           O  \\nATOM    776  N   TYR A 105       2.078  13.352  27.850  1.00  0.92           N  \\nATOM    777  CA  TYR A 105       3.390  13.962  27.727  1.00  0.92           C  \\nATOM    778  C   TYR A 105       3.600  14.559  26.342  1.00  0.92           C  \\nATOM    779  O   TYR A 105       3.283  13.979  25.311  1.00  0.92           O  \\nATOM    780  CB  TYR A 105       4.582  12.985  27.966  1.00  0.92           C  \\nATOM    781  CG  TYR A 105       4.688  12.560  29.404  1.00  0.92           C  \\nATOM    782  CD1 TYR A 105       5.553  13.190  30.318  1.00  0.92           C  \\nATOM    783  CD2 TYR A 105       3.910  11.491  29.848  1.00  0.92           C  \\nATOM    784  CE1 TYR A 105       5.577  12.784  31.666  1.00  0.92           C  \\nATOM    785  CE2 TYR A 105       3.941  11.078  31.186  1.00  0.92           C  \\nATOM    786  CZ  TYR A 105       4.761  11.747  32.092  1.00  0.92           C  \\nATOM    787  OH  TYR A 105       4.706  11.369  33.463  1.00  0.92           O  \\nATOM    788  N   VAL A 106       4.205  15.758  26.312  1.00  0.93           N  \\nATOM    789  CA  VAL A 106       4.753  16.366  25.118  1.00  0.93           C  \\nATOM    790  C   VAL A 106       6.261  16.400  25.298  1.00  0.93           C  \\nATOM    791  O   VAL A 106       6.794  16.998  26.231  1.00  0.93           O  \\nATOM    792  CB  VAL A 106       4.226  17.773  24.877  1.00  0.93           C  \\nATOM    793  CG1 VAL A 106       4.829  18.347  23.581  1.00  0.93           C  \\nATOM    794  CG2 VAL A 106       2.690  17.718  24.792  1.00  0.93           C  \\nATOM    795  N   LEU A 107       6.997  15.723  24.403  1.00  0.93           N  \\nATOM    796  CA  LEU A 107       8.436  15.609  24.497  1.00  0.93           C  \\nATOM    797  C   LEU A 107       9.074  16.375  23.343  1.00  0.93           C  \\nATOM    798  O   LEU A 107       8.794  16.119  22.177  1.00  0.93           O  \\nATOM    799  CB  LEU A 107       8.871  14.123  24.451  1.00  0.93           C  \\nATOM    800  CG  LEU A 107       8.239  13.218  25.529  1.00  0.93           C  \\nATOM    801  CD1 LEU A 107       8.668  11.762  25.300  1.00  0.93           C  \\nATOM    802  CD2 LEU A 107       8.596  13.662  26.952  1.00  0.93           C  \\nATOM    803  N   GLU A 108       9.957  17.348  23.654  1.00  0.90           N  \\nATOM    804  CA  GLU A 108      10.689  18.126  22.664  1.00  0.90           C  \\nATOM    805  C   GLU A 108      12.075  17.520  22.557  1.00  0.90           C  \\nATOM    806  O   GLU A 108      12.842  17.475  23.519  1.00  0.90           O  \\nATOM    807  CB  GLU A 108      10.735  19.642  23.036  1.00  0.90           C  \\nATOM    808  CG  GLU A 108      11.505  20.596  22.071  1.00  0.90           C  \\nATOM    809  CD  GLU A 108      11.567  22.069  22.506  1.00  0.90           C  \\nATOM    810  OE1 GLU A 108      11.198  22.379  23.672  1.00  0.90           O  \\nATOM    811  OE2 GLU A 108      11.967  22.942  21.685  1.00  0.90           O  \\nATOM    812  N   PHE A 109      12.413  16.949  21.392  1.00  0.89           N  \\nATOM    813  CA  PHE A 109      13.698  16.330  21.147  1.00  0.89           C  \\nATOM    814  C   PHE A 109      14.636  17.344  20.494  1.00  0.89           C  \\nATOM    815  O   PHE A 109      14.556  17.615  19.299  1.00  0.89           O  \\nATOM    816  CB  PHE A 109      13.466  15.090  20.251  1.00  0.89           C  \\nATOM    817  CG  PHE A 109      14.673  14.212  20.111  1.00  0.89           C  \\nATOM    818  CD1 PHE A 109      14.843  13.109  20.961  1.00  0.89           C  \\nATOM    819  CD2 PHE A 109      15.615  14.442  19.095  1.00  0.89           C  \\nATOM    820  CE1 PHE A 109      15.910  12.226  20.769  1.00  0.89           C  \\nATOM    821  CE2 PHE A 109      16.699  13.573  18.922  1.00  0.89           C  \\nATOM    822  CZ  PHE A 109      16.838  12.455  19.751  1.00  0.89           C  \\nATOM    823  N   ASP A 110      15.559  17.904  21.308  1.00  0.87           N  \\nATOM    824  CA  ASP A 110      16.429  18.996  20.923  1.00  0.87           C  \\nATOM    825  C   ASP A 110      17.774  18.505  20.429  1.00  0.87           C  \\nATOM    826  O   ASP A 110      18.454  17.701  21.049  1.00  0.87           O  \\nATOM    827  CB  ASP A 110      16.697  19.951  22.116  1.00  0.87           C  \\nATOM    828  CG  ASP A 110      15.450  20.750  22.431  1.00  0.87           C  \\nATOM    829  OD1 ASP A 110      14.761  21.117  21.451  1.00  0.87           O  \\nATOM    830  OD2 ASP A 110      15.211  21.049  23.633  1.00  0.87           O  \\nATOM    831  N   VAL A 111      18.204  19.029  19.266  1.00  0.87           N  \\nATOM    832  CA  VAL A 111      19.552  18.866  18.765  1.00  0.87           C  \\nATOM    833  C   VAL A 111      19.916  20.270  18.278  1.00  0.87           C  \\nATOM    834  O   VAL A 111      19.121  20.831  17.526  1.00  0.87           O  \\nATOM    835  CB  VAL A 111      19.652  17.831  17.643  1.00  0.87           C  \\nATOM    836  CG1 VAL A 111      21.079  17.767  17.068  1.00  0.87           C  \\nATOM    837  CG2 VAL A 111      19.277  16.446  18.202  1.00  0.87           C  \\nATOM    838  N   PRO A 112      21.005  20.933  18.678  1.00  0.87           N  \\nATOM    839  CA  PRO A 112      21.489  22.183  18.072  1.00  0.87           C  \\nATOM    840  C   PRO A 112      21.556  22.192  16.549  1.00  0.87           C  \\nATOM    841  O   PRO A 112      21.983  21.196  15.973  1.00  0.87           O  \\nATOM    842  CB  PRO A 112      22.894  22.367  18.678  1.00  0.87           C  \\nATOM    843  CG  PRO A 112      22.875  21.566  19.979  1.00  0.87           C  \\nATOM    844  CD  PRO A 112      21.963  20.395  19.641  1.00  0.87           C  \\nATOM    845  N   ASP A 113      21.226  23.326  15.889  1.00  0.83           N  \\nATOM    846  CA  ASP A 113      21.188  23.457  14.438  1.00  0.83           C  \\nATOM    847  C   ASP A 113      22.471  23.049  13.739  1.00  0.83           C  \\nATOM    848  O   ASP A 113      22.435  22.282  12.771  1.00  0.83           O  \\nATOM    849  CB  ASP A 113      20.874  24.926  14.050  1.00  0.83           C  \\nATOM    850  CG  ASP A 113      19.447  25.299  14.403  1.00  0.83           C  \\nATOM    851  OD1 ASP A 113      18.639  24.384  14.704  1.00  0.83           O  \\nATOM    852  OD2 ASP A 113      19.143  26.516  14.379  1.00  0.83           O  \\nATOM    853  N   GLU A 114      23.646  23.470  14.234  1.00  0.81           N  \\nATOM    854  CA  GLU A 114      24.935  23.092  13.667  1.00  0.81           C  \\nATOM    855  C   GLU A 114      25.145  21.583  13.684  1.00  0.81           C  \\nATOM    856  O   GLU A 114      25.558  20.976  12.676  1.00  0.81           O  \\nATOM    857  CB  GLU A 114      26.098  23.854  14.363  1.00  0.81           C  \\nATOM    858  CG  GLU A 114      26.123  25.381  14.098  1.00  0.81           C  \\nATOM    859  CD  GLU A 114      25.994  25.760  12.637  1.00  0.81           C  \\nATOM    860  OE1 GLU A 114      26.671  25.192  11.738  1.00  0.81           O  \\nATOM    861  OE2 GLU A 114      25.158  26.671  12.379  1.00  0.81           O  \\nATOM    862  N   LEU A 115      24.775  20.887  14.764  1.00  0.85           N  \\nATOM    863  CA  LEU A 115      24.868  19.440  14.859  1.00  0.85           C  \\nATOM    864  C   LEU A 115      23.859  18.711  13.970  1.00  0.85           C  \\nATOM    865  O   LEU A 115      24.134  17.622  13.457  1.00  0.85           O  \\nATOM    866  CB  LEU A 115      24.764  18.967  16.326  1.00  0.85           C  \\nATOM    867  CG  LEU A 115      25.826  19.584  17.263  1.00  0.85           C  \\nATOM    868  CD1 LEU A 115      25.612  19.135  18.713  1.00  0.85           C  \\nATOM    869  CD2 LEU A 115      27.259  19.244  16.833  1.00  0.85           C  \\nATOM    870  N   ILE A 116      22.666  19.295  13.730  1.00  0.87           N  \\nATOM    871  CA  ILE A 116      21.735  18.834  12.696  1.00  0.87           C  \\nATOM    872  C   ILE A 116      22.322  18.981  11.308  1.00  0.87           C  \\nATOM    873  O   ILE A 116      22.243  18.054  10.486  1.00  0.87           O  \\nATOM    874  CB  ILE A 116      20.376  19.533  12.729  1.00  0.87           C  \\nATOM    875  CG1 ILE A 116      19.674  19.219  14.062  1.00  0.87           C  \\nATOM    876  CG2 ILE A 116      19.510  19.094  11.522  1.00  0.87           C  \\nATOM    877  CD1 ILE A 116      18.327  19.918  14.241  1.00  0.87           C  \\nATOM    878  N   VAL A 117      22.961  20.124  11.006  1.00  0.86           N  \\nATOM    879  CA  VAL A 117      23.654  20.343   9.747  1.00  0.86           C  \\nATOM    880  C   VAL A 117      24.749  19.293   9.556  1.00  0.86           C  \\nATOM    881  O   VAL A 117      24.706  18.544   8.568  1.00  0.86           O  \\nATOM    882  CB  VAL A 117      24.177  21.776   9.608  1.00  0.86           C  \\nATOM    883  CG1 VAL A 117      24.995  21.954   8.319  1.00  0.86           C  \\nATOM    884  CG2 VAL A 117      22.998  22.769   9.575  1.00  0.86           C  \\nATOM    885  N   ASP A 118      25.639  19.069  10.538  1.00  0.82           N  \\nATOM    886  CA  ASP A 118      26.664  18.028  10.535  1.00  0.82           C  \\nATOM    887  C   ASP A 118      26.107  16.617  10.325  1.00  0.82           C  \\nATOM    888  O   ASP A 118      26.612  15.831   9.507  1.00  0.82           O  \\nATOM    889  CB  ASP A 118      27.429  18.001  11.888  1.00  0.82           C  \\nATOM    890  CG  ASP A 118      28.447  19.122  12.047  1.00  0.82           C  \\nATOM    891  OD1 ASP A 118      28.764  19.805  11.046  1.00  0.82           O  \\nATOM    892  OD2 ASP A 118      28.983  19.223  13.184  1.00  0.82           O  \\nATOM    893  N   ARG A 119      25.021  16.260  11.030  1.00  0.79           N  \\nATOM    894  CA  ARG A 119      24.300  14.997  10.939  1.00  0.79           C  \\nATOM    895  C   ARG A 119      23.705  14.719   9.568  1.00  0.79           C  \\nATOM    896  O   ARG A 119      23.632  13.562   9.131  1.00  0.79           O  \\nATOM    897  CB  ARG A 119      23.192  14.947  12.016  1.00  0.79           C  \\nATOM    898  CG  ARG A 119      22.408  13.622  12.094  1.00  0.79           C  \\nATOM    899  CD  ARG A 119      21.465  13.561  13.298  1.00  0.79           C  \\nATOM    900  NE  ARG A 119      20.882  12.178  13.377  1.00  0.79           N  \\nATOM    901  CZ  ARG A 119      19.707  11.794  12.859  1.00  0.79           C  \\nATOM    902  NH1 ARG A 119      18.949  12.605  12.143  1.00  0.79           N  \\nATOM    903  NH2 ARG A 119      19.274  10.557  13.082  1.00  0.79           N  \\nATOM    904  N   ILE A 120      23.242  15.755   8.859  1.00  0.85           N  \\nATOM    905  CA  ILE A 120      22.704  15.629   7.516  1.00  0.85           C  \\nATOM    906  C   ILE A 120      23.745  15.758   6.410  1.00  0.85           C  \\nATOM    907  O   ILE A 120      23.832  14.884   5.540  1.00  0.85           O  \\nATOM    908  CB  ILE A 120      21.531  16.573   7.322  1.00  0.85           C  \\nATOM    909  CG1 ILE A 120      20.400  16.136   8.282  1.00  0.85           C  \\nATOM    910  CG2 ILE A 120      21.066  16.552   5.854  1.00  0.85           C  \\nATOM    911  CD1 ILE A 120      19.078  16.854   8.025  1.00  0.85           C  \\nATOM    912  N   VAL A 121      24.607  16.795   6.391  1.00  0.85           N  \\nATOM    913  CA  VAL A 121      25.450  17.103   5.229  1.00  0.85           C  \\nATOM    914  C   VAL A 121      26.557  16.094   5.018  1.00  0.85           C  \\nATOM    915  O   VAL A 121      27.145  15.951   3.938  1.00  0.85           O  \\nATOM    916  CB  VAL A 121      26.084  18.491   5.268  1.00  0.85           C  \\nATOM    917  CG1 VAL A 121      24.991  19.564   5.403  1.00  0.85           C  \\nATOM    918  CG2 VAL A 121      27.157  18.618   6.367  1.00  0.85           C  \\nATOM    919  N   GLY A 122      26.861  15.322   6.071  1.00  0.87           N  \\nATOM    920  CA  GLY A 122      27.777  14.205   6.012  1.00  0.87           C  \\nATOM    921  C   GLY A 122      27.189  12.949   5.428  1.00  0.87           C  \\nATOM    922  O   GLY A 122      27.936  12.006   5.148  1.00  0.87           O  \\nATOM    923  N   ARG A 123      25.867  12.873   5.198  1.00  0.83           N  \\nATOM    924  CA  ARG A 123      25.236  11.726   4.569  1.00  0.83           C  \\nATOM    925  C   ARG A 123      25.668  11.480   3.133  1.00  0.83           C  \\nATOM    926  O   ARG A 123      25.899  12.399   2.355  1.00  0.83           O  \\nATOM    927  CB  ARG A 123      23.703  11.770   4.594  1.00  0.83           C  \\nATOM    928  CG  ARG A 123      23.078  11.508   5.969  1.00  0.83           C  \\nATOM    929  CD  ARG A 123      21.576  11.706   5.840  1.00  0.83           C  \\nATOM    930  NE  ARG A 123      20.948  11.465   7.169  1.00  0.83           N  \\nATOM    931  CZ  ARG A 123      19.679  11.817   7.400  1.00  0.83           C  \\nATOM    932  NH1 ARG A 123      19.121  11.554   8.575  1.00  0.83           N  \\nATOM    933  NH2 ARG A 123      18.959  12.425   6.455  1.00  0.83           N  \\nATOM    934  N   ARG A 124      25.779  10.207   2.731  1.00  0.84           N  \\nATOM    935  CA  ARG A 124      26.121   9.853   1.370  1.00  0.84           C  \\nATOM    936  C   ARG A 124      25.282   8.643   1.014  1.00  0.84           C  \\nATOM    937  O   ARG A 124      25.063   7.789   1.862  1.00  0.84           O  \\nATOM    938  CB  ARG A 124      27.618   9.473   1.211  1.00  0.84           C  \\nATOM    939  CG  ARG A 124      28.648  10.492   1.742  1.00  0.84           C  \\nATOM    940  CD  ARG A 124      28.931  11.631   0.764  1.00  0.84           C  \\nATOM    941  NE  ARG A 124      29.823  12.633   1.443  1.00  0.84           N  \\nATOM    942  CZ  ARG A 124      29.392  13.664   2.177  1.00  0.84           C  \\nATOM    943  NH1 ARG A 124      28.107  13.877   2.447  1.00  0.84           N  \\nATOM    944  NH2 ARG A 124      30.257  14.538   2.685  1.00  0.84           N  \\nATOM    945  N   VAL A 125      24.770   8.557  -0.229  1.00  0.88           N  \\nATOM    946  CA  VAL A 125      23.862   7.513  -0.669  1.00  0.88           C  \\nATOM    947  C   VAL A 125      24.397   6.836  -1.919  1.00  0.88           C  \\nATOM    948  O   VAL A 125      25.067   7.443  -2.749  1.00  0.88           O  \\nATOM    949  CB  VAL A 125      22.440   8.051  -0.897  1.00  0.88           C  \\nATOM    950  CG1 VAL A 125      22.385   9.049  -2.068  1.00  0.88           C  \\nATOM    951  CG2 VAL A 125      21.391   6.934  -1.074  1.00  0.88           C  \\nATOM    952  N   HIS A 126      24.106   5.530  -2.093  1.00  0.85           N  \\nATOM    953  CA  HIS A 126      24.167   4.885  -3.390  1.00  0.85           C  \\nATOM    954  C   HIS A 126      22.726   4.854  -3.891  1.00  0.85           C  \\nATOM    955  O   HIS A 126      21.905   4.076  -3.399  1.00  0.85           O  \\nATOM    956  CB  HIS A 126      24.784   3.463  -3.279  1.00  0.85           C  \\nATOM    957  CG  HIS A 126      25.062   2.778  -4.575  1.00  0.85           C  \\nATOM    958  ND1 HIS A 126      24.161   1.879  -5.107  1.00  0.85           N  \\nATOM    959  CD2 HIS A 126      26.070   3.051  -5.459  1.00  0.85           C  \\nATOM    960  CE1 HIS A 126      24.631   1.618  -6.325  1.00  0.85           C  \\nATOM    961  NE2 HIS A 126      25.765   2.305  -6.566  1.00  0.85           N  \\nATOM    962  N   ALA A 127      22.337   5.737  -4.835  1.00  0.88           N  \\nATOM    963  CA  ALA A 127      20.959   5.843  -5.315  1.00  0.88           C  \\nATOM    964  C   ALA A 127      20.332   4.581  -5.943  1.00  0.88           C  \\nATOM    965  O   ALA A 127      19.161   4.340  -5.651  1.00  0.88           O  \\nATOM    966  CB  ALA A 127      20.763   7.060  -6.246  1.00  0.88           C  \\nATOM    967  N   PRO A 128      20.985   3.738  -6.749  1.00  0.85           N  \\nATOM    968  CA  PRO A 128      20.415   2.475  -7.228  1.00  0.85           C  \\nATOM    969  C   PRO A 128      19.962   1.495  -6.154  1.00  0.85           C  \\nATOM    970  O   PRO A 128      19.054   0.709  -6.438  1.00  0.85           O  \\nATOM    971  CB  PRO A 128      21.550   1.866  -8.065  1.00  0.85           C  \\nATOM    972  CG  PRO A 128      22.353   3.061  -8.576  1.00  0.85           C  \\nATOM    973  CD  PRO A 128      22.244   4.055  -7.425  1.00  0.85           C  \\nATOM    974  N   SER A 129      20.605   1.474  -4.967  1.00  0.86           N  \\nATOM    975  CA  SER A 129      20.249   0.555  -3.872  1.00  0.86           C  \\nATOM    976  C   SER A 129      19.731   1.197  -2.594  1.00  0.86           C  \\nATOM    977  O   SER A 129      19.264   0.499  -1.695  1.00  0.86           O  \\nATOM    978  CB  SER A 129      21.359  -0.463  -3.551  1.00  0.86           C  \\nATOM    979  OG  SER A 129      22.552   0.157  -2.896  1.00  0.86           O  \\nATOM    980  N   GLY A 130      19.792   2.524  -2.471  1.00  0.91           N  \\nATOM    981  CA  GLY A 130      19.240   3.271  -1.341  1.00  0.91           C  \\nATOM    982  C   GLY A 130      20.051   3.217  -0.075  1.00  0.91           C  \\nATOM    983  O   GLY A 130      19.593   3.678   0.977  1.00  0.91           O  \\nATOM    984  N   ARG A 131      21.273   2.664  -0.122  1.00  0.84           N  \\nATOM    985  CA  ARG A 131      22.142   2.504   1.031  1.00  0.84           C  \\nATOM    986  C   ARG A 131      22.790   3.831   1.405  1.00  0.84           C  \\nATOM    987  O   ARG A 131      23.168   4.594   0.525  1.00  0.84           O  \\nATOM    988  CB  ARG A 131      23.201   1.414   0.745  1.00  0.84           C  \\nATOM    989  CG  ARG A 131      23.676   0.613   1.979  1.00  0.84           C  \\nATOM    990  CD  ARG A 131      24.397  -0.732   1.735  1.00  0.84           C  \\nATOM    991  NE  ARG A 131      24.917  -0.790   0.333  1.00  0.84           N  \\nATOM    992  CZ  ARG A 131      24.258  -1.372  -0.691  1.00  0.84           C  \\nATOM    993  NH1 ARG A 131      23.023  -1.817  -0.564  1.00  0.84           N  \\nATOM    994  NH2 ARG A 131      24.808  -1.414  -1.894  1.00  0.84           N  \\nATOM    995  N   VAL A 132      22.905   4.141   2.716  1.00  0.88           N  \\nATOM    996  CA  VAL A 132      23.321   5.446   3.204  1.00  0.88           C  \\nATOM    997  C   VAL A 132      24.455   5.291   4.216  1.00  0.88           C  \\nATOM    998  O   VAL A 132      24.542   4.298   4.929  1.00  0.88           O  \\nATOM    999  CB  VAL A 132      22.129   6.235   3.768  1.00  0.88           C  \\nATOM   1000  CG1 VAL A 132      21.562   5.549   5.024  1.00  0.88           C  \\nATOM   1001  CG2 VAL A 132      22.458   7.721   4.021  1.00  0.88           C  \\nATOM   1002  N   TYR A 133      25.361   6.291   4.273  1.00  0.84           N  \\nATOM   1003  CA  TYR A 133      26.584   6.323   5.059  1.00  0.84           C  \\nATOM   1004  C   TYR A 133      26.717   7.710   5.676  1.00  0.84           C  \\nATOM   1005  O   TYR A 133      25.940   8.608   5.359  1.00  0.84           O  \\nATOM   1006  CB  TYR A 133      27.832   6.132   4.162  1.00  0.84           C  \\nATOM   1007  CG  TYR A 133      27.817   4.766   3.554  1.00  0.84           C  \\nATOM   1008  CD1 TYR A 133      28.436   3.708   4.227  1.00  0.84           C  \\nATOM   1009  CD2 TYR A 133      27.147   4.510   2.349  1.00  0.84           C  \\nATOM   1010  CE1 TYR A 133      28.378   2.408   3.715  1.00  0.84           C  \\nATOM   1011  CE2 TYR A 133      27.054   3.204   1.841  1.00  0.84           C  \\nATOM   1012  CZ  TYR A 133      27.682   2.163   2.538  1.00  0.84           C  \\nATOM   1013  OH  TYR A 133      27.569   0.825   2.107  1.00  0.84           O  \\nATOM   1014  N   HIS A 134      27.712   7.947   6.566  1.00  0.80           N  \\nATOM   1015  CA  HIS A 134      27.998   9.291   7.050  1.00  0.80           C  \\nATOM   1016  C   HIS A 134      29.494   9.498   7.223  1.00  0.80           C  \\nATOM   1017  O   HIS A 134      30.169   8.658   7.808  1.00  0.80           O  \\nATOM   1018  CB  HIS A 134      27.319   9.550   8.409  1.00  0.80           C  \\nATOM   1019  CG  HIS A 134      27.284  10.989   8.812  1.00  0.80           C  \\nATOM   1020  ND1 HIS A 134      28.424  11.698   9.106  1.00  0.80           N  \\nATOM   1021  CD2 HIS A 134      26.184  11.806   8.828  1.00  0.80           C  \\nATOM   1022  CE1 HIS A 134      28.012  12.951   9.295  1.00  0.80           C  \\nATOM   1023  NE2 HIS A 134      26.681  13.040   9.137  1.00  0.80           N  \\nATOM   1024  N   VAL A 135      30.104  10.595   6.760  1.00  0.86           N  \\nATOM   1025  CA  VAL A 135      31.562  10.750   6.771  1.00  0.86           C  \\nATOM   1026  C   VAL A 135      32.271  10.706   8.118  1.00  0.86           C  \\nATOM   1027  O   VAL A 135      33.396  10.187   8.205  1.00  0.86           O  \\nATOM   1028  CB  VAL A 135      32.031  11.977   6.010  1.00  0.86           C  \\nATOM   1029  CG1 VAL A 135      31.430  11.926   4.599  1.00  0.86           C  \\nATOM   1030  CG2 VAL A 135      31.667  13.286   6.732  1.00  0.86           C  \\nATOM   1031  N   LYS A 136      31.684  11.254   9.185  1.00  0.79           N  \\nATOM   1032  CA  LYS A 136      32.315  11.318  10.491  1.00  0.79           C  \\nATOM   1033  C   LYS A 136      31.703  10.310  11.436  1.00  0.79           C  \\nATOM   1034  O   LYS A 136      32.417   9.693  12.242  1.00  0.79           O  \\nATOM   1035  CB  LYS A 136      32.174  12.751  11.085  1.00  0.79           C  \\nATOM   1036  CG  LYS A 136      32.600  12.862  12.561  1.00  0.79           C  \\nATOM   1037  CD  LYS A 136      32.696  14.292  13.117  1.00  0.79           C  \\nATOM   1038  CE  LYS A 136      34.100  14.887  13.011  1.00  0.79           C  \\nATOM   1039  NZ  LYS A 136      34.210  16.078  13.885  1.00  0.79           N  \\nATOM   1040  N   PHE A 137      30.382  10.081  11.384  1.00  0.70           N  \\nATOM   1041  CA  PHE A 137      29.721   9.301  12.423  1.00  0.70           C  \\nATOM   1042  C   PHE A 137      29.471   7.852  12.026  1.00  0.70           C  \\nATOM   1043  O   PHE A 137      29.183   7.010  12.884  1.00  0.70           O  \\nATOM   1044  CB  PHE A 137      28.358   9.932  12.806  1.00  0.70           C  \\nATOM   1045  CG  PHE A 137      28.466  11.401  13.129  1.00  0.70           C  \\nATOM   1046  CD1 PHE A 137      29.227  11.880  14.211  1.00  0.70           C  \\nATOM   1047  CD2 PHE A 137      27.761  12.326  12.346  1.00  0.70           C  \\nATOM   1048  CE1 PHE A 137      29.288  13.255  14.484  1.00  0.70           C  \\nATOM   1049  CE2 PHE A 137      27.847  13.700  12.594  1.00  0.70           C  \\nATOM   1050  CZ  PHE A 137      28.609  14.166  13.668  1.00  0.70           C  \\nATOM   1051  N   ASN A 138      29.609   7.497  10.742  1.00  0.81           N  \\nATOM   1052  CA  ASN A 138      29.459   6.132  10.260  1.00  0.81           C  \\nATOM   1053  C   ASN A 138      30.178   6.038   8.910  1.00  0.81           C  \\nATOM   1054  O   ASN A 138      29.495   5.967   7.878  1.00  0.81           O  \\nATOM   1055  CB  ASN A 138      27.943   5.750  10.157  1.00  0.81           C  \\nATOM   1056  CG  ASN A 138      27.724   4.327   9.642  1.00  0.81           C  \\nATOM   1057  OD1 ASN A 138      28.480   3.413   9.947  1.00  0.81           O  \\nATOM   1058  ND2 ASN A 138      26.665   4.137   8.809  1.00  0.81           N  \\nATOM   1059  N   PRO A 139      31.522   6.130   8.843  1.00  0.84           N  \\nATOM   1060  CA  PRO A 139      32.255   6.188   7.577  1.00  0.84           C  \\nATOM   1061  C   PRO A 139      32.039   4.947   6.714  1.00  0.84           C  \\nATOM   1062  O   PRO A 139      31.912   3.869   7.300  1.00  0.84           O  \\nATOM   1063  CB  PRO A 139      33.734   6.353   7.982  1.00  0.84           C  \\nATOM   1064  CG  PRO A 139      33.704   6.845   9.430  1.00  0.84           C  \\nATOM   1065  CD  PRO A 139      32.426   6.235   9.997  1.00  0.84           C  \\nATOM   1066  N   PRO A 140      31.975   5.008   5.381  1.00  0.88           N  \\nATOM   1067  CA  PRO A 140      32.029   3.807   4.556  1.00  0.88           C  \\nATOM   1068  C   PRO A 140      33.385   3.102   4.656  1.00  0.88           C  \\nATOM   1069  O   PRO A 140      34.368   3.689   5.100  1.00  0.88           O  \\nATOM   1070  CB  PRO A 140      31.770   4.369   3.151  1.00  0.88           C  \\nATOM   1071  CG  PRO A 140      32.480   5.721   3.157  1.00  0.88           C  \\nATOM   1072  CD  PRO A 140      32.329   6.202   4.605  1.00  0.88           C  \\nATOM   1073  N   LYS A 141      33.473   1.820   4.250  1.00  0.82           N  \\nATOM   1074  CA  LYS A 141      34.728   1.071   4.286  1.00  0.82           C  \\nATOM   1075  C   LYS A 141      35.835   1.592   3.387  1.00  0.82           C  \\nATOM   1076  O   LYS A 141      37.024   1.580   3.765  1.00  0.82           O  \\nATOM   1077  CB  LYS A 141      34.414  -0.371   3.869  1.00  0.82           C  \\nATOM   1078  CG  LYS A 141      35.166  -1.488   4.590  1.00  0.82           C  \\nATOM   1079  CD  LYS A 141      34.703  -2.862   4.066  1.00  0.82           C  \\nATOM   1080  CE  LYS A 141      33.231  -3.233   4.315  1.00  0.82           C  \\nATOM   1081  NZ  LYS A 141      32.967  -3.400   5.757  1.00  0.82           N  \\nATOM   1082  N   VAL A 142      35.499   2.036   2.177  1.00  0.83           N  \\nATOM   1083  CA  VAL A 142      36.409   2.633   1.227  1.00  0.83           C  \\nATOM   1084  C   VAL A 142      35.919   4.064   1.061  1.00  0.83           C  \\nATOM   1085  O   VAL A 142      34.772   4.291   0.672  1.00  0.83           O  \\nATOM   1086  CB  VAL A 142      36.425   1.897  -0.118  1.00  0.83           C  \\nATOM   1087  CG1 VAL A 142      37.513   2.498  -1.028  1.00  0.83           C  \\nATOM   1088  CG2 VAL A 142      36.670   0.387   0.090  1.00  0.83           C  \\nATOM   1089  N   GLU A 143      36.743   5.077   1.400  1.00  0.80           N  \\nATOM   1090  CA  GLU A 143      36.430   6.497   1.270  1.00  0.80           C  \\nATOM   1091  C   GLU A 143      35.682   6.932   0.003  1.00  0.80           C  \\nATOM   1092  O   GLU A 143      36.212   6.914  -1.111  1.00  0.80           O  \\nATOM   1093  CB  GLU A 143      37.733   7.312   1.419  1.00  0.80           C  \\nATOM   1094  CG  GLU A 143      38.134   7.567   2.892  1.00  0.80           C  \\nATOM   1095  CD  GLU A 143      39.620   7.308   3.135  1.00  0.80           C  \\nATOM   1096  OE1 GLU A 143      40.050   6.155   2.869  1.00  0.80           O  \\nATOM   1097  OE2 GLU A 143      40.320   8.248   3.588  1.00  0.80           O  \\nATOM   1098  N   GLY A 144      34.407   7.351   0.163  1.00  0.86           N  \\nATOM   1099  CA  GLY A 144      33.539   7.827  -0.919  1.00  0.86           C  \\nATOM   1100  C   GLY A 144      32.863   6.762  -1.723  1.00  0.86           C  \\nATOM   1101  O   GLY A 144      32.069   7.069  -2.631  1.00  0.86           O  \\nATOM   1102  N   LYS A 145      33.073   5.483  -1.429  1.00  0.79           N  \\nATOM   1103  CA  LYS A 145      32.571   4.398  -2.233  1.00  0.79           C  \\nATOM   1104  C   LYS A 145      31.623   3.549  -1.411  1.00  0.79           C  \\nATOM   1105  O   LYS A 145      31.825   3.316  -0.223  1.00  0.79           O  \\nATOM   1106  CB  LYS A 145      33.747   3.534  -2.742  1.00  0.79           C  \\nATOM   1107  CG  LYS A 145      34.760   4.298  -3.612  1.00  0.79           C  \\nATOM   1108  CD  LYS A 145      34.172   4.668  -4.976  1.00  0.79           C  \\nATOM   1109  CE  LYS A 145      35.174   5.287  -5.943  1.00  0.79           C  \\nATOM   1110  NZ  LYS A 145      34.488   5.577  -7.211  1.00  0.79           N  \\nATOM   1111  N   ASP A 146      30.531   3.054  -2.035  1.00  0.82           N  \\nATOM   1112  CA  ASP A 146      29.695   2.026  -1.436  1.00  0.82           C  \\nATOM   1113  C   ASP A 146      30.551   0.770  -1.285  1.00  0.82           C  \\nATOM   1114  O   ASP A 146      31.165   0.333  -2.258  1.00  0.82           O  \\nATOM   1115  CB  ASP A 146      28.485   1.758  -2.371  1.00  0.82           C  \\nATOM   1116  CG  ASP A 146      27.421   0.855  -1.781  1.00  0.82           C  \\nATOM   1117  OD1 ASP A 146      26.672   0.244  -2.604  1.00  0.82           O  \\nATOM   1118  OD2 ASP A 146      27.277   0.728  -0.531  1.00  0.82           O  \\nATOM   1119  N   ASP A 147      30.648   0.168  -0.081  1.00  0.79           N  \\nATOM   1120  CA  ASP A 147      31.570  -0.934   0.135  1.00  0.79           C  \\nATOM   1121  C   ASP A 147      31.151  -2.233  -0.526  1.00  0.79           C  \\nATOM   1122  O   ASP A 147      31.967  -3.152  -0.700  1.00  0.79           O  \\nATOM   1123  CB  ASP A 147      31.899  -1.147   1.627  1.00  0.79           C  \\nATOM   1124  CG  ASP A 147      30.758  -0.964   2.614  1.00  0.79           C  \\nATOM   1125  OD1 ASP A 147      31.026  -0.292   3.657  1.00  0.79           O  \\nATOM   1126  OD2 ASP A 147      29.680  -1.570   2.392  1.00  0.79           O  \\nATOM   1127  N   VAL A 148      29.892  -2.314  -0.948  1.00  0.85           N  \\nATOM   1128  CA  VAL A 148      29.414  -3.295  -1.897  1.00  0.85           C  \\nATOM   1129  C   VAL A 148      29.933  -3.055  -3.309  1.00  0.85           C  \\nATOM   1130  O   VAL A 148      30.836  -3.758  -3.790  1.00  0.85           O  \\nATOM   1131  CB  VAL A 148      27.891  -3.360  -1.911  1.00  0.85           C  \\nATOM   1132  CG1 VAL A 148      27.411  -4.430  -2.910  1.00  0.85           C  \\nATOM   1133  CG2 VAL A 148      27.383  -3.705  -0.500  1.00  0.85           C  \\nATOM   1134  N   THR A 149      29.348  -2.110  -4.063  1.00  0.85           N  \\nATOM   1135  CA  THR A 149      29.563  -1.994  -5.502  1.00  0.85           C  \\nATOM   1136  C   THR A 149      30.848  -1.272  -5.868  1.00  0.85           C  \\nATOM   1137  O   THR A 149      31.480  -1.581  -6.887  1.00  0.85           O  \\nATOM   1138  CB  THR A 149      28.304  -1.490  -6.247  1.00  0.85           C  \\nATOM   1139  OG1 THR A 149      28.562  -0.867  -7.494  1.00  0.85           O  \\nATOM   1140  CG2 THR A 149      27.507  -0.490  -5.421  1.00  0.85           C  \\nATOM   1141  N   GLY A 150      31.325  -0.302  -5.069  1.00  0.87           N  \\nATOM   1142  CA  GLY A 150      32.473   0.510  -5.467  1.00  0.87           C  \\nATOM   1143  C   GLY A 150      32.148   1.597  -6.461  1.00  0.87           C  \\nATOM   1144  O   GLY A 150      33.037   2.279  -6.982  1.00  0.87           O  \\nATOM   1145  N   GLU A 151      30.858   1.828  -6.744  1.00  0.82           N  \\nATOM   1146  CA  GLU A 151      30.398   3.039  -7.392  1.00  0.82           C  \\nATOM   1147  C   GLU A 151      30.571   4.258  -6.496  1.00  0.82           C  \\nATOM   1148  O   GLU A 151      30.609   4.173  -5.269  1.00  0.82           O  \\nATOM   1149  CB  GLU A 151      28.950   2.941  -7.897  1.00  0.82           C  \\nATOM   1150  CG  GLU A 151      28.811   2.044  -9.147  1.00  0.82           C  \\nATOM   1151  CD  GLU A 151      27.354   1.814  -9.545  1.00  0.82           C  \\nATOM   1152  OE1 GLU A 151      26.480   2.609  -9.113  1.00  0.82           O  \\nATOM   1153  OE2 GLU A 151      27.094   0.808 -10.249  1.00  0.82           O  \\nATOM   1154  N   GLU A 152      30.736   5.433  -7.132  1.00  0.79           N  \\nATOM   1155  CA  GLU A 152      30.856   6.707  -6.456  1.00  0.79           C  \\nATOM   1156  C   GLU A 152      29.599   7.076  -5.689  1.00  0.79           C  \\nATOM   1157  O   GLU A 152      28.514   7.161  -6.256  1.00  0.79           O  \\nATOM   1158  CB  GLU A 152      31.191   7.826  -7.475  1.00  0.79           C  \\nATOM   1159  CG  GLU A 152      31.744   9.101  -6.799  1.00  0.79           C  \\nATOM   1160  CD  GLU A 152      33.025   8.724  -6.078  1.00  0.79           C  \\nATOM   1161  OE1 GLU A 152      33.278   9.172  -4.925  1.00  0.79           O  \\nATOM   1162  OE2 GLU A 152      33.762   7.884  -6.692  1.00  0.79           O  \\nATOM   1163  N   LEU A 153      29.714   7.259  -4.356  1.00  0.87           N  \\nATOM   1164  CA  LEU A 153      28.598   7.689  -3.541  1.00  0.87           C  \\nATOM   1165  C   LEU A 153      28.203   9.120  -3.843  1.00  0.87           C  \\nATOM   1166  O   LEU A 153      29.015   9.973  -4.180  1.00  0.87           O  \\nATOM   1167  CB  LEU A 153      28.867   7.580  -2.025  1.00  0.87           C  \\nATOM   1168  CG  LEU A 153      29.026   6.152  -1.484  1.00  0.87           C  \\nATOM   1169  CD1 LEU A 153      29.435   6.188  -0.007  1.00  0.87           C  \\nATOM   1170  CD2 LEU A 153      27.750   5.324  -1.641  1.00  0.87           C  \\nATOM   1171  N   THR A 154      26.905   9.422  -3.701  1.00  0.85           N  \\nATOM   1172  CA  THR A 154      26.380  10.738  -3.997  1.00  0.85           C  \\nATOM   1173  C   THR A 154      25.818  11.336  -2.738  1.00  0.85           C  \\nATOM   1174  O   THR A 154      25.697  10.693  -1.699  1.00  0.85           O  \\nATOM   1175  CB  THR A 154      25.328  10.755  -5.101  1.00  0.85           C  \\nATOM   1176  OG1 THR A 154      24.201   9.935  -4.813  1.00  0.85           O  \\nATOM   1177  CG2 THR A 154      25.961  10.206  -6.384  1.00  0.85           C  \\nATOM   1178  N   THR A 155      25.474  12.627  -2.777  1.00  0.85           N  \\nATOM   1179  CA  THR A 155      24.764  13.288  -1.703  1.00  0.85           C  \\nATOM   1180  C   THR A 155      23.388  13.565  -2.232  1.00  0.85           C  \\nATOM   1181  O   THR A 155      23.194  13.876  -3.405  1.00  0.85           O  \\nATOM   1182  CB  THR A 155      25.396  14.588  -1.223  1.00  0.85           C  \\nATOM   1183  OG1 THR A 155      25.780  15.446  -2.290  1.00  0.85           O  \\nATOM   1184  CG2 THR A 155      26.676  14.217  -0.480  1.00  0.85           C  \\nATOM   1185  N   ARG A 156      22.350  13.390  -1.397  1.00  0.80           N  \\nATOM   1186  CA  ARG A 156      21.012  13.770  -1.791  1.00  0.80           C  \\nATOM   1187  C   ARG A 156      20.870  15.286  -1.855  1.00  0.80           C  \\nATOM   1188  O   ARG A 156      21.422  16.017  -1.037  1.00  0.80           O  \\nATOM   1189  CB  ARG A 156      19.973  13.165  -0.829  1.00  0.80           C  \\nATOM   1190  CG  ARG A 156      19.912  11.628  -0.898  1.00  0.80           C  \\nATOM   1191  CD  ARG A 156      19.084  11.032   0.239  1.00  0.80           C  \\nATOM   1192  NE  ARG A 156      18.832   9.588  -0.101  1.00  0.80           N  \\nATOM   1193  CZ  ARG A 156      18.432   8.674   0.794  1.00  0.80           C  \\nATOM   1194  NH1 ARG A 156      18.330   8.987   2.083  1.00  0.80           N  \\nATOM   1195  NH2 ARG A 156      18.113   7.442   0.403  1.00  0.80           N  \\nATOM   1196  N   LYS A 157      20.112  15.814  -2.835  1.00  0.77           N  \\nATOM   1197  CA  LYS A 157      19.932  17.252  -2.994  1.00  0.77           C  \\nATOM   1198  C   LYS A 157      19.211  17.942  -1.824  1.00  0.77           C  \\nATOM   1199  O   LYS A 157      19.493  19.105  -1.503  1.00  0.77           O  \\nATOM   1200  CB  LYS A 157      19.259  17.568  -4.353  1.00  0.77           C  \\nATOM   1201  CG  LYS A 157      17.750  17.292  -4.382  1.00  0.77           C  \\nATOM   1202  CD  LYS A 157      17.094  17.484  -5.758  1.00  0.77           C  \\nATOM   1203  CE  LYS A 157      15.753  18.223  -5.675  1.00  0.77           C  \\nATOM   1204  NZ  LYS A 157      15.957  19.668  -5.930  1.00  0.77           N  \\nATOM   1205  N   ASP A 158      18.293  17.241  -1.136  1.00  0.85           N  \\nATOM   1206  CA  ASP A 158      17.576  17.670   0.061  1.00  0.85           C  \\nATOM   1207  C   ASP A 158      18.475  17.812   1.281  1.00  0.85           C  \\nATOM   1208  O   ASP A 158      18.143  18.539   2.236  1.00  0.85           O  \\nATOM   1209  CB  ASP A 158      16.480  16.623   0.386  1.00  0.85           C  \\nATOM   1210  CG  ASP A 158      15.158  16.949  -0.292  1.00  0.85           C  \\nATOM   1211  OD1 ASP A 158      15.064  18.008  -0.968  1.00  0.85           O  \\nATOM   1212  OD2 ASP A 158      14.227  16.126  -0.128  1.00  0.85           O  \\nATOM   1213  N   ASP A 159      19.634  17.139   1.249  1.00  0.86           N  \\nATOM   1214  CA  ASP A 159      20.625  17.105   2.298  1.00  0.86           C  \\nATOM   1215  C   ASP A 159      21.715  18.156   2.018  1.00  0.86           C  \\nATOM   1216  O   ASP A 159      22.674  18.297   2.781  1.00  0.86           O  \\nATOM   1217  CB  ASP A 159      21.239  15.667   2.424  1.00  0.86           C  \\nATOM   1218  CG  ASP A 159      20.280  14.568   2.901  1.00  0.86           C  \\nATOM   1219  OD1 ASP A 159      19.171  14.873   3.409  1.00  0.86           O  \\nATOM   1220  OD2 ASP A 159      20.673  13.367   2.806  1.00  0.86           O  \\nATOM   1221  N   GLN A 160      21.594  18.974   0.936  1.00  0.78           N  \\nATOM   1222  CA  GLN A 160      22.438  20.148   0.729  1.00  0.78           C  \\nATOM   1223  C   GLN A 160      22.293  21.172   1.844  1.00  0.78           C  \\nATOM   1224  O   GLN A 160      21.193  21.486   2.291  1.00  0.78           O  \\nATOM   1225  CB  GLN A 160      22.218  20.850  -0.637  1.00  0.78           C  \\nATOM   1226  CG  GLN A 160      23.012  20.191  -1.789  1.00  0.78           C  \\nATOM   1227  CD  GLN A 160      22.753  20.889  -3.130  1.00  0.78           C  \\nATOM   1228  OE1 GLN A 160      21.891  21.741  -3.279  1.00  0.78           O  \\nATOM   1229  NE2 GLN A 160      23.558  20.508  -4.158  1.00  0.78           N  \\nATOM   1230  N   GLU A 161      23.430  21.740   2.304  1.00  0.80           N  \\nATOM   1231  CA  GLU A 161      23.490  22.513   3.535  1.00  0.80           C  \\nATOM   1232  C   GLU A 161      22.535  23.694   3.599  1.00  0.80           C  \\nATOM   1233  O   GLU A 161      21.803  23.863   4.583  1.00  0.80           O  \\nATOM   1234  CB  GLU A 161      24.937  22.999   3.787  1.00  0.80           C  \\nATOM   1235  CG  GLU A 161      25.060  23.826   5.089  1.00  0.80           C  \\nATOM   1236  CD  GLU A 161      26.485  24.028   5.601  1.00  0.80           C  \\nATOM   1237  OE1 GLU A 161      27.436  23.435   5.043  1.00  0.80           O  \\nATOM   1238  OE2 GLU A 161      26.606  24.794   6.598  1.00  0.80           O  \\nATOM   1239  N   GLU A 162      22.443  24.498   2.533  1.00  0.82           N  \\nATOM   1240  CA  GLU A 162      21.525  25.620   2.448  1.00  0.82           C  \\nATOM   1241  C   GLU A 162      20.060  25.231   2.527  1.00  0.82           C  \\nATOM   1242  O   GLU A 162      19.251  25.897   3.186  1.00  0.82           O  \\nATOM   1243  CB  GLU A 162      21.823  26.481   1.192  1.00  0.82           C  \\nATOM   1244  CG  GLU A 162      22.979  27.483   1.411  1.00  0.82           C  \\nATOM   1245  CD  GLU A 162      22.622  28.354   2.611  1.00  0.82           C  \\nATOM   1246  OE1 GLU A 162      21.504  28.949   2.566  1.00  0.82           O  \\nATOM   1247  OE2 GLU A 162      23.400  28.353   3.598  1.00  0.82           O  \\nATOM   1248  N   THR A 163      19.655  24.118   1.898  1.00  0.87           N  \\nATOM   1249  CA  THR A 163      18.304  23.567   2.024  1.00  0.87           C  \\nATOM   1250  C   THR A 163      18.012  23.128   3.451  1.00  0.87           C  \\nATOM   1251  O   THR A 163      16.946  23.418   4.007  1.00  0.87           O  \\nATOM   1252  CB  THR A 163      18.049  22.397   1.079  1.00  0.87           C  \\nATOM   1253  OG1 THR A 163      18.299  22.786  -0.267  1.00  0.87           O  \\nATOM   1254  CG2 THR A 163      16.583  21.945   1.136  1.00  0.87           C  \\nATOM   1255  N   VAL A 164      18.973  22.455   4.110  1.00  0.89           N  \\nATOM   1256  CA  VAL A 164      18.883  22.053   5.515  1.00  0.89           C  \\nATOM   1257  C   VAL A 164      18.755  23.241   6.460  1.00  0.89           C  \\nATOM   1258  O   VAL A 164      17.863  23.268   7.324  1.00  0.89           O  \\nATOM   1259  CB  VAL A 164      20.088  21.209   5.933  1.00  0.89           C  \\nATOM   1260  CG1 VAL A 164      20.050  20.835   7.429  1.00  0.89           C  \\nATOM   1261  CG2 VAL A 164      20.130  19.928   5.087  1.00  0.89           C  \\nATOM   1262  N   ARG A 165      19.594  24.277   6.299  1.00  0.81           N  \\nATOM   1263  CA  ARG A 165      19.555  25.497   7.092  1.00  0.81           C  \\nATOM   1264  C   ARG A 165      18.259  26.279   6.936  1.00  0.81           C  \\nATOM   1265  O   ARG A 165      17.676  26.726   7.934  1.00  0.81           O  \\nATOM   1266  CB  ARG A 165      20.764  26.421   6.791  1.00  0.81           C  \\nATOM   1267  CG  ARG A 165      22.119  25.860   7.279  1.00  0.81           C  \\nATOM   1268  CD  ARG A 165      23.290  26.845   7.134  1.00  0.81           C  \\nATOM   1269  NE  ARG A 165      24.534  26.181   7.640  1.00  0.81           N  \\nATOM   1270  CZ  ARG A 165      24.954  26.121   8.911  1.00  0.81           C  \\nATOM   1271  NH1 ARG A 165      24.257  26.601   9.930  1.00  0.81           N  \\nATOM   1272  NH2 ARG A 165      26.120  25.514   9.145  1.00  0.81           N  \\nATOM   1273  N   LYS A 166      17.728  26.431   5.709  1.00  0.85           N  \\nATOM   1274  CA  LYS A 166      16.430  27.056   5.484  1.00  0.85           C  \\nATOM   1275  C   LYS A 166      15.285  26.338   6.186  1.00  0.85           C  \\nATOM   1276  O   LYS A 166      14.473  26.981   6.875  1.00  0.85           O  \\nATOM   1277  CB  LYS A 166      16.118  27.177   3.976  1.00  0.85           C  \\nATOM   1278  CG  LYS A 166      16.941  28.272   3.282  1.00  0.85           C  \\nATOM   1279  CD  LYS A 166      16.376  28.604   1.894  1.00  0.85           C  \\nATOM   1280  CE  LYS A 166      17.167  29.682   1.155  1.00  0.85           C  \\nATOM   1281  NZ  LYS A 166      18.185  29.053   0.286  1.00  0.85           N  \\nATOM   1282  N   ARG A 167      15.238  24.997   6.127  1.00  0.80           N  \\nATOM   1283  CA  ARG A 167      14.243  24.195   6.830  1.00  0.80           C  \\nATOM   1284  C   ARG A 167      14.259  24.401   8.340  1.00  0.80           C  \\nATOM   1285  O   ARG A 167      13.202  24.475   8.988  1.00  0.80           O  \\nATOM   1286  CB  ARG A 167      14.441  22.676   6.578  1.00  0.80           C  \\nATOM   1287  CG  ARG A 167      13.998  22.182   5.187  1.00  0.80           C  \\nATOM   1288  CD  ARG A 167      13.652  20.684   5.139  1.00  0.80           C  \\nATOM   1289  NE  ARG A 167      14.858  19.873   5.538  1.00  0.80           N  \\nATOM   1290  CZ  ARG A 167      15.689  19.237   4.700  1.00  0.80           C  \\nATOM   1291  NH1 ARG A 167      15.563  19.280   3.374  1.00  0.80           N  \\nATOM   1292  NH2 ARG A 167      16.712  18.530   5.179  1.00  0.80           N  \\nATOM   1293  N   LEU A 168      15.443  24.488   8.954  1.00  0.87           N  \\nATOM   1294  CA  LEU A 168      15.599  24.764  10.373  1.00  0.87           C  \\nATOM   1295  C   LEU A 168      15.123  26.150  10.785  1.00  0.87           C  \\nATOM   1296  O   LEU A 168      14.417  26.301  11.794  1.00  0.87           O  \\nATOM   1297  CB  LEU A 168      17.055  24.524  10.816  1.00  0.87           C  \\nATOM   1298  CG  LEU A 168      17.482  23.047  10.743  1.00  0.87           C  \\nATOM   1299  CD1 LEU A 168      18.990  22.931  10.989  1.00  0.87           C  \\nATOM   1300  CD2 LEU A 168      16.705  22.192  11.751  1.00  0.87           C  \\nATOM   1301  N   VAL A 169      15.425  27.196   9.993  1.00  0.84           N  \\nATOM   1302  CA  VAL A 169      14.934  28.556  10.224  1.00  0.84           C  \\nATOM   1303  C   VAL A 169      13.407  28.620  10.221  1.00  0.84           C  \\nATOM   1304  O   VAL A 169      12.784  29.175  11.140  1.00  0.84           O  \\nATOM   1305  CB  VAL A 169      15.495  29.533   9.186  1.00  0.84           C  \\nATOM   1306  CG1 VAL A 169      14.815  30.915   9.257  1.00  0.84           C  \\nATOM   1307  CG2 VAL A 169      17.010  29.704   9.404  1.00  0.84           C  \\nATOM   1308  N   GLU A 170      12.759  27.996   9.222  1.00  0.80           N  \\nATOM   1309  CA  GLU A 170      11.310  27.894   9.132  1.00  0.80           C  \\nATOM   1310  C   GLU A 170      10.700  27.105  10.281  1.00  0.80           C  \\nATOM   1311  O   GLU A 170       9.708  27.528  10.898  1.00  0.80           O  \\nATOM   1312  CB  GLU A 170      10.889  27.227   7.805  1.00  0.80           C  \\nATOM   1313  CG  GLU A 170      11.252  28.040   6.540  1.00  0.80           C  \\nATOM   1314  CD  GLU A 170      10.923  27.295   5.242  1.00  0.80           C  \\nATOM   1315  OE1 GLU A 170      10.207  26.261   5.297  1.00  0.80           O  \\nATOM   1316  OE2 GLU A 170      11.408  27.764   4.179  1.00  0.80           O  \\nATOM   1317  N   TYR A 171      11.307  25.960  10.644  1.00  0.84           N  \\nATOM   1318  CA  TYR A 171      10.876  25.138  11.764  1.00  0.84           C  \\nATOM   1319  C   TYR A 171      10.892  25.897  13.085  1.00  0.84           C  \\nATOM   1320  O   TYR A 171       9.885  25.918  13.799  1.00  0.84           O  \\nATOM   1321  CB  TYR A 171      11.749  23.852  11.870  1.00  0.84           C  \\nATOM   1322  CG  TYR A 171      11.304  22.953  13.000  1.00  0.84           C  \\nATOM   1323  CD1 TYR A 171      10.240  22.052  12.838  1.00  0.84           C  \\nATOM   1324  CD2 TYR A 171      11.907  23.066  14.264  1.00  0.84           C  \\nATOM   1325  CE1 TYR A 171       9.799  21.270  13.916  1.00  0.84           C  \\nATOM   1326  CE2 TYR A 171      11.439  22.316  15.352  1.00  0.84           C  \\nATOM   1327  CZ  TYR A 171      10.395  21.403  15.173  1.00  0.84           C  \\nATOM   1328  OH  TYR A 171       9.932  20.634  16.260  1.00  0.84           O  \\nATOM   1329  N   HIS A 172      11.985  26.584  13.448  1.00  0.80           N  \\nATOM   1330  CA  HIS A 172      12.059  27.295  14.718  1.00  0.80           C  \\nATOM   1331  C   HIS A 172      11.055  28.424  14.847  1.00  0.80           C  \\nATOM   1332  O   HIS A 172      10.429  28.601  15.903  1.00  0.80           O  \\nATOM   1333  CB  HIS A 172      13.474  27.819  15.006  1.00  0.80           C  \\nATOM   1334  CG  HIS A 172      14.453  26.704  15.141  1.00  0.80           C  \\nATOM   1335  ND1 HIS A 172      14.201  25.701  16.046  1.00  0.80           N  \\nATOM   1336  CD2 HIS A 172      15.669  26.527  14.546  1.00  0.80           C  \\nATOM   1337  CE1 HIS A 172      15.273  24.921  15.998  1.00  0.80           C  \\nATOM   1338  NE2 HIS A 172      16.169  25.382  15.110  1.00  0.80           N  \\nATOM   1339  N   GLN A 173      10.834  29.187  13.765  1.00  0.77           N  \\nATOM   1340  CA  GLN A 173       9.824  30.234  13.705  1.00  0.77           C  \\nATOM   1341  C   GLN A 173       8.407  29.714  13.903  1.00  0.77           C  \\nATOM   1342  O   GLN A 173       7.607  30.300  14.647  1.00  0.77           O  \\nATOM   1343  CB  GLN A 173       9.915  30.980  12.350  1.00  0.77           C  \\nATOM   1344  CG  GLN A 173       8.896  32.132  12.160  1.00  0.77           C  \\nATOM   1345  CD  GLN A 173       9.113  33.259  13.173  1.00  0.77           C  \\nATOM   1346  OE1 GLN A 173      10.169  33.413  13.770  1.00  0.77           O  \\nATOM   1347  NE2 GLN A 173       8.063  34.100  13.364  1.00  0.77           N  \\nATOM   1348  N   MET A 174       8.054  28.596  13.254  1.00  0.81           N  \\nATOM   1349  CA  MET A 174       6.774  27.931  13.417  1.00  0.81           C  \\nATOM   1350  C   MET A 174       6.578  27.293  14.790  1.00  0.81           C  \\nATOM   1351  O   MET A 174       5.494  27.357  15.386  1.00  0.81           O  \\nATOM   1352  CB  MET A 174       6.603  26.879  12.299  1.00  0.81           C  \\nATOM   1353  CG  MET A 174       5.173  26.326  12.174  1.00  0.81           C  \\nATOM   1354  SD  MET A 174       4.878  25.285  10.710  1.00  0.81           S  \\nATOM   1355  CE  MET A 174       6.057  23.970  11.120  1.00  0.81           C  \\nATOM   1356  N   THR A 175       7.626  26.651  15.326  1.00  0.78           N  \\nATOM   1357  CA  THR A 175       7.557  25.779  16.497  1.00  0.78           C  \\nATOM   1358  C   THR A 175       7.710  26.504  17.825  1.00  0.78           C  \\nATOM   1359  O   THR A 175       7.106  26.104  18.821  1.00  0.78           O  \\nATOM   1360  CB  THR A 175       8.571  24.645  16.395  1.00  0.78           C  \\nATOM   1361  OG1 THR A 175       8.301  23.868  15.233  1.00  0.78           O  \\nATOM   1362  CG2 THR A 175       8.505  23.659  17.562  1.00  0.78           C  \\nATOM   1363  N   ALA A 176       8.441  27.638  17.906  1.00  0.75           N  \\nATOM   1364  CA  ALA A 176       8.573  28.423  19.134  1.00  0.75           C  \\nATOM   1365  C   ALA A 176       7.262  28.825  19.845  1.00  0.75           C  \\nATOM   1366  O   ALA A 176       7.199  28.655  21.069  1.00  0.75           O  \\nATOM   1367  CB  ALA A 176       9.429  29.683  18.883  1.00  0.75           C  \\nATOM   1368  N   PRO A 177       6.177  29.298  19.211  1.00  0.80           N  \\nATOM   1369  CA  PRO A 177       4.907  29.513  19.899  1.00  0.80           C  \\nATOM   1370  C   PRO A 177       4.278  28.217  20.385  1.00  0.80           C  \\nATOM   1371  O   PRO A 177       3.657  28.214  21.443  1.00  0.80           O  \\nATOM   1372  CB  PRO A 177       4.034  30.259  18.873  1.00  0.80           C  \\nATOM   1373  CG  PRO A 177       4.663  29.951  17.517  1.00  0.80           C  \\nATOM   1374  CD  PRO A 177       6.144  29.804  17.836  1.00  0.80           C  \\nATOM   1375  N   LEU A 178       4.416  27.105  19.641  1.00  0.83           N  \\nATOM   1376  CA  LEU A 178       3.875  25.806  20.012  1.00  0.83           C  \\nATOM   1377  C   LEU A 178       4.527  25.215  21.255  1.00  0.83           C  \\nATOM   1378  O   LEU A 178       3.864  24.637  22.124  1.00  0.83           O  \\nATOM   1379  CB  LEU A 178       3.962  24.815  18.834  1.00  0.83           C  \\nATOM   1380  CG  LEU A 178       3.320  23.443  19.103  1.00  0.83           C  \\nATOM   1381  CD1 LEU A 178       1.813  23.558  19.357  1.00  0.83           C  \\nATOM   1382  CD2 LEU A 178       3.618  22.496  17.939  1.00  0.83           C  \\nATOM   1383  N   ILE A 179       5.849  25.378  21.404  1.00  0.84           N  \\nATOM   1384  CA  ILE A 179       6.576  25.023  22.617  1.00  0.84           C  \\nATOM   1385  C   ILE A 179       6.089  25.820  23.821  1.00  0.84           C  \\nATOM   1386  O   ILE A 179       5.841  25.283  24.910  1.00  0.84           O  \\nATOM   1387  CB  ILE A 179       8.070  25.235  22.430  1.00  0.84           C  \\nATOM   1388  CG1 ILE A 179       8.655  24.366  21.299  1.00  0.84           C  \\nATOM   1389  CG2 ILE A 179       8.839  24.952  23.727  1.00  0.84           C  \\nATOM   1390  CD1 ILE A 179       8.263  22.887  21.300  1.00  0.84           C  \\nATOM   1391  N   GLY A 180       5.876  27.137  23.641  1.00  0.86           N  \\nATOM   1392  CA  GLY A 180       5.324  27.998  24.680  1.00  0.86           C  \\nATOM   1393  C   GLY A 180       3.897  27.673  25.056  1.00  0.86           C  \\nATOM   1394  O   GLY A 180       3.526  27.771  26.231  1.00  0.86           O  \\nATOM   1395  N   TYR A 181       3.078  27.232  24.082  1.00  0.85           N  \\nATOM   1396  CA  TYR A 181       1.742  26.685  24.283  1.00  0.85           C  \\nATOM   1397  C   TYR A 181       1.762  25.446  25.180  1.00  0.85           C  \\nATOM   1398  O   TYR A 181       1.097  25.398  26.212  1.00  0.85           O  \\nATOM   1399  CB  TYR A 181       1.109  26.368  22.890  1.00  0.85           C  \\nATOM   1400  CG  TYR A 181      -0.245  25.703  22.941  1.00  0.85           C  \\nATOM   1401  CD1 TYR A 181      -0.351  24.303  23.019  1.00  0.85           C  \\nATOM   1402  CD2 TYR A 181      -1.420  26.466  22.903  1.00  0.85           C  \\nATOM   1403  CE1 TYR A 181      -1.606  23.689  23.167  1.00  0.85           C  \\nATOM   1404  CE2 TYR A 181      -2.675  25.848  22.997  1.00  0.85           C  \\nATOM   1405  CZ  TYR A 181      -2.763  24.465  23.150  1.00  0.85           C  \\nATOM   1406  OH  TYR A 181      -4.052  23.900  23.216  1.00  0.85           O  \\nATOM   1407  N   TYR A 182       2.583  24.428  24.852  1.00  0.85           N  \\nATOM   1408  CA  TYR A 182       2.650  23.202  25.638  1.00  0.85           C  \\nATOM   1409  C   TYR A 182       3.239  23.362  27.017  1.00  0.85           C  \\nATOM   1410  O   TYR A 182       2.814  22.693  27.965  1.00  0.85           O  \\nATOM   1411  CB  TYR A 182       3.381  22.065  24.898  1.00  0.85           C  \\nATOM   1412  CG  TYR A 182       2.513  21.522  23.805  1.00  0.85           C  \\nATOM   1413  CD1 TYR A 182       1.218  21.053  24.088  1.00  0.85           C  \\nATOM   1414  CD2 TYR A 182       3.007  21.410  22.498  1.00  0.85           C  \\nATOM   1415  CE1 TYR A 182       0.435  20.487  23.079  1.00  0.85           C  \\nATOM   1416  CE2 TYR A 182       2.230  20.819  21.492  1.00  0.85           C  \\nATOM   1417  CZ  TYR A 182       0.940  20.372  21.787  1.00  0.85           C  \\nATOM   1418  OH  TYR A 182       0.124  19.811  20.798  1.00  0.85           O  \\nATOM   1419  N   SER A 183       4.211  24.270  27.184  1.00  0.85           N  \\nATOM   1420  CA  SER A 183       4.701  24.655  28.501  1.00  0.85           C  \\nATOM   1421  C   SER A 183       3.593  25.267  29.364  1.00  0.85           C  \\nATOM   1422  O   SER A 183       3.402  24.875  30.517  1.00  0.85           O  \\nATOM   1423  CB  SER A 183       5.931  25.594  28.390  1.00  0.85           C  \\nATOM   1424  OG  SER A 183       6.582  25.755  29.653  1.00  0.85           O  \\nATOM   1425  N   LYS A 184       2.758  26.169  28.804  1.00  0.82           N  \\nATOM   1426  CA  LYS A 184       1.570  26.705  29.470  1.00  0.82           C  \\nATOM   1427  C   LYS A 184       0.499  25.669  29.798  1.00  0.82           C  \\nATOM   1428  O   LYS A 184      -0.084  25.675  30.893  1.00  0.82           O  \\nATOM   1429  CB  LYS A 184       0.953  27.860  28.644  1.00  0.82           C  \\nATOM   1430  CG  LYS A 184       1.799  29.143  28.679  1.00  0.82           C  \\nATOM   1431  CD  LYS A 184       1.621  29.878  30.014  1.00  0.82           C  \\nATOM   1432  CE  LYS A 184       2.596  31.025  30.280  1.00  0.82           C  \\nATOM   1433  NZ  LYS A 184       3.950  30.469  30.487  1.00  0.82           N  \\nATOM   1434  N   GLU A 185       0.219  24.719  28.894  1.00  0.82           N  \\nATOM   1435  CA  GLU A 185      -0.661  23.587  29.161  1.00  0.82           C  \\nATOM   1436  C   GLU A 185      -0.157  22.720  30.315  1.00  0.82           C  \\nATOM   1437  O   GLU A 185      -0.944  22.237  31.139  1.00  0.82           O  \\nATOM   1438  CB  GLU A 185      -0.852  22.697  27.908  1.00  0.82           C  \\nATOM   1439  CG  GLU A 185      -1.636  23.341  26.748  1.00  0.82           C  \\nATOM   1440  CD  GLU A 185      -3.075  23.580  27.105  1.00  0.82           C  \\nATOM   1441  OE1 GLU A 185      -3.821  22.613  27.428  1.00  0.82           O  \\nATOM   1442  OE2 GLU A 185      -3.539  24.741  27.012  1.00  0.82           O  \\nATOM   1443  N   ALA A 186       1.169  22.512  30.417  1.00  0.87           N  \\nATOM   1444  CA  ALA A 186       1.816  21.873  31.550  1.00  0.87           C  \\nATOM   1445  C   ALA A 186       1.703  22.646  32.868  1.00  0.87           C  \\nATOM   1446  O   ALA A 186       1.383  22.058  33.911  1.00  0.87           O  \\nATOM   1447  CB  ALA A 186       3.283  21.551  31.212  1.00  0.87           C  \\nATOM   1448  N   GLU A 187       1.887  23.982  32.857  1.00  0.80           N  \\nATOM   1449  CA  GLU A 187       1.660  24.894  33.987  1.00  0.80           C  \\nATOM   1450  C   GLU A 187       0.214  24.834  34.494  1.00  0.80           C  \\nATOM   1451  O   GLU A 187      -0.048  24.880  35.706  1.00  0.80           O  \\nATOM   1452  CB  GLU A 187       2.006  26.377  33.622  1.00  0.80           C  \\nATOM   1453  CG  GLU A 187       3.514  26.731  33.405  1.00  0.80           C  \\nATOM   1454  CD  GLU A 187       3.786  28.152  32.870  1.00  0.80           C  \\nATOM   1455  OE1 GLU A 187       2.853  28.990  32.786  1.00  0.80           O  \\nATOM   1456  OE2 GLU A 187       4.938  28.457  32.456  1.00  0.80           O  \\nATOM   1457  N   ALA A 188      -0.768  24.689  33.590  1.00  0.85           N  \\nATOM   1458  CA  ALA A 188      -2.175  24.523  33.909  1.00  0.85           C  \\nATOM   1459  C   ALA A 188      -2.540  23.119  34.393  1.00  0.85           C  \\nATOM   1460  O   ALA A 188      -3.651  22.898  34.898  1.00  0.85           O  \\nATOM   1461  CB  ALA A 188      -3.025  24.908  32.679  1.00  0.85           C  \\nATOM   1462  N   GLY A 189      -1.635  22.129  34.284  1.00  0.86           N  \\nATOM   1463  CA  GLY A 189      -1.884  20.758  34.730  1.00  0.86           C  \\nATOM   1464  C   GLY A 189      -2.582  19.882  33.728  1.00  0.86           C  \\nATOM   1465  O   GLY A 189      -2.989  18.760  34.060  1.00  0.86           O  \\nATOM   1466  N   ASN A 190      -2.736  20.333  32.474  1.00  0.83           N  \\nATOM   1467  CA  ASN A 190      -3.439  19.600  31.432  1.00  0.83           C  \\nATOM   1468  C   ASN A 190      -2.583  18.515  30.792  1.00  0.83           C  \\nATOM   1469  O   ASN A 190      -3.107  17.585  30.178  1.00  0.83           O  \\nATOM   1470  CB  ASN A 190      -3.873  20.523  30.258  1.00  0.83           C  \\nATOM   1471  CG  ASN A 190      -4.870  21.592  30.734  1.00  0.83           C  \\nATOM   1472  OD1 ASN A 190      -5.558  21.472  31.688  1.00  0.83           O  \\nATOM   1473  ND2 ASN A 190      -4.925  22.707  29.922  1.00  0.83           N  \\nATOM   1474  N   THR A 191      -1.253  18.611  30.881  1.00  0.87           N  \\nATOM   1475  CA  THR A 191      -0.315  17.699  30.234  1.00  0.87           C  \\nATOM   1476  C   THR A 191       0.994  17.853  30.957  1.00  0.87           C  \\nATOM   1477  O   THR A 191       1.121  18.657  31.872  1.00  0.87           O  \\nATOM   1478  CB  THR A 191      -0.131  17.905  28.720  1.00  0.87           C  \\nATOM   1479  OG1 THR A 191       0.647  16.875  28.102  1.00  0.87           O  \\nATOM   1480  CG2 THR A 191       0.555  19.232  28.373  1.00  0.87           C  \\nATOM   1481  N   LYS A 192       2.011  17.072  30.584  1.00  0.84           N  \\nATOM   1482  CA  LYS A 192       3.365  17.293  31.014  1.00  0.84           C  \\nATOM   1483  C   LYS A 192       4.229  17.591  29.814  1.00  0.84           C  \\nATOM   1484  O   LYS A 192       4.015  17.113  28.707  1.00  0.84           O  \\nATOM   1485  CB  LYS A 192       3.904  16.084  31.789  1.00  0.84           C  \\nATOM   1486  CG  LYS A 192       3.164  15.925  33.121  1.00  0.84           C  \\nATOM   1487  CD  LYS A 192       3.546  14.631  33.832  1.00  0.84           C  \\nATOM   1488  CE  LYS A 192       2.828  14.451  35.159  1.00  0.84           C  \\nATOM   1489  NZ  LYS A 192       3.279  13.184  35.765  1.00  0.84           N  \\nATOM   1490  N   TYR A 193       5.239  18.441  30.004  1.00  0.88           N  \\nATOM   1491  CA  TYR A 193       6.072  18.900  28.925  1.00  0.88           C  \\nATOM   1492  C   TYR A 193       7.517  18.741  29.376  1.00  0.88           C  \\nATOM   1493  O   TYR A 193       7.860  19.075  30.510  1.00  0.88           O  \\nATOM   1494  CB  TYR A 193       5.653  20.349  28.569  1.00  0.88           C  \\nATOM   1495  CG  TYR A 193       6.685  21.058  27.769  1.00  0.88           C  \\nATOM   1496  CD1 TYR A 193       6.938  20.776  26.415  1.00  0.88           C  \\nATOM   1497  CD2 TYR A 193       7.508  21.949  28.462  1.00  0.88           C  \\nATOM   1498  CE1 TYR A 193       8.004  21.413  25.758  1.00  0.88           C  \\nATOM   1499  CE2 TYR A 193       8.571  22.565  27.820  1.00  0.88           C  \\nATOM   1500  CZ  TYR A 193       8.796  22.313  26.478  1.00  0.88           C  \\nATOM   1501  OH  TYR A 193       9.853  23.023  25.939  1.00  0.88           O  \\nATOM   1502  N   ALA A 194       8.391  18.204  28.502  1.00  0.92           N  \\nATOM   1503  CA  ALA A 194       9.800  18.102  28.805  1.00  0.92           C  \\nATOM   1504  C   ALA A 194      10.658  18.207  27.554  1.00  0.92           C  \\nATOM   1505  O   ALA A 194      10.340  17.661  26.498  1.00  0.92           O  \\nATOM   1506  CB  ALA A 194      10.110  16.775  29.522  1.00  0.92           C  \\nATOM   1507  N   LYS A 195      11.800  18.913  27.665  1.00  0.87           N  \\nATOM   1508  CA  LYS A 195      12.837  18.921  26.654  1.00  0.87           C  \\nATOM   1509  C   LYS A 195      13.791  17.769  26.899  1.00  0.87           C  \\nATOM   1510  O   LYS A 195      14.017  17.344  28.026  1.00  0.87           O  \\nATOM   1511  CB  LYS A 195      13.663  20.220  26.660  1.00  0.87           C  \\nATOM   1512  CG  LYS A 195      12.834  21.468  26.377  1.00  0.87           C  \\nATOM   1513  CD  LYS A 195      13.709  22.721  26.460  1.00  0.87           C  \\nATOM   1514  CE  LYS A 195      12.953  24.004  26.141  1.00  0.87           C  \\nATOM   1515  NZ  LYS A 195      12.735  24.123  24.691  1.00  0.87           N  \\nATOM   1516  N   VAL A 196      14.356  17.234  25.808  1.00  0.89           N  \\nATOM   1517  CA  VAL A 196      15.222  16.080  25.817  1.00  0.89           C  \\nATOM   1518  C   VAL A 196      16.447  16.441  25.010  1.00  0.89           C  \\nATOM   1519  O   VAL A 196      16.343  16.844  23.856  1.00  0.89           O  \\nATOM   1520  CB  VAL A 196      14.542  14.877  25.166  1.00  0.89           C  \\nATOM   1521  CG1 VAL A 196      15.458  13.644  25.195  1.00  0.89           C  \\nATOM   1522  CG2 VAL A 196      13.212  14.580  25.881  1.00  0.89           C  \\nATOM   1523  N   ASP A 197      17.656  16.285  25.590  1.00  0.85           N  \\nATOM   1524  CA  ASP A 197      18.890  16.485  24.865  1.00  0.85           C  \\nATOM   1525  C   ASP A 197      19.136  15.275  23.975  1.00  0.85           C  \\nATOM   1526  O   ASP A 197      19.493  14.179  24.437  1.00  0.85           O  \\nATOM   1527  CB  ASP A 197      20.030  16.744  25.876  1.00  0.85           C  \\nATOM   1528  CG  ASP A 197      21.371  17.056  25.225  1.00  0.85           C  \\nATOM   1529  OD1 ASP A 197      21.421  17.238  23.976  1.00  0.85           O  \\nATOM   1530  OD2 ASP A 197      22.368  17.092  25.996  1.00  0.85           O  \\nATOM   1531  N   GLY A 198      18.897  15.436  22.671  1.00  0.88           N  \\nATOM   1532  CA  GLY A 198      19.015  14.408  21.654  1.00  0.88           C  \\nATOM   1533  C   GLY A 198      20.408  14.212  21.122  1.00  0.88           C  \\nATOM   1534  O   GLY A 198      20.607  13.385  20.227  1.00  0.88           O  \\nATOM   1535  N   THR A 199      21.410  14.982  21.600  1.00  0.84           N  \\nATOM   1536  CA  THR A 199      22.799  14.832  21.142  1.00  0.84           C  \\nATOM   1537  C   THR A 199      23.499  13.687  21.817  1.00  0.84           C  \\nATOM   1538  O   THR A 199      24.492  13.148  21.307  1.00  0.84           O  \\nATOM   1539  CB  THR A 199      23.706  16.057  21.310  1.00  0.84           C  \\nATOM   1540  OG1 THR A 199      24.107  16.311  22.658  1.00  0.84           O  \\nATOM   1541  CG2 THR A 199      22.988  17.303  20.804  1.00  0.84           C  \\nATOM   1542  N   LYS A 200      23.012  13.299  22.999  1.00  0.80           N  \\nATOM   1543  CA  LYS A 200      23.605  12.279  23.827  1.00  0.80           C  \\nATOM   1544  C   LYS A 200      23.585  10.897  23.162  1.00  0.80           C  \\nATOM   1545  O   LYS A 200      22.711  10.634  22.332  1.00  0.80           O  \\nATOM   1546  CB  LYS A 200      22.921  12.197  25.204  1.00  0.80           C  \\nATOM   1547  CG  LYS A 200      22.931  13.491  26.027  1.00  0.80           C  \\nATOM   1548  CD  LYS A 200      22.121  13.284  27.314  1.00  0.80           C  \\nATOM   1549  CE  LYS A 200      22.215  14.425  28.322  1.00  0.80           C  \\nATOM   1550  NZ  LYS A 200      21.590  13.968  29.578  1.00  0.80           N  \\nATOM   1551  N   PRO A 201      24.495   9.958  23.436  1.00  0.81           N  \\nATOM   1552  CA  PRO A 201      24.299   8.542  23.100  1.00  0.81           C  \\nATOM   1553  C   PRO A 201      22.893   8.004  23.396  1.00  0.81           C  \\nATOM   1554  O   PRO A 201      22.279   8.399  24.373  1.00  0.81           O  \\nATOM   1555  CB  PRO A 201      25.382   7.810  23.905  1.00  0.81           C  \\nATOM   1556  CG  PRO A 201      26.453   8.867  24.190  1.00  0.81           C  \\nATOM   1557  CD  PRO A 201      25.635  10.145  24.338  1.00  0.81           C  \\nATOM   1558  N   VAL A 202      22.362   7.076  22.570  1.00  0.85           N  \\nATOM   1559  CA  VAL A 202      20.971   6.628  22.648  1.00  0.85           C  \\nATOM   1560  C   VAL A 202      20.532   6.056  23.991  1.00  0.85           C  \\nATOM   1561  O   VAL A 202      19.362   6.186  24.380  1.00  0.85           O  \\nATOM   1562  CB  VAL A 202      20.631   5.631  21.545  1.00  0.85           C  \\nATOM   1563  CG1 VAL A 202      20.689   6.340  20.182  1.00  0.85           C  \\nATOM   1564  CG2 VAL A 202      21.560   4.404  21.580  1.00  0.85           C  \\nATOM   1565  N   ALA A 203      21.433   5.398  24.732  1.00  0.85           N  \\nATOM   1566  CA  ALA A 203      21.218   4.926  26.088  1.00  0.85           C  \\nATOM   1567  C   ALA A 203      21.038   6.047  27.113  1.00  0.85           C  \\nATOM   1568  O   ALA A 203      20.161   5.960  27.979  1.00  0.85           O  \\nATOM   1569  CB  ALA A 203      22.342   3.952  26.493  1.00  0.85           C  \\nATOM   1570  N   GLU A 204      21.824   7.132  27.015  1.00  0.81           N  \\nATOM   1571  CA  GLU A 204      21.717   8.325  27.842  1.00  0.81           C  \\nATOM   1572  C   GLU A 204      20.395   9.045  27.612  1.00  0.81           C  \\nATOM   1573  O   GLU A 204      19.668   9.383  28.549  1.00  0.81           O  \\nATOM   1574  CB  GLU A 204      22.887   9.286  27.519  1.00  0.81           C  \\nATOM   1575  CG  GLU A 204      24.305   8.731  27.803  1.00  0.81           C  \\nATOM   1576  CD  GLU A 204      24.649   8.664  29.292  1.00  0.81           C  \\nATOM   1577  OE1 GLU A 204      23.972   9.353  30.095  1.00  0.81           O  \\nATOM   1578  OE2 GLU A 204      25.623   7.939  29.613  1.00  0.81           O  \\nATOM   1579  N   VAL A 205      19.992   9.212  26.333  1.00  0.86           N  \\nATOM   1580  CA  VAL A 205      18.693   9.783  25.960  1.00  0.86           C  \\nATOM   1581  C   VAL A 205      17.544   8.957  26.515  1.00  0.86           C  \\nATOM   1582  O   VAL A 205      16.556   9.489  27.048  1.00  0.86           O  \\nATOM   1583  CB  VAL A 205      18.507   9.892  24.448  1.00  0.86           C  \\nATOM   1584  CG1 VAL A 205      17.106  10.438  24.103  1.00  0.86           C  \\nATOM   1585  CG2 VAL A 205      19.561  10.829  23.849  1.00  0.86           C  \\nATOM   1586  N   ARG A 206      17.646   7.627  26.431  1.00  0.79           N  \\nATOM   1587  CA  ARG A 206      16.693   6.695  27.002  1.00  0.79           C  \\nATOM   1588  C   ARG A 206      16.570   6.811  28.517  1.00  0.79           C  \\nATOM   1589  O   ARG A 206      15.456   6.844  29.054  1.00  0.79           O  \\nATOM   1590  CB  ARG A 206      17.095   5.255  26.622  1.00  0.79           C  \\nATOM   1591  CG  ARG A 206      16.049   4.194  27.004  1.00  0.79           C  \\nATOM   1592  CD  ARG A 206      16.466   2.726  26.875  1.00  0.79           C  \\nATOM   1593  NE  ARG A 206      17.228   2.601  25.595  1.00  0.79           N  \\nATOM   1594  CZ  ARG A 206      18.510   2.219  25.513  1.00  0.79           C  \\nATOM   1595  NH1 ARG A 206      19.177   1.793  26.585  1.00  0.79           N  \\nATOM   1596  NH2 ARG A 206      19.147   2.293  24.349  1.00  0.79           N  \\nATOM   1597  N   ALA A 207      17.693   6.920  29.247  1.00  0.87           N  \\nATOM   1598  CA  ALA A 207      17.725   7.155  30.682  1.00  0.87           C  \\nATOM   1599  C   ALA A 207      17.076   8.481  31.090  1.00  0.87           C  \\nATOM   1600  O   ALA A 207      16.275   8.534  32.032  1.00  0.87           O  \\nATOM   1601  CB  ALA A 207      19.178   7.080  31.191  1.00  0.87           C  \\nATOM   1602  N   ASP A 208      17.347   9.572  30.345  1.00  0.85           N  \\nATOM   1603  CA  ASP A 208      16.681  10.856  30.513  1.00  0.85           C  \\nATOM   1604  C   ASP A 208      15.159  10.755  30.303  1.00  0.85           C  \\nATOM   1605  O   ASP A 208      14.371  11.289  31.094  1.00  0.85           O  \\nATOM   1606  CB  ASP A 208      17.266  11.928  29.556  1.00  0.85           C  \\nATOM   1607  CG  ASP A 208      18.648  12.444  29.917  1.00  0.85           C  \\nATOM   1608  OD1 ASP A 208      19.233  12.153  30.984  1.00  0.85           O  \\nATOM   1609  OD2 ASP A 208      19.146  13.252  29.083  1.00  0.85           O  \\nATOM   1610  N   LEU A 209      14.699  10.019  29.267  1.00  0.87           N  \\nATOM   1611  CA  LEU A 209      13.290   9.720  29.024  1.00  0.87           C  \\nATOM   1612  C   LEU A 209      12.617   8.925  30.139  1.00  0.87           C  \\nATOM   1613  O   LEU A 209      11.501   9.257  30.551  1.00  0.87           O  \\nATOM   1614  CB  LEU A 209      13.067   8.966  27.686  1.00  0.87           C  \\nATOM   1615  CG  LEU A 209      13.312   9.793  26.410  1.00  0.87           C  \\nATOM   1616  CD1 LEU A 209      13.295   8.891  25.168  1.00  0.87           C  \\nATOM   1617  CD2 LEU A 209      12.284  10.917  26.260  1.00  0.87           C  \\nATOM   1618  N   GLU A 210      13.270   7.888  30.692  1.00  0.80           N  \\nATOM   1619  CA  GLU A 210      12.776   7.116  31.831  1.00  0.80           C  \\nATOM   1620  C   GLU A 210      12.584   8.004  33.062  1.00  0.80           C  \\nATOM   1621  O   GLU A 210      11.566   7.936  33.759  1.00  0.80           O  \\nATOM   1622  CB  GLU A 210      13.736   5.938  32.164  1.00  0.80           C  \\nATOM   1623  CG  GLU A 210      13.916   4.871  31.040  1.00  0.80           C  \\nATOM   1624  CD  GLU A 210      12.907   3.730  31.010  1.00  0.80           C  \\nATOM   1625  OE1 GLU A 210      11.779   3.931  31.516  1.00  0.80           O  \\nATOM   1626  OE2 GLU A 210      13.215   2.656  30.424  1.00  0.80           O  \\nATOM   1627  N   LYS A 211      13.521   8.931  33.340  1.00  0.82           N  \\nATOM   1628  CA  LYS A 211      13.361   9.905  34.413  1.00  0.82           C  \\nATOM   1629  C   LYS A 211      12.191  10.880  34.217  1.00  0.82           C  \\nATOM   1630  O   LYS A 211      11.515  11.287  35.182  1.00  0.82           O  \\nATOM   1631  CB  LYS A 211      14.642  10.741  34.646  1.00  0.82           C  \\nATOM   1632  CG  LYS A 211      14.591  11.542  35.964  1.00  0.82           C  \\nATOM   1633  CD  LYS A 211      15.396  12.846  35.927  1.00  0.82           C  \\nATOM   1634  CE  LYS A 211      14.902  13.898  36.927  1.00  0.82           C  \\nATOM   1635  NZ  LYS A 211      13.548  14.366  36.539  1.00  0.82           N  \\nATOM   1636  N   ILE A 212      11.944  11.353  32.990  1.00  0.85           N  \\nATOM   1637  CA  ILE A 212      10.787  12.182  32.649  1.00  0.85           C  \\nATOM   1638  C   ILE A 212       9.470  11.439  32.788  1.00  0.85           C  \\nATOM   1639  O   ILE A 212       8.490  11.973  33.330  1.00  0.85           O  \\nATOM   1640  CB  ILE A 212      10.918  12.721  31.220  1.00  0.85           C  \\nATOM   1641  CG1 ILE A 212      12.068  13.746  31.130  1.00  0.85           C  \\nATOM   1642  CG2 ILE A 212       9.600  13.347  30.705  1.00  0.85           C  \\nATOM   1643  CD1 ILE A 212      12.571  13.982  29.702  1.00  0.85           C  \\nATOM   1644  N   LEU A 213       9.400  10.205  32.281  1.00  0.86           N  \\nATOM   1645  CA  LEU A 213       8.149   9.493  32.128  1.00  0.86           C  \\nATOM   1646  C   LEU A 213       7.701   8.729  33.368  1.00  0.86           C  \\nATOM   1647  O   LEU A 213       6.476   8.585  33.585  1.00  0.86           O  \\nATOM   1648  CB  LEU A 213       8.273   8.517  30.934  1.00  0.86           C  \\nATOM   1649  CG  LEU A 213       8.396   9.181  29.544  1.00  0.86           C  \\nATOM   1650  CD1 LEU A 213       8.806   8.163  28.468  1.00  0.86           C  \\nATOM   1651  CD2 LEU A 213       7.097   9.887  29.143  1.00  0.86           C  \\nATOM   1652  N   GLY A 214       8.633   8.267  34.209  1.00  0.86           N  \\nATOM   1653  CA  GLY A 214       8.393   7.381  35.346  1.00  0.86           C  \\nATOM   1654  C   GLY A 214       8.167   5.882  35.007  1.00  0.86           C  \\nATOM   1655  O   GLY A 214       8.342   5.440  33.831  1.00  0.86           O  \\nATOM   1656  OXT GLY A 214       7.832   5.140  35.974  1.00  0.86           O  \\n\",\"pdb\");\n\tviewer_1762785482869049.setStyle({\"chain\": \"B\"},{\"style\": {}});\n\tviewer_1762785482869049.setStyle({\"chain\": \"A\"},{\"cartoon\": {\"color\": \"grey\", \"opacity\": 0.75}});\n\tviewer_1762785482869049.setStyle({\"chain\": \"A\", \"resi\": [5, 6, 7, 8, 9, 10, 12, 13, 14, 28, 84, 120, 123, 158, 167]},{\"stick\": {\"color\": \"green\"}, \"cartoon\": {\"color\": \"grey\", \"opacity\": 0.75}});\nviewer_1762785482869049.render();\n});\n</script>",
      "text/html": [
       "<div id=\"3dmolviewer_1762785482869049\"  style=\"position: relative; width: 800px; height: 800px;\">\n",
       "        <p id=\"3dmolwarning_1762785482869049\" style=\"background-color:#ffcccc;color:black\">3Dmol.js failed to load for some reason.  Please check your browser console for error messages.<br></p>\n",
       "        </div>\n",
       "<script>\n",
       "\n",
       "var loadScriptAsync = function(uri){\n",
       "  return new Promise((resolve, reject) => {\n",
       "    //this is to ignore the existence of requirejs amd\n",
       "    var savedexports, savedmodule;\n",
       "    if (typeof exports !== 'undefined') savedexports = exports;\n",
       "    else exports = {}\n",
       "    if (typeof module !== 'undefined') savedmodule = module;\n",
       "    else module = {}\n",
       "\n",
       "    var tag = document.createElement('script');\n",
       "    tag.src = uri;\n",
       "    tag.async = true;\n",
       "    tag.onload = () => {\n",
       "        exports = savedexports;\n",
       "        module = savedmodule;\n",
       "        resolve();\n",
       "    };\n",
       "  var firstScriptTag = document.getElementsByTagName('script')[0];\n",
       "  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n",
       "});\n",
       "};\n",
       "\n",
       "if(typeof $3Dmolpromise === 'undefined') {\n",
       "$3Dmolpromise = null;\n",
       "  $3Dmolpromise = loadScriptAsync('https://cdn.jsdelivr.net/npm/3dmol@2.5.3/build/3Dmol-min.js');\n",
       "}\n",
       "\n",
       "var viewer_1762785482869049 = null;\n",
       "var warn = document.getElementById(\"3dmolwarning_1762785482869049\");\n",
       "if(warn) {\n",
       "    warn.parentNode.removeChild(warn);\n",
       "}\n",
       "$3Dmolpromise.then(function() {\n",
       "viewer_1762785482869049 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_1762785482869049\"),{backgroundColor:\"white\"});\n",
       "viewer_1762785482869049.zoomTo();\n",
       "\tviewer_1762785482869049.setCameraParameters({\"fov\": 35, \"z\": 100});\n",
       "\tviewer_1762785482869049.addModel(\"\\n     RDKit          3D\\n\\n 31 33  0  0  0  0  0  0  0  0999 V2000\\n   10.6521   10.3343   14.0011 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   10.1833   11.3892   14.6904 N   0  0  0  0  0  0  0  0  0  0  0  0\\n   10.6380   12.6361   14.5690 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   11.6702   12.8291   13.6578 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.1618   11.7567   12.9432 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   11.6487   10.5360   13.1250 N   0  0  0  0  0  0  0  0  0  0  0  0\\n   13.1405   12.1864   12.1387 N   0  0  0  0  0  0  0  0  0  0  0  0\\n   13.2527   13.5084   12.3533 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.3594   13.9108   13.2791 N   0  0  0  0  0  0  0  0  0  0  0  0\\n   13.9404   11.4328   11.2147 C   0  0  1  0  0  0  0  0  0  0  0  0\\n   13.0970   10.5567   10.3101 C   0  0  1  0  0  0  0  0  0  0  0  0\\n   14.0970   10.2330    9.2200 C   0  0  1  0  0  0  0  0  0  0  0  0\\n   15.1022   11.3646    9.3602 C   0  0  2  0  0  0  0  0  0  0  0  0\\n   14.5365   12.2749   10.2455 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   15.3278   11.9485    7.9957 C   0  0  0  0  0  0  0  0  0  0  0  0\\n   14.1414   12.3144    7.3719 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   14.1306   13.9155    6.8790 P   0  0  1  0  0  5  0  0  0  0  0  0\\n   15.3201   14.5881    7.5527 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   14.3876   14.0796    5.2137 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.7538   14.7495    7.3818 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.5295   16.1870    6.5309 P   0  0  2  0  0  5  0  0  0  0  0  0\\n   13.8242   16.9944    6.5506 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   11.2268   17.0794    7.0973 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.2339   15.8127    4.9069 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.2035   17.1749    3.9374 P   0  0  0  0  0  5  0  0  0  0  0  0\\n   13.4776   17.9913    4.0286 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   11.8921   16.7016    2.3363 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   10.8691   18.1237    4.4366 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   14.7604    9.0468    9.4777 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   12.6152    9.4366   10.9550 O   0  0  0  0  0  0  0  0  0  0  0  0\\n   10.1436   13.7407   15.2941 N   0  0  0  0  0  0  0  0  0  0  0  0\\n  1  2  2  0\\n  2  3  1  0\\n  3  4  2  0\\n  4  5  1  0\\n  5  6  2  0\\n  5  7  1  0\\n  7  8  1  0\\n  8  9  2  0\\n 10  7  1  1\\n 10 11  1  0\\n 11 12  1  0\\n 12 13  1  0\\n 13 14  1  0\\n 13 15  1  6\\n 15 16  1  0\\n 16 17  1  0\\n 17 18  2  0\\n 17 19  1  6\\n 17 20  1  0\\n 20 21  1  0\\n 21 22  2  0\\n 21 23  1  1\\n 21 24  1  0\\n 24 25  1  0\\n 25 26  2  0\\n 25 27  1  0\\n 25 28  1  0\\n 12 29  1  1\\n 11 30  1  1\\n  3 31  1  0\\n  6  1  1  0\\n  9  4  1  0\\n 14 10  1  0\\nM  END\\n$$$$\\n\",\"sdf\");\n",
       "\tviewer_1762785482869049.setStyle({\"model\": 0},{\"stick\": {\"color\": \"red\"}});\n",
       "\tviewer_1762785482869049.zoomTo();\n",
       "\tviewer_1762785482869049.addModel(\"ATOM      1  N   MET A   1      -0.707   5.264  27.934  1.00  0.84           N  \\nATOM      2  CA  MET A   1       0.264   6.383  27.765  1.00  0.84           C  \\nATOM      3  C   MET A   1       0.229   7.063  26.413  1.00  0.84           C  \\nATOM      4  O   MET A   1       0.446   6.432  25.379  1.00  0.84           O  \\nATOM      5  CB  MET A   1       1.689   5.853  28.038  1.00  0.84           C  \\nATOM      6  CG  MET A   1       2.845   6.837  27.760  1.00  0.84           C  \\nATOM      7  SD  MET A   1       3.113   8.251  28.874  1.00  0.84           S  \\nATOM      8  CE  MET A   1       1.860   8.129  30.182  1.00  0.84           C  \\nATOM      9  N   ARG A   2      -0.057   8.357  26.364  1.00  0.83           N  \\nATOM     10  CA  ARG A   2      -0.271   9.136  25.174  1.00  0.83           C  \\nATOM     11  C   ARG A   2       0.821  10.187  25.088  1.00  0.83           C  \\nATOM     12  O   ARG A   2       0.955  11.051  25.951  1.00  0.83           O  \\nATOM     13  CB  ARG A   2      -1.662   9.798  25.293  1.00  0.83           C  \\nATOM     14  CG  ARG A   2      -2.814   8.789  25.500  1.00  0.83           C  \\nATOM     15  CD  ARG A   2      -4.011   9.369  26.262  1.00  0.83           C  \\nATOM     16  NE  ARG A   2      -4.715  10.321  25.357  1.00  0.83           N  \\nATOM     17  CZ  ARG A   2      -5.729  10.021  24.526  1.00  0.83           C  \\nATOM     18  NH1 ARG A   2      -6.216   8.798  24.395  1.00  0.83           N  \\nATOM     19  NH2 ARG A   2      -6.235  11.025  23.812  1.00  0.83           N  \\nATOM     20  N   ILE A   3       1.662  10.110  24.045  1.00  0.95           N  \\nATOM     21  CA  ILE A   3       2.851  10.935  23.939  1.00  0.95           C  \\nATOM     22  C   ILE A   3       2.838  11.666  22.614  1.00  0.95           C  \\nATOM     23  O   ILE A   3       2.588  11.083  21.565  1.00  0.95           O  \\nATOM     24  CB  ILE A   3       4.130  10.105  24.079  1.00  0.95           C  \\nATOM     25  CG1 ILE A   3       4.218   9.476  25.485  1.00  0.95           C  \\nATOM     26  CG2 ILE A   3       5.387  10.958  23.817  1.00  0.95           C  \\nATOM     27  CD1 ILE A   3       5.269   8.369  25.609  1.00  0.95           C  \\nATOM     28  N   ILE A   4       3.124  12.980  22.639  1.00  0.95           N  \\nATOM     29  CA  ILE A   4       3.469  13.750  21.458  1.00  0.95           C  \\nATOM     30  C   ILE A   4       4.976  13.891  21.462  1.00  0.95           C  \\nATOM     31  O   ILE A   4       5.582  14.290  22.454  1.00  0.95           O  \\nATOM     32  CB  ILE A   4       2.787  15.114  21.428  1.00  0.95           C  \\nATOM     33  CG1 ILE A   4       1.297  14.924  21.088  1.00  0.95           C  \\nATOM     34  CG2 ILE A   4       3.440  16.059  20.400  1.00  0.95           C  \\nATOM     35  CD1 ILE A   4       0.440  16.165  21.345  1.00  0.95           C  \\nATOM     36  N   LEU A   5       5.625  13.543  20.340  1.00  0.94           N  \\nATOM     37  CA  LEU A   5       7.059  13.667  20.205  1.00  0.94           C  \\nATOM     38  C   LEU A   5       7.365  14.698  19.126  1.00  0.94           C  \\nATOM     39  O   LEU A   5       7.021  14.544  17.955  1.00  0.94           O  \\nATOM     40  CB  LEU A   5       7.683  12.298  19.874  1.00  0.94           C  \\nATOM     41  CG  LEU A   5       9.218  12.250  19.775  1.00  0.94           C  \\nATOM     42  CD1 LEU A   5       9.935  12.652  21.071  1.00  0.94           C  \\nATOM     43  CD2 LEU A   5       9.654  10.846  19.339  1.00  0.94           C  \\nATOM     44  N   LEU A   6       8.016  15.809  19.517  1.00  0.92           N  \\nATOM     45  CA  LEU A   6       8.330  16.918  18.638  1.00  0.92           C  \\nATOM     46  C   LEU A   6       9.827  16.961  18.409  1.00  0.92           C  \\nATOM     47  O   LEU A   6      10.626  16.514  19.222  1.00  0.92           O  \\nATOM     48  CB  LEU A   6       7.895  18.282  19.226  1.00  0.92           C  \\nATOM     49  CG  LEU A   6       6.400  18.390  19.577  1.00  0.92           C  \\nATOM     50  CD1 LEU A   6       6.091  19.775  20.160  1.00  0.92           C  \\nATOM     51  CD2 LEU A   6       5.480  18.089  18.388  1.00  0.92           C  \\nATOM     52  N   GLY A   7      10.268  17.491  17.257  1.00  0.93           N  \\nATOM     53  CA  GLY A   7      11.694  17.583  17.001  1.00  0.93           C  \\nATOM     54  C   GLY A   7      11.928  18.120  15.625  1.00  0.93           C  \\nATOM     55  O   GLY A   7      11.191  17.815  14.690  1.00  0.93           O  \\nATOM     56  N   ALA A   8      12.976  18.951  15.484  1.00  0.95           N  \\nATOM     57  CA  ALA A   8      13.364  19.596  14.249  1.00  0.95           C  \\nATOM     58  C   ALA A   8      13.677  18.618  13.110  1.00  0.95           C  \\nATOM     59  O   ALA A   8      14.126  17.495  13.376  1.00  0.95           O  \\nATOM     60  CB  ALA A   8      14.550  20.541  14.530  1.00  0.95           C  \\nATOM     61  N   PRO A   9      13.433  18.937  11.830  1.00  0.90           N  \\nATOM     62  CA  PRO A   9      13.850  18.086  10.722  1.00  0.90           C  \\nATOM     63  C   PRO A   9      15.341  17.815  10.751  1.00  0.90           C  \\nATOM     64  O   PRO A   9      16.148  18.739  10.639  1.00  0.90           O  \\nATOM     65  CB  PRO A   9      13.400  18.846   9.464  1.00  0.90           C  \\nATOM     66  CG  PRO A   9      13.403  20.311   9.898  1.00  0.90           C  \\nATOM     67  CD  PRO A   9      12.964  20.242  11.359  1.00  0.90           C  \\nATOM     68  N   GLY A  10      15.730  16.539  10.870  1.00  0.86           N  \\nATOM     69  CA  GLY A  10      17.128  16.157  10.977  1.00  0.86           C  \\nATOM     70  C   GLY A  10      17.620  15.900  12.363  1.00  0.86           C  \\nATOM     71  O   GLY A  10      18.674  15.270  12.509  1.00  0.86           O  \\nATOM     72  N   ALA A  11      16.885  16.278  13.414  1.00  0.89           N  \\nATOM     73  CA  ALA A  11      17.268  16.014  14.794  1.00  0.89           C  \\nATOM     74  C   ALA A  11      17.355  14.525  15.073  1.00  0.89           C  \\nATOM     75  O   ALA A  11      18.306  14.038  15.723  1.00  0.89           O  \\nATOM     76  CB  ALA A  11      16.314  16.732  15.768  1.00  0.89           C  \\nATOM     77  N   GLY A  12      16.434  13.716  14.534  1.00  0.80           N  \\nATOM     78  CA  GLY A  12      16.474  12.268  14.694  1.00  0.80           C  \\nATOM     79  C   GLY A  12      15.686  11.741  15.852  1.00  0.80           C  \\nATOM     80  O   GLY A  12      16.034  10.686  16.389  1.00  0.80           O  \\nATOM     81  N   LYS A  13      14.572  12.374  16.227  1.00  0.78           N  \\nATOM     82  CA  LYS A  13      13.675  11.954  17.307  1.00  0.78           C  \\nATOM     83  C   LYS A  13      13.165  10.524  17.204  1.00  0.78           C  \\nATOM     84  O   LYS A  13      12.868   9.860  18.212  1.00  0.78           O  \\nATOM     85  CB  LYS A  13      12.467  12.918  17.391  1.00  0.78           C  \\nATOM     86  CG  LYS A  13      11.535  12.914  16.168  1.00  0.78           C  \\nATOM     87  CD  LYS A  13      10.395  13.937  16.299  1.00  0.78           C  \\nATOM     88  CE  LYS A  13       9.875  14.523  14.982  1.00  0.78           C  \\nATOM     89  NZ  LYS A  13       9.513  13.437  14.061  1.00  0.78           N  \\nATOM     90  N   GLY A  14      13.112   9.980  15.981  1.00  0.85           N  \\nATOM     91  CA  GLY A  14      12.661   8.626  15.680  1.00  0.85           C  \\nATOM     92  C   GLY A  14      13.592   7.544  16.179  1.00  0.85           C  \\nATOM     93  O   GLY A  14      13.216   6.365  16.209  1.00  0.85           O  \\nATOM     94  N   THR A  15      14.801   7.894  16.648  1.00  0.79           N  \\nATOM     95  CA  THR A  15      15.762   6.987  17.287  1.00  0.79           C  \\nATOM     96  C   THR A  15      15.247   6.421  18.596  1.00  0.79           C  \\nATOM     97  O   THR A  15      15.692   5.353  19.040  1.00  0.79           O  \\nATOM     98  CB  THR A  15      17.137   7.609  17.565  1.00  0.79           C  \\nATOM     99  OG1 THR A  15      17.040   8.777  18.374  1.00  0.79           O  \\nATOM    100  CG2 THR A  15      17.824   8.003  16.247  1.00  0.79           C  \\nATOM    101  N   GLN A  16      14.286   7.096  19.246  1.00  0.83           N  \\nATOM    102  CA  GLN A  16      13.679   6.632  20.477  1.00  0.83           C  \\nATOM    103  C   GLN A  16      12.272   6.088  20.295  1.00  0.83           C  \\nATOM    104  O   GLN A  16      11.631   5.672  21.265  1.00  0.83           O  \\nATOM    105  CB  GLN A  16      13.666   7.765  21.525  1.00  0.83           C  \\nATOM    106  CG  GLN A  16      15.084   8.224  21.927  1.00  0.83           C  \\nATOM    107  CD  GLN A  16      15.889   7.105  22.592  1.00  0.83           C  \\nATOM    108  OE1 GLN A  16      15.377   6.152  23.173  1.00  0.83           O  \\nATOM    109  NE2 GLN A  16      17.237   7.249  22.529  1.00  0.83           N  \\nATOM    110  N   ALA A  17      11.745   6.023  19.055  1.00  0.87           N  \\nATOM    111  CA  ALA A  17      10.391   5.552  18.816  1.00  0.87           C  \\nATOM    112  C   ALA A  17      10.161   4.101  19.233  1.00  0.87           C  \\nATOM    113  O   ALA A  17       9.195   3.792  19.935  1.00  0.87           O  \\nATOM    114  CB  ALA A  17      10.013   5.743  17.337  1.00  0.87           C  \\nATOM    115  N   GLN A  18      11.076   3.181  18.875  1.00  0.79           N  \\nATOM    116  CA  GLN A  18      11.001   1.770  19.244  1.00  0.79           C  \\nATOM    117  C   GLN A  18      11.091   1.522  20.736  1.00  0.79           C  \\nATOM    118  O   GLN A  18      10.329   0.707  21.279  1.00  0.79           O  \\nATOM    119  CB  GLN A  18      12.022   0.916  18.464  1.00  0.79           C  \\nATOM    120  CG  GLN A  18      11.613   0.780  16.981  1.00  0.79           C  \\nATOM    121  CD  GLN A  18      12.646  -0.023  16.183  1.00  0.79           C  \\nATOM    122  OE1 GLN A  18      13.470  -0.741  16.707  1.00  0.79           O  \\nATOM    123  NE2 GLN A  18      12.575   0.118  14.829  1.00  0.79           N  \\nATOM    124  N   PHE A  19      11.949   2.264  21.452  1.00  0.87           N  \\nATOM    125  CA  PHE A  19      12.016   2.252  22.905  1.00  0.87           C  \\nATOM    126  C   PHE A  19      10.667   2.619  23.528  1.00  0.87           C  \\nATOM    127  O   PHE A  19      10.157   1.912  24.403  1.00  0.87           O  \\nATOM    128  CB  PHE A  19      13.143   3.233  23.367  1.00  0.87           C  \\nATOM    129  CG  PHE A  19      13.035   3.617  24.820  1.00  0.87           C  \\nATOM    130  CD1 PHE A  19      13.117   2.639  25.823  1.00  0.87           C  \\nATOM    131  CD2 PHE A  19      12.670   4.930  25.173  1.00  0.87           C  \\nATOM    132  CE1 PHE A  19      12.836   2.967  27.154  1.00  0.87           C  \\nATOM    133  CE2 PHE A  19      12.374   5.252  26.502  1.00  0.87           C  \\nATOM    134  CZ  PHE A  19      12.465   4.272  27.491  1.00  0.87           C  \\nATOM    135  N   ILE A  20      10.033   3.713  23.072  1.00  0.88           N  \\nATOM    136  CA  ILE A  20       8.736   4.143  23.577  1.00  0.88           C  \\nATOM    137  C   ILE A  20       7.626   3.143  23.265  1.00  0.88           C  \\nATOM    138  O   ILE A  20       6.806   2.803  24.128  1.00  0.88           O  \\nATOM    139  CB  ILE A  20       8.395   5.542  23.074  1.00  0.88           C  \\nATOM    140  CG1 ILE A  20       9.373   6.565  23.696  1.00  0.88           C  \\nATOM    141  CG2 ILE A  20       6.932   5.909  23.397  1.00  0.88           C  \\nATOM    142  CD1 ILE A  20       9.305   7.952  23.050  1.00  0.88           C  \\nATOM    143  N   MET A  21       7.589   2.608  22.031  1.00  0.86           N  \\nATOM    144  CA  MET A  21       6.621   1.603  21.624  1.00  0.86           C  \\nATOM    145  C   MET A  21       6.715   0.318  22.432  1.00  0.86           C  \\nATOM    146  O   MET A  21       5.694  -0.205  22.893  1.00  0.86           O  \\nATOM    147  CB  MET A  21       6.770   1.241  20.128  1.00  0.86           C  \\nATOM    148  CG  MET A  21       6.336   2.349  19.152  1.00  0.86           C  \\nATOM    149  SD  MET A  21       6.084   1.750  17.454  1.00  0.86           S  \\nATOM    150  CE  MET A  21       7.834   1.578  17.025  1.00  0.86           C  \\nATOM    151  N   GLU A  22       7.931  -0.204  22.653  1.00  0.84           N  \\nATOM    152  CA  GLU A  22       8.172  -1.371  23.489  1.00  0.84           C  \\nATOM    153  C   GLU A  22       7.847  -1.127  24.954  1.00  0.84           C  \\nATOM    154  O   GLU A  22       7.208  -1.970  25.610  1.00  0.84           O  \\nATOM    155  CB  GLU A  22       9.612  -1.898  23.310  1.00  0.84           C  \\nATOM    156  CG  GLU A  22       9.895  -3.204  24.093  1.00  0.84           C  \\nATOM    157  CD  GLU A  22      11.235  -3.859  23.749  1.00  0.84           C  \\nATOM    158  OE1 GLU A  22      11.879  -3.446  22.751  1.00  0.84           O  \\nATOM    159  OE2 GLU A  22      11.614  -4.797  24.496  1.00  0.84           O  \\nATOM    160  N   LYS A  23       8.206   0.041  25.514  1.00  0.84           N  \\nATOM    161  CA  LYS A  23       7.943   0.402  26.904  1.00  0.84           C  \\nATOM    162  C   LYS A  23       6.470   0.387  27.289  1.00  0.84           C  \\nATOM    163  O   LYS A  23       6.112  -0.045  28.398  1.00  0.84           O  \\nATOM    164  CB  LYS A  23       8.534   1.800  27.239  1.00  0.84           C  \\nATOM    165  CG  LYS A  23       8.352   2.219  28.715  1.00  0.84           C  \\nATOM    166  CD  LYS A  23       9.080   3.520  29.098  1.00  0.84           C  \\nATOM    167  CE  LYS A  23       8.801   4.046  30.524  1.00  0.84           C  \\nATOM    168  NZ  LYS A  23       9.332   3.167  31.585  1.00  0.84           N  \\nATOM    169  N   TYR A  24       5.579   0.874  26.415  1.00  0.84           N  \\nATOM    170  CA  TYR A  24       4.167   1.009  26.733  1.00  0.84           C  \\nATOM    171  C   TYR A  24       3.237   0.110  25.942  1.00  0.84           C  \\nATOM    172  O   TYR A  24       2.035   0.040  26.242  1.00  0.84           O  \\nATOM    173  CB  TYR A  24       3.739   2.460  26.446  1.00  0.84           C  \\nATOM    174  CG  TYR A  24       4.457   3.383  27.371  1.00  0.84           C  \\nATOM    175  CD1 TYR A  24       4.123   3.389  28.730  1.00  0.84           C  \\nATOM    176  CD2 TYR A  24       5.463   4.241  26.905  1.00  0.84           C  \\nATOM    177  CE1 TYR A  24       4.746   4.282  29.603  1.00  0.84           C  \\nATOM    178  CE2 TYR A  24       6.106   5.126  27.781  1.00  0.84           C  \\nATOM    179  CZ  TYR A  24       5.716   5.158  29.125  1.00  0.84           C  \\nATOM    180  OH  TYR A  24       6.230   6.080  30.040  1.00  0.84           O  \\nATOM    181  N   GLY A  25       3.721  -0.599  24.914  1.00  0.90           N  \\nATOM    182  CA  GLY A  25       2.877  -1.415  24.044  1.00  0.90           C  \\nATOM    183  C   GLY A  25       1.942  -0.579  23.209  1.00  0.90           C  \\nATOM    184  O   GLY A  25       0.736  -0.856  23.153  1.00  0.90           O  \\nATOM    185  N   ILE A  26       2.463   0.496  22.598  1.00  0.89           N  \\nATOM    186  CA  ILE A  26       1.692   1.510  21.898  1.00  0.89           C  \\nATOM    187  C   ILE A  26       2.224   1.680  20.486  1.00  0.89           C  \\nATOM    188  O   ILE A  26       3.413   1.447  20.267  1.00  0.89           O  \\nATOM    189  CB  ILE A  26       1.688   2.867  22.609  1.00  0.89           C  \\nATOM    190  CG1 ILE A  26       3.093   3.497  22.794  1.00  0.89           C  \\nATOM    191  CG2 ILE A  26       0.914   2.682  23.924  1.00  0.89           C  \\nATOM    192  CD1 ILE A  26       3.097   4.819  23.573  1.00  0.89           C  \\nATOM    193  N   PRO A  27       1.434   2.060  19.486  1.00  0.94           N  \\nATOM    194  CA  PRO A  27       1.948   2.276  18.143  1.00  0.94           C  \\nATOM    195  C   PRO A  27       2.489   3.684  17.979  1.00  0.94           C  \\nATOM    196  O   PRO A  27       2.091   4.615  18.677  1.00  0.94           O  \\nATOM    197  CB  PRO A  27       0.722   2.055  17.242  1.00  0.94           C  \\nATOM    198  CG  PRO A  27      -0.466   2.458  18.113  1.00  0.94           C  \\nATOM    199  CD  PRO A  27      -0.031   2.036  19.516  1.00  0.94           C  \\nATOM    200  N   GLN A  28       3.419   3.846  17.023  1.00  0.89           N  \\nATOM    201  CA  GLN A  28       3.857   5.130  16.526  1.00  0.89           C  \\nATOM    202  C   GLN A  28       2.894   5.530  15.421  1.00  0.89           C  \\nATOM    203  O   GLN A  28       2.607   4.752  14.515  1.00  0.89           O  \\nATOM    204  CB  GLN A  28       5.302   5.026  15.994  1.00  0.89           C  \\nATOM    205  CG  GLN A  28       5.917   6.330  15.435  1.00  0.89           C  \\nATOM    206  CD  GLN A  28       7.346   6.045  14.944  1.00  0.89           C  \\nATOM    207  OE1 GLN A  28       7.785   4.908  14.869  1.00  0.89           O  \\nATOM    208  NE2 GLN A  28       8.120   7.104  14.594  1.00  0.89           N  \\nATOM    209  N   ILE A  29       2.310   6.737  15.512  1.00  0.88           N  \\nATOM    210  CA  ILE A  29       1.329   7.227  14.567  1.00  0.88           C  \\nATOM    211  C   ILE A  29       1.792   8.584  14.071  1.00  0.88           C  \\nATOM    212  O   ILE A  29       2.444   9.347  14.785  1.00  0.88           O  \\nATOM    213  CB  ILE A  29      -0.102   7.280  15.132  1.00  0.88           C  \\nATOM    214  CG1 ILE A  29      -0.282   8.339  16.245  1.00  0.88           C  \\nATOM    215  CG2 ILE A  29      -0.487   5.867  15.619  1.00  0.88           C  \\nATOM    216  CD1 ILE A  29      -1.710   8.472  16.780  1.00  0.88           C  \\nATOM    217  N   SER A  30       1.473   8.933  12.815  1.00  0.90           N  \\nATOM    218  CA  SER A  30       1.903  10.187  12.227  1.00  0.90           C  \\nATOM    219  C   SER A  30       0.840  10.618  11.255  1.00  0.90           C  \\nATOM    220  O   SER A  30       0.062   9.802  10.765  1.00  0.90           O  \\nATOM    221  CB  SER A  30       3.295  10.149  11.545  1.00  0.90           C  \\nATOM    222  OG  SER A  30       3.243   9.504  10.236  1.00  0.90           O  \\nATOM    223  N   THR A  31       0.771  11.923  10.947  1.00  0.89           N  \\nATOM    224  CA  THR A  31      -0.174  12.471   9.977  1.00  0.89           C  \\nATOM    225  C   THR A  31       0.011  11.885   8.596  1.00  0.89           C  \\nATOM    226  O   THR A  31      -0.959  11.630   7.884  1.00  0.89           O  \\nATOM    227  CB  THR A  31      -0.139  13.986   9.877  1.00  0.89           C  \\nATOM    228  OG1 THR A  31       1.192  14.510   9.668  1.00  0.89           O  \\nATOM    229  CG2 THR A  31      -0.617  14.571  11.208  1.00  0.89           C  \\nATOM    230  N   GLY A  32       1.265  11.621   8.197  1.00  0.85           N  \\nATOM    231  CA  GLY A  32       1.586  10.969   6.933  1.00  0.85           C  \\nATOM    232  C   GLY A  32       1.125   9.532   6.846  1.00  0.85           C  \\nATOM    233  O   GLY A  32       0.557   9.125   5.828  1.00  0.85           O  \\nATOM    234  N   ASP A  33       1.334   8.722   7.895  1.00  0.83           N  \\nATOM    235  CA  ASP A  33       0.866   7.341   7.952  1.00  0.83           C  \\nATOM    236  C   ASP A  33      -0.646   7.227   7.978  1.00  0.83           C  \\nATOM    237  O   ASP A  33      -1.239   6.419   7.250  1.00  0.83           O  \\nATOM    238  CB  ASP A  33       1.479   6.597   9.165  1.00  0.83           C  \\nATOM    239  CG  ASP A  33       2.934   6.244   8.906  1.00  0.83           C  \\nATOM    240  OD1 ASP A  33       3.440   6.587   7.796  1.00  0.83           O  \\nATOM    241  OD2 ASP A  33       3.559   5.634   9.807  1.00  0.83           O  \\nATOM    242  N   MET A  34      -1.320   8.078   8.771  1.00  0.81           N  \\nATOM    243  CA  MET A  34      -2.771   8.142   8.836  1.00  0.81           C  \\nATOM    244  C   MET A  34      -3.384   8.470   7.495  1.00  0.81           C  \\nATOM    245  O   MET A  34      -4.330   7.816   7.041  1.00  0.81           O  \\nATOM    246  CB  MET A  34      -3.231   9.267   9.792  1.00  0.81           C  \\nATOM    247  CG  MET A  34      -3.002   8.989  11.279  1.00  0.81           C  \\nATOM    248  SD  MET A  34      -3.515  10.396  12.292  1.00  0.81           S  \\nATOM    249  CE  MET A  34      -2.144  10.358  13.475  1.00  0.81           C  \\nATOM    250  N   LEU A  35      -2.819   9.469   6.811  1.00  0.83           N  \\nATOM    251  CA  LEU A  35      -3.248   9.900   5.503  1.00  0.83           C  \\nATOM    252  C   LEU A  35      -3.103   8.818   4.443  1.00  0.83           C  \\nATOM    253  O   LEU A  35      -4.048   8.532   3.695  1.00  0.83           O  \\nATOM    254  CB  LEU A  35      -2.415  11.145   5.148  1.00  0.83           C  \\nATOM    255  CG  LEU A  35      -2.800  11.866   3.858  1.00  0.83           C  \\nATOM    256  CD1 LEU A  35      -4.256  12.323   3.889  1.00  0.83           C  \\nATOM    257  CD2 LEU A  35      -1.881  13.072   3.633  1.00  0.83           C  \\nATOM    258  N   ARG A  36      -1.952   8.127   4.393  1.00  0.75           N  \\nATOM    259  CA  ARG A  36      -1.722   6.997   3.506  1.00  0.75           C  \\nATOM    260  C   ARG A  36      -2.640   5.811   3.766  1.00  0.75           C  \\nATOM    261  O   ARG A  36      -3.193   5.217   2.828  1.00  0.75           O  \\nATOM    262  CB  ARG A  36      -0.259   6.506   3.621  1.00  0.75           C  \\nATOM    263  CG  ARG A  36       0.765   7.442   2.956  1.00  0.75           C  \\nATOM    264  CD  ARG A  36       2.196   6.883   2.958  1.00  0.75           C  \\nATOM    265  NE  ARG A  36       2.798   7.053   4.325  1.00  0.75           N  \\nATOM    266  CZ  ARG A  36       3.498   8.139   4.706  1.00  0.75           C  \\nATOM    267  NH1 ARG A  36       3.613   9.204   3.930  1.00  0.75           N  \\nATOM    268  NH2 ARG A  36       4.066   8.178   5.907  1.00  0.75           N  \\nATOM    269  N   ALA A  37      -2.833   5.435   5.039  1.00  0.86           N  \\nATOM    270  CA  ALA A  37      -3.729   4.367   5.437  1.00  0.86           C  \\nATOM    271  C   ALA A  37      -5.193   4.659   5.128  1.00  0.86           C  \\nATOM    272  O   ALA A  37      -5.906   3.788   4.603  1.00  0.86           O  \\nATOM    273  CB  ALA A  37      -3.542   4.060   6.935  1.00  0.86           C  \\nATOM    274  N   ALA A  38      -5.678   5.880   5.399  1.00  0.85           N  \\nATOM    275  CA  ALA A  38      -7.026   6.307   5.070  1.00  0.85           C  \\nATOM    276  C   ALA A  38      -7.300   6.315   3.571  1.00  0.85           C  \\nATOM    277  O   ALA A  38      -8.311   5.771   3.105  1.00  0.85           O  \\nATOM    278  CB  ALA A  38      -7.287   7.699   5.676  1.00  0.85           C  \\nATOM    279  N   VAL A  39      -6.397   6.861   2.741  1.00  0.83           N  \\nATOM    280  CA  VAL A  39      -6.548   6.862   1.283  1.00  0.83           C  \\nATOM    281  C   VAL A  39      -6.618   5.460   0.708  1.00  0.83           C  \\nATOM    282  O   VAL A  39      -7.464   5.154  -0.149  1.00  0.83           O  \\nATOM    283  CB  VAL A  39      -5.427   7.640   0.604  1.00  0.83           C  \\nATOM    284  CG1 VAL A  39      -5.445   7.455  -0.926  1.00  0.83           C  \\nATOM    285  CG2 VAL A  39      -5.606   9.131   0.931  1.00  0.83           C  \\nATOM    286  N   LYS A  40      -5.753   4.558   1.184  1.00  0.80           N  \\nATOM    287  CA  LYS A  40      -5.744   3.158   0.810  1.00  0.80           C  \\nATOM    288  C   LYS A  40      -6.986   2.389   1.235  1.00  0.80           C  \\nATOM    289  O   LYS A  40      -7.427   1.466   0.535  1.00  0.80           O  \\nATOM    290  CB  LYS A  40      -4.483   2.470   1.373  1.00  0.80           C  \\nATOM    291  CG  LYS A  40      -4.310   1.026   0.877  1.00  0.80           C  \\nATOM    292  CD  LYS A  40      -2.999   0.376   1.337  1.00  0.80           C  \\nATOM    293  CE  LYS A  40      -2.869  -1.070   0.856  1.00  0.80           C  \\nATOM    294  NZ  LYS A  40      -1.624  -1.667   1.387  1.00  0.80           N  \\nATOM    295  N   SER A  41      -7.569   2.710   2.400  1.00  0.83           N  \\nATOM    296  CA  SER A  41      -8.744   2.030   2.934  1.00  0.83           C  \\nATOM    297  C   SER A  41     -10.009   2.305   2.156  1.00  0.83           C  \\nATOM    298  O   SER A  41     -10.924   1.462   2.115  1.00  0.83           O  \\nATOM    299  CB  SER A  41      -8.974   2.307   4.451  1.00  0.83           C  \\nATOM    300  OG  SER A  41      -9.526   3.597   4.720  1.00  0.83           O  \\nATOM    301  N   GLY A  42     -10.127   3.488   1.536  1.00  0.84           N  \\nATOM    302  CA  GLY A  42     -11.339   3.915   0.854  1.00  0.84           C  \\nATOM    303  C   GLY A  42     -12.409   4.388   1.809  1.00  0.84           C  \\nATOM    304  O   GLY A  42     -13.570   4.525   1.422  1.00  0.84           O  \\nATOM    305  N   SER A  43     -12.049   4.626   3.093  1.00  0.83           N  \\nATOM    306  CA  SER A  43     -12.886   5.270   4.112  1.00  0.83           C  \\nATOM    307  C   SER A  43     -13.446   6.615   3.681  1.00  0.83           C  \\nATOM    308  O   SER A  43     -12.953   7.238   2.752  1.00  0.83           O  \\nATOM    309  CB  SER A  43     -12.235   5.412   5.532  1.00  0.83           C  \\nATOM    310  OG  SER A  43     -11.199   6.400   5.617  1.00  0.83           O  \\nATOM    311  N   GLU A  44     -14.484   7.141   4.370  1.00  0.82           N  \\nATOM    312  CA  GLU A  44     -15.048   8.446   4.048  1.00  0.82           C  \\nATOM    313  C   GLU A  44     -14.029   9.585   4.144  1.00  0.82           C  \\nATOM    314  O   GLU A  44     -13.885  10.417   3.241  1.00  0.82           O  \\nATOM    315  CB  GLU A  44     -16.256   8.723   4.980  1.00  0.82           C  \\nATOM    316  CG  GLU A  44     -17.462   7.789   4.717  1.00  0.82           C  \\nATOM    317  CD  GLU A  44     -17.938   7.934   3.274  1.00  0.82           C  \\nATOM    318  OE1 GLU A  44     -18.288   9.077   2.884  1.00  0.82           O  \\nATOM    319  OE2 GLU A  44     -17.923   6.907   2.551  1.00  0.82           O  \\nATOM    320  N   LEU A  45     -13.216   9.596   5.218  1.00  0.85           N  \\nATOM    321  CA  LEU A  45     -12.085  10.502   5.366  1.00  0.85           C  \\nATOM    322  C   LEU A  45     -10.989  10.256   4.344  1.00  0.85           C  \\nATOM    323  O   LEU A  45     -10.399  11.195   3.791  1.00  0.85           O  \\nATOM    324  CB  LEU A  45     -11.485  10.428   6.786  1.00  0.85           C  \\nATOM    325  CG  LEU A  45     -12.424  10.877   7.922  1.00  0.85           C  \\nATOM    326  CD1 LEU A  45     -11.679  10.764   9.255  1.00  0.85           C  \\nATOM    327  CD2 LEU A  45     -12.937  12.308   7.733  1.00  0.85           C  \\nATOM    328  N   GLY A  46     -10.702   8.985   4.034  1.00  0.81           N  \\nATOM    329  CA  GLY A  46      -9.751   8.595   3.007  1.00  0.81           C  \\nATOM    330  C   GLY A  46     -10.116   8.984   1.614  1.00  0.81           C  \\nATOM    331  O   GLY A  46      -9.259   9.378   0.821  1.00  0.81           O  \\nATOM    332  N   LYS A  47     -11.407   8.923   1.270  1.00  0.79           N  \\nATOM    333  CA  LYS A  47     -11.904   9.420   0.008  1.00  0.79           C  \\nATOM    334  C   LYS A  47     -11.755  10.930  -0.110  1.00  0.79           C  \\nATOM    335  O   LYS A  47     -11.256  11.432  -1.127  1.00  0.79           O  \\nATOM    336  CB  LYS A  47     -13.381   9.037  -0.236  1.00  0.79           C  \\nATOM    337  CG  LYS A  47     -13.750   9.249  -1.712  1.00  0.79           C  \\nATOM    338  CD  LYS A  47     -15.222   9.594  -1.957  1.00  0.79           C  \\nATOM    339  CE  LYS A  47     -15.539   9.852  -3.439  1.00  0.79           C  \\nATOM    340  NZ  LYS A  47     -14.629  10.844  -4.046  1.00  0.79           N  \\nATOM    341  N   GLN A  48     -12.117  11.694   0.937  1.00  0.79           N  \\nATOM    342  CA  GLN A  48     -11.940  13.141   0.956  1.00  0.79           C  \\nATOM    343  C   GLN A  48     -10.482  13.543   0.835  1.00  0.79           C  \\nATOM    344  O   GLN A  48     -10.119  14.457   0.074  1.00  0.79           O  \\nATOM    345  CB  GLN A  48     -12.504  13.766   2.256  1.00  0.79           C  \\nATOM    346  CG  GLN A  48     -12.382  15.310   2.280  1.00  0.79           C  \\nATOM    347  CD  GLN A  48     -13.011  15.929   3.531  1.00  0.79           C  \\nATOM    348  OE1 GLN A  48     -13.595  15.271   4.372  1.00  0.79           O  \\nATOM    349  NE2 GLN A  48     -12.874  17.281   3.642  1.00  0.79           N  \\nATOM    350  N   ALA A  49      -9.598  12.854   1.562  1.00  0.88           N  \\nATOM    351  CA  ALA A  49      -8.169  13.030   1.480  1.00  0.88           C  \\nATOM    352  C   ALA A  49      -7.596  12.706   0.116  1.00  0.88           C  \\nATOM    353  O   ALA A  49      -6.781  13.472  -0.416  1.00  0.88           O  \\nATOM    354  CB  ALA A  49      -7.489  12.147   2.532  1.00  0.88           C  \\nATOM    355  N   LYS A  50      -8.025  11.598  -0.504  1.00  0.79           N  \\nATOM    356  CA  LYS A  50      -7.604  11.213  -1.837  1.00  0.79           C  \\nATOM    357  C   LYS A  50      -7.973  12.228  -2.906  1.00  0.79           C  \\nATOM    358  O   LYS A  50      -7.108  12.654  -3.677  1.00  0.79           O  \\nATOM    359  CB  LYS A  50      -8.195   9.837  -2.218  1.00  0.79           C  \\nATOM    360  CG  LYS A  50      -7.549   9.265  -3.485  1.00  0.79           C  \\nATOM    361  CD  LYS A  50      -8.056   7.869  -3.869  1.00  0.79           C  \\nATOM    362  CE  LYS A  50      -7.331   7.288  -5.086  1.00  0.79           C  \\nATOM    363  NZ  LYS A  50      -7.591   8.104  -6.280  1.00  0.79           N  \\nATOM    364  N   ASP A  51      -9.223  12.716  -2.930  1.00  0.80           N  \\nATOM    365  CA  ASP A  51      -9.688  13.726  -3.868  1.00  0.80           C  \\nATOM    366  C   ASP A  51      -8.878  15.035  -3.750  1.00  0.80           C  \\nATOM    367  O   ASP A  51      -8.513  15.663  -4.760  1.00  0.80           O  \\nATOM    368  CB  ASP A  51     -11.202  13.993  -3.627  1.00  0.80           C  \\nATOM    369  CG  ASP A  51     -12.189  12.889  -4.006  1.00  0.80           C  \\nATOM    370  OD1 ASP A  51     -11.875  11.845  -4.635  1.00  0.80           O  \\nATOM    371  OD2 ASP A  51     -13.381  13.088  -3.632  1.00  0.80           O  \\nATOM    372  N   ILE A  52      -8.532  15.478  -2.526  1.00  0.79           N  \\nATOM    373  CA  ILE A  52      -7.645  16.619  -2.285  1.00  0.79           C  \\nATOM    374  C   ILE A  52      -6.217  16.398  -2.776  1.00  0.79           C  \\nATOM    375  O   ILE A  52      -5.645  17.238  -3.490  1.00  0.79           O  \\nATOM    376  CB  ILE A  52      -7.631  16.974  -0.794  1.00  0.79           C  \\nATOM    377  CG1 ILE A  52      -8.994  17.571  -0.383  1.00  0.79           C  \\nATOM    378  CG2 ILE A  52      -6.499  17.959  -0.443  1.00  0.79           C  \\nATOM    379  CD1 ILE A  52      -9.166  17.747   1.130  1.00  0.79           C  \\nATOM    380  N   MET A  53      -5.599  15.263  -2.425  1.00  0.76           N  \\nATOM    381  CA  MET A  53      -4.230  14.937  -2.797  1.00  0.76           C  \\nATOM    382  C   MET A  53      -4.038  14.688  -4.285  1.00  0.76           C  \\nATOM    383  O   MET A  53      -3.056  15.156  -4.881  1.00  0.76           O  \\nATOM    384  CB  MET A  53      -3.730  13.706  -2.027  1.00  0.76           C  \\nATOM    385  CG  MET A  53      -3.567  13.914  -0.514  1.00  0.76           C  \\nATOM    386  SD  MET A  53      -3.318  12.341   0.360  1.00  0.76           S  \\nATOM    387  CE  MET A  53      -1.898  11.706  -0.580  1.00  0.76           C  \\nATOM    388  N   ASP A  54      -4.986  13.996  -4.937  1.00  0.79           N  \\nATOM    389  CA  ASP A  54      -4.996  13.726  -6.367  1.00  0.79           C  \\nATOM    390  C   ASP A  54      -5.092  15.018  -7.190  1.00  0.79           C  \\nATOM    391  O   ASP A  54      -4.652  15.073  -8.349  1.00  0.79           O  \\nATOM    392  CB  ASP A  54      -6.156  12.762  -6.739  1.00  0.79           C  \\nATOM    393  CG  ASP A  54      -5.940  11.303  -6.360  1.00  0.79           C  \\nATOM    394  OD1 ASP A  54      -4.884  10.889  -5.828  1.00  0.79           O  \\nATOM    395  OD2 ASP A  54      -6.895  10.527  -6.646  1.00  0.79           O  \\nATOM    396  N   ALA A  55      -5.612  16.116  -6.609  1.00  0.82           N  \\nATOM    397  CA  ALA A  55      -5.656  17.425  -7.232  1.00  0.82           C  \\nATOM    398  C   ALA A  55      -4.455  18.303  -6.873  1.00  0.82           C  \\nATOM    399  O   ALA A  55      -4.425  19.495  -7.213  1.00  0.82           O  \\nATOM    400  CB  ALA A  55      -6.951  18.157  -6.821  1.00  0.82           C  \\nATOM    401  N   GLY A  56      -3.428  17.767  -6.183  1.00  0.81           N  \\nATOM    402  CA  GLY A  56      -2.221  18.514  -5.814  1.00  0.81           C  \\nATOM    403  C   GLY A  56      -2.447  19.576  -4.779  1.00  0.81           C  \\nATOM    404  O   GLY A  56      -1.771  20.613  -4.758  1.00  0.81           O  \\nATOM    405  N   LYS A  57      -3.416  19.363  -3.888  1.00  0.76           N  \\nATOM    406  CA  LYS A  57      -3.765  20.286  -2.841  1.00  0.76           C  \\nATOM    407  C   LYS A  57      -3.400  19.702  -1.494  1.00  0.76           C  \\nATOM    408  O   LYS A  57      -3.217  18.501  -1.314  1.00  0.76           O  \\nATOM    409  CB  LYS A  57      -5.274  20.620  -2.877  1.00  0.76           C  \\nATOM    410  CG  LYS A  57      -5.685  21.435  -4.113  1.00  0.76           C  \\nATOM    411  CD  LYS A  57      -7.204  21.669  -4.162  1.00  0.76           C  \\nATOM    412  CE  LYS A  57      -7.699  22.401  -5.410  1.00  0.76           C  \\nATOM    413  NZ  LYS A  57      -7.070  23.735  -5.473  1.00  0.76           N  \\nATOM    414  N   LEU A  58      -3.278  20.583  -0.487  1.00  0.79           N  \\nATOM    415  CA  LEU A  58      -3.057  20.164   0.875  1.00  0.79           C  \\nATOM    416  C   LEU A  58      -4.380  19.852   1.539  1.00  0.79           C  \\nATOM    417  O   LEU A  58      -5.377  20.546   1.375  1.00  0.79           O  \\nATOM    418  CB  LEU A  58      -2.324  21.244   1.702  1.00  0.79           C  \\nATOM    419  CG  LEU A  58      -0.793  21.100   1.726  1.00  0.79           C  \\nATOM    420  CD1 LEU A  58      -0.180  22.317   2.427  1.00  0.79           C  \\nATOM    421  CD2 LEU A  58      -0.357  19.816   2.444  1.00  0.79           C  \\nATOM    422  N   VAL A  59      -4.394  18.757   2.326  1.00  0.80           N  \\nATOM    423  CA  VAL A  59      -5.463  18.452   3.249  1.00  0.80           C  \\nATOM    424  C   VAL A  59      -5.616  19.539   4.304  1.00  0.80           C  \\nATOM    425  O   VAL A  59      -4.651  20.116   4.795  1.00  0.80           O  \\nATOM    426  CB  VAL A  59      -5.312  17.073   3.888  1.00  0.80           C  \\nATOM    427  CG1 VAL A  59      -5.448  15.991   2.799  1.00  0.80           C  \\nATOM    428  CG2 VAL A  59      -3.975  16.942   4.640  1.00  0.80           C  \\nATOM    429  N   THR A  60      -6.873  19.874   4.644  1.00  0.81           N  \\nATOM    430  CA  THR A  60      -7.205  20.858   5.664  1.00  0.81           C  \\nATOM    431  C   THR A  60      -6.826  20.395   7.062  1.00  0.81           C  \\nATOM    432  O   THR A  60      -6.830  19.202   7.351  1.00  0.81           O  \\nATOM    433  CB  THR A  60      -8.687  21.248   5.653  1.00  0.81           C  \\nATOM    434  OG1 THR A  60      -9.547  20.129   5.856  1.00  0.81           O  \\nATOM    435  CG2 THR A  60      -9.048  21.843   4.285  1.00  0.81           C  \\nATOM    436  N   ASP A  61      -6.535  21.336   7.998  1.00  0.82           N  \\nATOM    437  CA  ASP A  61      -6.327  21.015   9.407  1.00  0.82           C  \\nATOM    438  C   ASP A  61      -7.511  20.249  10.006  1.00  0.82           C  \\nATOM    439  O   ASP A  61      -7.327  19.291  10.754  1.00  0.82           O  \\nATOM    440  CB  ASP A  61      -6.065  22.294  10.253  1.00  0.82           C  \\nATOM    441  CG  ASP A  61      -4.612  22.762  10.222  1.00  0.82           C  \\nATOM    442  OD1 ASP A  61      -3.727  22.015   9.729  1.00  0.82           O  \\nATOM    443  OD2 ASP A  61      -4.359  23.872  10.757  1.00  0.82           O  \\nATOM    444  N   GLU A  62      -8.760  20.599   9.635  1.00  0.80           N  \\nATOM    445  CA  GLU A  62      -9.954  19.882  10.068  1.00  0.80           C  \\nATOM    446  C   GLU A  62      -9.955  18.407   9.679  1.00  0.80           C  \\nATOM    447  O   GLU A  62     -10.196  17.523  10.514  1.00  0.80           O  \\nATOM    448  CB  GLU A  62     -11.210  20.541   9.458  1.00  0.80           C  \\nATOM    449  CG  GLU A  62     -12.548  19.884   9.878  1.00  0.80           C  \\nATOM    450  CD  GLU A  62     -13.756  20.521   9.189  1.00  0.80           C  \\nATOM    451  OE1 GLU A  62     -13.566  21.535   8.466  1.00  0.80           O  \\nATOM    452  OE2 GLU A  62     -14.872  19.973   9.373  1.00  0.80           O  \\nATOM    453  N   LEU A  63      -9.615  18.089   8.417  1.00  0.84           N  \\nATOM    454  CA  LEU A  63      -9.479  16.720   7.950  1.00  0.84           C  \\nATOM    455  C   LEU A  63      -8.342  15.974   8.638  1.00  0.84           C  \\nATOM    456  O   LEU A  63      -8.492  14.815   9.043  1.00  0.84           O  \\nATOM    457  CB  LEU A  63      -9.316  16.674   6.416  1.00  0.84           C  \\nATOM    458  CG  LEU A  63      -9.176  15.265   5.810  1.00  0.84           C  \\nATOM    459  CD1 LEU A  63     -10.315  14.328   6.215  1.00  0.84           C  \\nATOM    460  CD2 LEU A  63      -9.109  15.354   4.287  1.00  0.84           C  \\nATOM    461  N   VAL A  64      -7.183  16.627   8.843  1.00  0.86           N  \\nATOM    462  CA  VAL A  64      -6.071  16.053   9.598  1.00  0.86           C  \\nATOM    463  C   VAL A  64      -6.464  15.711  11.031  1.00  0.86           C  \\nATOM    464  O   VAL A  64      -6.204  14.601  11.505  1.00  0.86           O  \\nATOM    465  CB  VAL A  64      -4.841  16.962   9.582  1.00  0.86           C  \\nATOM    466  CG1 VAL A  64      -3.715  16.428  10.489  1.00  0.86           C  \\nATOM    467  CG2 VAL A  64      -4.319  17.069   8.140  1.00  0.86           C  \\nATOM    468  N   ILE A  65      -7.160  16.608  11.749  1.00  0.84           N  \\nATOM    469  CA  ILE A  65      -7.674  16.358  13.096  1.00  0.84           C  \\nATOM    470  C   ILE A  65      -8.691  15.226  13.151  1.00  0.84           C  \\nATOM    471  O   ILE A  65      -8.647  14.373  14.046  1.00  0.84           O  \\nATOM    472  CB  ILE A  65      -8.252  17.624  13.718  1.00  0.84           C  \\nATOM    473  CG1 ILE A  65      -7.159  18.701  13.894  1.00  0.84           C  \\nATOM    474  CG2 ILE A  65      -8.968  17.347  15.061  1.00  0.84           C  \\nATOM    475  CD1 ILE A  65      -6.103  18.387  14.947  1.00  0.84           C  \\nATOM    476  N   ALA A  66      -9.615  15.152  12.174  1.00  0.87           N  \\nATOM    477  CA  ALA A  66     -10.570  14.063  12.053  1.00  0.87           C  \\nATOM    478  C   ALA A  66      -9.887  12.703  11.884  1.00  0.87           C  \\nATOM    479  O   ALA A  66     -10.249  11.727  12.565  1.00  0.87           O  \\nATOM    480  CB  ALA A  66     -11.545  14.361  10.894  1.00  0.87           C  \\nATOM    481  N   LEU A  67      -8.834  12.622  11.054  1.00  0.82           N  \\nATOM    482  CA  LEU A  67      -7.967  11.457  10.913  1.00  0.82           C  \\nATOM    483  C   LEU A  67      -7.241  11.074  12.197  1.00  0.82           C  \\nATOM    484  O   LEU A  67      -7.156   9.884  12.537  1.00  0.82           O  \\nATOM    485  CB  LEU A  67      -6.923  11.668   9.788  1.00  0.82           C  \\nATOM    486  CG  LEU A  67      -7.504  11.652   8.361  1.00  0.82           C  \\nATOM    487  CD1 LEU A  67      -6.483  12.185   7.344  1.00  0.82           C  \\nATOM    488  CD2 LEU A  67      -7.955  10.242   7.970  1.00  0.82           C  \\nATOM    489  N   VAL A  68      -6.717  12.038  12.981  1.00  0.82           N  \\nATOM    490  CA  VAL A  68      -6.115  11.764  14.290  1.00  0.82           C  \\nATOM    491  C   VAL A  68      -7.123  11.173  15.253  1.00  0.82           C  \\nATOM    492  O   VAL A  68      -6.858  10.155  15.913  1.00  0.82           O  \\nATOM    493  CB  VAL A  68      -5.466  12.989  14.944  1.00  0.82           C  \\nATOM    494  CG1 VAL A  68      -4.857  12.624  16.314  1.00  0.82           C  \\nATOM    495  CG2 VAL A  68      -4.348  13.551  14.056  1.00  0.82           C  \\nATOM    496  N   LYS A  69      -8.329  11.750  15.325  1.00  0.77           N  \\nATOM    497  CA  LYS A  69      -9.410  11.272  16.170  1.00  0.77           C  \\nATOM    498  C   LYS A  69      -9.866   9.861  15.819  1.00  0.77           C  \\nATOM    499  O   LYS A  69     -10.051   9.012  16.706  1.00  0.77           O  \\nATOM    500  CB  LYS A  69     -10.617  12.235  16.076  1.00  0.77           C  \\nATOM    501  CG  LYS A  69     -11.838  11.755  16.871  1.00  0.77           C  \\nATOM    502  CD  LYS A  69     -12.969  12.784  16.952  1.00  0.77           C  \\nATOM    503  CE  LYS A  69     -14.188  12.205  17.671  1.00  0.77           C  \\nATOM    504  NZ  LYS A  69     -15.175  13.271  17.924  1.00  0.77           N  \\nATOM    505  N   GLU A  70     -10.032   9.567  14.522  1.00  0.76           N  \\nATOM    506  CA  GLU A  70     -10.337   8.242  13.996  1.00  0.76           C  \\nATOM    507  C   GLU A  70      -9.227   7.245  14.295  1.00  0.76           C  \\nATOM    508  O   GLU A  70      -9.473   6.115  14.765  1.00  0.76           O  \\nATOM    509  CB  GLU A  70     -10.548   8.333  12.461  1.00  0.76           C  \\nATOM    510  CG  GLU A  70     -10.899   6.994  11.762  1.00  0.76           C  \\nATOM    511  CD  GLU A  70     -11.006   7.112  10.236  1.00  0.76           C  \\nATOM    512  OE1 GLU A  70      -9.975   7.425   9.586  1.00  0.76           O  \\nATOM    513  OE2 GLU A  70     -12.115   6.863   9.695  1.00  0.76           O  \\nATOM    514  N   ARG A  71      -7.961   7.626  14.095  1.00  0.75           N  \\nATOM    515  CA  ARG A  71      -6.793   6.788  14.316  1.00  0.75           C  \\nATOM    516  C   ARG A  71      -6.616   6.308  15.745  1.00  0.75           C  \\nATOM    517  O   ARG A  71      -6.241   5.142  15.965  1.00  0.75           O  \\nATOM    518  CB  ARG A  71      -5.484   7.502  13.905  1.00  0.75           C  \\nATOM    519  CG  ARG A  71      -4.208   6.643  14.043  1.00  0.75           C  \\nATOM    520  CD  ARG A  71      -4.168   5.448  13.094  1.00  0.75           C  \\nATOM    521  NE  ARG A  71      -3.022   4.587  13.506  1.00  0.75           N  \\nATOM    522  CZ  ARG A  71      -3.146   3.516  14.286  1.00  0.75           C  \\nATOM    523  NH1 ARG A  71      -4.250   3.232  14.985  1.00  0.75           N  \\nATOM    524  NH2 ARG A  71      -2.120   2.669  14.392  1.00  0.75           N  \\nATOM    525  N   ILE A  72      -6.853   7.164  16.745  1.00  0.80           N  \\nATOM    526  CA  ILE A  72      -6.619   6.844  18.147  1.00  0.80           C  \\nATOM    527  C   ILE A  72      -7.801   6.152  18.787  1.00  0.80           C  \\nATOM    528  O   ILE A  72      -7.702   5.683  19.932  1.00  0.80           O  \\nATOM    529  CB  ILE A  72      -6.211   8.062  18.966  1.00  0.80           C  \\nATOM    530  CG1 ILE A  72      -7.349   9.096  19.114  1.00  0.80           C  \\nATOM    531  CG2 ILE A  72      -4.934   8.653  18.333  1.00  0.80           C  \\nATOM    532  CD1 ILE A  72      -7.005  10.257  20.046  1.00  0.80           C  \\nATOM    533  N   ALA A  73      -8.925   6.030  18.066  1.00  0.82           N  \\nATOM    534  CA  ALA A  73     -10.076   5.244  18.461  1.00  0.82           C  \\nATOM    535  C   ALA A  73      -9.932   3.782  18.051  1.00  0.82           C  \\nATOM    536  O   ALA A  73     -10.702   2.925  18.499  1.00  0.82           O  \\nATOM    537  CB  ALA A  73     -11.351   5.820  17.811  1.00  0.82           C  \\nATOM    538  N   GLN A  74      -8.943   3.457  17.199  1.00  0.78           N  \\nATOM    539  CA  GLN A  74      -8.627   2.091  16.811  1.00  0.78           C  \\nATOM    540  C   GLN A  74      -8.084   1.254  17.970  1.00  0.78           C  \\nATOM    541  O   GLN A  74      -7.470   1.738  18.913  1.00  0.78           O  \\nATOM    542  CB  GLN A  74      -7.665   1.995  15.598  1.00  0.78           C  \\nATOM    543  CG  GLN A  74      -8.083   2.792  14.335  1.00  0.78           C  \\nATOM    544  CD  GLN A  74      -9.524   2.511  13.902  1.00  0.78           C  \\nATOM    545  OE1 GLN A  74      -9.859   1.421  13.457  1.00  0.78           O  \\nATOM    546  NE2 GLN A  74     -10.402   3.539  14.007  1.00  0.78           N  \\nATOM    547  N   GLU A  75      -8.311  -0.074  17.911  1.00  0.78           N  \\nATOM    548  CA  GLU A  75      -8.023  -1.032  18.961  1.00  0.78           C  \\nATOM    549  C   GLU A  75      -6.561  -1.104  19.375  1.00  0.78           C  \\nATOM    550  O   GLU A  75      -6.262  -1.421  20.537  1.00  0.78           O  \\nATOM    551  CB  GLU A  75      -8.479  -2.459  18.550  1.00  0.78           C  \\nATOM    552  CG  GLU A  75      -9.999  -2.624  18.289  1.00  0.78           C  \\nATOM    553  CD  GLU A  75     -10.484  -2.008  16.972  1.00  0.78           C  \\nATOM    554  OE1 GLU A  75      -9.626  -1.627  16.136  1.00  0.78           O  \\nATOM    555  OE2 GLU A  75     -11.723  -1.892  16.834  1.00  0.78           O  \\nATOM    556  N   ASP A  76      -5.623  -0.805  18.456  1.00  0.86           N  \\nATOM    557  CA  ASP A  76      -4.188  -0.802  18.691  1.00  0.86           C  \\nATOM    558  C   ASP A  76      -3.744   0.340  19.604  1.00  0.86           C  \\nATOM    559  O   ASP A  76      -2.708   0.247  20.275  1.00  0.86           O  \\nATOM    560  CB  ASP A  76      -3.377  -0.902  17.368  1.00  0.86           C  \\nATOM    561  CG  ASP A  76      -3.480   0.285  16.453  1.00  0.86           C  \\nATOM    562  OD1 ASP A  76      -4.435   1.104  16.526  1.00  0.86           O  \\nATOM    563  OD2 ASP A  76      -2.554   0.405  15.600  1.00  0.86           O  \\nATOM    564  N   CYS A  77      -4.555   1.407  19.729  1.00  0.88           N  \\nATOM    565  CA  CYS A  77      -4.266   2.568  20.552  1.00  0.88           C  \\nATOM    566  C   CYS A  77      -4.921   2.497  21.930  1.00  0.88           C  \\nATOM    567  O   CYS A  77      -4.973   3.482  22.673  1.00  0.88           O  \\nATOM    568  CB  CYS A  77      -4.663   3.888  19.841  1.00  0.88           C  \\nATOM    569  SG  CYS A  77      -3.657   4.226  18.354  1.00  0.88           S  \\nATOM    570  N   ARG A  78      -5.394   1.304  22.354  1.00  0.78           N  \\nATOM    571  CA  ARG A  78      -5.975   1.053  23.673  1.00  0.78           C  \\nATOM    572  C   ARG A  78      -5.043   1.366  24.835  1.00  0.78           C  \\nATOM    573  O   ARG A  78      -5.463   1.904  25.865  1.00  0.78           O  \\nATOM    574  CB  ARG A  78      -6.470  -0.413  23.789  1.00  0.78           C  \\nATOM    575  CG  ARG A  78      -7.220  -0.709  25.107  1.00  0.78           C  \\nATOM    576  CD  ARG A  78      -8.148  -1.931  25.115  1.00  0.78           C  \\nATOM    577  NE  ARG A  78      -9.204  -1.703  24.068  1.00  0.78           N  \\nATOM    578  CZ  ARG A  78      -9.289  -2.356  22.900  1.00  0.78           C  \\nATOM    579  NH1 ARG A  78     -10.192  -1.983  21.994  1.00  0.78           N  \\nATOM    580  NH2 ARG A  78      -8.451  -3.336  22.586  1.00  0.78           N  \\nATOM    581  N   ASN A  79      -3.738   1.093  24.681  1.00  0.82           N  \\nATOM    582  CA  ASN A  79      -2.751   1.370  25.706  1.00  0.82           C  \\nATOM    583  C   ASN A  79      -2.205   2.787  25.550  1.00  0.82           C  \\nATOM    584  O   ASN A  79      -1.388   3.253  26.352  1.00  0.82           O  \\nATOM    585  CB  ASN A  79      -1.551   0.398  25.590  1.00  0.82           C  \\nATOM    586  CG  ASN A  79      -1.979  -1.052  25.792  1.00  0.82           C  \\nATOM    587  OD1 ASN A  79      -3.028  -1.371  26.347  1.00  0.82           O  \\nATOM    588  ND2 ASN A  79      -1.115  -1.977  25.308  1.00  0.82           N  \\nATOM    589  N   GLY A  80      -2.656   3.537  24.526  1.00  0.91           N  \\nATOM    590  CA  GLY A  80      -2.143   4.850  24.178  1.00  0.91           C  \\nATOM    591  C   GLY A  80      -1.504   4.880  22.820  1.00  0.91           C  \\nATOM    592  O   GLY A  80      -1.747   4.028  21.974  1.00  0.91           O  \\nATOM    593  N   PHE A  81      -0.673   5.899  22.567  1.00  0.93           N  \\nATOM    594  CA  PHE A  81      -0.081   6.090  21.262  1.00  0.93           C  \\nATOM    595  C   PHE A  81       1.117   7.007  21.365  1.00  0.93           C  \\nATOM    596  O   PHE A  81       1.245   7.804  22.296  1.00  0.93           O  \\nATOM    597  CB  PHE A  81      -1.079   6.630  20.197  1.00  0.93           C  \\nATOM    598  CG  PHE A  81      -1.783   7.884  20.637  1.00  0.93           C  \\nATOM    599  CD1 PHE A  81      -1.189   9.145  20.462  1.00  0.93           C  \\nATOM    600  CD2 PHE A  81      -3.052   7.807  21.232  1.00  0.93           C  \\nATOM    601  CE1 PHE A  81      -1.846  10.305  20.891  1.00  0.93           C  \\nATOM    602  CE2 PHE A  81      -3.711   8.967  21.648  1.00  0.93           C  \\nATOM    603  CZ  PHE A  81      -3.110  10.216  21.481  1.00  0.93           C  \\nATOM    604  N   LEU A  82       2.017   6.918  20.377  1.00  0.95           N  \\nATOM    605  CA  LEU A  82       3.108   7.848  20.205  1.00  0.95           C  \\nATOM    606  C   LEU A  82       2.823   8.637  18.943  1.00  0.95           C  \\nATOM    607  O   LEU A  82       2.886   8.123  17.827  1.00  0.95           O  \\nATOM    608  CB  LEU A  82       4.440   7.076  20.106  1.00  0.95           C  \\nATOM    609  CG  LEU A  82       5.688   7.902  19.758  1.00  0.95           C  \\nATOM    610  CD1 LEU A  82       5.988   8.973  20.807  1.00  0.95           C  \\nATOM    611  CD2 LEU A  82       6.896   6.977  19.559  1.00  0.95           C  \\nATOM    612  N   LEU A  83       2.467   9.920  19.085  1.00  0.94           N  \\nATOM    613  CA  LEU A  83       2.127  10.790  17.980  1.00  0.94           C  \\nATOM    614  C   LEU A  83       3.374  11.552  17.579  1.00  0.94           C  \\nATOM    615  O   LEU A  83       3.913  12.379  18.309  1.00  0.94           O  \\nATOM    616  CB  LEU A  83       0.941  11.712  18.351  1.00  0.94           C  \\nATOM    617  CG  LEU A  83       0.492  12.756  17.307  1.00  0.94           C  \\nATOM    618  CD1 LEU A  83       0.144  12.153  15.943  1.00  0.94           C  \\nATOM    619  CD2 LEU A  83      -0.715  13.537  17.847  1.00  0.94           C  \\nATOM    620  N   ASP A  84       3.890  11.221  16.386  1.00  0.91           N  \\nATOM    621  CA  ASP A  84       5.167  11.698  15.919  1.00  0.91           C  \\nATOM    622  C   ASP A  84       5.015  12.944  15.044  1.00  0.91           C  \\nATOM    623  O   ASP A  84       4.424  12.922  13.962  1.00  0.91           O  \\nATOM    624  CB  ASP A  84       5.860  10.516  15.195  1.00  0.91           C  \\nATOM    625  CG  ASP A  84       7.319  10.776  14.897  1.00  0.91           C  \\nATOM    626  OD1 ASP A  84       7.990   9.855  14.373  1.00  0.91           O  \\nATOM    627  OD2 ASP A  84       7.810  11.898  15.178  1.00  0.91           O  \\nATOM    628  N   GLY A  85       5.567  14.089  15.510  1.00  0.85           N  \\nATOM    629  CA  GLY A  85       5.732  15.285  14.683  1.00  0.85           C  \\nATOM    630  C   GLY A  85       4.484  16.064  14.390  1.00  0.85           C  \\nATOM    631  O   GLY A  85       4.410  16.786  13.379  1.00  0.85           O  \\nATOM    632  N   PHE A  86       3.465  15.952  15.240  1.00  0.88           N  \\nATOM    633  CA  PHE A  86       2.234  16.691  15.108  1.00  0.88           C  \\nATOM    634  C   PHE A  86       1.711  16.961  16.512  1.00  0.88           C  \\nATOM    635  O   PHE A  86       1.875  16.090  17.370  1.00  0.88           O  \\nATOM    636  CB  PHE A  86       1.218  15.882  14.260  1.00  0.88           C  \\nATOM    637  CG  PHE A  86      -0.038  16.655  13.968  1.00  0.88           C  \\nATOM    638  CD1 PHE A  86      -0.064  17.582  12.915  1.00  0.88           C  \\nATOM    639  CD2 PHE A  86      -1.191  16.490  14.752  1.00  0.88           C  \\nATOM    640  CE1 PHE A  86      -1.234  18.286  12.615  1.00  0.88           C  \\nATOM    641  CE2 PHE A  86      -2.348  17.231  14.483  1.00  0.88           C  \\nATOM    642  CZ  PHE A  86      -2.376  18.115  13.401  1.00  0.88           C  \\nATOM    643  N   PRO A  87       1.066  18.082  16.830  1.00  0.91           N  \\nATOM    644  CA  PRO A  87       0.908  19.314  16.053  1.00  0.91           C  \\nATOM    645  C   PRO A  87       2.165  19.988  15.608  1.00  0.91           C  \\nATOM    646  O   PRO A  87       3.234  19.742  16.152  1.00  0.91           O  \\nATOM    647  CB  PRO A  87       0.132  20.254  16.974  1.00  0.91           C  \\nATOM    648  CG  PRO A  87      -0.702  19.268  17.764  1.00  0.91           C  \\nATOM    649  CD  PRO A  87       0.273  18.129  18.042  1.00  0.91           C  \\nATOM    650  N   ARG A  88       2.027  20.858  14.602  1.00  0.77           N  \\nATOM    651  CA  ARG A  88       3.120  21.668  14.132  1.00  0.77           C  \\nATOM    652  C   ARG A  88       2.849  23.137  14.378  1.00  0.77           C  \\nATOM    653  O   ARG A  88       3.755  23.969  14.298  1.00  0.77           O  \\nATOM    654  CB  ARG A  88       3.271  21.401  12.622  1.00  0.77           C  \\nATOM    655  CG  ARG A  88       3.620  19.928  12.330  1.00  0.77           C  \\nATOM    656  CD  ARG A  88       3.870  19.626  10.853  1.00  0.77           C  \\nATOM    657  NE  ARG A  88       2.561  19.264  10.219  1.00  0.77           N  \\nATOM    658  CZ  ARG A  88       2.129  18.010  10.011  1.00  0.77           C  \\nATOM    659  NH1 ARG A  88       2.806  16.955  10.461  1.00  0.77           N  \\nATOM    660  NH2 ARG A  88       0.959  17.841   9.405  1.00  0.77           N  \\nATOM    661  N   THR A  89       1.607  23.510  14.732  1.00  0.81           N  \\nATOM    662  CA  THR A  89       1.218  24.897  14.938  1.00  0.81           C  \\nATOM    663  C   THR A  89       0.318  24.991  16.158  1.00  0.81           C  \\nATOM    664  O   THR A  89      -0.258  23.993  16.592  1.00  0.81           O  \\nATOM    665  CB  THR A  89       0.501  25.531  13.732  1.00  0.81           C  \\nATOM    666  OG1 THR A  89      -0.804  25.002  13.517  1.00  0.81           O  \\nATOM    667  CG2 THR A  89       1.292  25.296  12.439  1.00  0.81           C  \\nATOM    668  N   ILE A  90       0.148  26.203  16.752  1.00  0.84           N  \\nATOM    669  CA  ILE A  90      -0.893  26.436  17.765  1.00  0.84           C  \\nATOM    670  C   ILE A  90      -2.311  26.151  17.240  1.00  0.84           C  \\nATOM    671  O   ILE A  90      -2.998  25.380  17.906  1.00  0.84           O  \\nATOM    672  CB  ILE A  90      -0.793  27.805  18.460  1.00  0.84           C  \\nATOM    673  CG1 ILE A  90       0.492  27.892  19.314  1.00  0.84           C  \\nATOM    674  CG2 ILE A  90      -2.042  28.072  19.331  1.00  0.84           C  \\nATOM    675  CD1 ILE A  90       0.662  29.240  20.027  1.00  0.84           C  \\nATOM    676  N   PRO A  91      -2.793  26.603  16.068  1.00  0.85           N  \\nATOM    677  CA  PRO A  91      -4.132  26.257  15.581  1.00  0.85           C  \\nATOM    678  C   PRO A  91      -4.429  24.770  15.483  1.00  0.85           C  \\nATOM    679  O   PRO A  91      -5.574  24.365  15.715  1.00  0.85           O  \\nATOM    680  CB  PRO A  91      -4.205  26.957  14.218  1.00  0.85           C  \\nATOM    681  CG  PRO A  91      -3.361  28.215  14.406  1.00  0.85           C  \\nATOM    682  CD  PRO A  91      -2.223  27.715  15.295  1.00  0.85           C  \\nATOM    683  N   GLN A  92      -3.438  23.931  15.144  1.00  0.83           N  \\nATOM    684  CA  GLN A  92      -3.554  22.484  15.191  1.00  0.83           C  \\nATOM    685  C   GLN A  92      -3.678  21.917  16.603  1.00  0.83           C  \\nATOM    686  O   GLN A  92      -4.494  21.024  16.860  1.00  0.83           O  \\nATOM    687  CB  GLN A  92      -2.365  21.849  14.443  1.00  0.83           C  \\nATOM    688  CG  GLN A  92      -2.458  22.087  12.924  1.00  0.83           C  \\nATOM    689  CD  GLN A  92      -1.177  21.703  12.196  1.00  0.83           C  \\nATOM    690  OE1 GLN A  92      -0.142  21.355  12.782  1.00  0.83           O  \\nATOM    691  NE2 GLN A  92      -1.241  21.756  10.847  1.00  0.83           N  \\nATOM    692  N   ALA A  93      -2.893  22.426  17.570  1.00  0.92           N  \\nATOM    693  CA  ALA A  93      -3.009  22.087  18.980  1.00  0.92           C  \\nATOM    694  C   ALA A  93      -4.347  22.509  19.589  1.00  0.92           C  \\nATOM    695  O   ALA A  93      -4.990  21.732  20.311  1.00  0.92           O  \\nATOM    696  CB  ALA A  93      -1.824  22.672  19.770  1.00  0.92           C  \\nATOM    697  N   ASP A  94      -4.839  23.712  19.246  1.00  0.87           N  \\nATOM    698  CA  ASP A  94      -6.165  24.202  19.591  1.00  0.87           C  \\nATOM    699  C   ASP A  94      -7.272  23.297  19.065  1.00  0.87           C  \\nATOM    700  O   ASP A  94      -8.185  22.899  19.802  1.00  0.87           O  \\nATOM    701  CB  ASP A  94      -6.379  25.614  18.996  1.00  0.87           C  \\nATOM    702  CG  ASP A  94      -5.529  26.673  19.684  1.00  0.87           C  \\nATOM    703  OD1 ASP A  94      -5.095  26.446  20.846  1.00  0.87           O  \\nATOM    704  OD2 ASP A  94      -5.335  27.739  19.046  1.00  0.87           O  \\nATOM    705  N   ALA A  95      -7.188  22.880  17.794  1.00  0.89           N  \\nATOM    706  CA  ALA A  95      -8.115  21.954  17.175  1.00  0.89           C  \\nATOM    707  C   ALA A  95      -8.141  20.563  17.818  1.00  0.89           C  \\nATOM    708  O   ALA A  95      -9.207  19.941  17.941  1.00  0.89           O  \\nATOM    709  CB  ALA A  95      -7.840  21.872  15.666  1.00  0.89           C  \\nATOM    710  N   MET A  96      -6.993  20.031  18.274  1.00  0.83           N  \\nATOM    711  CA  MET A  96      -6.935  18.821  19.087  1.00  0.83           C  \\nATOM    712  C   MET A  96      -7.641  18.960  20.415  1.00  0.83           C  \\nATOM    713  O   MET A  96      -8.362  18.045  20.843  1.00  0.83           O  \\nATOM    714  CB  MET A  96      -5.488  18.373  19.369  1.00  0.83           C  \\nATOM    715  CG  MET A  96      -4.822  17.801  18.116  1.00  0.83           C  \\nATOM    716  SD  MET A  96      -3.154  17.137  18.349  1.00  0.83           S  \\nATOM    717  CE  MET A  96      -3.335  15.709  19.446  1.00  0.83           C  \\nATOM    718  N   LYS A  97      -7.487  20.095  21.102  1.00  0.81           N  \\nATOM    719  CA  LYS A  97      -8.231  20.369  22.320  1.00  0.81           C  \\nATOM    720  C   LYS A  97      -9.743  20.449  22.101  1.00  0.81           C  \\nATOM    721  O   LYS A  97     -10.507  19.794  22.829  1.00  0.81           O  \\nATOM    722  CB  LYS A  97      -7.719  21.654  23.007  1.00  0.81           C  \\nATOM    723  CG  LYS A  97      -8.451  21.960  24.323  1.00  0.81           C  \\nATOM    724  CD  LYS A  97      -7.872  23.156  25.092  1.00  0.81           C  \\nATOM    725  CE  LYS A  97      -6.496  22.876  25.687  1.00  0.81           C  \\nATOM    726  NZ  LYS A  97      -5.994  24.039  26.436  1.00  0.81           N  \\nATOM    727  N   GLU A  98     -10.203  21.166  21.061  1.00  0.79           N  \\nATOM    728  CA  GLU A  98     -11.604  21.284  20.654  1.00  0.79           C  \\nATOM    729  C   GLU A  98     -12.224  19.938  20.313  1.00  0.79           C  \\nATOM    730  O   GLU A  98     -13.396  19.654  20.630  1.00  0.79           O  \\nATOM    731  CB  GLU A  98     -11.746  22.248  19.444  1.00  0.79           C  \\nATOM    732  CG  GLU A  98     -11.464  23.734  19.775  1.00  0.79           C  \\nATOM    733  CD  GLU A  98     -12.552  24.323  20.671  1.00  0.79           C  \\nATOM    734  OE1 GLU A  98     -13.658  24.611  20.146  1.00  0.79           O  \\nATOM    735  OE2 GLU A  98     -12.273  24.488  21.887  1.00  0.79           O  \\nATOM    736  N   ALA A  99     -11.468  19.033  19.682  1.00  0.83           N  \\nATOM    737  CA  ALA A  99     -11.898  17.691  19.348  1.00  0.83           C  \\nATOM    738  C   ALA A  99     -11.873  16.703  20.518  1.00  0.83           C  \\nATOM    739  O   ALA A  99     -12.441  15.598  20.412  1.00  0.83           O  \\nATOM    740  CB  ALA A  99     -11.017  17.163  18.204  1.00  0.83           C  \\nATOM    741  N   GLY A 100     -11.235  17.060  21.647  1.00  0.83           N  \\nATOM    742  CA  GLY A 100     -11.119  16.238  22.852  1.00  0.83           C  \\nATOM    743  C   GLY A 100      -9.917  15.338  22.902  1.00  0.83           C  \\nATOM    744  O   GLY A 100      -9.936  14.283  23.551  1.00  0.83           O  \\nATOM    745  N   ILE A 101      -8.811  15.701  22.242  1.00  0.80           N  \\nATOM    746  CA  ILE A 101      -7.614  14.886  22.187  1.00  0.80           C  \\nATOM    747  C   ILE A 101      -6.540  15.473  23.081  1.00  0.80           C  \\nATOM    748  O   ILE A 101      -5.733  16.333  22.692  1.00  0.80           O  \\nATOM    749  CB  ILE A 101      -7.104  14.677  20.765  1.00  0.80           C  \\nATOM    750  CG1 ILE A 101      -8.252  14.226  19.829  1.00  0.80           C  \\nATOM    751  CG2 ILE A 101      -5.953  13.646  20.786  1.00  0.80           C  \\nATOM    752  CD1 ILE A 101      -7.870  14.253  18.349  1.00  0.80           C  \\nATOM    753  N   ASN A 102      -6.465  14.985  24.324  1.00  0.84           N  \\nATOM    754  CA  ASN A 102      -5.418  15.289  25.275  1.00  0.84           C  \\nATOM    755  C   ASN A 102      -4.344  14.193  25.203  1.00  0.84           C  \\nATOM    756  O   ASN A 102      -4.615  13.069  24.766  1.00  0.84           O  \\nATOM    757  CB  ASN A 102      -6.082  15.414  26.669  1.00  0.84           C  \\nATOM    758  CG  ASN A 102      -5.097  15.644  27.812  1.00  0.84           C  \\nATOM    759  OD1 ASN A 102      -4.773  14.694  28.494  1.00  0.84           O  \\nATOM    760  ND2 ASN A 102      -4.608  16.895  27.993  1.00  0.84           N  \\nATOM    761  N   VAL A 103      -3.097  14.499  25.618  1.00  0.90           N  \\nATOM    762  CA  VAL A 103      -2.009  13.549  25.748  1.00  0.90           C  \\nATOM    763  C   VAL A 103      -1.363  13.782  27.097  1.00  0.90           C  \\nATOM    764  O   VAL A 103      -1.479  14.870  27.659  1.00  0.90           O  \\nATOM    765  CB  VAL A 103      -0.936  13.631  24.654  1.00  0.90           C  \\nATOM    766  CG1 VAL A 103      -1.601  13.413  23.284  1.00  0.90           C  \\nATOM    767  CG2 VAL A 103      -0.122  14.937  24.719  1.00  0.90           C  \\nATOM    768  N   ASP A 104      -0.668  12.753  27.624  1.00  0.91           N  \\nATOM    769  CA  ASP A 104      -0.039  12.735  28.926  1.00  0.91           C  \\nATOM    770  C   ASP A 104       1.285  13.497  28.931  1.00  0.91           C  \\nATOM    771  O   ASP A 104       1.622  14.209  29.886  1.00  0.91           O  \\nATOM    772  CB  ASP A 104       0.223  11.258  29.334  1.00  0.91           C  \\nATOM    773  CG  ASP A 104      -1.029  10.384  29.311  1.00  0.91           C  \\nATOM    774  OD1 ASP A 104      -2.010  10.676  30.029  1.00  0.91           O  \\nATOM    775  OD2 ASP A 104      -0.999   9.354  28.579  1.00  0.91           O  \\nATOM    776  N   TYR A 105       2.078  13.352  27.850  1.00  0.92           N  \\nATOM    777  CA  TYR A 105       3.390  13.962  27.727  1.00  0.92           C  \\nATOM    778  C   TYR A 105       3.600  14.559  26.342  1.00  0.92           C  \\nATOM    779  O   TYR A 105       3.283  13.979  25.311  1.00  0.92           O  \\nATOM    780  CB  TYR A 105       4.582  12.985  27.966  1.00  0.92           C  \\nATOM    781  CG  TYR A 105       4.688  12.560  29.404  1.00  0.92           C  \\nATOM    782  CD1 TYR A 105       5.553  13.190  30.318  1.00  0.92           C  \\nATOM    783  CD2 TYR A 105       3.910  11.491  29.848  1.00  0.92           C  \\nATOM    784  CE1 TYR A 105       5.577  12.784  31.666  1.00  0.92           C  \\nATOM    785  CE2 TYR A 105       3.941  11.078  31.186  1.00  0.92           C  \\nATOM    786  CZ  TYR A 105       4.761  11.747  32.092  1.00  0.92           C  \\nATOM    787  OH  TYR A 105       4.706  11.369  33.463  1.00  0.92           O  \\nATOM    788  N   VAL A 106       4.205  15.758  26.312  1.00  0.93           N  \\nATOM    789  CA  VAL A 106       4.753  16.366  25.118  1.00  0.93           C  \\nATOM    790  C   VAL A 106       6.261  16.400  25.298  1.00  0.93           C  \\nATOM    791  O   VAL A 106       6.794  16.998  26.231  1.00  0.93           O  \\nATOM    792  CB  VAL A 106       4.226  17.773  24.877  1.00  0.93           C  \\nATOM    793  CG1 VAL A 106       4.829  18.347  23.581  1.00  0.93           C  \\nATOM    794  CG2 VAL A 106       2.690  17.718  24.792  1.00  0.93           C  \\nATOM    795  N   LEU A 107       6.997  15.723  24.403  1.00  0.93           N  \\nATOM    796  CA  LEU A 107       8.436  15.609  24.497  1.00  0.93           C  \\nATOM    797  C   LEU A 107       9.074  16.375  23.343  1.00  0.93           C  \\nATOM    798  O   LEU A 107       8.794  16.119  22.177  1.00  0.93           O  \\nATOM    799  CB  LEU A 107       8.871  14.123  24.451  1.00  0.93           C  \\nATOM    800  CG  LEU A 107       8.239  13.218  25.529  1.00  0.93           C  \\nATOM    801  CD1 LEU A 107       8.668  11.762  25.300  1.00  0.93           C  \\nATOM    802  CD2 LEU A 107       8.596  13.662  26.952  1.00  0.93           C  \\nATOM    803  N   GLU A 108       9.957  17.348  23.654  1.00  0.90           N  \\nATOM    804  CA  GLU A 108      10.689  18.126  22.664  1.00  0.90           C  \\nATOM    805  C   GLU A 108      12.075  17.520  22.557  1.00  0.90           C  \\nATOM    806  O   GLU A 108      12.842  17.475  23.519  1.00  0.90           O  \\nATOM    807  CB  GLU A 108      10.735  19.642  23.036  1.00  0.90           C  \\nATOM    808  CG  GLU A 108      11.505  20.596  22.071  1.00  0.90           C  \\nATOM    809  CD  GLU A 108      11.567  22.069  22.506  1.00  0.90           C  \\nATOM    810  OE1 GLU A 108      11.198  22.379  23.672  1.00  0.90           O  \\nATOM    811  OE2 GLU A 108      11.967  22.942  21.685  1.00  0.90           O  \\nATOM    812  N   PHE A 109      12.413  16.949  21.392  1.00  0.89           N  \\nATOM    813  CA  PHE A 109      13.698  16.330  21.147  1.00  0.89           C  \\nATOM    814  C   PHE A 109      14.636  17.344  20.494  1.00  0.89           C  \\nATOM    815  O   PHE A 109      14.556  17.615  19.299  1.00  0.89           O  \\nATOM    816  CB  PHE A 109      13.466  15.090  20.251  1.00  0.89           C  \\nATOM    817  CG  PHE A 109      14.673  14.212  20.111  1.00  0.89           C  \\nATOM    818  CD1 PHE A 109      14.843  13.109  20.961  1.00  0.89           C  \\nATOM    819  CD2 PHE A 109      15.615  14.442  19.095  1.00  0.89           C  \\nATOM    820  CE1 PHE A 109      15.910  12.226  20.769  1.00  0.89           C  \\nATOM    821  CE2 PHE A 109      16.699  13.573  18.922  1.00  0.89           C  \\nATOM    822  CZ  PHE A 109      16.838  12.455  19.751  1.00  0.89           C  \\nATOM    823  N   ASP A 110      15.559  17.904  21.308  1.00  0.87           N  \\nATOM    824  CA  ASP A 110      16.429  18.996  20.923  1.00  0.87           C  \\nATOM    825  C   ASP A 110      17.774  18.505  20.429  1.00  0.87           C  \\nATOM    826  O   ASP A 110      18.454  17.701  21.049  1.00  0.87           O  \\nATOM    827  CB  ASP A 110      16.697  19.951  22.116  1.00  0.87           C  \\nATOM    828  CG  ASP A 110      15.450  20.750  22.431  1.00  0.87           C  \\nATOM    829  OD1 ASP A 110      14.761  21.117  21.451  1.00  0.87           O  \\nATOM    830  OD2 ASP A 110      15.211  21.049  23.633  1.00  0.87           O  \\nATOM    831  N   VAL A 111      18.204  19.029  19.266  1.00  0.87           N  \\nATOM    832  CA  VAL A 111      19.552  18.866  18.765  1.00  0.87           C  \\nATOM    833  C   VAL A 111      19.916  20.270  18.278  1.00  0.87           C  \\nATOM    834  O   VAL A 111      19.121  20.831  17.526  1.00  0.87           O  \\nATOM    835  CB  VAL A 111      19.652  17.831  17.643  1.00  0.87           C  \\nATOM    836  CG1 VAL A 111      21.079  17.767  17.068  1.00  0.87           C  \\nATOM    837  CG2 VAL A 111      19.277  16.446  18.202  1.00  0.87           C  \\nATOM    838  N   PRO A 112      21.005  20.933  18.678  1.00  0.87           N  \\nATOM    839  CA  PRO A 112      21.489  22.183  18.072  1.00  0.87           C  \\nATOM    840  C   PRO A 112      21.556  22.192  16.549  1.00  0.87           C  \\nATOM    841  O   PRO A 112      21.983  21.196  15.973  1.00  0.87           O  \\nATOM    842  CB  PRO A 112      22.894  22.367  18.678  1.00  0.87           C  \\nATOM    843  CG  PRO A 112      22.875  21.566  19.979  1.00  0.87           C  \\nATOM    844  CD  PRO A 112      21.963  20.395  19.641  1.00  0.87           C  \\nATOM    845  N   ASP A 113      21.226  23.326  15.889  1.00  0.83           N  \\nATOM    846  CA  ASP A 113      21.188  23.457  14.438  1.00  0.83           C  \\nATOM    847  C   ASP A 113      22.471  23.049  13.739  1.00  0.83           C  \\nATOM    848  O   ASP A 113      22.435  22.282  12.771  1.00  0.83           O  \\nATOM    849  CB  ASP A 113      20.874  24.926  14.050  1.00  0.83           C  \\nATOM    850  CG  ASP A 113      19.447  25.299  14.403  1.00  0.83           C  \\nATOM    851  OD1 ASP A 113      18.639  24.384  14.704  1.00  0.83           O  \\nATOM    852  OD2 ASP A 113      19.143  26.516  14.379  1.00  0.83           O  \\nATOM    853  N   GLU A 114      23.646  23.470  14.234  1.00  0.81           N  \\nATOM    854  CA  GLU A 114      24.935  23.092  13.667  1.00  0.81           C  \\nATOM    855  C   GLU A 114      25.145  21.583  13.684  1.00  0.81           C  \\nATOM    856  O   GLU A 114      25.558  20.976  12.676  1.00  0.81           O  \\nATOM    857  CB  GLU A 114      26.098  23.854  14.363  1.00  0.81           C  \\nATOM    858  CG  GLU A 114      26.123  25.381  14.098  1.00  0.81           C  \\nATOM    859  CD  GLU A 114      25.994  25.760  12.637  1.00  0.81           C  \\nATOM    860  OE1 GLU A 114      26.671  25.192  11.738  1.00  0.81           O  \\nATOM    861  OE2 GLU A 114      25.158  26.671  12.379  1.00  0.81           O  \\nATOM    862  N   LEU A 115      24.775  20.887  14.764  1.00  0.85           N  \\nATOM    863  CA  LEU A 115      24.868  19.440  14.859  1.00  0.85           C  \\nATOM    864  C   LEU A 115      23.859  18.711  13.970  1.00  0.85           C  \\nATOM    865  O   LEU A 115      24.134  17.622  13.457  1.00  0.85           O  \\nATOM    866  CB  LEU A 115      24.764  18.967  16.326  1.00  0.85           C  \\nATOM    867  CG  LEU A 115      25.826  19.584  17.263  1.00  0.85           C  \\nATOM    868  CD1 LEU A 115      25.612  19.135  18.713  1.00  0.85           C  \\nATOM    869  CD2 LEU A 115      27.259  19.244  16.833  1.00  0.85           C  \\nATOM    870  N   ILE A 116      22.666  19.295  13.730  1.00  0.87           N  \\nATOM    871  CA  ILE A 116      21.735  18.834  12.696  1.00  0.87           C  \\nATOM    872  C   ILE A 116      22.322  18.981  11.308  1.00  0.87           C  \\nATOM    873  O   ILE A 116      22.243  18.054  10.486  1.00  0.87           O  \\nATOM    874  CB  ILE A 116      20.376  19.533  12.729  1.00  0.87           C  \\nATOM    875  CG1 ILE A 116      19.674  19.219  14.062  1.00  0.87           C  \\nATOM    876  CG2 ILE A 116      19.510  19.094  11.522  1.00  0.87           C  \\nATOM    877  CD1 ILE A 116      18.327  19.918  14.241  1.00  0.87           C  \\nATOM    878  N   VAL A 117      22.961  20.124  11.006  1.00  0.86           N  \\nATOM    879  CA  VAL A 117      23.654  20.343   9.747  1.00  0.86           C  \\nATOM    880  C   VAL A 117      24.749  19.293   9.556  1.00  0.86           C  \\nATOM    881  O   VAL A 117      24.706  18.544   8.568  1.00  0.86           O  \\nATOM    882  CB  VAL A 117      24.177  21.776   9.608  1.00  0.86           C  \\nATOM    883  CG1 VAL A 117      24.995  21.954   8.319  1.00  0.86           C  \\nATOM    884  CG2 VAL A 117      22.998  22.769   9.575  1.00  0.86           C  \\nATOM    885  N   ASP A 118      25.639  19.069  10.538  1.00  0.82           N  \\nATOM    886  CA  ASP A 118      26.664  18.028  10.535  1.00  0.82           C  \\nATOM    887  C   ASP A 118      26.107  16.617  10.325  1.00  0.82           C  \\nATOM    888  O   ASP A 118      26.612  15.831   9.507  1.00  0.82           O  \\nATOM    889  CB  ASP A 118      27.429  18.001  11.888  1.00  0.82           C  \\nATOM    890  CG  ASP A 118      28.447  19.122  12.047  1.00  0.82           C  \\nATOM    891  OD1 ASP A 118      28.764  19.805  11.046  1.00  0.82           O  \\nATOM    892  OD2 ASP A 118      28.983  19.223  13.184  1.00  0.82           O  \\nATOM    893  N   ARG A 119      25.021  16.260  11.030  1.00  0.79           N  \\nATOM    894  CA  ARG A 119      24.300  14.997  10.939  1.00  0.79           C  \\nATOM    895  C   ARG A 119      23.705  14.719   9.568  1.00  0.79           C  \\nATOM    896  O   ARG A 119      23.632  13.562   9.131  1.00  0.79           O  \\nATOM    897  CB  ARG A 119      23.192  14.947  12.016  1.00  0.79           C  \\nATOM    898  CG  ARG A 119      22.408  13.622  12.094  1.00  0.79           C  \\nATOM    899  CD  ARG A 119      21.465  13.561  13.298  1.00  0.79           C  \\nATOM    900  NE  ARG A 119      20.882  12.178  13.377  1.00  0.79           N  \\nATOM    901  CZ  ARG A 119      19.707  11.794  12.859  1.00  0.79           C  \\nATOM    902  NH1 ARG A 119      18.949  12.605  12.143  1.00  0.79           N  \\nATOM    903  NH2 ARG A 119      19.274  10.557  13.082  1.00  0.79           N  \\nATOM    904  N   ILE A 120      23.242  15.755   8.859  1.00  0.85           N  \\nATOM    905  CA  ILE A 120      22.704  15.629   7.516  1.00  0.85           C  \\nATOM    906  C   ILE A 120      23.745  15.758   6.410  1.00  0.85           C  \\nATOM    907  O   ILE A 120      23.832  14.884   5.540  1.00  0.85           O  \\nATOM    908  CB  ILE A 120      21.531  16.573   7.322  1.00  0.85           C  \\nATOM    909  CG1 ILE A 120      20.400  16.136   8.282  1.00  0.85           C  \\nATOM    910  CG2 ILE A 120      21.066  16.552   5.854  1.00  0.85           C  \\nATOM    911  CD1 ILE A 120      19.078  16.854   8.025  1.00  0.85           C  \\nATOM    912  N   VAL A 121      24.607  16.795   6.391  1.00  0.85           N  \\nATOM    913  CA  VAL A 121      25.450  17.103   5.229  1.00  0.85           C  \\nATOM    914  C   VAL A 121      26.557  16.094   5.018  1.00  0.85           C  \\nATOM    915  O   VAL A 121      27.145  15.951   3.938  1.00  0.85           O  \\nATOM    916  CB  VAL A 121      26.084  18.491   5.268  1.00  0.85           C  \\nATOM    917  CG1 VAL A 121      24.991  19.564   5.403  1.00  0.85           C  \\nATOM    918  CG2 VAL A 121      27.157  18.618   6.367  1.00  0.85           C  \\nATOM    919  N   GLY A 122      26.861  15.322   6.071  1.00  0.87           N  \\nATOM    920  CA  GLY A 122      27.777  14.205   6.012  1.00  0.87           C  \\nATOM    921  C   GLY A 122      27.189  12.949   5.428  1.00  0.87           C  \\nATOM    922  O   GLY A 122      27.936  12.006   5.148  1.00  0.87           O  \\nATOM    923  N   ARG A 123      25.867  12.873   5.198  1.00  0.83           N  \\nATOM    924  CA  ARG A 123      25.236  11.726   4.569  1.00  0.83           C  \\nATOM    925  C   ARG A 123      25.668  11.480   3.133  1.00  0.83           C  \\nATOM    926  O   ARG A 123      25.899  12.399   2.355  1.00  0.83           O  \\nATOM    927  CB  ARG A 123      23.703  11.770   4.594  1.00  0.83           C  \\nATOM    928  CG  ARG A 123      23.078  11.508   5.969  1.00  0.83           C  \\nATOM    929  CD  ARG A 123      21.576  11.706   5.840  1.00  0.83           C  \\nATOM    930  NE  ARG A 123      20.948  11.465   7.169  1.00  0.83           N  \\nATOM    931  CZ  ARG A 123      19.679  11.817   7.400  1.00  0.83           C  \\nATOM    932  NH1 ARG A 123      19.121  11.554   8.575  1.00  0.83           N  \\nATOM    933  NH2 ARG A 123      18.959  12.425   6.455  1.00  0.83           N  \\nATOM    934  N   ARG A 124      25.779  10.207   2.731  1.00  0.84           N  \\nATOM    935  CA  ARG A 124      26.121   9.853   1.370  1.00  0.84           C  \\nATOM    936  C   ARG A 124      25.282   8.643   1.014  1.00  0.84           C  \\nATOM    937  O   ARG A 124      25.063   7.789   1.862  1.00  0.84           O  \\nATOM    938  CB  ARG A 124      27.618   9.473   1.211  1.00  0.84           C  \\nATOM    939  CG  ARG A 124      28.648  10.492   1.742  1.00  0.84           C  \\nATOM    940  CD  ARG A 124      28.931  11.631   0.764  1.00  0.84           C  \\nATOM    941  NE  ARG A 124      29.823  12.633   1.443  1.00  0.84           N  \\nATOM    942  CZ  ARG A 124      29.392  13.664   2.177  1.00  0.84           C  \\nATOM    943  NH1 ARG A 124      28.107  13.877   2.447  1.00  0.84           N  \\nATOM    944  NH2 ARG A 124      30.257  14.538   2.685  1.00  0.84           N  \\nATOM    945  N   VAL A 125      24.770   8.557  -0.229  1.00  0.88           N  \\nATOM    946  CA  VAL A 125      23.862   7.513  -0.669  1.00  0.88           C  \\nATOM    947  C   VAL A 125      24.397   6.836  -1.919  1.00  0.88           C  \\nATOM    948  O   VAL A 125      25.067   7.443  -2.749  1.00  0.88           O  \\nATOM    949  CB  VAL A 125      22.440   8.051  -0.897  1.00  0.88           C  \\nATOM    950  CG1 VAL A 125      22.385   9.049  -2.068  1.00  0.88           C  \\nATOM    951  CG2 VAL A 125      21.391   6.934  -1.074  1.00  0.88           C  \\nATOM    952  N   HIS A 126      24.106   5.530  -2.093  1.00  0.85           N  \\nATOM    953  CA  HIS A 126      24.167   4.885  -3.390  1.00  0.85           C  \\nATOM    954  C   HIS A 126      22.726   4.854  -3.891  1.00  0.85           C  \\nATOM    955  O   HIS A 126      21.905   4.076  -3.399  1.00  0.85           O  \\nATOM    956  CB  HIS A 126      24.784   3.463  -3.279  1.00  0.85           C  \\nATOM    957  CG  HIS A 126      25.062   2.778  -4.575  1.00  0.85           C  \\nATOM    958  ND1 HIS A 126      24.161   1.879  -5.107  1.00  0.85           N  \\nATOM    959  CD2 HIS A 126      26.070   3.051  -5.459  1.00  0.85           C  \\nATOM    960  CE1 HIS A 126      24.631   1.618  -6.325  1.00  0.85           C  \\nATOM    961  NE2 HIS A 126      25.765   2.305  -6.566  1.00  0.85           N  \\nATOM    962  N   ALA A 127      22.337   5.737  -4.835  1.00  0.88           N  \\nATOM    963  CA  ALA A 127      20.959   5.843  -5.315  1.00  0.88           C  \\nATOM    964  C   ALA A 127      20.332   4.581  -5.943  1.00  0.88           C  \\nATOM    965  O   ALA A 127      19.161   4.340  -5.651  1.00  0.88           O  \\nATOM    966  CB  ALA A 127      20.763   7.060  -6.246  1.00  0.88           C  \\nATOM    967  N   PRO A 128      20.985   3.738  -6.749  1.00  0.85           N  \\nATOM    968  CA  PRO A 128      20.415   2.475  -7.228  1.00  0.85           C  \\nATOM    969  C   PRO A 128      19.962   1.495  -6.154  1.00  0.85           C  \\nATOM    970  O   PRO A 128      19.054   0.709  -6.438  1.00  0.85           O  \\nATOM    971  CB  PRO A 128      21.550   1.866  -8.065  1.00  0.85           C  \\nATOM    972  CG  PRO A 128      22.353   3.061  -8.576  1.00  0.85           C  \\nATOM    973  CD  PRO A 128      22.244   4.055  -7.425  1.00  0.85           C  \\nATOM    974  N   SER A 129      20.605   1.474  -4.967  1.00  0.86           N  \\nATOM    975  CA  SER A 129      20.249   0.555  -3.872  1.00  0.86           C  \\nATOM    976  C   SER A 129      19.731   1.197  -2.594  1.00  0.86           C  \\nATOM    977  O   SER A 129      19.264   0.499  -1.695  1.00  0.86           O  \\nATOM    978  CB  SER A 129      21.359  -0.463  -3.551  1.00  0.86           C  \\nATOM    979  OG  SER A 129      22.552   0.157  -2.896  1.00  0.86           O  \\nATOM    980  N   GLY A 130      19.792   2.524  -2.471  1.00  0.91           N  \\nATOM    981  CA  GLY A 130      19.240   3.271  -1.341  1.00  0.91           C  \\nATOM    982  C   GLY A 130      20.051   3.217  -0.075  1.00  0.91           C  \\nATOM    983  O   GLY A 130      19.593   3.678   0.977  1.00  0.91           O  \\nATOM    984  N   ARG A 131      21.273   2.664  -0.122  1.00  0.84           N  \\nATOM    985  CA  ARG A 131      22.142   2.504   1.031  1.00  0.84           C  \\nATOM    986  C   ARG A 131      22.790   3.831   1.405  1.00  0.84           C  \\nATOM    987  O   ARG A 131      23.168   4.594   0.525  1.00  0.84           O  \\nATOM    988  CB  ARG A 131      23.201   1.414   0.745  1.00  0.84           C  \\nATOM    989  CG  ARG A 131      23.676   0.613   1.979  1.00  0.84           C  \\nATOM    990  CD  ARG A 131      24.397  -0.732   1.735  1.00  0.84           C  \\nATOM    991  NE  ARG A 131      24.917  -0.790   0.333  1.00  0.84           N  \\nATOM    992  CZ  ARG A 131      24.258  -1.372  -0.691  1.00  0.84           C  \\nATOM    993  NH1 ARG A 131      23.023  -1.817  -0.564  1.00  0.84           N  \\nATOM    994  NH2 ARG A 131      24.808  -1.414  -1.894  1.00  0.84           N  \\nATOM    995  N   VAL A 132      22.905   4.141   2.716  1.00  0.88           N  \\nATOM    996  CA  VAL A 132      23.321   5.446   3.204  1.00  0.88           C  \\nATOM    997  C   VAL A 132      24.455   5.291   4.216  1.00  0.88           C  \\nATOM    998  O   VAL A 132      24.542   4.298   4.929  1.00  0.88           O  \\nATOM    999  CB  VAL A 132      22.129   6.235   3.768  1.00  0.88           C  \\nATOM   1000  CG1 VAL A 132      21.562   5.549   5.024  1.00  0.88           C  \\nATOM   1001  CG2 VAL A 132      22.458   7.721   4.021  1.00  0.88           C  \\nATOM   1002  N   TYR A 133      25.361   6.291   4.273  1.00  0.84           N  \\nATOM   1003  CA  TYR A 133      26.584   6.323   5.059  1.00  0.84           C  \\nATOM   1004  C   TYR A 133      26.717   7.710   5.676  1.00  0.84           C  \\nATOM   1005  O   TYR A 133      25.940   8.608   5.359  1.00  0.84           O  \\nATOM   1006  CB  TYR A 133      27.832   6.132   4.162  1.00  0.84           C  \\nATOM   1007  CG  TYR A 133      27.817   4.766   3.554  1.00  0.84           C  \\nATOM   1008  CD1 TYR A 133      28.436   3.708   4.227  1.00  0.84           C  \\nATOM   1009  CD2 TYR A 133      27.147   4.510   2.349  1.00  0.84           C  \\nATOM   1010  CE1 TYR A 133      28.378   2.408   3.715  1.00  0.84           C  \\nATOM   1011  CE2 TYR A 133      27.054   3.204   1.841  1.00  0.84           C  \\nATOM   1012  CZ  TYR A 133      27.682   2.163   2.538  1.00  0.84           C  \\nATOM   1013  OH  TYR A 133      27.569   0.825   2.107  1.00  0.84           O  \\nATOM   1014  N   HIS A 134      27.712   7.947   6.566  1.00  0.80           N  \\nATOM   1015  CA  HIS A 134      27.998   9.291   7.050  1.00  0.80           C  \\nATOM   1016  C   HIS A 134      29.494   9.498   7.223  1.00  0.80           C  \\nATOM   1017  O   HIS A 134      30.169   8.658   7.808  1.00  0.80           O  \\nATOM   1018  CB  HIS A 134      27.319   9.550   8.409  1.00  0.80           C  \\nATOM   1019  CG  HIS A 134      27.284  10.989   8.812  1.00  0.80           C  \\nATOM   1020  ND1 HIS A 134      28.424  11.698   9.106  1.00  0.80           N  \\nATOM   1021  CD2 HIS A 134      26.184  11.806   8.828  1.00  0.80           C  \\nATOM   1022  CE1 HIS A 134      28.012  12.951   9.295  1.00  0.80           C  \\nATOM   1023  NE2 HIS A 134      26.681  13.040   9.137  1.00  0.80           N  \\nATOM   1024  N   VAL A 135      30.104  10.595   6.760  1.00  0.86           N  \\nATOM   1025  CA  VAL A 135      31.562  10.750   6.771  1.00  0.86           C  \\nATOM   1026  C   VAL A 135      32.271  10.706   8.118  1.00  0.86           C  \\nATOM   1027  O   VAL A 135      33.396  10.187   8.205  1.00  0.86           O  \\nATOM   1028  CB  VAL A 135      32.031  11.977   6.010  1.00  0.86           C  \\nATOM   1029  CG1 VAL A 135      31.430  11.926   4.599  1.00  0.86           C  \\nATOM   1030  CG2 VAL A 135      31.667  13.286   6.732  1.00  0.86           C  \\nATOM   1031  N   LYS A 136      31.684  11.254   9.185  1.00  0.79           N  \\nATOM   1032  CA  LYS A 136      32.315  11.318  10.491  1.00  0.79           C  \\nATOM   1033  C   LYS A 136      31.703  10.310  11.436  1.00  0.79           C  \\nATOM   1034  O   LYS A 136      32.417   9.693  12.242  1.00  0.79           O  \\nATOM   1035  CB  LYS A 136      32.174  12.751  11.085  1.00  0.79           C  \\nATOM   1036  CG  LYS A 136      32.600  12.862  12.561  1.00  0.79           C  \\nATOM   1037  CD  LYS A 136      32.696  14.292  13.117  1.00  0.79           C  \\nATOM   1038  CE  LYS A 136      34.100  14.887  13.011  1.00  0.79           C  \\nATOM   1039  NZ  LYS A 136      34.210  16.078  13.885  1.00  0.79           N  \\nATOM   1040  N   PHE A 137      30.382  10.081  11.384  1.00  0.70           N  \\nATOM   1041  CA  PHE A 137      29.721   9.301  12.423  1.00  0.70           C  \\nATOM   1042  C   PHE A 137      29.471   7.852  12.026  1.00  0.70           C  \\nATOM   1043  O   PHE A 137      29.183   7.010  12.884  1.00  0.70           O  \\nATOM   1044  CB  PHE A 137      28.358   9.932  12.806  1.00  0.70           C  \\nATOM   1045  CG  PHE A 137      28.466  11.401  13.129  1.00  0.70           C  \\nATOM   1046  CD1 PHE A 137      29.227  11.880  14.211  1.00  0.70           C  \\nATOM   1047  CD2 PHE A 137      27.761  12.326  12.346  1.00  0.70           C  \\nATOM   1048  CE1 PHE A 137      29.288  13.255  14.484  1.00  0.70           C  \\nATOM   1049  CE2 PHE A 137      27.847  13.700  12.594  1.00  0.70           C  \\nATOM   1050  CZ  PHE A 137      28.609  14.166  13.668  1.00  0.70           C  \\nATOM   1051  N   ASN A 138      29.609   7.497  10.742  1.00  0.81           N  \\nATOM   1052  CA  ASN A 138      29.459   6.132  10.260  1.00  0.81           C  \\nATOM   1053  C   ASN A 138      30.178   6.038   8.910  1.00  0.81           C  \\nATOM   1054  O   ASN A 138      29.495   5.967   7.878  1.00  0.81           O  \\nATOM   1055  CB  ASN A 138      27.943   5.750  10.157  1.00  0.81           C  \\nATOM   1056  CG  ASN A 138      27.724   4.327   9.642  1.00  0.81           C  \\nATOM   1057  OD1 ASN A 138      28.480   3.413   9.947  1.00  0.81           O  \\nATOM   1058  ND2 ASN A 138      26.665   4.137   8.809  1.00  0.81           N  \\nATOM   1059  N   PRO A 139      31.522   6.130   8.843  1.00  0.84           N  \\nATOM   1060  CA  PRO A 139      32.255   6.188   7.577  1.00  0.84           C  \\nATOM   1061  C   PRO A 139      32.039   4.947   6.714  1.00  0.84           C  \\nATOM   1062  O   PRO A 139      31.912   3.869   7.300  1.00  0.84           O  \\nATOM   1063  CB  PRO A 139      33.734   6.353   7.982  1.00  0.84           C  \\nATOM   1064  CG  PRO A 139      33.704   6.845   9.430  1.00  0.84           C  \\nATOM   1065  CD  PRO A 139      32.426   6.235   9.997  1.00  0.84           C  \\nATOM   1066  N   PRO A 140      31.975   5.008   5.381  1.00  0.88           N  \\nATOM   1067  CA  PRO A 140      32.029   3.807   4.556  1.00  0.88           C  \\nATOM   1068  C   PRO A 140      33.385   3.102   4.656  1.00  0.88           C  \\nATOM   1069  O   PRO A 140      34.368   3.689   5.100  1.00  0.88           O  \\nATOM   1070  CB  PRO A 140      31.770   4.369   3.151  1.00  0.88           C  \\nATOM   1071  CG  PRO A 140      32.480   5.721   3.157  1.00  0.88           C  \\nATOM   1072  CD  PRO A 140      32.329   6.202   4.605  1.00  0.88           C  \\nATOM   1073  N   LYS A 141      33.473   1.820   4.250  1.00  0.82           N  \\nATOM   1074  CA  LYS A 141      34.728   1.071   4.286  1.00  0.82           C  \\nATOM   1075  C   LYS A 141      35.835   1.592   3.387  1.00  0.82           C  \\nATOM   1076  O   LYS A 141      37.024   1.580   3.765  1.00  0.82           O  \\nATOM   1077  CB  LYS A 141      34.414  -0.371   3.869  1.00  0.82           C  \\nATOM   1078  CG  LYS A 141      35.166  -1.488   4.590  1.00  0.82           C  \\nATOM   1079  CD  LYS A 141      34.703  -2.862   4.066  1.00  0.82           C  \\nATOM   1080  CE  LYS A 141      33.231  -3.233   4.315  1.00  0.82           C  \\nATOM   1081  NZ  LYS A 141      32.967  -3.400   5.757  1.00  0.82           N  \\nATOM   1082  N   VAL A 142      35.499   2.036   2.177  1.00  0.83           N  \\nATOM   1083  CA  VAL A 142      36.409   2.633   1.227  1.00  0.83           C  \\nATOM   1084  C   VAL A 142      35.919   4.064   1.061  1.00  0.83           C  \\nATOM   1085  O   VAL A 142      34.772   4.291   0.672  1.00  0.83           O  \\nATOM   1086  CB  VAL A 142      36.425   1.897  -0.118  1.00  0.83           C  \\nATOM   1087  CG1 VAL A 142      37.513   2.498  -1.028  1.00  0.83           C  \\nATOM   1088  CG2 VAL A 142      36.670   0.387   0.090  1.00  0.83           C  \\nATOM   1089  N   GLU A 143      36.743   5.077   1.400  1.00  0.80           N  \\nATOM   1090  CA  GLU A 143      36.430   6.497   1.270  1.00  0.80           C  \\nATOM   1091  C   GLU A 143      35.682   6.932   0.003  1.00  0.80           C  \\nATOM   1092  O   GLU A 143      36.212   6.914  -1.111  1.00  0.80           O  \\nATOM   1093  CB  GLU A 143      37.733   7.312   1.419  1.00  0.80           C  \\nATOM   1094  CG  GLU A 143      38.134   7.567   2.892  1.00  0.80           C  \\nATOM   1095  CD  GLU A 143      39.620   7.308   3.135  1.00  0.80           C  \\nATOM   1096  OE1 GLU A 143      40.050   6.155   2.869  1.00  0.80           O  \\nATOM   1097  OE2 GLU A 143      40.320   8.248   3.588  1.00  0.80           O  \\nATOM   1098  N   GLY A 144      34.407   7.351   0.163  1.00  0.86           N  \\nATOM   1099  CA  GLY A 144      33.539   7.827  -0.919  1.00  0.86           C  \\nATOM   1100  C   GLY A 144      32.863   6.762  -1.723  1.00  0.86           C  \\nATOM   1101  O   GLY A 144      32.069   7.069  -2.631  1.00  0.86           O  \\nATOM   1102  N   LYS A 145      33.073   5.483  -1.429  1.00  0.79           N  \\nATOM   1103  CA  LYS A 145      32.571   4.398  -2.233  1.00  0.79           C  \\nATOM   1104  C   LYS A 145      31.623   3.549  -1.411  1.00  0.79           C  \\nATOM   1105  O   LYS A 145      31.825   3.316  -0.223  1.00  0.79           O  \\nATOM   1106  CB  LYS A 145      33.747   3.534  -2.742  1.00  0.79           C  \\nATOM   1107  CG  LYS A 145      34.760   4.298  -3.612  1.00  0.79           C  \\nATOM   1108  CD  LYS A 145      34.172   4.668  -4.976  1.00  0.79           C  \\nATOM   1109  CE  LYS A 145      35.174   5.287  -5.943  1.00  0.79           C  \\nATOM   1110  NZ  LYS A 145      34.488   5.577  -7.211  1.00  0.79           N  \\nATOM   1111  N   ASP A 146      30.531   3.054  -2.035  1.00  0.82           N  \\nATOM   1112  CA  ASP A 146      29.695   2.026  -1.436  1.00  0.82           C  \\nATOM   1113  C   ASP A 146      30.551   0.770  -1.285  1.00  0.82           C  \\nATOM   1114  O   ASP A 146      31.165   0.333  -2.258  1.00  0.82           O  \\nATOM   1115  CB  ASP A 146      28.485   1.758  -2.371  1.00  0.82           C  \\nATOM   1116  CG  ASP A 146      27.421   0.855  -1.781  1.00  0.82           C  \\nATOM   1117  OD1 ASP A 146      26.672   0.244  -2.604  1.00  0.82           O  \\nATOM   1118  OD2 ASP A 146      27.277   0.728  -0.531  1.00  0.82           O  \\nATOM   1119  N   ASP A 147      30.648   0.168  -0.081  1.00  0.79           N  \\nATOM   1120  CA  ASP A 147      31.570  -0.934   0.135  1.00  0.79           C  \\nATOM   1121  C   ASP A 147      31.151  -2.233  -0.526  1.00  0.79           C  \\nATOM   1122  O   ASP A 147      31.967  -3.152  -0.700  1.00  0.79           O  \\nATOM   1123  CB  ASP A 147      31.899  -1.147   1.627  1.00  0.79           C  \\nATOM   1124  CG  ASP A 147      30.758  -0.964   2.614  1.00  0.79           C  \\nATOM   1125  OD1 ASP A 147      31.026  -0.292   3.657  1.00  0.79           O  \\nATOM   1126  OD2 ASP A 147      29.680  -1.570   2.392  1.00  0.79           O  \\nATOM   1127  N   VAL A 148      29.892  -2.314  -0.948  1.00  0.85           N  \\nATOM   1128  CA  VAL A 148      29.414  -3.295  -1.897  1.00  0.85           C  \\nATOM   1129  C   VAL A 148      29.933  -3.055  -3.309  1.00  0.85           C  \\nATOM   1130  O   VAL A 148      30.836  -3.758  -3.790  1.00  0.85           O  \\nATOM   1131  CB  VAL A 148      27.891  -3.360  -1.911  1.00  0.85           C  \\nATOM   1132  CG1 VAL A 148      27.411  -4.430  -2.910  1.00  0.85           C  \\nATOM   1133  CG2 VAL A 148      27.383  -3.705  -0.500  1.00  0.85           C  \\nATOM   1134  N   THR A 149      29.348  -2.110  -4.063  1.00  0.85           N  \\nATOM   1135  CA  THR A 149      29.563  -1.994  -5.502  1.00  0.85           C  \\nATOM   1136  C   THR A 149      30.848  -1.272  -5.868  1.00  0.85           C  \\nATOM   1137  O   THR A 149      31.480  -1.581  -6.887  1.00  0.85           O  \\nATOM   1138  CB  THR A 149      28.304  -1.490  -6.247  1.00  0.85           C  \\nATOM   1139  OG1 THR A 149      28.562  -0.867  -7.494  1.00  0.85           O  \\nATOM   1140  CG2 THR A 149      27.507  -0.490  -5.421  1.00  0.85           C  \\nATOM   1141  N   GLY A 150      31.325  -0.302  -5.069  1.00  0.87           N  \\nATOM   1142  CA  GLY A 150      32.473   0.510  -5.467  1.00  0.87           C  \\nATOM   1143  C   GLY A 150      32.148   1.597  -6.461  1.00  0.87           C  \\nATOM   1144  O   GLY A 150      33.037   2.279  -6.982  1.00  0.87           O  \\nATOM   1145  N   GLU A 151      30.858   1.828  -6.744  1.00  0.82           N  \\nATOM   1146  CA  GLU A 151      30.398   3.039  -7.392  1.00  0.82           C  \\nATOM   1147  C   GLU A 151      30.571   4.258  -6.496  1.00  0.82           C  \\nATOM   1148  O   GLU A 151      30.609   4.173  -5.269  1.00  0.82           O  \\nATOM   1149  CB  GLU A 151      28.950   2.941  -7.897  1.00  0.82           C  \\nATOM   1150  CG  GLU A 151      28.811   2.044  -9.147  1.00  0.82           C  \\nATOM   1151  CD  GLU A 151      27.354   1.814  -9.545  1.00  0.82           C  \\nATOM   1152  OE1 GLU A 151      26.480   2.609  -9.113  1.00  0.82           O  \\nATOM   1153  OE2 GLU A 151      27.094   0.808 -10.249  1.00  0.82           O  \\nATOM   1154  N   GLU A 152      30.736   5.433  -7.132  1.00  0.79           N  \\nATOM   1155  CA  GLU A 152      30.856   6.707  -6.456  1.00  0.79           C  \\nATOM   1156  C   GLU A 152      29.599   7.076  -5.689  1.00  0.79           C  \\nATOM   1157  O   GLU A 152      28.514   7.161  -6.256  1.00  0.79           O  \\nATOM   1158  CB  GLU A 152      31.191   7.826  -7.475  1.00  0.79           C  \\nATOM   1159  CG  GLU A 152      31.744   9.101  -6.799  1.00  0.79           C  \\nATOM   1160  CD  GLU A 152      33.025   8.724  -6.078  1.00  0.79           C  \\nATOM   1161  OE1 GLU A 152      33.278   9.172  -4.925  1.00  0.79           O  \\nATOM   1162  OE2 GLU A 152      33.762   7.884  -6.692  1.00  0.79           O  \\nATOM   1163  N   LEU A 153      29.714   7.259  -4.356  1.00  0.87           N  \\nATOM   1164  CA  LEU A 153      28.598   7.689  -3.541  1.00  0.87           C  \\nATOM   1165  C   LEU A 153      28.203   9.120  -3.843  1.00  0.87           C  \\nATOM   1166  O   LEU A 153      29.015   9.973  -4.180  1.00  0.87           O  \\nATOM   1167  CB  LEU A 153      28.867   7.580  -2.025  1.00  0.87           C  \\nATOM   1168  CG  LEU A 153      29.026   6.152  -1.484  1.00  0.87           C  \\nATOM   1169  CD1 LEU A 153      29.435   6.188  -0.007  1.00  0.87           C  \\nATOM   1170  CD2 LEU A 153      27.750   5.324  -1.641  1.00  0.87           C  \\nATOM   1171  N   THR A 154      26.905   9.422  -3.701  1.00  0.85           N  \\nATOM   1172  CA  THR A 154      26.380  10.738  -3.997  1.00  0.85           C  \\nATOM   1173  C   THR A 154      25.818  11.336  -2.738  1.00  0.85           C  \\nATOM   1174  O   THR A 154      25.697  10.693  -1.699  1.00  0.85           O  \\nATOM   1175  CB  THR A 154      25.328  10.755  -5.101  1.00  0.85           C  \\nATOM   1176  OG1 THR A 154      24.201   9.935  -4.813  1.00  0.85           O  \\nATOM   1177  CG2 THR A 154      25.961  10.206  -6.384  1.00  0.85           C  \\nATOM   1178  N   THR A 155      25.474  12.627  -2.777  1.00  0.85           N  \\nATOM   1179  CA  THR A 155      24.764  13.288  -1.703  1.00  0.85           C  \\nATOM   1180  C   THR A 155      23.388  13.565  -2.232  1.00  0.85           C  \\nATOM   1181  O   THR A 155      23.194  13.876  -3.405  1.00  0.85           O  \\nATOM   1182  CB  THR A 155      25.396  14.588  -1.223  1.00  0.85           C  \\nATOM   1183  OG1 THR A 155      25.780  15.446  -2.290  1.00  0.85           O  \\nATOM   1184  CG2 THR A 155      26.676  14.217  -0.480  1.00  0.85           C  \\nATOM   1185  N   ARG A 156      22.350  13.390  -1.397  1.00  0.80           N  \\nATOM   1186  CA  ARG A 156      21.012  13.770  -1.791  1.00  0.80           C  \\nATOM   1187  C   ARG A 156      20.870  15.286  -1.855  1.00  0.80           C  \\nATOM   1188  O   ARG A 156      21.422  16.017  -1.037  1.00  0.80           O  \\nATOM   1189  CB  ARG A 156      19.973  13.165  -0.829  1.00  0.80           C  \\nATOM   1190  CG  ARG A 156      19.912  11.628  -0.898  1.00  0.80           C  \\nATOM   1191  CD  ARG A 156      19.084  11.032   0.239  1.00  0.80           C  \\nATOM   1192  NE  ARG A 156      18.832   9.588  -0.101  1.00  0.80           N  \\nATOM   1193  CZ  ARG A 156      18.432   8.674   0.794  1.00  0.80           C  \\nATOM   1194  NH1 ARG A 156      18.330   8.987   2.083  1.00  0.80           N  \\nATOM   1195  NH2 ARG A 156      18.113   7.442   0.403  1.00  0.80           N  \\nATOM   1196  N   LYS A 157      20.112  15.814  -2.835  1.00  0.77           N  \\nATOM   1197  CA  LYS A 157      19.932  17.252  -2.994  1.00  0.77           C  \\nATOM   1198  C   LYS A 157      19.211  17.942  -1.824  1.00  0.77           C  \\nATOM   1199  O   LYS A 157      19.493  19.105  -1.503  1.00  0.77           O  \\nATOM   1200  CB  LYS A 157      19.259  17.568  -4.353  1.00  0.77           C  \\nATOM   1201  CG  LYS A 157      17.750  17.292  -4.382  1.00  0.77           C  \\nATOM   1202  CD  LYS A 157      17.094  17.484  -5.758  1.00  0.77           C  \\nATOM   1203  CE  LYS A 157      15.753  18.223  -5.675  1.00  0.77           C  \\nATOM   1204  NZ  LYS A 157      15.957  19.668  -5.930  1.00  0.77           N  \\nATOM   1205  N   ASP A 158      18.293  17.241  -1.136  1.00  0.85           N  \\nATOM   1206  CA  ASP A 158      17.576  17.670   0.061  1.00  0.85           C  \\nATOM   1207  C   ASP A 158      18.475  17.812   1.281  1.00  0.85           C  \\nATOM   1208  O   ASP A 158      18.143  18.539   2.236  1.00  0.85           O  \\nATOM   1209  CB  ASP A 158      16.480  16.623   0.386  1.00  0.85           C  \\nATOM   1210  CG  ASP A 158      15.158  16.949  -0.292  1.00  0.85           C  \\nATOM   1211  OD1 ASP A 158      15.064  18.008  -0.968  1.00  0.85           O  \\nATOM   1212  OD2 ASP A 158      14.227  16.126  -0.128  1.00  0.85           O  \\nATOM   1213  N   ASP A 159      19.634  17.139   1.249  1.00  0.86           N  \\nATOM   1214  CA  ASP A 159      20.625  17.105   2.298  1.00  0.86           C  \\nATOM   1215  C   ASP A 159      21.715  18.156   2.018  1.00  0.86           C  \\nATOM   1216  O   ASP A 159      22.674  18.297   2.781  1.00  0.86           O  \\nATOM   1217  CB  ASP A 159      21.239  15.667   2.424  1.00  0.86           C  \\nATOM   1218  CG  ASP A 159      20.280  14.568   2.901  1.00  0.86           C  \\nATOM   1219  OD1 ASP A 159      19.171  14.873   3.409  1.00  0.86           O  \\nATOM   1220  OD2 ASP A 159      20.673  13.367   2.806  1.00  0.86           O  \\nATOM   1221  N   GLN A 160      21.594  18.974   0.936  1.00  0.78           N  \\nATOM   1222  CA  GLN A 160      22.438  20.148   0.729  1.00  0.78           C  \\nATOM   1223  C   GLN A 160      22.293  21.172   1.844  1.00  0.78           C  \\nATOM   1224  O   GLN A 160      21.193  21.486   2.291  1.00  0.78           O  \\nATOM   1225  CB  GLN A 160      22.218  20.850  -0.637  1.00  0.78           C  \\nATOM   1226  CG  GLN A 160      23.012  20.191  -1.789  1.00  0.78           C  \\nATOM   1227  CD  GLN A 160      22.753  20.889  -3.130  1.00  0.78           C  \\nATOM   1228  OE1 GLN A 160      21.891  21.741  -3.279  1.00  0.78           O  \\nATOM   1229  NE2 GLN A 160      23.558  20.508  -4.158  1.00  0.78           N  \\nATOM   1230  N   GLU A 161      23.430  21.740   2.304  1.00  0.80           N  \\nATOM   1231  CA  GLU A 161      23.490  22.513   3.535  1.00  0.80           C  \\nATOM   1232  C   GLU A 161      22.535  23.694   3.599  1.00  0.80           C  \\nATOM   1233  O   GLU A 161      21.803  23.863   4.583  1.00  0.80           O  \\nATOM   1234  CB  GLU A 161      24.937  22.999   3.787  1.00  0.80           C  \\nATOM   1235  CG  GLU A 161      25.060  23.826   5.089  1.00  0.80           C  \\nATOM   1236  CD  GLU A 161      26.485  24.028   5.601  1.00  0.80           C  \\nATOM   1237  OE1 GLU A 161      27.436  23.435   5.043  1.00  0.80           O  \\nATOM   1238  OE2 GLU A 161      26.606  24.794   6.598  1.00  0.80           O  \\nATOM   1239  N   GLU A 162      22.443  24.498   2.533  1.00  0.82           N  \\nATOM   1240  CA  GLU A 162      21.525  25.620   2.448  1.00  0.82           C  \\nATOM   1241  C   GLU A 162      20.060  25.231   2.527  1.00  0.82           C  \\nATOM   1242  O   GLU A 162      19.251  25.897   3.186  1.00  0.82           O  \\nATOM   1243  CB  GLU A 162      21.823  26.481   1.192  1.00  0.82           C  \\nATOM   1244  CG  GLU A 162      22.979  27.483   1.411  1.00  0.82           C  \\nATOM   1245  CD  GLU A 162      22.622  28.354   2.611  1.00  0.82           C  \\nATOM   1246  OE1 GLU A 162      21.504  28.949   2.566  1.00  0.82           O  \\nATOM   1247  OE2 GLU A 162      23.400  28.353   3.598  1.00  0.82           O  \\nATOM   1248  N   THR A 163      19.655  24.118   1.898  1.00  0.87           N  \\nATOM   1249  CA  THR A 163      18.304  23.567   2.024  1.00  0.87           C  \\nATOM   1250  C   THR A 163      18.012  23.128   3.451  1.00  0.87           C  \\nATOM   1251  O   THR A 163      16.946  23.418   4.007  1.00  0.87           O  \\nATOM   1252  CB  THR A 163      18.049  22.397   1.079  1.00  0.87           C  \\nATOM   1253  OG1 THR A 163      18.299  22.786  -0.267  1.00  0.87           O  \\nATOM   1254  CG2 THR A 163      16.583  21.945   1.136  1.00  0.87           C  \\nATOM   1255  N   VAL A 164      18.973  22.455   4.110  1.00  0.89           N  \\nATOM   1256  CA  VAL A 164      18.883  22.053   5.515  1.00  0.89           C  \\nATOM   1257  C   VAL A 164      18.755  23.241   6.460  1.00  0.89           C  \\nATOM   1258  O   VAL A 164      17.863  23.268   7.324  1.00  0.89           O  \\nATOM   1259  CB  VAL A 164      20.088  21.209   5.933  1.00  0.89           C  \\nATOM   1260  CG1 VAL A 164      20.050  20.835   7.429  1.00  0.89           C  \\nATOM   1261  CG2 VAL A 164      20.130  19.928   5.087  1.00  0.89           C  \\nATOM   1262  N   ARG A 165      19.594  24.277   6.299  1.00  0.81           N  \\nATOM   1263  CA  ARG A 165      19.555  25.497   7.092  1.00  0.81           C  \\nATOM   1264  C   ARG A 165      18.259  26.279   6.936  1.00  0.81           C  \\nATOM   1265  O   ARG A 165      17.676  26.726   7.934  1.00  0.81           O  \\nATOM   1266  CB  ARG A 165      20.764  26.421   6.791  1.00  0.81           C  \\nATOM   1267  CG  ARG A 165      22.119  25.860   7.279  1.00  0.81           C  \\nATOM   1268  CD  ARG A 165      23.290  26.845   7.134  1.00  0.81           C  \\nATOM   1269  NE  ARG A 165      24.534  26.181   7.640  1.00  0.81           N  \\nATOM   1270  CZ  ARG A 165      24.954  26.121   8.911  1.00  0.81           C  \\nATOM   1271  NH1 ARG A 165      24.257  26.601   9.930  1.00  0.81           N  \\nATOM   1272  NH2 ARG A 165      26.120  25.514   9.145  1.00  0.81           N  \\nATOM   1273  N   LYS A 166      17.728  26.431   5.709  1.00  0.85           N  \\nATOM   1274  CA  LYS A 166      16.430  27.056   5.484  1.00  0.85           C  \\nATOM   1275  C   LYS A 166      15.285  26.338   6.186  1.00  0.85           C  \\nATOM   1276  O   LYS A 166      14.473  26.981   6.875  1.00  0.85           O  \\nATOM   1277  CB  LYS A 166      16.118  27.177   3.976  1.00  0.85           C  \\nATOM   1278  CG  LYS A 166      16.941  28.272   3.282  1.00  0.85           C  \\nATOM   1279  CD  LYS A 166      16.376  28.604   1.894  1.00  0.85           C  \\nATOM   1280  CE  LYS A 166      17.167  29.682   1.155  1.00  0.85           C  \\nATOM   1281  NZ  LYS A 166      18.185  29.053   0.286  1.00  0.85           N  \\nATOM   1282  N   ARG A 167      15.238  24.997   6.127  1.00  0.80           N  \\nATOM   1283  CA  ARG A 167      14.243  24.195   6.830  1.00  0.80           C  \\nATOM   1284  C   ARG A 167      14.259  24.401   8.340  1.00  0.80           C  \\nATOM   1285  O   ARG A 167      13.202  24.475   8.988  1.00  0.80           O  \\nATOM   1286  CB  ARG A 167      14.441  22.676   6.578  1.00  0.80           C  \\nATOM   1287  CG  ARG A 167      13.998  22.182   5.187  1.00  0.80           C  \\nATOM   1288  CD  ARG A 167      13.652  20.684   5.139  1.00  0.80           C  \\nATOM   1289  NE  ARG A 167      14.858  19.873   5.538  1.00  0.80           N  \\nATOM   1290  CZ  ARG A 167      15.689  19.237   4.700  1.00  0.80           C  \\nATOM   1291  NH1 ARG A 167      15.563  19.280   3.374  1.00  0.80           N  \\nATOM   1292  NH2 ARG A 167      16.712  18.530   5.179  1.00  0.80           N  \\nATOM   1293  N   LEU A 168      15.443  24.488   8.954  1.00  0.87           N  \\nATOM   1294  CA  LEU A 168      15.599  24.764  10.373  1.00  0.87           C  \\nATOM   1295  C   LEU A 168      15.123  26.150  10.785  1.00  0.87           C  \\nATOM   1296  O   LEU A 168      14.417  26.301  11.794  1.00  0.87           O  \\nATOM   1297  CB  LEU A 168      17.055  24.524  10.816  1.00  0.87           C  \\nATOM   1298  CG  LEU A 168      17.482  23.047  10.743  1.00  0.87           C  \\nATOM   1299  CD1 LEU A 168      18.990  22.931  10.989  1.00  0.87           C  \\nATOM   1300  CD2 LEU A 168      16.705  22.192  11.751  1.00  0.87           C  \\nATOM   1301  N   VAL A 169      15.425  27.196   9.993  1.00  0.84           N  \\nATOM   1302  CA  VAL A 169      14.934  28.556  10.224  1.00  0.84           C  \\nATOM   1303  C   VAL A 169      13.407  28.620  10.221  1.00  0.84           C  \\nATOM   1304  O   VAL A 169      12.784  29.175  11.140  1.00  0.84           O  \\nATOM   1305  CB  VAL A 169      15.495  29.533   9.186  1.00  0.84           C  \\nATOM   1306  CG1 VAL A 169      14.815  30.915   9.257  1.00  0.84           C  \\nATOM   1307  CG2 VAL A 169      17.010  29.704   9.404  1.00  0.84           C  \\nATOM   1308  N   GLU A 170      12.759  27.996   9.222  1.00  0.80           N  \\nATOM   1309  CA  GLU A 170      11.310  27.894   9.132  1.00  0.80           C  \\nATOM   1310  C   GLU A 170      10.700  27.105  10.281  1.00  0.80           C  \\nATOM   1311  O   GLU A 170       9.708  27.528  10.898  1.00  0.80           O  \\nATOM   1312  CB  GLU A 170      10.889  27.227   7.805  1.00  0.80           C  \\nATOM   1313  CG  GLU A 170      11.252  28.040   6.540  1.00  0.80           C  \\nATOM   1314  CD  GLU A 170      10.923  27.295   5.242  1.00  0.80           C  \\nATOM   1315  OE1 GLU A 170      10.207  26.261   5.297  1.00  0.80           O  \\nATOM   1316  OE2 GLU A 170      11.408  27.764   4.179  1.00  0.80           O  \\nATOM   1317  N   TYR A 171      11.307  25.960  10.644  1.00  0.84           N  \\nATOM   1318  CA  TYR A 171      10.876  25.138  11.764  1.00  0.84           C  \\nATOM   1319  C   TYR A 171      10.892  25.897  13.085  1.00  0.84           C  \\nATOM   1320  O   TYR A 171       9.885  25.918  13.799  1.00  0.84           O  \\nATOM   1321  CB  TYR A 171      11.749  23.852  11.870  1.00  0.84           C  \\nATOM   1322  CG  TYR A 171      11.304  22.953  13.000  1.00  0.84           C  \\nATOM   1323  CD1 TYR A 171      10.240  22.052  12.838  1.00  0.84           C  \\nATOM   1324  CD2 TYR A 171      11.907  23.066  14.264  1.00  0.84           C  \\nATOM   1325  CE1 TYR A 171       9.799  21.270  13.916  1.00  0.84           C  \\nATOM   1326  CE2 TYR A 171      11.439  22.316  15.352  1.00  0.84           C  \\nATOM   1327  CZ  TYR A 171      10.395  21.403  15.173  1.00  0.84           C  \\nATOM   1328  OH  TYR A 171       9.932  20.634  16.260  1.00  0.84           O  \\nATOM   1329  N   HIS A 172      11.985  26.584  13.448  1.00  0.80           N  \\nATOM   1330  CA  HIS A 172      12.059  27.295  14.718  1.00  0.80           C  \\nATOM   1331  C   HIS A 172      11.055  28.424  14.847  1.00  0.80           C  \\nATOM   1332  O   HIS A 172      10.429  28.601  15.903  1.00  0.80           O  \\nATOM   1333  CB  HIS A 172      13.474  27.819  15.006  1.00  0.80           C  \\nATOM   1334  CG  HIS A 172      14.453  26.704  15.141  1.00  0.80           C  \\nATOM   1335  ND1 HIS A 172      14.201  25.701  16.046  1.00  0.80           N  \\nATOM   1336  CD2 HIS A 172      15.669  26.527  14.546  1.00  0.80           C  \\nATOM   1337  CE1 HIS A 172      15.273  24.921  15.998  1.00  0.80           C  \\nATOM   1338  NE2 HIS A 172      16.169  25.382  15.110  1.00  0.80           N  \\nATOM   1339  N   GLN A 173      10.834  29.187  13.765  1.00  0.77           N  \\nATOM   1340  CA  GLN A 173       9.824  30.234  13.705  1.00  0.77           C  \\nATOM   1341  C   GLN A 173       8.407  29.714  13.903  1.00  0.77           C  \\nATOM   1342  O   GLN A 173       7.607  30.300  14.647  1.00  0.77           O  \\nATOM   1343  CB  GLN A 173       9.915  30.980  12.350  1.00  0.77           C  \\nATOM   1344  CG  GLN A 173       8.896  32.132  12.160  1.00  0.77           C  \\nATOM   1345  CD  GLN A 173       9.113  33.259  13.173  1.00  0.77           C  \\nATOM   1346  OE1 GLN A 173      10.169  33.413  13.770  1.00  0.77           O  \\nATOM   1347  NE2 GLN A 173       8.063  34.100  13.364  1.00  0.77           N  \\nATOM   1348  N   MET A 174       8.054  28.596  13.254  1.00  0.81           N  \\nATOM   1349  CA  MET A 174       6.774  27.931  13.417  1.00  0.81           C  \\nATOM   1350  C   MET A 174       6.578  27.293  14.790  1.00  0.81           C  \\nATOM   1351  O   MET A 174       5.494  27.357  15.386  1.00  0.81           O  \\nATOM   1352  CB  MET A 174       6.603  26.879  12.299  1.00  0.81           C  \\nATOM   1353  CG  MET A 174       5.173  26.326  12.174  1.00  0.81           C  \\nATOM   1354  SD  MET A 174       4.878  25.285  10.710  1.00  0.81           S  \\nATOM   1355  CE  MET A 174       6.057  23.970  11.120  1.00  0.81           C  \\nATOM   1356  N   THR A 175       7.626  26.651  15.326  1.00  0.78           N  \\nATOM   1357  CA  THR A 175       7.557  25.779  16.497  1.00  0.78           C  \\nATOM   1358  C   THR A 175       7.710  26.504  17.825  1.00  0.78           C  \\nATOM   1359  O   THR A 175       7.106  26.104  18.821  1.00  0.78           O  \\nATOM   1360  CB  THR A 175       8.571  24.645  16.395  1.00  0.78           C  \\nATOM   1361  OG1 THR A 175       8.301  23.868  15.233  1.00  0.78           O  \\nATOM   1362  CG2 THR A 175       8.505  23.659  17.562  1.00  0.78           C  \\nATOM   1363  N   ALA A 176       8.441  27.638  17.906  1.00  0.75           N  \\nATOM   1364  CA  ALA A 176       8.573  28.423  19.134  1.00  0.75           C  \\nATOM   1365  C   ALA A 176       7.262  28.825  19.845  1.00  0.75           C  \\nATOM   1366  O   ALA A 176       7.199  28.655  21.069  1.00  0.75           O  \\nATOM   1367  CB  ALA A 176       9.429  29.683  18.883  1.00  0.75           C  \\nATOM   1368  N   PRO A 177       6.177  29.298  19.211  1.00  0.80           N  \\nATOM   1369  CA  PRO A 177       4.907  29.513  19.899  1.00  0.80           C  \\nATOM   1370  C   PRO A 177       4.278  28.217  20.385  1.00  0.80           C  \\nATOM   1371  O   PRO A 177       3.657  28.214  21.443  1.00  0.80           O  \\nATOM   1372  CB  PRO A 177       4.034  30.259  18.873  1.00  0.80           C  \\nATOM   1373  CG  PRO A 177       4.663  29.951  17.517  1.00  0.80           C  \\nATOM   1374  CD  PRO A 177       6.144  29.804  17.836  1.00  0.80           C  \\nATOM   1375  N   LEU A 178       4.416  27.105  19.641  1.00  0.83           N  \\nATOM   1376  CA  LEU A 178       3.875  25.806  20.012  1.00  0.83           C  \\nATOM   1377  C   LEU A 178       4.527  25.215  21.255  1.00  0.83           C  \\nATOM   1378  O   LEU A 178       3.864  24.637  22.124  1.00  0.83           O  \\nATOM   1379  CB  LEU A 178       3.962  24.815  18.834  1.00  0.83           C  \\nATOM   1380  CG  LEU A 178       3.320  23.443  19.103  1.00  0.83           C  \\nATOM   1381  CD1 LEU A 178       1.813  23.558  19.357  1.00  0.83           C  \\nATOM   1382  CD2 LEU A 178       3.618  22.496  17.939  1.00  0.83           C  \\nATOM   1383  N   ILE A 179       5.849  25.378  21.404  1.00  0.84           N  \\nATOM   1384  CA  ILE A 179       6.576  25.023  22.617  1.00  0.84           C  \\nATOM   1385  C   ILE A 179       6.089  25.820  23.821  1.00  0.84           C  \\nATOM   1386  O   ILE A 179       5.841  25.283  24.910  1.00  0.84           O  \\nATOM   1387  CB  ILE A 179       8.070  25.235  22.430  1.00  0.84           C  \\nATOM   1388  CG1 ILE A 179       8.655  24.366  21.299  1.00  0.84           C  \\nATOM   1389  CG2 ILE A 179       8.839  24.952  23.727  1.00  0.84           C  \\nATOM   1390  CD1 ILE A 179       8.263  22.887  21.300  1.00  0.84           C  \\nATOM   1391  N   GLY A 180       5.876  27.137  23.641  1.00  0.86           N  \\nATOM   1392  CA  GLY A 180       5.324  27.998  24.680  1.00  0.86           C  \\nATOM   1393  C   GLY A 180       3.897  27.673  25.056  1.00  0.86           C  \\nATOM   1394  O   GLY A 180       3.526  27.771  26.231  1.00  0.86           O  \\nATOM   1395  N   TYR A 181       3.078  27.232  24.082  1.00  0.85           N  \\nATOM   1396  CA  TYR A 181       1.742  26.685  24.283  1.00  0.85           C  \\nATOM   1397  C   TYR A 181       1.762  25.446  25.180  1.00  0.85           C  \\nATOM   1398  O   TYR A 181       1.097  25.398  26.212  1.00  0.85           O  \\nATOM   1399  CB  TYR A 181       1.109  26.368  22.890  1.00  0.85           C  \\nATOM   1400  CG  TYR A 181      -0.245  25.703  22.941  1.00  0.85           C  \\nATOM   1401  CD1 TYR A 181      -0.351  24.303  23.019  1.00  0.85           C  \\nATOM   1402  CD2 TYR A 181      -1.420  26.466  22.903  1.00  0.85           C  \\nATOM   1403  CE1 TYR A 181      -1.606  23.689  23.167  1.00  0.85           C  \\nATOM   1404  CE2 TYR A 181      -2.675  25.848  22.997  1.00  0.85           C  \\nATOM   1405  CZ  TYR A 181      -2.763  24.465  23.150  1.00  0.85           C  \\nATOM   1406  OH  TYR A 181      -4.052  23.900  23.216  1.00  0.85           O  \\nATOM   1407  N   TYR A 182       2.583  24.428  24.852  1.00  0.85           N  \\nATOM   1408  CA  TYR A 182       2.650  23.202  25.638  1.00  0.85           C  \\nATOM   1409  C   TYR A 182       3.239  23.362  27.017  1.00  0.85           C  \\nATOM   1410  O   TYR A 182       2.814  22.693  27.965  1.00  0.85           O  \\nATOM   1411  CB  TYR A 182       3.381  22.065  24.898  1.00  0.85           C  \\nATOM   1412  CG  TYR A 182       2.513  21.522  23.805  1.00  0.85           C  \\nATOM   1413  CD1 TYR A 182       1.218  21.053  24.088  1.00  0.85           C  \\nATOM   1414  CD2 TYR A 182       3.007  21.410  22.498  1.00  0.85           C  \\nATOM   1415  CE1 TYR A 182       0.435  20.487  23.079  1.00  0.85           C  \\nATOM   1416  CE2 TYR A 182       2.230  20.819  21.492  1.00  0.85           C  \\nATOM   1417  CZ  TYR A 182       0.940  20.372  21.787  1.00  0.85           C  \\nATOM   1418  OH  TYR A 182       0.124  19.811  20.798  1.00  0.85           O  \\nATOM   1419  N   SER A 183       4.211  24.270  27.184  1.00  0.85           N  \\nATOM   1420  CA  SER A 183       4.701  24.655  28.501  1.00  0.85           C  \\nATOM   1421  C   SER A 183       3.593  25.267  29.364  1.00  0.85           C  \\nATOM   1422  O   SER A 183       3.402  24.875  30.517  1.00  0.85           O  \\nATOM   1423  CB  SER A 183       5.931  25.594  28.390  1.00  0.85           C  \\nATOM   1424  OG  SER A 183       6.582  25.755  29.653  1.00  0.85           O  \\nATOM   1425  N   LYS A 184       2.758  26.169  28.804  1.00  0.82           N  \\nATOM   1426  CA  LYS A 184       1.570  26.705  29.470  1.00  0.82           C  \\nATOM   1427  C   LYS A 184       0.499  25.669  29.798  1.00  0.82           C  \\nATOM   1428  O   LYS A 184      -0.084  25.675  30.893  1.00  0.82           O  \\nATOM   1429  CB  LYS A 184       0.953  27.860  28.644  1.00  0.82           C  \\nATOM   1430  CG  LYS A 184       1.799  29.143  28.679  1.00  0.82           C  \\nATOM   1431  CD  LYS A 184       1.621  29.878  30.014  1.00  0.82           C  \\nATOM   1432  CE  LYS A 184       2.596  31.025  30.280  1.00  0.82           C  \\nATOM   1433  NZ  LYS A 184       3.950  30.469  30.487  1.00  0.82           N  \\nATOM   1434  N   GLU A 185       0.219  24.719  28.894  1.00  0.82           N  \\nATOM   1435  CA  GLU A 185      -0.661  23.587  29.161  1.00  0.82           C  \\nATOM   1436  C   GLU A 185      -0.157  22.720  30.315  1.00  0.82           C  \\nATOM   1437  O   GLU A 185      -0.944  22.237  31.139  1.00  0.82           O  \\nATOM   1438  CB  GLU A 185      -0.852  22.697  27.908  1.00  0.82           C  \\nATOM   1439  CG  GLU A 185      -1.636  23.341  26.748  1.00  0.82           C  \\nATOM   1440  CD  GLU A 185      -3.075  23.580  27.105  1.00  0.82           C  \\nATOM   1441  OE1 GLU A 185      -3.821  22.613  27.428  1.00  0.82           O  \\nATOM   1442  OE2 GLU A 185      -3.539  24.741  27.012  1.00  0.82           O  \\nATOM   1443  N   ALA A 186       1.169  22.512  30.417  1.00  0.87           N  \\nATOM   1444  CA  ALA A 186       1.816  21.873  31.550  1.00  0.87           C  \\nATOM   1445  C   ALA A 186       1.703  22.646  32.868  1.00  0.87           C  \\nATOM   1446  O   ALA A 186       1.383  22.058  33.911  1.00  0.87           O  \\nATOM   1447  CB  ALA A 186       3.283  21.551  31.212  1.00  0.87           C  \\nATOM   1448  N   GLU A 187       1.887  23.982  32.857  1.00  0.80           N  \\nATOM   1449  CA  GLU A 187       1.660  24.894  33.987  1.00  0.80           C  \\nATOM   1450  C   GLU A 187       0.214  24.834  34.494  1.00  0.80           C  \\nATOM   1451  O   GLU A 187      -0.048  24.880  35.706  1.00  0.80           O  \\nATOM   1452  CB  GLU A 187       2.006  26.377  33.622  1.00  0.80           C  \\nATOM   1453  CG  GLU A 187       3.514  26.731  33.405  1.00  0.80           C  \\nATOM   1454  CD  GLU A 187       3.786  28.152  32.870  1.00  0.80           C  \\nATOM   1455  OE1 GLU A 187       2.853  28.990  32.786  1.00  0.80           O  \\nATOM   1456  OE2 GLU A 187       4.938  28.457  32.456  1.00  0.80           O  \\nATOM   1457  N   ALA A 188      -0.768  24.689  33.590  1.00  0.85           N  \\nATOM   1458  CA  ALA A 188      -2.175  24.523  33.909  1.00  0.85           C  \\nATOM   1459  C   ALA A 188      -2.540  23.119  34.393  1.00  0.85           C  \\nATOM   1460  O   ALA A 188      -3.651  22.898  34.898  1.00  0.85           O  \\nATOM   1461  CB  ALA A 188      -3.025  24.908  32.679  1.00  0.85           C  \\nATOM   1462  N   GLY A 189      -1.635  22.129  34.284  1.00  0.86           N  \\nATOM   1463  CA  GLY A 189      -1.884  20.758  34.730  1.00  0.86           C  \\nATOM   1464  C   GLY A 189      -2.582  19.882  33.728  1.00  0.86           C  \\nATOM   1465  O   GLY A 189      -2.989  18.760  34.060  1.00  0.86           O  \\nATOM   1466  N   ASN A 190      -2.736  20.333  32.474  1.00  0.83           N  \\nATOM   1467  CA  ASN A 190      -3.439  19.600  31.432  1.00  0.83           C  \\nATOM   1468  C   ASN A 190      -2.583  18.515  30.792  1.00  0.83           C  \\nATOM   1469  O   ASN A 190      -3.107  17.585  30.178  1.00  0.83           O  \\nATOM   1470  CB  ASN A 190      -3.873  20.523  30.258  1.00  0.83           C  \\nATOM   1471  CG  ASN A 190      -4.870  21.592  30.734  1.00  0.83           C  \\nATOM   1472  OD1 ASN A 190      -5.558  21.472  31.688  1.00  0.83           O  \\nATOM   1473  ND2 ASN A 190      -4.925  22.707  29.922  1.00  0.83           N  \\nATOM   1474  N   THR A 191      -1.253  18.611  30.881  1.00  0.87           N  \\nATOM   1475  CA  THR A 191      -0.315  17.699  30.234  1.00  0.87           C  \\nATOM   1476  C   THR A 191       0.994  17.853  30.957  1.00  0.87           C  \\nATOM   1477  O   THR A 191       1.121  18.657  31.872  1.00  0.87           O  \\nATOM   1478  CB  THR A 191      -0.131  17.905  28.720  1.00  0.87           C  \\nATOM   1479  OG1 THR A 191       0.647  16.875  28.102  1.00  0.87           O  \\nATOM   1480  CG2 THR A 191       0.555  19.232  28.373  1.00  0.87           C  \\nATOM   1481  N   LYS A 192       2.011  17.072  30.584  1.00  0.84           N  \\nATOM   1482  CA  LYS A 192       3.365  17.293  31.014  1.00  0.84           C  \\nATOM   1483  C   LYS A 192       4.229  17.591  29.814  1.00  0.84           C  \\nATOM   1484  O   LYS A 192       4.015  17.113  28.707  1.00  0.84           O  \\nATOM   1485  CB  LYS A 192       3.904  16.084  31.789  1.00  0.84           C  \\nATOM   1486  CG  LYS A 192       3.164  15.925  33.121  1.00  0.84           C  \\nATOM   1487  CD  LYS A 192       3.546  14.631  33.832  1.00  0.84           C  \\nATOM   1488  CE  LYS A 192       2.828  14.451  35.159  1.00  0.84           C  \\nATOM   1489  NZ  LYS A 192       3.279  13.184  35.765  1.00  0.84           N  \\nATOM   1490  N   TYR A 193       5.239  18.441  30.004  1.00  0.88           N  \\nATOM   1491  CA  TYR A 193       6.072  18.900  28.925  1.00  0.88           C  \\nATOM   1492  C   TYR A 193       7.517  18.741  29.376  1.00  0.88           C  \\nATOM   1493  O   TYR A 193       7.860  19.075  30.510  1.00  0.88           O  \\nATOM   1494  CB  TYR A 193       5.653  20.349  28.569  1.00  0.88           C  \\nATOM   1495  CG  TYR A 193       6.685  21.058  27.769  1.00  0.88           C  \\nATOM   1496  CD1 TYR A 193       6.938  20.776  26.415  1.00  0.88           C  \\nATOM   1497  CD2 TYR A 193       7.508  21.949  28.462  1.00  0.88           C  \\nATOM   1498  CE1 TYR A 193       8.004  21.413  25.758  1.00  0.88           C  \\nATOM   1499  CE2 TYR A 193       8.571  22.565  27.820  1.00  0.88           C  \\nATOM   1500  CZ  TYR A 193       8.796  22.313  26.478  1.00  0.88           C  \\nATOM   1501  OH  TYR A 193       9.853  23.023  25.939  1.00  0.88           O  \\nATOM   1502  N   ALA A 194       8.391  18.204  28.502  1.00  0.92           N  \\nATOM   1503  CA  ALA A 194       9.800  18.102  28.805  1.00  0.92           C  \\nATOM   1504  C   ALA A 194      10.658  18.207  27.554  1.00  0.92           C  \\nATOM   1505  O   ALA A 194      10.340  17.661  26.498  1.00  0.92           O  \\nATOM   1506  CB  ALA A 194      10.110  16.775  29.522  1.00  0.92           C  \\nATOM   1507  N   LYS A 195      11.800  18.913  27.665  1.00  0.87           N  \\nATOM   1508  CA  LYS A 195      12.837  18.921  26.654  1.00  0.87           C  \\nATOM   1509  C   LYS A 195      13.791  17.769  26.899  1.00  0.87           C  \\nATOM   1510  O   LYS A 195      14.017  17.344  28.026  1.00  0.87           O  \\nATOM   1511  CB  LYS A 195      13.663  20.220  26.660  1.00  0.87           C  \\nATOM   1512  CG  LYS A 195      12.834  21.468  26.377  1.00  0.87           C  \\nATOM   1513  CD  LYS A 195      13.709  22.721  26.460  1.00  0.87           C  \\nATOM   1514  CE  LYS A 195      12.953  24.004  26.141  1.00  0.87           C  \\nATOM   1515  NZ  LYS A 195      12.735  24.123  24.691  1.00  0.87           N  \\nATOM   1516  N   VAL A 196      14.356  17.234  25.808  1.00  0.89           N  \\nATOM   1517  CA  VAL A 196      15.222  16.080  25.817  1.00  0.89           C  \\nATOM   1518  C   VAL A 196      16.447  16.441  25.010  1.00  0.89           C  \\nATOM   1519  O   VAL A 196      16.343  16.844  23.856  1.00  0.89           O  \\nATOM   1520  CB  VAL A 196      14.542  14.877  25.166  1.00  0.89           C  \\nATOM   1521  CG1 VAL A 196      15.458  13.644  25.195  1.00  0.89           C  \\nATOM   1522  CG2 VAL A 196      13.212  14.580  25.881  1.00  0.89           C  \\nATOM   1523  N   ASP A 197      17.656  16.285  25.590  1.00  0.85           N  \\nATOM   1524  CA  ASP A 197      18.890  16.485  24.865  1.00  0.85           C  \\nATOM   1525  C   ASP A 197      19.136  15.275  23.975  1.00  0.85           C  \\nATOM   1526  O   ASP A 197      19.493  14.179  24.437  1.00  0.85           O  \\nATOM   1527  CB  ASP A 197      20.030  16.744  25.876  1.00  0.85           C  \\nATOM   1528  CG  ASP A 197      21.371  17.056  25.225  1.00  0.85           C  \\nATOM   1529  OD1 ASP A 197      21.421  17.238  23.976  1.00  0.85           O  \\nATOM   1530  OD2 ASP A 197      22.368  17.092  25.996  1.00  0.85           O  \\nATOM   1531  N   GLY A 198      18.897  15.436  22.671  1.00  0.88           N  \\nATOM   1532  CA  GLY A 198      19.015  14.408  21.654  1.00  0.88           C  \\nATOM   1533  C   GLY A 198      20.408  14.212  21.122  1.00  0.88           C  \\nATOM   1534  O   GLY A 198      20.607  13.385  20.227  1.00  0.88           O  \\nATOM   1535  N   THR A 199      21.410  14.982  21.600  1.00  0.84           N  \\nATOM   1536  CA  THR A 199      22.799  14.832  21.142  1.00  0.84           C  \\nATOM   1537  C   THR A 199      23.499  13.687  21.817  1.00  0.84           C  \\nATOM   1538  O   THR A 199      24.492  13.148  21.307  1.00  0.84           O  \\nATOM   1539  CB  THR A 199      23.706  16.057  21.310  1.00  0.84           C  \\nATOM   1540  OG1 THR A 199      24.107  16.311  22.658  1.00  0.84           O  \\nATOM   1541  CG2 THR A 199      22.988  17.303  20.804  1.00  0.84           C  \\nATOM   1542  N   LYS A 200      23.012  13.299  22.999  1.00  0.80           N  \\nATOM   1543  CA  LYS A 200      23.605  12.279  23.827  1.00  0.80           C  \\nATOM   1544  C   LYS A 200      23.585  10.897  23.162  1.00  0.80           C  \\nATOM   1545  O   LYS A 200      22.711  10.634  22.332  1.00  0.80           O  \\nATOM   1546  CB  LYS A 200      22.921  12.197  25.204  1.00  0.80           C  \\nATOM   1547  CG  LYS A 200      22.931  13.491  26.027  1.00  0.80           C  \\nATOM   1548  CD  LYS A 200      22.121  13.284  27.314  1.00  0.80           C  \\nATOM   1549  CE  LYS A 200      22.215  14.425  28.322  1.00  0.80           C  \\nATOM   1550  NZ  LYS A 200      21.590  13.968  29.578  1.00  0.80           N  \\nATOM   1551  N   PRO A 201      24.495   9.958  23.436  1.00  0.81           N  \\nATOM   1552  CA  PRO A 201      24.299   8.542  23.100  1.00  0.81           C  \\nATOM   1553  C   PRO A 201      22.893   8.004  23.396  1.00  0.81           C  \\nATOM   1554  O   PRO A 201      22.279   8.399  24.373  1.00  0.81           O  \\nATOM   1555  CB  PRO A 201      25.382   7.810  23.905  1.00  0.81           C  \\nATOM   1556  CG  PRO A 201      26.453   8.867  24.190  1.00  0.81           C  \\nATOM   1557  CD  PRO A 201      25.635  10.145  24.338  1.00  0.81           C  \\nATOM   1558  N   VAL A 202      22.362   7.076  22.570  1.00  0.85           N  \\nATOM   1559  CA  VAL A 202      20.971   6.628  22.648  1.00  0.85           C  \\nATOM   1560  C   VAL A 202      20.532   6.056  23.991  1.00  0.85           C  \\nATOM   1561  O   VAL A 202      19.362   6.186  24.380  1.00  0.85           O  \\nATOM   1562  CB  VAL A 202      20.631   5.631  21.545  1.00  0.85           C  \\nATOM   1563  CG1 VAL A 202      20.689   6.340  20.182  1.00  0.85           C  \\nATOM   1564  CG2 VAL A 202      21.560   4.404  21.580  1.00  0.85           C  \\nATOM   1565  N   ALA A 203      21.433   5.398  24.732  1.00  0.85           N  \\nATOM   1566  CA  ALA A 203      21.218   4.926  26.088  1.00  0.85           C  \\nATOM   1567  C   ALA A 203      21.038   6.047  27.113  1.00  0.85           C  \\nATOM   1568  O   ALA A 203      20.161   5.960  27.979  1.00  0.85           O  \\nATOM   1569  CB  ALA A 203      22.342   3.952  26.493  1.00  0.85           C  \\nATOM   1570  N   GLU A 204      21.824   7.132  27.015  1.00  0.81           N  \\nATOM   1571  CA  GLU A 204      21.717   8.325  27.842  1.00  0.81           C  \\nATOM   1572  C   GLU A 204      20.395   9.045  27.612  1.00  0.81           C  \\nATOM   1573  O   GLU A 204      19.668   9.383  28.549  1.00  0.81           O  \\nATOM   1574  CB  GLU A 204      22.887   9.286  27.519  1.00  0.81           C  \\nATOM   1575  CG  GLU A 204      24.305   8.731  27.803  1.00  0.81           C  \\nATOM   1576  CD  GLU A 204      24.649   8.664  29.292  1.00  0.81           C  \\nATOM   1577  OE1 GLU A 204      23.972   9.353  30.095  1.00  0.81           O  \\nATOM   1578  OE2 GLU A 204      25.623   7.939  29.613  1.00  0.81           O  \\nATOM   1579  N   VAL A 205      19.992   9.212  26.333  1.00  0.86           N  \\nATOM   1580  CA  VAL A 205      18.693   9.783  25.960  1.00  0.86           C  \\nATOM   1581  C   VAL A 205      17.544   8.957  26.515  1.00  0.86           C  \\nATOM   1582  O   VAL A 205      16.556   9.489  27.048  1.00  0.86           O  \\nATOM   1583  CB  VAL A 205      18.507   9.892  24.448  1.00  0.86           C  \\nATOM   1584  CG1 VAL A 205      17.106  10.438  24.103  1.00  0.86           C  \\nATOM   1585  CG2 VAL A 205      19.561  10.829  23.849  1.00  0.86           C  \\nATOM   1586  N   ARG A 206      17.646   7.627  26.431  1.00  0.79           N  \\nATOM   1587  CA  ARG A 206      16.693   6.695  27.002  1.00  0.79           C  \\nATOM   1588  C   ARG A 206      16.570   6.811  28.517  1.00  0.79           C  \\nATOM   1589  O   ARG A 206      15.456   6.844  29.054  1.00  0.79           O  \\nATOM   1590  CB  ARG A 206      17.095   5.255  26.622  1.00  0.79           C  \\nATOM   1591  CG  ARG A 206      16.049   4.194  27.004  1.00  0.79           C  \\nATOM   1592  CD  ARG A 206      16.466   2.726  26.875  1.00  0.79           C  \\nATOM   1593  NE  ARG A 206      17.228   2.601  25.595  1.00  0.79           N  \\nATOM   1594  CZ  ARG A 206      18.510   2.219  25.513  1.00  0.79           C  \\nATOM   1595  NH1 ARG A 206      19.177   1.793  26.585  1.00  0.79           N  \\nATOM   1596  NH2 ARG A 206      19.147   2.293  24.349  1.00  0.79           N  \\nATOM   1597  N   ALA A 207      17.693   6.920  29.247  1.00  0.87           N  \\nATOM   1598  CA  ALA A 207      17.725   7.155  30.682  1.00  0.87           C  \\nATOM   1599  C   ALA A 207      17.076   8.481  31.090  1.00  0.87           C  \\nATOM   1600  O   ALA A 207      16.275   8.534  32.032  1.00  0.87           O  \\nATOM   1601  CB  ALA A 207      19.178   7.080  31.191  1.00  0.87           C  \\nATOM   1602  N   ASP A 208      17.347   9.572  30.345  1.00  0.85           N  \\nATOM   1603  CA  ASP A 208      16.681  10.856  30.513  1.00  0.85           C  \\nATOM   1604  C   ASP A 208      15.159  10.755  30.303  1.00  0.85           C  \\nATOM   1605  O   ASP A 208      14.371  11.289  31.094  1.00  0.85           O  \\nATOM   1606  CB  ASP A 208      17.266  11.928  29.556  1.00  0.85           C  \\nATOM   1607  CG  ASP A 208      18.648  12.444  29.917  1.00  0.85           C  \\nATOM   1608  OD1 ASP A 208      19.233  12.153  30.984  1.00  0.85           O  \\nATOM   1609  OD2 ASP A 208      19.146  13.252  29.083  1.00  0.85           O  \\nATOM   1610  N   LEU A 209      14.699  10.019  29.267  1.00  0.87           N  \\nATOM   1611  CA  LEU A 209      13.290   9.720  29.024  1.00  0.87           C  \\nATOM   1612  C   LEU A 209      12.617   8.925  30.139  1.00  0.87           C  \\nATOM   1613  O   LEU A 209      11.501   9.257  30.551  1.00  0.87           O  \\nATOM   1614  CB  LEU A 209      13.067   8.966  27.686  1.00  0.87           C  \\nATOM   1615  CG  LEU A 209      13.312   9.793  26.410  1.00  0.87           C  \\nATOM   1616  CD1 LEU A 209      13.295   8.891  25.168  1.00  0.87           C  \\nATOM   1617  CD2 LEU A 209      12.284  10.917  26.260  1.00  0.87           C  \\nATOM   1618  N   GLU A 210      13.270   7.888  30.692  1.00  0.80           N  \\nATOM   1619  CA  GLU A 210      12.776   7.116  31.831  1.00  0.80           C  \\nATOM   1620  C   GLU A 210      12.584   8.004  33.062  1.00  0.80           C  \\nATOM   1621  O   GLU A 210      11.566   7.936  33.759  1.00  0.80           O  \\nATOM   1622  CB  GLU A 210      13.736   5.938  32.164  1.00  0.80           C  \\nATOM   1623  CG  GLU A 210      13.916   4.871  31.040  1.00  0.80           C  \\nATOM   1624  CD  GLU A 210      12.907   3.730  31.010  1.00  0.80           C  \\nATOM   1625  OE1 GLU A 210      11.779   3.931  31.516  1.00  0.80           O  \\nATOM   1626  OE2 GLU A 210      13.215   2.656  30.424  1.00  0.80           O  \\nATOM   1627  N   LYS A 211      13.521   8.931  33.340  1.00  0.82           N  \\nATOM   1628  CA  LYS A 211      13.361   9.905  34.413  1.00  0.82           C  \\nATOM   1629  C   LYS A 211      12.191  10.880  34.217  1.00  0.82           C  \\nATOM   1630  O   LYS A 211      11.515  11.287  35.182  1.00  0.82           O  \\nATOM   1631  CB  LYS A 211      14.642  10.741  34.646  1.00  0.82           C  \\nATOM   1632  CG  LYS A 211      14.591  11.542  35.964  1.00  0.82           C  \\nATOM   1633  CD  LYS A 211      15.396  12.846  35.927  1.00  0.82           C  \\nATOM   1634  CE  LYS A 211      14.902  13.898  36.927  1.00  0.82           C  \\nATOM   1635  NZ  LYS A 211      13.548  14.366  36.539  1.00  0.82           N  \\nATOM   1636  N   ILE A 212      11.944  11.353  32.990  1.00  0.85           N  \\nATOM   1637  CA  ILE A 212      10.787  12.182  32.649  1.00  0.85           C  \\nATOM   1638  C   ILE A 212       9.470  11.439  32.788  1.00  0.85           C  \\nATOM   1639  O   ILE A 212       8.490  11.973  33.330  1.00  0.85           O  \\nATOM   1640  CB  ILE A 212      10.918  12.721  31.220  1.00  0.85           C  \\nATOM   1641  CG1 ILE A 212      12.068  13.746  31.130  1.00  0.85           C  \\nATOM   1642  CG2 ILE A 212       9.600  13.347  30.705  1.00  0.85           C  \\nATOM   1643  CD1 ILE A 212      12.571  13.982  29.702  1.00  0.85           C  \\nATOM   1644  N   LEU A 213       9.400  10.205  32.281  1.00  0.86           N  \\nATOM   1645  CA  LEU A 213       8.149   9.493  32.128  1.00  0.86           C  \\nATOM   1646  C   LEU A 213       7.701   8.729  33.368  1.00  0.86           C  \\nATOM   1647  O   LEU A 213       6.476   8.585  33.585  1.00  0.86           O  \\nATOM   1648  CB  LEU A 213       8.273   8.517  30.934  1.00  0.86           C  \\nATOM   1649  CG  LEU A 213       8.396   9.181  29.544  1.00  0.86           C  \\nATOM   1650  CD1 LEU A 213       8.806   8.163  28.468  1.00  0.86           C  \\nATOM   1651  CD2 LEU A 213       7.097   9.887  29.143  1.00  0.86           C  \\nATOM   1652  N   GLY A 214       8.633   8.267  34.209  1.00  0.86           N  \\nATOM   1653  CA  GLY A 214       8.393   7.381  35.346  1.00  0.86           C  \\nATOM   1654  C   GLY A 214       8.167   5.882  35.007  1.00  0.86           C  \\nATOM   1655  O   GLY A 214       8.342   5.440  33.831  1.00  0.86           O  \\nATOM   1656  OXT GLY A 214       7.832   5.140  35.974  1.00  0.86           O  \\n\",\"pdb\");\n",
       "\tviewer_1762785482869049.setStyle({\"chain\": \"B\"},{\"style\": {}});\n",
       "\tviewer_1762785482869049.setStyle({\"chain\": \"A\"},{\"cartoon\": {\"color\": \"grey\", \"opacity\": 0.75}});\n",
       "\tviewer_1762785482869049.setStyle({\"chain\": \"A\", \"resi\": [5, 6, 7, 8, 9, 10, 12, 13, 14, 28, 84, 120, 123, 158, 167]},{\"stick\": {\"color\": \"green\"}, \"cartoon\": {\"color\": \"grey\", \"opacity\": 0.75}});\n",
       "viewer_1762785482869049.render();\n",
       "});\n",
       "</script>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<py3Dmol.view at 0x10a039220>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### TODO: display the protein binding the ligand, highlighting the binding residues\n",
    "\n",
    "display_protein_ligand(file_protein_open, binding_residues, ligand_file_open, 'A')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac71d5e2",
   "metadata": {},
   "source": [
    "Analyze the ligand docking on the closed conformation (best model from the last exercise)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "295569f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "### TODO: do the same analysis for complex with close conformation \n",
    "best_model_closed = # enter here your best model, e.g. model_01\n",
    "# we take the ligand model with the highest rank (1), e.g. rank1_confidence-0.73.sdf\n",
    "ligand_file_closed = #enter the ligand file\n",
    "ligand_closed = # use a function to parse a ligand sdf_file\n",
    "\n",
    "file_protein_closed = #enter the file of the best model\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7adcd588",
   "metadata": {},
   "outputs": [],
   "source": [
    "### TODO: explain the observations"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pib",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
