{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Processing FASTA files\n",
    "\n",
    "Exercise for processing DNA sequences.\n",
    "\n",
    "* **Contact:** mate.balajti@unibas.ch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "jupyter": {
     "is_executing": true
    }
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "\n",
    "filename1 = \"sequences.fasta\"\n",
    "filename2 = \"genome.fasta\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 1.1 (3 points)\n",
    "\n",
    "Write a function `parse_fasta()` that takes a path to a FASTA file as input\n",
    "and returns a tuple of two lists, the first containing sequence headers\n",
    "stripped of the leading `>`, and the second containing the actual sequences.\n",
    "\n",
    "> **Notes:**\n",
    ">  \n",
    "> * Please write the parser from scratch and do _not_ use existing FASTA\n",
    ">   parsers, such as the one provided by Biopython.\n",
    "> * Ensure that wrapped sequences are handled such that fragments of a given\n",
    ">   sequence are concatenated, without white space, in the order they appear\n",
    ">   in the file. Make use of the leading `>` character to separate records\n",
    ">   from each other.\n",
    "> * Ensure that the number of items in the returned lists correspond to the\n",
    ">   original number of records in the input file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(['chr1', 'chr2', 'chr3', 'chr4'],\n",
       " ['TATTTACGCTCCCGCACATATACCGTCGTAAGCGACTGGCTGCTCGCGACCAACCGCCTCGCTACGATTGAGTCGAGAGCGGGTAGACCGGAGGCCGTTCGGCGCATAACTTGTCTTGTTAATTAGGCTGCCTGTGAACGATATTCAATTATGGCACGTGGCGCATTAAGCTCCACGCCACATTCCAACTTCTAGTAGTATGCTGAAACACCGTGCGGGGGACAGAGAGCCATGGCAAATTCAGTTCGGTGTAGACCATACTTTATAACTGGACAACAAGTCCGAGTCCTTCACCTGATGCGCCCAGAACCCCGCATTTCTGTACCTATAGCACCGGCGCACACCCCTCTCACATGCTGGGCAGCAGAGTACTTCTCACTCGATCAGAGAAACCAAACCAGTTGATTGAAGTTCGCATAAGCGTGTTTGCTGTTGCTTGCGGCTCTCATGCACATAGTCGTTCAAGCTTACCCTCCACCGTCGGGCGACAATGCTCGGGGCGCGCATCAGACCGTATTGACTTTTTAGTACAGATTGGACGCAGAGGTTGAGTCATCGTATTCATGTGATAATCTCAACAGGTCCGTCTTGTTATCAAACTTAGTAGTGGGGAGGTGAGTGGTGCAATGTGGAAGTGCCGTAGCCAGCCCGCGATCCACGCACTATTTCAGGGACCGAGGTAACTTAAACCACCCGTTGATGGACATACTGGTTATAGATTTTAGTGCGGGAGTAGGCCTCGTCGCAGCGCAAAGGATGAAGTTTACTAACTGGAGTGGTCAGAAGTTGCCGCCTGTGAAGGCGGCACACGCTTAGTCCGCAATATTATGGGACTGGGGAGGCGACGGTATCCGAAGGTGCTCTGATCAGGATAGCCTCTAGACAGTAATCATGGGTTCTCCATTACGGAGTATTATGGCGGACCGATCGGTCACGTTAATTTGGCTAACTATACGTTCATATGGAGCACACTTCGCGGACCCGAGATCACTGGCCTGTTATTTATTTAGATCCTTCAAGGGTAACGACTACTACGTATAGCGTTAAATATCTCATGAGACCTCACAGAATCATTTTCGACTACAACTGCCCTCGGAGGCCTCCACAAGGCATTCATCCCGCGTTACTTCTGCCGCACTACGTGTCACCACTCAGCAACATTCTGATGGCTCTGCGAGAATTTGCGAGGATGGTAATATGATCCTTACGGAGCCGAAGGAAGCAGTCCACGTGGTCCGTTACTTTTCTAGATACGACGAATAGTTACTCCGAGGCTCTTCGTGCGTGGGTTGCCGGAACCTACGCATGCCACAGTCTAACCCACCCTAGAAGGGCTAGCCGGTAGCTGAGAGGGCCTGGGAGGCTTCCAGCTGTAAGGCTGAGTACACGCCGCTTGATGTCCACTGCGTCTTCCAATTCCTAAGACCCACTTATACGAGACATTCCTTGGTGACTGACGAAACGAGAAGCGCCTCCGCCGCCGGTCTAAGAATGTGGCCTGGTCAAGGACGGTAGGCGCCTAAACTAGGTGCAGTGAGGTACCCTTCCTACCTTGGGTTGCGCCGCAGGGGATTGACAAACACCTTTGTCAACTCATGATTAGAACAAAGCCCACGCGACCAGTGATGCCGCATCGCCTACGCTGAACCGTCGCAAAGTGTGCGGAATACTCTTCTGTCTGCTTGGTACTTTATGATAACGTGAGAGGAACCGAACGCGACCGGCCGTCGTTACATCATAGTAAAGCTACAGCCCGTTCACCCCCGGCAATAATACGACTCTTCCGAAAGGAAACTCAACACAATTGGCCGCTTCTAACATATAAATGCGGTTGGTGCCATTCCCTGCGGAACAATTTAGGACGGATAGTGAGGAACAAGTGCCATGCTTCACGCCAACTACTCCTAATGAAGACTCGTATTCCAATGAAACGAAAGGGTCTCACAGGACACGCACAGCCAAGCGTGGCAGGATCAATAACTAAGACAAGCATCGAGTATGACGCAGCCCCACTCGAATTTGTAGCCCCATGCGAGTCCACAGGGATTTTCTGCAGTAGGTGATACGAAACGTATAATTTTTTCCCCTGGCAGCGAAGGAACCATAAAGAAGTTGGTCTGCCCTAA',\n",
       "  'ATCCATCTACGCCTAGTGATATTCACGAGCAAACGCCTGCCAAGATACGTAAGACACTCATGTAAATAGTCCGTAGCCCCGACATCACCGGTCCACCTCGAGCAGCGCGCATTGATAAGGTGGGGATTGTGCCTATAGTATCGGTGGTGTATGTGTAACGCGTCCGGCAACGTCGTTTCATACCGCACTCTCGATTACGTGGTTCATAAGTATCTAAAGTATTTTCTTTCGCGGCGAATCTTTGACATAACACTATGCGATGCCCCTCCAGTTCATTCATAGCTGGAGGGGTAGATACCCCAAGTACCTTATTAGACGACCTAGTTACCTAGAAGTTGGTACGGGTGCCGCCACAGCGAGTTGTGTAGGGATAAGCATGAACATTCACAGACGCGCGAGAGGGTGGTGGACGGATGCCTTGCCTTTTTCCATGCGAGACAGACGAAGCTAATCGCCATTATCACTGCGTTGATGGGACTTTGCGTACTAGCGCTCAGTTATATCTAGCGGTAGTTATACCCTATGTGTTTATTGTTCCTTCTGGCCACCTCCGCGCTACCTTGCCTTCACGCAAGCAGCAGAACAGTATTCCGCCATAAATTGGTCTAGTCCTTCTTGCCAGCGATTACAAGCTGGGGAGGCGTTCAGTCTTTCCGAAAAGCGTTTTAAAGCTCCTCCCCTCAACCCTTCGACCTACCTGAGCAAATGGAATTAATAGTGGCTATCTAACTGGTCCACCATCATGTAGAGCCAAGGATACGTATGGTCATTAGAGAACTAATGTACACAGTAGGCGAGTAAAGCGTGTGCAGATCTTCCACCCACAGGTATATTTTACGGAGGTCTGGCTCCCCGGTCCGTAACAGGTCCACCGCCTCCGGGTAAGTTGGCGCGTATTGTGCTCGTTGCTGACCGGCGACGAACACTTCCACACCTCGTGGTTGTACGGTTTGATGGGCGCTAGTTTGTATTGCTCTAGAAGGCCCCGTAAGTCAGCCATCACGCACGTCACCGCAATTGTTGCCTCGTAACGAGTGATTTGAGCGCCCCGCTTACTTCAGTGACCACGAGAACGGGAGGACCTTCCACATGTGCCTCCGCCGAAGCGGCACCGAGACTGTGGTCCGTCTCTGCCATACGTGTTGGACAGTTGCAAACCGCGTCTCCCCTTAAATTTGGCAGTTGAGTCCCCACGTCAAACGTCCACCTCTGTCCTTGGCGCCGATTGTTAGGGAGAAGGTAACTTCTCCGGAGTAATGTGTACTCAGGTCGAGCATTGGCTCCGTTATAGTATAGGTCCCGTAGGACTACAGTGGTTCGGGGACCTTCCGGTGAACAGCTTGCGTGAAAGTGACCGTAGAAGGGGTAACTTGTAAACCTCGGGCGTGCGTCTGAATCAGTCGAGTAGTATCCTCCGCATAGCCCGGGCGTATGTATGAGAGATGTGGCCAGAAGTCGAAATCAGCGGTACAGGTTGCACTCATTAAGAAGCGAGTTTGGTGAGAGGCGAGACCGCAAGCACACGTGTGAGTCTTTTTGATCGGTCCTCTTCGGTCGGTGGCACTGAGGCTTGAAGCCATGTATAACGGCGCCGAGAACTTGGCGGCTAAGACTCATGGGCAATCGATGTTGTTAGAGGCTTGCACAAAAAACTCGAGCTTATGTCTATAACAAAACGCTCAGAATCCCTCTCAAACCGCGGGGCAATGCGGCCGCAACGATCTAGGGTCGGGCCTAGTGTTGTCTTTAATGCAGATAAGCACAGTTCGAAGGAGTATATCACAGAACCGCCTTCTGGAATCGACTTGAAAATCTGAACTTCCGGCCGATTTGAATCGCGAACCCATTTTGGACTCTGCTCGGTTATACGATGGAAATGTGCTGACATCCGGCGACAGTTAGCAGTGACCATGAAACCTTGGGACCGTTGCGCCTGCACACACACGATAATGAACGCGGCCAAGGTCATTGATTTAGCCTGCTGATCTCGCTGGCCTCATACGCTTGAGAAAGTCTTCGGCTACCCACTGACGCGCGACTCCATTGCATCGAGTAAAGAAGCTACCTTTGCCGGTGGGGCGGGCCGTGGGAATCAGCCGCATCATAAAGTGTCCAGTCCAGAGCAGT',\n",
       "  'GTACCACACCATCGACTGTTGCACCTATTCGTCTGGATCAGCACTAGTTAGATCTTGGCGAAAGTGAACAACCCAAAGACTAGGGTAAATCCTAGCGCCTACTCAGAGACACTTGAGGATGGGACCATAGCAACCTTACCCCTCCACGGTTATAAGTTTCGGGAGAGTCCGGCGTTTTGGGATAGCGAACCCTGGGACGAGGCGCGGCTCTGTGCGTAATACGATACTGGCAGTGTGTACTAGGGACGAGGCTTCGCAGCAACTGAATACATAAGACCTGACTACGCCAAATCCAGCCATGAAAGATCTTAGTCTAATGCTCCTACCACCCCACGTACTTCCGTTAATCATAAGTCAAGGCAGACGTACCAACACGCCACGTAGAGCGTAAGACTGAGTCGTAATTAAACACTGACCGCGTATTGCGGCAGAACCAGTTAGGCCCGTCGCTTACAACTCCACGCACTCAGTGTCCGACCTGCAGTCCTAGCACTGACATATGACTGGAGATCTACTTCGCCGGATCTCCCATCATCTGGTGTGGAGAGTGTAGTCGGATGAGCACGTCCCCCGGGAAGAGGTCCGAGTATTTGCGAGGCCTTCCTTTTATATGATCGCCCTGAATGCAGAGCAGGCGGGTGAGAAAGACGCCATGGTGCCGGGGACCCAGGTCGCTCTCACATGACATGCTCACAATTATACTTTAATAAAGCCTAGAATTGGTCACTTGATGTCTACATTACACCAGCTTGCCAAATGATTGCAACGCGTTACAGTGCAACAGACGATCGATTTTATCTAAGCACTAATGTCAAGAGGAGTCAACTACACCCAAGGCCTTCCTACTAGTGGTGTCAGCAAACGGGAACAGGTCTGAGTTTCCATGTGACAAGCTCGCTGTCTTCTTGGCCTTTTAAGACGGCTTTAGCCTGTGAGGGAGAAAGTGGCCTCTCTCACTCCTAACTAGAGTTCCTCCGAGACCGACCGAAGTGGATAGTACACATTTCGTGTTATTGGCTCTCATTTGTTTATTCTTGTCTAGGCATAGAAGTTGTCCCCACTAGCCTGGTCTCGTTCAGGCGTAGTATATGGAATACATAGTCGATGGTCTAGTGAAACTCCTTTAATACCCCCATAAATCTCCTTCCGTCAGTAGTTCTGCGAAGACTACGATTCCTCCTAAGTGATGGATAAACGCTCACTAATGTCGGATGGGACATGAAAGACATTAGATTTGCGACATTCGATAAGCCCCATTACCTCTTAGCTACTGCTGCAGCGCACTCTCTGGTCAATCACAATAACGCGGCGTGCGATGCACCCCGGCAGAGGGACTTTCGTACATTACGGACCCGCGGTTTGTAGTTCTTCATCTGCAATGTCTCATTGCTAAGGTTCTGTTGTTGGTTGGGGGGGGACGCTTTGAGAGGCCTCCACCTCAACCGAGCTAGTGCTTGGGCATGTGATCCTATTCGCCTCATATCATATTGATCATCCAGTTCTAGTTGCCCGGTGACGAGCTCCTATGTATTCCTCTCGAATCTATATGGGCGGACCTAGTGTCCGCCTCGCCCTTTGTACACCATGAGGAAAAACCGGTGCAGTCGGGGGAATACGGACCAGGCAGTGCAATTGCAAAATCCCGGCGGGGTTCCGGGATTACACAAAGGTATGCAGCCCAGTGTCCAGTTGTGAACATTGCGGGATCGGGCCTGCAAGCCGCCGCCTAGGCAAGTGGCACAAATATCTTGGCTTCGAGGCACACTTCACGGTTTACTAGCAGTCCCATAAGCGAAGGACTAGTAAATCCCGCCGTTATCCTTCGTCCACAACATGGTTGGCTATGTGGTGAGTGAACTCTCCTTAAGCCATAATAACGATGTAACGCGTCCTTTACGGTGCGAAGTCAGCCTGTTCGCAATCAGCAAGGAACTAGACCCTGAAGGAAGCGTCGGAAGTGAGAGCCTCCACTATCGGAATCCCCTTGCACTCCCCGTACACGTCGTATGTGCACCGGGAGCTCTCTTCGCGCTTCCCGCTGTGCTTTCTACCGGTTCAGAGCATACATGGAGTTTGGCATTGTGCTGTTAAGCTATAGTAACGAAGTGCGTGGCCTACTCGGAA',\n",
       "  'ACCCATCGACCACATTATATGAACTGAACGCCTTGACTTATCGACAAGTAACTCGCCCACATGAAATTAAATATGGGGTGATCTACCCGATTCACCTGGTGCCAACCCTCAGACGGCTGACCGCATATTAACGAGGTACATCAACAAGACACGAGGATCTACCGATGGCCGGGGGTTTCATAATCGGGGACACGAAATAAGGTGAAATATCGTGCACTGCTAGTGCCTCGGGCAGACCGTCTACCCTAAACGCGAGCCGATGGCCTGTCTGGATGCTGCATTTCGTACCAACAGTGTACCCCGCTACAGGCGAGATGGTTCCTCAGGACTCGTGGGGGTGTAACCATTGACCAGTGTCGAGAAAATTCAATAACCAGGGGCGTGAAGCCTGGTGTGCTCTATAGAGACGGCCTCTTACGTTAGAAACTCAACGATGCTGTAGTGTAAGGCAAATCCCTATCCTCTGGGCTACTAGAGGCTAGCAAATGTCCGCCGGGTCTGCATCACAGGGCTAATCCGCACCCTTCCGAGTAGCCTTTGGACACATCAGTTGTTCTGTTAGCAGCGCGGCTGGAATTAGAGATTACAAATTAGACCGCGACTGCGTCGGCGCACTTACACAAATTCCGAACACCGATGTGTGCCGAAACCCTCGTACAGTGGAGGACGTCGGACAGAAAGAACGTCATTAATTCCAGCGTAGACATCGATGGGGTTTGGCGCTTATTTCGCGGCTGAGTGCGGCAGCCATTCCTGCGGTGAAATCGCCCCAATACCGCTCTAAATTAACCATTAGGGGCCGACTTGCACGTATCTCGAAGGACGAGACCTAAATCCCCCACAATTCCTATCCACTCCATGTTAGCATAATTCCATGAGTTACGTGGCGGCTAAGCGTAAGTCAGCTCTGGTATTTTGGATCTGATCTACATTAATTAGTAGCAGTGGTTCTTACCGGGCAGCGAAGGAACTCGTAGTTCTAGGAATCATGTGTGCCCCCCCTGCGGACTTGTCCTATATTACAAATTTGCGAACGCCTTGAGAATGAATACAGTATACTTCGAAGTGCGCGCCTTTGAAATAGCGACAGCTCCCTAAAGATCGGTGCGCGAAGTGCTTGTTGACGGCCACCGTTAGACCATGATGTCGGGAAGCTTGCCGTAGTACTATTGACGTACCAATAACATCACCCTATGACAGGGCACTCCCCCCAATTGCACGGTTGACACTCTGTTCATTCGCTGAGATTGATGACTCTGCTTTTGAACGTTGCTTCTGTCCAAATGATTATGTGTCCGGTAACTATAAACGTGCTACGATGTCTGGATTAGAACAGCAAGCGGGATCCAACAGCCATCCCTTCTAACAGCACAATGATAGTACACACATACGGGAGTCCCCCTGAGGTGTCGCAATTATGCATCATCGAGCAACTATATTTTCATGCCCGCACCTTTGAGAATACATAGGAACATTACGTTTCCTGGGGATATGTCGTTGAATATCTAAACTGGCCGCTCATAGGGTCCATACAAGCTCAGTGCTGACTCTAGGTGAAGAGGTGAGGTAAACGTTCTTCAATCCCGTACGAACGTGCGGATCGCGCGGGCTCTCACCAGCTATCTTACTATTGCCCTCTGTGGAAGGGACGGACATTATCTAACTGGCATTATCTGTAACACAACGCTACTCCGCCCTTTGCAGAACTAATACAACTCTTTTGACATATCATCTGACAGCTTCGTTGAATTAATAATGTCACCGACGCGTGTCCGTAAGAGGCTACAGCCCTAGGGGTTCACCAACATTGGGGTGGATAGAGTGCTCTCGGGTTGCATCGACTCGCCGAAGGGTCTCTGTTTCCGTAAATGACCAGATCCGTAAGCACAGGTTGTGTGTTATGGATTCAAAATAACCCAAGGAGCTCATCCGTGCTCAGCACAGCGATAAGGTAGTAGTGATCGGCAGCTACTGATGCATGGGTCGTGACGGCACTTCTAGCTCTTAGTTCATCCCCAATGCCGGAAGTTGTAGCAAACCACTGTTCGGCTCGTGGAATGAATCGATCTGTCTCAGCCTTCATGGCTTCATAAGATGGGGATGATAAAGAAGCAAGTGTACACAGGCGTATACCC'])"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def parse_fasta(path: str) -> tuple[list[str], list[str]]:\n",
    "\n",
    "    file = open(path)\n",
    "    #print(file.read())\n",
    "\n",
    "    headerList = []\n",
    "    sequenceList = []\n",
    "    listCounter = -1\n",
    "\n",
    "    for line in file: \n",
    "        if line[0] == \">\":\n",
    "            #headerList.append(line[1:])\n",
    "            # In order to not print the \\n characters ill just print the following 4 chars\n",
    "            headerList.append(line[1:5])\n",
    "            listCounter += 1 # first run through will be 0 for chr1\n",
    "            sequenceList.append(\"\")\n",
    "        \n",
    "        else:\n",
    "            sequenceList[listCounter] += line[0:-1] # \\n is 1 character not 2!\n",
    "            \n",
    "    file.close()\n",
    "\n",
    "    header = headerList\n",
    "    sequence = sequenceList\n",
    "    #print(\"Header: \", header)\n",
    "    #print(\"Sequences: \", sequence)\n",
    "    return header, sequence\n",
    "\n",
    "parse_fasta(\"genome.fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 1.2 (2 points)\n",
    "\n",
    "Write a function `discard_ambiguous_seqs()` that takes two arguments, a list of sequence headers and list of strings as input \n",
    "and returns a tuple of two lists with only those headers and strings that exclusively consist of letters of the\n",
    "\"DNA alphabet\" (`A`, `C`, `G`, `T`).\n",
    "\n",
    "> **Notes:**\n",
    ">  \n",
    "> * Make sure your implementation is case-insensitive, i.e., sequences\n",
    ">   containing lowercase DNA characters, even if mixed with uppercase\n",
    ">   characters, are valid as well."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: Sequence 1 contains illegal characters.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(['chr2', 'chr3'], ['GTAC', 'AAAA'])"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def discard_ambiguous_seqs(header: list[str], sequence: list[str]) -> \\\n",
    "  tuple[: list[str], : list[str]]:\n",
    "    \n",
    "    r_header = []\n",
    "    r_sequence = []\n",
    "    \n",
    "    if len(header) != len(sequence): \n",
    "        print(\"Error: List sizes of the two inputs mismatch. \")\n",
    "        return\n",
    "    \n",
    "    counter = 0\n",
    "\n",
    "    for element in header: \n",
    "        #print(element)\n",
    "        error = False\n",
    "        for char in sequence[counter]:\n",
    "           #print(char)\n",
    "           if char == \"A\" or char == \"a\" or char == \"G\" or char == \"g\" or char == \"T\" or char == \"t\" or char == \"C\" or char == \"c\":\n",
    "               continue\n",
    "           else: \n",
    "               print(\"Error: Sequence \" + str(counter + 1) + \" contains illegal characters.\")\n",
    "               error = True\n",
    "               break\n",
    "        \n",
    "        if error == False: \n",
    "            #print(\"No error\")\n",
    "            r_header.append(header[counter])\n",
    "            r_sequence.append(sequence[counter])\n",
    "            \n",
    "        counter += 1\n",
    "    \n",
    "    return r_header, r_sequence\n",
    "\n",
    "\n",
    "list1 = [\"chr1\", \"chr2\", \"chr3\"]\n",
    "list2 = [\"ABCD\", \"GTAC\", \"AAAA\"]\n",
    "\n",
    "discard_ambiguous_seqs(list1, list2)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 1.3 (2 points)\n",
    "\n",
    "Write a function `nucleotide_frequencies()` that takes a list of strings as\n",
    "input, and which prints out the total frequency of each nucleotide across\n",
    "all input sequences. Use the following example as a template to format your\n",
    "output:\n",
    "\n",
    "```console\n",
    "A: 0.31\n",
    "C: 0.21\n",
    "G: 0.17\n",
    "T: 0.31\n",
    "```\n",
    "\n",
    "> **Notes:**\n",
    ">\n",
    "> * Note how numbers are rounded in the example and format decimals printed\n",
    ">   by your solution in the same manner, i.e., rounded to two significant\n",
    ">   digits.\n",
    "> * The function does not require any specific return value. In case you are\n",
    ">   not aware of how Python deals with functions without an explicit `return`\n",
    ">   statement, look up the behavior in relevant documentation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A: 0.33\n",
      "C: 0.28\n",
      "G: 0.22\n",
      "T: 0.17\n",
      "Total: 18\n"
     ]
    }
   ],
   "source": [
    "def nucleotide_frequencies(seqs: list[str]) -> None:\n",
    "\n",
    "    # alt shift click to add carets\n",
    "    # ctrl shift arrows to add carets up or down\n",
    "    # alt shift up down to copy up down\n",
    "    # alt control up down to move line up down\n",
    "\n",
    "    number_g = 0\n",
    "    number_t = 0\n",
    "    number_a = 0\n",
    "    number_c = 0\n",
    "    number_char = 0\n",
    "\n",
    "    for char in seqs: \n",
    "        #print(char)\n",
    "\n",
    "        if char == \"G\" or char == \"g\": \n",
    "            number_g += 1\n",
    "        if char == \"T\" or char == \"t\": \n",
    "            number_t += 1\n",
    "        if char == \"A\" or char == \"a\": \n",
    "            number_a += 1\n",
    "        if char == \"C\" or char == \"c\": \n",
    "            number_c += 1\n",
    "\n",
    "        number_char += 1\n",
    "\n",
    "    percentage_a = round((number_a / number_char),2) #* 100 round(0.123, 2) = 0.12\n",
    "    percentage_c = round((number_c / number_char),2) #* 100\n",
    "    percentage_g = round((number_g / number_char),2) #* 100\n",
    "    percentage_t = round((number_t / number_char),2) #* 100\n",
    "\n",
    "    print(\"A: \" + str(percentage_a))\n",
    "    print(\"C: \" + str(percentage_c))\n",
    "    print(\"G: \" + str(percentage_g))\n",
    "    print(\"T: \" + str(percentage_t))\n",
    "    print(\"Total: \" + str(number_char))\n",
    "\n",
    "nucleotide_frequencies(\"AAACacgtcggctagcta\")\n",
    "\n",
    "        \n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 1.4 (3 points)\n",
    "\n",
    "Write a function `map_reads()` that takes as input two FASTA files, the first\n",
    "containing short read sequences (\"query\"), and the second containing reference\n",
    "sequences. The function should \n",
    "* read the files, \n",
    "* discard _query_ sequences that contain non-DNA characters, \n",
    "* print the nucleotide fractions for both files to\n",
    "the console \n",
    "* and returns a dictionary of dictionaries, where the outer\n",
    "dictionary uses the names of query sequences as its keys, and the inner\n",
    "dictionary uses reference sequence names as keys and a list of 1-based indices\n",
    "indicating at which position (counting from left to right) in the reference\n",
    "sequence the query sequence occurs as an exact substring.\n",
    "\n",
    "Execute the function, passing `sequences.fasta` and `genome.fasta` as input.\n",
    "Inspect the returned \"hits\" object (the dictionary of dicionaries). Interpret\n",
    "the results in at least 2-3 bullet points. What's special about query sequence\n",
    "`sequence4`?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: Sequence 3 contains illegal characters.\n",
      "cleanListFile1: \n",
      "(['sequence1', 'sequence2', 'sequence4'], ['GAAGTTTACTAaCTGGAGTGGTCAGaAGTTGCCGCCTGTG', 'GCCCGGGCGTATGTATGAGAGATGTGGCCAGAAGTCGAAA', 'TTTGAG'])\n",
      "Percentages of File sequences.fasta\n",
      "\n",
      "\n",
      "A: 0.23\n",
      "C: 0.17\n",
      "G: 0.33\n",
      "T: 0.28\n",
      "Total: 40\n",
      "\n",
      "\n",
      "A: 0.28\n",
      "C: 0.17\n",
      "G: 0.38\n",
      "T: 0.17\n",
      "Total: 40\n",
      "\n",
      "\n",
      "A: 0.17\n",
      "C: 0.0\n",
      "G: 0.33\n",
      "T: 0.5\n",
      "Total: 6\n",
      "\n",
      "\n",
      "Percentages of File genome.fasta\n",
      "\n",
      "\n",
      "A: 0.25\n",
      "C: 0.26\n",
      "G: 0.25\n",
      "T: 0.24\n",
      "Total: 2126\n",
      "\n",
      "\n",
      "A: 0.24\n",
      "C: 0.26\n",
      "G: 0.26\n",
      "T: 0.25\n",
      "Total: 2127\n",
      "\n",
      "\n",
      "A: 0.24\n",
      "C: 0.26\n",
      "G: 0.25\n",
      "T: 0.25\n",
      "Total: 2124\n",
      "\n",
      "\n",
      "A: 0.26\n",
      "C: 0.25\n",
      "G: 0.24\n",
      "T: 0.25\n",
      "Total: 2135\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'sequence1': {'chr1': [], 'chr2': [], 'chr3': [], 'chr4': []},\n",
       " 'sequence2': {'chr1': [], 'chr2': 1422, 'chr3': [], 'chr4': []},\n",
       " 'sequence4': {'chr1': [], 'chr2': 1039, 'chr3': 1422, 'chr4': 1455}}"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def map_reads(filename1: str, filename2: str) -> dict[dict[str: list[int]]]:\n",
    "\n",
    "    file1 = open(filename1)\n",
    "    file2 = open(filename2)\n",
    "\n",
    "    # Discard Queries\n",
    "\n",
    "    list1 = []\n",
    "    list2 = []\n",
    "\n",
    "    for line in file1: \n",
    "        if line[0] == \">\": \n",
    "            list1.append(line[1:-1])\n",
    "        else: \n",
    "            list2.append(line[:-1])\n",
    "\n",
    "    #print(list1)\n",
    "    #print(list2)\n",
    "\n",
    "    cleanListFile1 = discard_ambiguous_seqs(list1, list2)\n",
    "\n",
    "    print(\"cleanListFile1: \")\n",
    "    print(cleanListFile1)\n",
    "\n",
    "    # Fractions\n",
    "\n",
    "    print(\"Percentages of File \" + filename1)\n",
    "    print(\"\\n\")\n",
    "\n",
    "    for i in range(len(cleanListFile1[1])):\n",
    "        nucleotide_frequencies(cleanListFile1[1][i])\n",
    "        print(\"\\n\")\n",
    "\n",
    "  \n",
    "    print(\"Percentages of File \" + filename2)\n",
    "    print(\"\\n\")\n",
    "\n",
    "    listFile2 = parse_fasta(filename2)\n",
    "    cleanListFile2 = discard_ambiguous_seqs(listFile2[0], listFile2[1]) # i guess it can be assumed that this file is clean\n",
    "    #print(cleanListFile2)\n",
    "    for i in range(len(cleanListFile2[1])):\n",
    "        nucleotide_frequencies(cleanListFile2[1][i])\n",
    "        print(\"\\n\")\n",
    "    \n",
    "    # Sequence Dictionaries\n",
    "    # set up double dict\n",
    "\n",
    "    findings = {}\n",
    "    \n",
    "    countOutisde = 0\n",
    "    countInside = 0\n",
    "\n",
    "    for i in cleanListFile1[0]:\n",
    "        findings[i] = {}\n",
    "        countInside = 0\n",
    "        for k in cleanListFile2[0]:\n",
    "            position = cleanListFile2[1][countInside].find(cleanListFile1[1][countOutisde])\n",
    "            if position != -1: \n",
    "                findings[i][k] = position + 1 # positioning from find starts at 0\n",
    "            else: \n",
    "                findings[i][k] = []\n",
    "            \n",
    "            countInside += 1\n",
    "\n",
    "        countOutisde += 1\n",
    "\n",
    "\n",
    "    # finder\n",
    "\n",
    "    # for i in range(len(cleanListFile2[1])): # check how many genomes there are\n",
    "    #     counter = 0\n",
    "\n",
    "    #     for char in cleanListFile2[1][i]: #look through genome list chr[i]\n",
    "\n",
    "    #         if char == cleanListFile1[1][0][counter]:\n",
    "    #             counter += 1\n",
    "    #             continue\n",
    "    #         else:\n",
    "    #             counter = 0 # reset counter\n",
    "    #             break\n",
    "\n",
    "    # ich benutze einfach sequence.find(string) in der dict builder loop\n",
    "\n",
    "    \n",
    "\n",
    "    \n",
    "    file1.close()\n",
    "    file2.close()\n",
    "\n",
    "    return findings\n",
    "\n",
    "map_reads(\"sequences.fasta\", \"genome.fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Inspect the returned \"hits\" object (the dictionary of dicionaries). Interpret\n",
    "the results in at least 2-3 bullet points. What's special about query sequence\n",
    "`sequence4`?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sequence 1 is not found in any of the genomes. \n",
    "Sequence 2 is found only in genome2 at pos 1422. \n",
    "Sequence 3 has been removed from the findinds as it included an illegal character \"D\"\n",
    "Sequence 4 has been found in genome 2, 3 and 4. \n",
    "\n",
    "Sequence 4 has been found the most as it is also the shortest of the 3 sequences. Thus the probability of it occurring is higher. \n",
    "\n",
    "Its worthy to note that all these findings show the first found occurrence of the sequence, so there is no way of telling if they occur multiple times in the genome with this version of the code. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
